//
// $Id: cv.hpp 3625 2012-05-14 22:26:50Z chambm $
//
//
// Original author: Matt Chambers <matt.chambers .@. vanderbilt.edu>
//
// Copyright 2008 Spielberg Family Center for Applied Proteomics
//   Cedars Sinai Medical Center, Los Angeles, California  90048
// Copyright 2008 Vanderbilt University - Nashville, TN 37232
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// This file was generated by cvgen_cli.
//
// Do not edit this file! Your changes will be lost next time cvgen_cli is run -
// see pwiz/scripts/misc/update_cv.bat for info on how that works.
// Instead, edit cvgen_cli.cpp itself.
//


#ifndef _CV_HPP_CLI_
#define _CV_HPP_CLI_


#include "pwiz/data/common/cv.hpp"
#include "SharedCLI.hpp"


// [psi-ms.obo]
//   format-version: 1.2
//   data-version: 3.25.0
//   date: 09:05:2012 16:37
//   saved-by: Gerhard Mayer
//   auto-generated-by: OBO-Edit 2.1
//   import: http://unit-ontology.googlecode.com/svn/trunk/unit.obo
//   default-namespace: MS
//   remark: namespace: MS
//   remark: version: 3.25.0
//   remark: release date: 2012-05-09
//   remark: coverage: Mass spectrometer output files and spectra interpretation
//   remark: creator: Matt Chambers <matt.chambers <-at-> vanderbilt.edu>
//   remark: creator: Andreas Bertsch <bertsch <-at-> informatik.uni-tuebingen.de>
//   remark: creator: Marius Kallhardt <Marius.Kallhardt <-at-> bdal.de>
//   remark: creator: Eric Deutsch <edeutsch <-at-> systemsbiology.org>
//   remark: creator: Fredrik Levander <fredrik.levander <-at-> immun.lth.se>
//   remark: creator: Gerhard Mayer <mayerg97 <-at-> rub.de>
//   remark: publisher: HUPO Proteomics Standards Initiative Mass Spectrometry Standards Working Group and HUPO Proteomics Standards Initiative Proteomics Informatics Working Group
//   remark: When appropriate the definition and synonyms of a term are reported exactly as in the chapter 12 of IUPAC orange book. See http://www.iupac.org/projects/2003/2003-056-2-500.html.
//
// [unimod.obo]
//   format-version: 1.2
//   date: 14:07:2011 12:49
//   saved-by: psi-pi_team
//   default-namespace: UNIMOD
//
// [unit.obo]
//   format-version: 1.2
//   date: 12:10:2011 11:21
//   saved-by: George Gkoutos
//   auto-generated-by: OBO-Edit 2.1-beta13
//   subsetdef: unit_group_slim "unit group slim"
//   subsetdef: unit_slim "unit slim"
//   default-namespace: unit.ontology
//   namespace-id-rule: * UO:$sequence(7,0,9999999)$
//   import: http://purl.obolibrary.org/obo/pato.obo
//


namespace pwiz {
namespace CLI {
namespace cv {


/// <summary>enumeration of controlled vocabulary (CV) terms, generated from OBO file(s)</summary>
public enum class CVID
{
    CVID_Unknown = -1,

    /// <summary>Proteomics Standards Initiative Mass Spectrometry Vocabularies: Proteomics Standards Initiative Mass Spectrometry Vocabularies.</summary>
    MS_Proteomics_Standards_Initiative_Mass_Spectrometry_Vocabularies = 0,

    /// <summary>sample number: A reference number relevant to the sample under study.</summary>
    MS_sample_number = 1000001,

    /// <summary>sample name: A reference string relevant to the sample under study.</summary>
    MS_sample_name = 1000002,

    /// <summary>sample state: The chemical phase of a pure sample, or the state of a mixed sample.</summary>
    MS_sample_state = 1000003,

    /// <summary>sample mass: Total mass of sample used.</summary>
    MS_sample_mass = 1000004,

    /// <summary>sample volume: Total volume of solution used.</summary>
    MS_sample_volume = 1000005,

    /// <summary>sample concentration: Concentration of sample in picomol/ul, femtomol/ul or attomol/ul solution used.</summary>
    MS_sample_concentration = 1000006,

    /// <summary>inlet type: The nature of the sample inlet.</summary>
    MS_inlet_type = 1000007,

    /// <summary>ionization type: The method by which gas phase ions are generated from the sample.</summary>
    MS_ionization_type = 1000008,

    /// <summary>ionization mode: Whether positive or negative ions are selected for analysis by the spectrometer.</summary>
    MS_ionization_mode_OBSOLETE = 1000009,

    /// <summary>analyzer type: The common name of the particular analyzer stage being described. Synonym of mass analyzer, should be obsoleted.</summary>
    MS_analyzer_type = 1000010,

    /// <summary>mass resolution: Smallest mass difference between two equal magnitude peaks so that the valley between them is a specified fraction of the peak height.</summary>
    MS_mass_resolution = 1000011,

    /// <summary>resolution measurement method: Which of the available standard measures is used to define whether two peaks are separate.</summary>
    MS_resolution_measurement_method = 1000012,

    /// <summary>resolution type: Specify the nature of resolution for the mass analyzer. Resolution is usually either constant with respect to m/z or proportional to m/z.</summary>
    MS_resolution_type = 1000013,

    /// <summary>accuracy: Accuracy is the degree of conformity of a measured mass to its actual value.</summary>
    MS_accuracy = 1000014,

    /// <summary>scan rate: Rate in Th/sec for scanning analyzers.</summary>
    MS_scan_rate = 1000015,

    /// <summary>scan start time: The time that an analyzer started a scan, relative to the start of the MS run.</summary>
    MS_scan_start_time = 1000016,

    /// <summary>Scan Function: Describes the type of mass analysis being performed. Two primary modes are: typical acquisition over a range of masses (Mass Scan), and Selected Ion Detection. The primary difference is that Selected Ion Detection produces a single value for the signal at the selected mass rather than producing a mass spectrum.</summary>
    MS_Scan_Function = 1000017,

    /// <summary>scan direction: Direction in terms of m/z of the scan for scanning analyzers (low to high, or high to low).</summary>
    MS_scan_direction = 1000018,

    /// <summary>scan law: Describes the function in control of the m/z scan (for scanning instruments). Commonly the scan function is linear, but in principle any function can be used.</summary>
    MS_scan_law = 1000019,

    /// <summary>scanning method: Describes the acquisition data type produced by a tandem mass spectrometry experiment.</summary>
    MS_scanning_method = 1000020,

    /// <summary>reflectron state: Status of the reflectron, turned on or off.</summary>
    MS_reflectron_state = 1000021,

    /// <summary>TOF Total Path Length: The length of the field free drift space in a time of flight mass spectrometer.</summary>
    MS_TOF_Total_Path_Length = 1000022,

    /// <summary>isolation width: The total width (i.e. not half for plus-or-minus) of the gate applied around a selected precursor ion.</summary>
    MS_isolation_width = 1000023,

    /// <summary>final MS exponent: Final MS level achieved when performing PFF with the ion trap (e.g. MS E10).</summary>
    MS_final_MS_exponent = 1000024,

    /// <summary>magnetic field strength: A property of space that produces a force on a charged particle equal to qv x B where q is the particle charge and v its velocity.</summary>
    MS_magnetic_field_strength = 1000025,

    /// <summary>B: A property of space that produces a force on a charged particle equal to qv x B where q is the particle charge and v its velocity.</summary>
    MS_B = MS_magnetic_field_strength,

    /// <summary>detector type: Type of detector used in the mass spectrometer.</summary>
    MS_detector_type = 1000026,

    /// <summary>detector acquisition mode: Method by which detector signal is acquired by the data system.</summary>
    MS_detector_acquisition_mode = 1000027,

    /// <summary>detector resolution: The resolving power of the detector to detect the smallest difference between two ions so that the valley between them is a specified fraction of the peak height.</summary>
    MS_detector_resolution = 1000028,

    /// <summary>sampling frequency: The rate of signal sampling (measurement) with respect to time.</summary>
    MS_sampling_frequency = 1000029,

    /// <summary>vendor: Name of instrument vendor.</summary>
    MS_vendor_OBSOLETE = 1000030,

    /// <summary>instrument model: Instrument model name not including the vendor's name.</summary>
    MS_instrument_model = 1000031,

    /// <summary>customization: Free text description of a single customization made to the instrument; for several modifications, use several entries.</summary>
    MS_customization = 1000032,

    /// <summary>deisotoping: The removal of isotope peaks to represent the fragment ion as one data point and is commonly done to reduce complexity. It is done in conjunction with the charge state deconvolution.</summary>
    MS_deisotoping = 1000033,

    /// <summary>charge deconvolution: The determination of the mass of an ion based on the mass spectral peaks that represent multiple-charge ions.</summary>
    MS_charge_deconvolution = 1000034,

    /// <summary>peak picking: Spectral peak processing conducted on the acquired data to convert profile data to centroided data.</summary>
    MS_peak_picking = 1000035,

    /// <summary>scan mode: OBSOLETE.</summary>
    MS_scan_mode_OBSOLETE = 1000036,

    /// <summary>polarity: Terms to describe the polarity setting of the instrument.</summary>
    MS_polarity = 1000037,

    /// <summary>minute: Acquisition time in minutes.</summary>
    MS_minute_OBSOLETE = 1000038,

    /// <summary>second: Acquisition time in seconds.</summary>
    MS_second_OBSOLETE = 1000039,

    /// <summary>m/z: Three-character symbol m/z is used to denote the quantity formed by dividing the mass of an ion in unified atomic mass units by its charge number (regardless of sign). The symbol is written in italicized lower case letters with no spaces. Note 1: The term mass-to-charge-ratio is deprecated. Mass-to-charge ratio has been used for the abscissa of a mass spectrum, although the quantity measured is not the quotient of the ion's mass to its electric charge. The three-character symbol m/z is recommended for the quantity that is the independent variable in a mass spectrum Note 2: The proposed unit thomson (Th) is deprecated.</summary>
    MS_m_z = 1000040,

    /// <summary>mass-to-charge ratio: Three-character symbol m/z is used to denote the quantity formed by dividing the mass of an ion in unified atomic mass units by its charge number (regardless of sign). The symbol is written in italicized lower case letters with no spaces. Note 1: The term mass-to-charge-ratio is deprecated. Mass-to-charge ratio has been used for the abscissa of a mass spectrum, although the quantity measured is not the quotient of the ion's mass to its electric charge. The three-character symbol m/z is recommended for the quantity that is the independent variable in a mass spectrum Note 2: The proposed unit thomson (Th) is deprecated.</summary>
    MS_mass_to_charge_ratio = MS_m_z,

    /// <summary>Th: Three-character symbol m/z is used to denote the quantity formed by dividing the mass of an ion in unified atomic mass units by its charge number (regardless of sign). The symbol is written in italicized lower case letters with no spaces. Note 1: The term mass-to-charge-ratio is deprecated. Mass-to-charge ratio has been used for the abscissa of a mass spectrum, although the quantity measured is not the quotient of the ion's mass to its electric charge. The three-character symbol m/z is recommended for the quantity that is the independent variable in a mass spectrum Note 2: The proposed unit thomson (Th) is deprecated.</summary>
    MS_Th = MS_m_z,

    /// <summary>thomson: Three-character symbol m/z is used to denote the quantity formed by dividing the mass of an ion in unified atomic mass units by its charge number (regardless of sign). The symbol is written in italicized lower case letters with no spaces. Note 1: The term mass-to-charge-ratio is deprecated. Mass-to-charge ratio has been used for the abscissa of a mass spectrum, although the quantity measured is not the quotient of the ion's mass to its electric charge. The three-character symbol m/z is recommended for the quantity that is the independent variable in a mass spectrum Note 2: The proposed unit thomson (Th) is deprecated.</summary>
    MS_thomson = MS_m_z,

    /// <summary>charge state: The charge state of the ion, single or multiple and positive or negatively charged.</summary>
    MS_charge_state = 1000041,

    /// <summary>z: The charge state of the ion, single or multiple and positive or negatively charged.</summary>
    MS_z = MS_charge_state,

    /// <summary>peak intensity: Intensity of ions as measured by the height or area of a peak in a mass spectrum.</summary>
    MS_peak_intensity = 1000042,

    /// <summary>intensity unit: Intensity units are commonly arbitrary. Detected in counts per second (cps) when using counting detectors, but measured in volts when using analog detectors.</summary>
    MS_intensity_unit = 1000043,

    /// <summary>dissociation method: Fragmentation method used for dissociation or fragmentation.</summary>
    MS_dissociation_method = 1000044,

    /// <summary>collision energy: Energy for an ion experiencing collision with a stationary gas particle resulting in dissociation of the ion.</summary>
    MS_collision_energy = 1000045,

    /// <summary>energy unit: Energy units are represented in either eV or Joules.</summary>
    MS_energy_unit_OBSOLETE = 1000046,

    /// <summary>emulsion: State if the sample is in emulsion form.</summary>
    MS_emulsion = 1000047,

    /// <summary>gas: State if the sample is in gaseous form.</summary>
    MS_gas = 1000048,

    /// <summary>liquid: State if the sample is in liquid form.</summary>
    MS_liquid = 1000049,

    /// <summary>solid: State if the sample is in solid form.</summary>
    MS_solid = 1000050,

    /// <summary>solution: State if the sample is in solution form.</summary>
    MS_solution = 1000051,

    /// <summary>suspension: State if the sample is in suspension form.</summary>
    MS_suspension = 1000052,

    /// <summary>sample batch: Sample batch lot identifier.</summary>
    MS_sample_batch = 1000053,

    /// <summary>chromatography: Chromatographic conditions used to obtain the sample.</summary>
    MS_chromatography = 1000054,

    /// <summary>continuous flow fast atom bombardment: Fast atom bombardment ionization in which the analyte in solution is entrained in a flowing liquid matrix.</summary>
    MS_continuous_flow_fast_atom_bombardment = 1000055,

    /// <summary>CF-FAB: Fast atom bombardment ionization in which the analyte in solution is entrained in a flowing liquid matrix.</summary>
    MS_CF_FAB = MS_continuous_flow_fast_atom_bombardment,

    /// <summary>direct inlet: The sample is directly inserted into the ion source, usually on the end of a heatable probe.</summary>
    MS_direct_inlet = 1000056,

    /// <summary>electrospray inlet: Inlet used for introducing the liquid sample into an electrospray ionization source.</summary>
    MS_electrospray_inlet = 1000057,

    /// <summary>flow injection analysis: Sample is directly injected or infused into the ionization source.</summary>
    MS_flow_injection_analysis = 1000058,

    /// <summary>inductively coupled plasma: A gas discharge ion source in which the energy to the plasma is supplied by electromagnetic induction.</summary>
    MS_inductively_coupled_plasma = 1000059,

    /// <summary>infusion: The continuous flow of solution of a sample into the ionization source.</summary>
    MS_infusion = 1000060,

    /// <summary>jet separator: A device that separates carrier gas from gaseous analyte molecules on the basis of diffusivity.</summary>
    MS_jet_separator = 1000061,

    /// <summary>membrane separator: A device to separate carrier molecules from analyte molecules on the basis of ease of diffusion across a semipermeable membrane.</summary>
    MS_membrane_separator = 1000062,

    /// <summary>moving belt: Continuous moving surface in the form of a belt which passes through an ionsource carrying analyte molecules.</summary>
    MS_moving_belt = 1000063,

    /// <summary>moving wire: Continuous moving surface in the form of a wire which passes through an ionsource carrying analyte molecules.</summary>
    MS_moving_wire = 1000064,

    /// <summary>open split: A division of flowing stream of liquid into two streams.</summary>
    MS_open_split = 1000065,

    /// <summary>particle beam: Method for generating ions from a solution of an analyte.</summary>
    MS_particle_beam = 1000066,

    /// <summary>reservoir: A sample inlet method involving a reservoir.</summary>
    MS_reservoir = 1000067,

    /// <summary>septum: A disc composed of a flexible material that seals the entrance to the reservoir. Can also be enterance to the vaccum chamber.</summary>
    MS_septum = 1000068,

    /// <summary>thermospray inlet: A method for generating gas phase ions from a solution of an analyte by rapid heating of the sample.</summary>
    MS_thermospray_inlet = 1000069,

    /// <summary>atmospheric pressure chemical ionization: Chemical ionization that takes place at atmospheric pressure as opposed to the reduced pressure is normally used for chemical ionization.</summary>
    MS_atmospheric_pressure_chemical_ionization = 1000070,

    /// <summary>APCI: Chemical ionization that takes place at atmospheric pressure as opposed to the reduced pressure is normally used for chemical ionization.</summary>
    MS_APCI = MS_atmospheric_pressure_chemical_ionization,

    /// <summary>chemical ionization: The formation of a new ion by the reaction of a neutral species with an ion. The process may involve transfer of an electron, a proton or other charged species between the reactants. When a positive ion results from chemical ionization the term may be used without qualification. When a negative ion results the term negative ion chemical ionization should be used. Note that this term is not synonymous with chemi-ionization.</summary>
    MS_chemical_ionization = 1000071,

    /// <summary>CI: The formation of a new ion by the reaction of a neutral species with an ion. The process may involve transfer of an electron, a proton or other charged species between the reactants. When a positive ion results from chemical ionization the term may be used without qualification. When a negative ion results the term negative ion chemical ionization should be used. Note that this term is not synonymous with chemi-ionization.</summary>
    MS_CI = MS_chemical_ionization,

    /// <summary>Electronic Ionization: The ionization of an atom or molecule by electrons that are typically accelerated to energies between 50 and 150 eV. Usually 70 eV electrons are used to produce positive ions. The term 'electron impact' is not recommended. </summary>
    MS_Electronic_Ionization_OBSOLETE = 1000072,

    /// <summary>EI: The ionization of an atom or molecule by electrons that are typically accelerated to energies between 50 and 150 eV. Usually 70 eV electrons are used to produce positive ions. The term 'electron impact' is not recommended. </summary>
    MS_EI_OBSOLETE = MS_Electronic_Ionization_OBSOLETE,

    /// <summary>electrospray ionization: A process in which ionized species in the gas phase are produced from an analyte-containing solution via highly charged fine droplets, by means of spraying the solution from a narrow-bore needle tip at atmospheric pressure in the presence of a high electric field. When a pressurized gas is used to aid in the formation of a stable spray, the term pneumatically assisted electrospray ionization is used. The term ion spray is not recommended.</summary>
    MS_electrospray_ionization = 1000073,

    /// <summary>ESI: A process in which ionized species in the gas phase are produced from an analyte-containing solution via highly charged fine droplets, by means of spraying the solution from a narrow-bore needle tip at atmospheric pressure in the presence of a high electric field. When a pressurized gas is used to aid in the formation of a stable spray, the term pneumatically assisted electrospray ionization is used. The term ion spray is not recommended.</summary>
    MS_ESI = MS_electrospray_ionization,

    /// <summary>fast atom bombardment ionization: The ionization of any species by the interaction of a focused beam of neutral atoms having a translational energy of several thousand eV with a sample that is typically dissolved in a solvent matrix. See also secondary ionization.</summary>
    MS_fast_atom_bombardment_ionization = 1000074,

    /// <summary>FAB: The ionization of any species by the interaction of a focused beam of neutral atoms having a translational energy of several thousand eV with a sample that is typically dissolved in a solvent matrix. See also secondary ionization.</summary>
    MS_FAB = MS_fast_atom_bombardment_ionization,

    /// <summary>matrix-assisted laser desorption ionization: The formation of gas-phase ions from molecules that are present in a solid or solvent matrix that is irradiated with a pulsed laser. See also laser desorption/ionization.</summary>
    MS_matrix_assisted_laser_desorption_ionization = 1000075,

    /// <summary>MALDI: The formation of gas-phase ions from molecules that are present in a solid or solvent matrix that is irradiated with a pulsed laser. See also laser desorption/ionization.</summary>
    MS_MALDI = MS_matrix_assisted_laser_desorption_ionization,

    /// <summary>negative ion mode: OBSOLETE.</summary>
    MS_negative_ion_mode_OBSOLETE = 1000076,

    /// <summary>positive ion mode: OBSOLETE.</summary>
    MS_positive_ion_mode_OBSOLETE = 1000077,

    /// <summary>axial ejection linear ion trap: A linear ion trap mass spectrometer where ions are ejected along the axis of the analyzer.</summary>
    MS_axial_ejection_linear_ion_trap = 1000078,

    /// <summary>fourier transform ion cyclotron resonance mass spectrometer: A mass spectrometer based on the principle of ion cyclotron resonance in which an ion in a magnetic field moves in a circular orbit at a frequency characteristic of its m/z value. Ions are coherently excited to a larger radius orbit using a pulse of radio frequency energy and their image charge is detected on receiver plates as a time domain signal. Fourier transformation of the time domain signal results in a frequency domain signal which is converted to a mass spectrum based in the inverse relationship between frequency and m/z.</summary>
    MS_fourier_transform_ion_cyclotron_resonance_mass_spectrometer = 1000079,

    /// <summary>FT_ICR: A mass spectrometer based on the principle of ion cyclotron resonance in which an ion in a magnetic field moves in a circular orbit at a frequency characteristic of its m/z value. Ions are coherently excited to a larger radius orbit using a pulse of radio frequency energy and their image charge is detected on receiver plates as a time domain signal. Fourier transformation of the time domain signal results in a frequency domain signal which is converted to a mass spectrum based in the inverse relationship between frequency and m/z.</summary>
    MS_FT_ICR = MS_fourier_transform_ion_cyclotron_resonance_mass_spectrometer,

    /// <summary>magnetic sector: A device that produces a magnetic field perpendicular to a charged particle beam that deflects the beam to an extent that is proportional to the particle momentum per unit charge. For a monoenergetic beam, the deflection is proportional to m/z.</summary>
    MS_magnetic_sector = 1000080,

    /// <summary>quadrupole: A mass spectrometer that consists of four parallel rods whose centers form the corners of a square and whose opposing poles are connected. The voltage applied to the rods is a superposition of a static potential and a sinusoidal radio frequency potential. The motion of an ion in the x and y dimensions is described by the Matthieu equation whose solutions show that ions in a particular m/z range can be transmitted along the z axis.</summary>
    MS_quadrupole = 1000081,

    /// <summary>quadrupole ion trap: Quadrupole Ion Trap mass analyzer captures the ions in a three dimensional ion trap and then selectively ejects them by varying the RF and DC potentials.</summary>
    MS_quadrupole_ion_trap = 1000082,

    /// <summary>Paul Ion trap: Quadrupole Ion Trap mass analyzer captures the ions in a three dimensional ion trap and then selectively ejects them by varying the RF and DC potentials.</summary>
    MS_Paul_Ion_trap = MS_quadrupole_ion_trap,

    /// <summary>QIT: Quadrupole Ion Trap mass analyzer captures the ions in a three dimensional ion trap and then selectively ejects them by varying the RF and DC potentials.</summary>
    MS_QIT = MS_quadrupole_ion_trap,

    /// <summary>Quistor: Quadrupole Ion Trap mass analyzer captures the ions in a three dimensional ion trap and then selectively ejects them by varying the RF and DC potentials.</summary>
    MS_Quistor = MS_quadrupole_ion_trap,

    /// <summary>radial ejection linear ion trap: A linear ion trap mass spectrometer where ions are ejected along the radius of the analyzer.</summary>
    MS_radial_ejection_linear_ion_trap = 1000083,

    /// <summary>time-of-flight: Instrument that separates ions by m/z in a field-free region after acceleration to a fixed acceleration energy.</summary>
    MS_time_of_flight = 1000084,

    /// <summary>TOF: Instrument that separates ions by m/z in a field-free region after acceleration to a fixed acceleration energy.</summary>
    MS_TOF = MS_time_of_flight,

    /// <summary>baseline: An attribute of resolution when recording the detector response in absence of the analyte.</summary>
    MS_baseline = 1000085,

    /// <summary>full width at half-maximum: A measure of resolution represented as width of the peak at half peak height.</summary>
    MS_full_width_at_half_maximum = 1000086,

    /// <summary>FWHM: A measure of resolution represented as width of the peak at half peak height.</summary>
    MS_FWHM = MS_full_width_at_half_maximum,

    /// <summary>ten percent valley: An attribute of resolution when the ratio between adjacent signals is 10% of the signal height.</summary>
    MS_ten_percent_valley = 1000087,

    /// <summary>constant: When resolution is constant with respect to m/z.</summary>
    MS_constant = 1000088,

    /// <summary>proportional: When resolution is proportional with respect to m/z.</summary>
    MS_proportional = 1000089,

    /// <summary>mass scan: A variation of instrument where a selected mass is scanned.</summary>
    MS_mass_scan = 1000090,

    /// <summary>selected ion detection: Please see Single Ion Monitoring.</summary>
    MS_selected_ion_detection = 1000091,

    /// <summary>decreasing m/z scan: High to low direction in terms of m/z of the scan for scanning analyzers.</summary>
    MS_decreasing_m_z_scan = 1000092,

    /// <summary>increasing m/z scan: Low to high direction in terms of m/z of the scan for scanning analyzers.</summary>
    MS_increasing_m_z_scan = 1000093,

    /// <summary>exponential: The mass scan is done in exponential mode.</summary>
    MS_exponential = 1000094,

    /// <summary>linear: The mass scan is done in linear mode.</summary>
    MS_linear = 1000095,

    /// <summary>quadratic: The mass scan is done in quadratic mode.</summary>
    MS_quadratic = 1000096,

    /// <summary>constant neutral mass loss: A spectrum formed of all product ions that have been produced with a selected m/z decrement from any precursor ions. The spectrum shown correlates to the precursor ion spectrum. See also neutral loss spectrum.</summary>
    MS_constant_neutral_mass_loss = 1000097,

    /// <summary>multiple ion monitoring: Data acquired when monitoring the ion current of a few specific m/z values. Remap to MS:1000205 -Selected Ion Monitoring.</summary>
    MS_multiple_ion_monitoring = 1000098,

    /// <summary>multiple reaction monitoring: This term is not recommended. See Selected Reaction Monitoring.</summary>
    MS_multiple_reaction_monitoring_OBSOLETE = 1000099,

    /// <summary>MRM: This term is not recommended. See Selected Reaction Monitoring.</summary>
    MS_MRM_OBSOLETE = MS_multiple_reaction_monitoring_OBSOLETE,

    /// <summary>precursor ion scan: The specific scan function or process that will record a precursor ion spectrum.</summary>
    MS_precursor_ion_scan = 1000100,

    /// <summary>product ion scan: The specific scan function or process that records product ion spectrum.</summary>
    MS_product_ion_scan = 1000101,

    /// <summary>single ion monitoring: The operation of a mass spectrometer to monitor a single ion rather than scanning entire mass spectrum.</summary>
    MS_single_ion_monitoring_OBSOLETE = 1000102,

    /// <summary>single reaction monitoring: This term is not recommended. See Selected Reaction Monitoring.</summary>
    MS_single_reaction_monitoring_OBSOLETE = 1000103,

    /// <summary>None  ??: None.</summary>
    MS_None_____OBSOLETE = 1000104,

    /// <summary>reflectron off: Reflectron is off.</summary>
    MS_reflectron_off = 1000105,

    /// <summary>reflectron on: Reflectron is on.</summary>
    MS_reflectron_on = 1000106,

    /// <summary>channeltron: A horn-shaped (or cone-shaped) continuous dynode particle multiplier. The ion strikes the inner surface of the device and induces the production of secondary electrons that in turn impinge on the inner surfaces to produce more secondary electrons. This avalanche effect produces an increase in signal in the final measured current pulse.</summary>
    MS_channeltron = 1000107,

    /// <summary>Channeltron Detector: A horn-shaped (or cone-shaped) continuous dynode particle multiplier. The ion strikes the inner surface of the device and induces the production of secondary electrons that in turn impinge on the inner surfaces to produce more secondary electrons. This avalanche effect produces an increase in signal in the final measured current pulse.</summary>
    MS_Channeltron_Detector = MS_channeltron,

    /// <summary>conversion dynode electron multiplier: A surface that is held at high potential so that ions striking the surface produce electrons that are subsequently detected.</summary>
    MS_conversion_dynode_electron_multiplier = 1000108,

    /// <summary>conversion dynode photomultiplier: A detector in which ions strike a conversion dynode to produce electrons that in turn generate photons through a phosphorescent screen that are detected by a photomultiplier.</summary>
    MS_conversion_dynode_photomultiplier = 1000109,

    /// <summary>daly detector: Detector consisting of a conversion dynode, scintillator and photomultiplier. The metal knob at high potential emits secondary electrons when ions impinge on the surface. The secondary electrons are accelerated onto the scintillator that produces light that is then detected by the photomultiplier detector.</summary>
    MS_daly_detector = 1000110,

    /// <summary>Daly: Detector consisting of a conversion dynode, scintillator and photomultiplier. The metal knob at high potential emits secondary electrons when ions impinge on the surface. The secondary electrons are accelerated onto the scintillator that produces light that is then detected by the photomultiplier detector.</summary>
    MS_Daly = MS_daly_detector,

    /// <summary>electron multiplier tube: A device to amplify the current of a beam or packet of charged particles or photons by incidence upon the surface of an electrode to produce secondary electrons.</summary>
    MS_electron_multiplier_tube = 1000111,

    /// <summary>EMT: A device to amplify the current of a beam or packet of charged particles or photons by incidence upon the surface of an electrode to produce secondary electrons.</summary>
    MS_EMT = MS_electron_multiplier_tube,

    /// <summary>faraday cup: A conducting cup or chamber that intercepts a charged particle beam and is electrically connected to a current measuring device.</summary>
    MS_faraday_cup = 1000112,

    /// <summary>focal plane array: An array of detectors for spatially disperse ion beams in which all ions simultaneously impinge on the detector plane.</summary>
    MS_focal_plane_array = 1000113,

    /// <summary>microchannel plate detector: A thin plate that contains a closely spaced array of channels that each act as a continuous dynode particle multiplier. A charged particle, fast neutral particle, or photon striking the plate causes a cascade of secondary electrons that ultimately exits the opposite side of the plate.</summary>
    MS_microchannel_plate_detector = 1000114,

    /// <summary>multichannel plate: A thin plate that contains a closely spaced array of channels that each act as a continuous dynode particle multiplier. A charged particle, fast neutral particle, or photon striking the plate causes a cascade of secondary electrons that ultimately exits the opposite side of the plate.</summary>
    MS_multichannel_plate = MS_microchannel_plate_detector,

    /// <summary>multi-collector: A detector system commonly used in inductively coupled plasma mass spectrometers.</summary>
    MS_multi_collector = 1000115,

    /// <summary>photomultiplier: A detector for conversion of the ion/electron signal into photon(s) which are then amplified and detected.</summary>
    MS_photomultiplier = 1000116,

    /// <summary>PMT: A detector for conversion of the ion/electron signal into photon(s) which are then amplified and detected.</summary>
    MS_PMT = MS_photomultiplier,

    /// <summary>analog-digital converter: Analog-to-digital converter (abbreviated ADC, A/D or A to D) is an electronic integrated circuit (i/c) that converts continuous signals to discrete digital numbers.</summary>
    MS_analog_digital_converter = 1000117,

    /// <summary>ADC: Analog-to-digital converter (abbreviated ADC, A/D or A to D) is an electronic integrated circuit (i/c) that converts continuous signals to discrete digital numbers.</summary>
    MS_ADC = MS_analog_digital_converter,

    /// <summary>pulse counting: Definition to do.</summary>
    MS_pulse_counting = 1000118,

    /// <summary>time-digital converter: A device for converting a signal of sporadic pluses into a digital representation of their time indices.</summary>
    MS_time_digital_converter = 1000119,

    /// <summary>TDC: A device for converting a signal of sporadic pluses into a digital representation of their time indices.</summary>
    MS_TDC = MS_time_digital_converter,

    /// <summary>transient recorder: A detector acquisition mode used for detecting transient signals.</summary>
    MS_transient_recorder = 1000120,

    /// <summary>AB SCIEX instrument model: The brand of instruments from the joint venture between Applied Biosystems and MDS Analytical Technologies (formerly MDS SCIEX). Previously branded as \"Applied Biosystems|MDS SCIEX\".</summary>
    MS_AB_SCIEX_instrument_model = 1000121,

    /// <summary>Bruker Daltonics instrument model: Bruker Daltonics instrument model.</summary>
    MS_Bruker_Daltonics_instrument_model = 1000122,

    /// <summary>IonSpec instrument model: IonSpec corporation instrument model.</summary>
    MS_IonSpec_instrument_model = 1000123,

    /// <summary>Shimadzu instrument model: Shimadzu corporation instrument model.</summary>
    MS_Shimadzu_instrument_model = 1000124,

    /// <summary>Thermo Finnigan instrument model: ThermoFinnigan from Thermo Electron Corporation instrument model.</summary>
    MS_Thermo_Finnigan_instrument_model = 1000125,

    /// <summary>Waters instrument model: Waters Corporation instrument model.</summary>
    MS_Waters_instrument_model = 1000126,

    /// <summary>centroid spectrum: Processing of profile data to produce spectra that contains discrete peaks of zero width. Often used to reduce the size of dataset.</summary>
    MS_centroid_spectrum = 1000127,

    /// <summary>Discrete Mass Spectrum: Processing of profile data to produce spectra that contains discrete peaks of zero width. Often used to reduce the size of dataset.</summary>
    MS_Discrete_Mass_Spectrum = MS_centroid_spectrum,

    /// <summary>profile spectrum: A profile mass spectrum is created when data is recorded with ion current (counts per second) on one axis and mass/charge ratio on another axis.</summary>
    MS_profile_spectrum = 1000128,

    /// <summary>continuous mass spectrum: A profile mass spectrum is created when data is recorded with ion current (counts per second) on one axis and mass/charge ratio on another axis.</summary>
    MS_continuous_mass_spectrum = MS_profile_spectrum,

    /// <summary>Continuum Mass Spectrum: A profile mass spectrum is created when data is recorded with ion current (counts per second) on one axis and mass/charge ratio on another axis.</summary>
    MS_Continuum_Mass_Spectrum = MS_profile_spectrum,

    /// <summary>negative scan: Polarity of the scan is negative.</summary>
    MS_negative_scan = 1000129,

    /// <summary>positive scan: Polarity of the scan is positive.</summary>
    MS_positive_scan = 1000130,

    /// <summary>number of counts: The number of counted events observed in one or a group of elements of a detector.</summary>
    MS_number_of_counts = 1000131,

    /// <summary>percent of base peak: The magnitude of a peak or measurement element expressed in terms of the percentage of the magnitude of the base peak intensity.</summary>
    MS_percent_of_base_peak = 1000132,

    /// <summary>collision-induced dissociation: The dissociation of an ion after collisional excitation. The term collisional-activated dissociation is not recommended.</summary>
    MS_collision_induced_dissociation = 1000133,

    /// <summary>CID: The dissociation of an ion after collisional excitation. The term collisional-activated dissociation is not recommended.</summary>
    MS_CID = MS_collision_induced_dissociation,

    /// <summary>CAD: The dissociation of an ion after collisional excitation. The term collisional-activated dissociation is not recommended.</summary>
    MS_CAD = MS_collision_induced_dissociation,

    /// <summary>collision activated dissociation: The dissociation of an ion after collisional excitation. The term collisional-activated dissociation is not recommended.</summary>
    MS_collision_activated_dissociation = MS_collision_induced_dissociation,

    /// <summary>plasma desorption: The ionization of material in a solid sample by bombarding it with ionic or neutral atoms formed as a result of the fission of a suitable nuclide, typically 252Cf. Synonymous with fission fragment ionization.</summary>
    MS_plasma_desorption = 1000134,

    /// <summary>PD: The ionization of material in a solid sample by bombarding it with ionic or neutral atoms formed as a result of the fission of a suitable nuclide, typically 252Cf. Synonymous with fission fragment ionization.</summary>
    MS_PD = MS_plasma_desorption,

    /// <summary>post-source decay: A technique specific to reflectron time-of-flight mass spectrometers where product ions of metastable transitions or collision-induced dissociations generated in the drift tube prior to entering the reflectron are m/z separated to yield product ion spectra.</summary>
    MS_post_source_decay = 1000135,

    /// <summary>PSD: A technique specific to reflectron time-of-flight mass spectrometers where product ions of metastable transitions or collision-induced dissociations generated in the drift tube prior to entering the reflectron are m/z separated to yield product ion spectra.</summary>
    MS_PSD = MS_post_source_decay,

    /// <summary>surface-induced dissociation: Fragmentation that results from the collision of an ion with a surface.</summary>
    MS_surface_induced_dissociation = 1000136,

    /// <summary>SID: Fragmentation that results from the collision of an ion with a surface.</summary>
    MS_SID = MS_surface_induced_dissociation,

    /// <summary>electron volt: A non-SI unit of energy (eV) defined as the energy acquired by a particle containing one unit of charge through a potential difference of one volt. An electron-volt is equal to 1.602 176 53(14) x 10^-19 J.</summary>
    MS_electron_volt_OBSOLETE = 1000137,

    /// <summary>eV: A non-SI unit of energy (eV) defined as the energy acquired by a particle containing one unit of charge through a potential difference of one volt. An electron-volt is equal to 1.602 176 53(14) x 10^-19 J.</summary>
    MS_eV_OBSOLETE = MS_electron_volt_OBSOLETE,

    /// <summary>percent collision energy: Collision energy required to fragment an ion represented as a percent value.</summary>
    MS_percent_collision_energy = 1000138,

    /// <summary>4000 QTRAP: Applied Biosystems/MDS SCIEX Q 4000 TRAP MS.</summary>
    MS_4000_QTRAP = 1000139,

    /// <summary>4700 Proteomics Analyzer: Applied Biosystems/MDS SCIEX 4700 Proteomics Analyzer MS.</summary>
    MS_4700_Proteomics_Analyzer = 1000140,

    /// <summary>apex IV: Bruker Daltonics' apex IV: ESI, MALDI, Nanospray, APCI, APPI, Qh-FT_ICR.</summary>
    MS_apex_IV = 1000141,

    /// <summary>apex Q: Bruker Daltonics' apex Q: ESI, MALDI, Nanospray, APCI, APPI, Qh-FT_ICR.</summary>
    MS_apex_Q = 1000142,

    /// <summary>API 150EX: Applied Biosystems/MDS SCIEX API 150EX MS.</summary>
    MS_API_150EX = 1000143,

    /// <summary>API 150EX Prep: Applied Biosystems/MDS SCIEX API 150EX Prep MS.</summary>
    MS_API_150EX_Prep = 1000144,

    /// <summary>API 2000: Applied Biosystems/MDS SCIEX API 2000 MS.</summary>
    MS_API_2000 = 1000145,

    /// <summary>API 3000: Applied Biosystems/MDS SCIEX API 3000 MS.</summary>
    MS_API_3000 = 1000146,

    /// <summary>API 4000: Applied Biosystems/MDS SCIEX API 4000 MS.</summary>
    MS_API_4000 = 1000147,

    /// <summary>autoflex II: Bruker Daltonics' autoflex II: MALDI TOF.</summary>
    MS_autoflex_II = 1000148,

    /// <summary>autoflex TOF/TOF: Bruker Daltonics' autoflex TOF/TOF MS: MALDI TOF.</summary>
    MS_autoflex_TOF_TOF = 1000149,

    /// <summary>Auto Spec Ultima NT: Waters AutoSpec Ultima NT MS.</summary>
    MS_Auto_Spec_Ultima_NT = 1000150,

    /// <summary>BioTOF II: Bruker Daltonics' BioTOF II: ESI TOF.</summary>
    MS_BioTOF_II = 1000151,

    /// <summary>BioTOF-Q: Bruker Daltonics' BioTOF-Q: ESI Q-TOF.</summary>
    MS_BioTOF_Q = 1000152,

    /// <summary>DELTA plusAdvantage: ThermoFinnigan DELTA plusAdvantage MS.</summary>
    MS_DELTA_plusAdvantage = 1000153,

    /// <summary>DELTAplusXP: ThermoFinnigan DELTAplusXP MS.</summary>
    MS_DELTAplusXP = 1000154,

    /// <summary>ELEMENT2: ThermoFinnigan ELEMENT2 MS.</summary>
    MS_ELEMENT2 = 1000155,

    /// <summary>esquire 4000: Bruker Daltonics' esquire 4000: linear ion trap, ESI, MALDI, Nanospray, APCI, APPI.</summary>
    MS_esquire_4000 = 1000156,

    /// <summary>esquire 6000: Bruker Daltonics' esquire 6000: linear ion trap, ESI, MALDI, Nanospray, APCI, APPI.</summary>
    MS_esquire_6000 = 1000157,

    /// <summary>explorer: IonSpec Explorer MS.</summary>
    MS_explorer = 1000158,

    /// <summary>GCT: Waters GCT MS.</summary>
    MS_GCT = 1000159,

    /// <summary>HCT: Bruker Daltonics' HCT: ESI Q-TOF, Nanospray, APCI, APPI.</summary>
    MS_HCT = 1000160,

    /// <summary>HCTplus: Bruker Daltonics' HCTplus: ESI Q-TOF, Nanospray, APCI, APPI.</summary>
    MS_HCTplus = 1000161,

    /// <summary>HiRes ESI: IonSpec HiResESI MS.</summary>
    MS_HiRes_ESI = 1000162,

    /// <summary>HiRes MALDI: IonSpec HiResMALDI MS.</summary>
    MS_HiRes_MALDI = 1000163,

    /// <summary>IsoPrime: Waters IsoPrime MS.</summary>
    MS_IsoPrime = 1000164,

    /// <summary>IsoProbe: Waters IsoProbe MS.</summary>
    MS_IsoProbe = 1000165,

    /// <summary>IsoProbe T: Waters IsoProbe T MS.</summary>
    MS_IsoProbe_T = 1000166,

    /// <summary>LCQ Advantage: ThermoFinnigan LCQ Advantage MS.</summary>
    MS_LCQ_Advantage = 1000167,

    /// <summary>LCQ Classic: ThermoFinnigan LCQ Classic MS.</summary>
    MS_LCQ_Classic = 1000168,

    /// <summary>LCQ Deca XP Plus: ThermoFinnigan LCQ Deca XP Plus MS.</summary>
    MS_LCQ_Deca_XP_Plus = 1000169,

    /// <summary>M@LDI L: Waters MALDI L MS.</summary>
    MS_M_LDI_L = 1000170,

    /// <summary>M@LDI LR: Waters MALDI LR MS.</summary>
    MS_M_LDI_LR = 1000171,

    /// <summary>MAT253: ThermoFinnigan MAT253 MS.</summary>
    MS_MAT253 = 1000172,

    /// <summary>MAT900XP: ThermoFinnigan MAT900XP MS.</summary>
    MS_MAT900XP = 1000173,

    /// <summary>MAT900XP Trap: ThermoFinnigan MAT900XP Trap MS.</summary>
    MS_MAT900XP_Trap = 1000174,

    /// <summary>MAT95XP: ThermoFinnigan MAT95XP MS.</summary>
    MS_MAT95XP = 1000175,

    /// <summary>MAT95XP Trap: ThermoFinnigan MAT95XP Trap MS.</summary>
    MS_MAT95XP_Trap = 1000176,

    /// <summary>microflex: Bruker Daltonics' microflex: MALDI TOF.</summary>
    MS_microflex = 1000177,

    /// <summary>microTOF LC: Bruker Daltonics' microTOF LC: ESI TOF, Nanospray, APCI, APPI.</summary>
    MS_microTOF_LC = 1000178,

    /// <summary>neptune: ThermoFinnigan NEPTUNE MS.</summary>
    MS_neptune = 1000179,

    /// <summary>NG-5400: Waters NG-5400 MS.</summary>
    MS_NG_5400 = 1000180,

    /// <summary>OMEGA: IonSpec OMEGA MS.</summary>
    MS_OMEGA = 1000181,

    /// <summary>OMEGA-2001: IonSpec OMEGA-2001 MS.</summary>
    MS_OMEGA_2001 = 1000182,

    /// <summary>OmniFlex: Bruker Daltonics' OmniFlex: MALDI TOF.</summary>
    MS_OmniFlex = 1000183,

    /// <summary>Platform ICP: Waters Platform ICP MS.</summary>
    MS_Platform_ICP = 1000184,

    /// <summary>PolarisQ: ThermoFinnigan PolarisQ MS.</summary>
    MS_PolarisQ = 1000185,

    /// <summary>proteomics solution 1: Applied Biosystems/MDS SCIEX Proteomics Solution 1 MS.</summary>
    MS_proteomics_solution_1 = 1000186,

    /// <summary>Q TRAP: Applied Biosystems/MDS SCIEX Q TRAP MS.</summary>
    MS_Q_TRAP = 1000187,

    /// <summary>Q-Tof micro: Waters Q-Tof micro MS.</summary>
    MS_Q_Tof_micro = 1000188,

    /// <summary>Q-Tof ultima: Waters Q-Tof Ultima MS.</summary>
    MS_Q_Tof_ultima = 1000189,

    /// <summary>QSTAR: Applied Biosystems/MDS SCIEX QSTAR MS.</summary>
    MS_QSTAR = 1000190,

    /// <summary>quattro micro: Waters Quattro micro MS.</summary>
    MS_quattro_micro = 1000191,

    /// <summary>Quattro UItima: Waters Quattro Uitima MS.</summary>
    MS_Quattro_UItima = 1000192,

    /// <summary>Surveyor MSQ: ThermoFinnigan Surveyor MSQ MS.</summary>
    MS_Surveyor_MSQ = 1000193,

    /// <summary>SymBiot I: Applied Biosystems/MDS SCIEX SymBiot I MS.</summary>
    MS_SymBiot_I = 1000194,

    /// <summary>SymBiot XVI: Applied Biosystems/MDS SCIEX SymBiot XVI MS.</summary>
    MS_SymBiot_XVI = 1000195,

    /// <summary>TEMPUS TOF: ThermoFinnigan TEMPUS TOF MS.</summary>
    MS_TEMPUS_TOF = 1000196,

    /// <summary>TRACE DSQ: ThermoFinnigan TRACE DSQ MS.</summary>
    MS_TRACE_DSQ = 1000197,

    /// <summary>TRITON: ThermoFinnigan TRITON MS.</summary>
    MS_TRITON = 1000198,

    /// <summary>TSQ Quantum: ThermoFinnigan TSQ Quantum MS.</summary>
    MS_TSQ_Quantum = 1000199,

    /// <summary>ultima: IonSpec Ultima MS.</summary>
    MS_ultima = 1000200,

    /// <summary>ultraflex: Bruker Daltonics' ultraflex: MALDI TOF.</summary>
    MS_ultraflex = 1000201,

    /// <summary>ultraflex TOF/TOF: Bruker Daltonics' ultraflex TOF/TOF: MALDI TOF.</summary>
    MS_ultraflex_TOF_TOF = 1000202,

    /// <summary>Voyager-DE PRO: Applied Biosystems/MDS SCIEX Voyager-DE PRO MS.</summary>
    MS_Voyager_DE_PRO = 1000203,

    /// <summary>Voyager-DE STR: Applied Biosystems/MDS SCIEX Voyager-DE STR MS.</summary>
    MS_Voyager_DE_STR = 1000204,

    /// <summary>selected ion monitoring: The operation of a mass spectrometer in which the intensities of several specific m/z values are recorded rather than the entire mass spectrum.</summary>
    MS_selected_ion_monitoring = 1000205,

    /// <summary>Multiple Ion Monitoring: The operation of a mass spectrometer in which the intensities of several specific m/z values are recorded rather than the entire mass spectrum.</summary>
    MS_Multiple_Ion_Monitoring = MS_selected_ion_monitoring,

    /// <summary>SIM: The operation of a mass spectrometer in which the intensities of several specific m/z values are recorded rather than the entire mass spectrum.</summary>
    MS_SIM = MS_selected_ion_monitoring,

    /// <summary>selected reaction monitoring: Data acquired from specific product ions corresponding to m/z selected precursor ions recorded via multiple stages of mass spectrometry. Selected reaction monitoring can be performed in time or in space.</summary>
    MS_selected_reaction_monitoring = 1000206,

    /// <summary>SRM: Data acquired from specific product ions corresponding to m/z selected precursor ions recorded via multiple stages of mass spectrometry. Selected reaction monitoring can be performed in time or in space.</summary>
    MS_SRM = MS_selected_reaction_monitoring,

    /// <summary>accurate mass: An experimentally determined mass that is can be to determine a unique elemental formula. For ions less than 200 u, a measurement with 5 ppm accuracy is sufficient to determine the elemental composition.</summary>
    MS_accurate_mass = 1000207,

    /// <summary>average mass: The mass of an ion or molecule calculated using the average mass of each element weighted for its natural isotopic abundance.</summary>
    MS_average_mass = 1000208,

    /// <summary>appearance energy: The minimum energy that must be imparted to an atom or molecule to produce a specified ion. The term appearance potential is not recommended.</summary>
    MS_appearance_energy = 1000209,

    /// <summary>AE: The minimum energy that must be imparted to an atom or molecule to produce a specified ion. The term appearance potential is not recommended.</summary>
    MS_AE = MS_appearance_energy,

    /// <summary>base peak: The peak in a mass spectrum that has the greatest intensity. This term may be applied to the spectra of pure substances or mixtures.</summary>
    MS_base_peak = 1000210,

    /// <summary>BP: The peak in a mass spectrum that has the greatest intensity. This term may be applied to the spectra of pure substances or mixtures.</summary>
    MS_BP = MS_base_peak,

    /// <summary>OBSOLETE charge number: OBSOLETE. The total charge on an ion divided by the electron charge e. OBSOLETED 2009-10-27 since this was viewed as a duplication of 00041 charge state.</summary>
    MS_OBSOLETE_charge_number_OBSOLETE = 1000211,

    /// <summary>z: OBSOLETE. The total charge on an ion divided by the electron charge e. OBSOLETED 2009-10-27 since this was viewed as a duplication of 00041 charge state.</summary>
    MS_z_OBSOLETE = MS_OBSOLETE_charge_number_OBSOLETE,

    /// <summary>dalton: A non-SI unit of mass (symbol Da) that is equal to the unified atomic mass unit: 1.660 538 86(28) x 10^-27 kg.</summary>
    MS_dalton_OBSOLETE = 1000212,

    /// <summary>Da: A non-SI unit of mass (symbol Da) that is equal to the unified atomic mass unit: 1.660 538 86(28) x 10^-27 kg.</summary>
    MS_Da_OBSOLETE = MS_dalton_OBSOLETE,

    /// <summary>electron affinity: The electron affinity of M is the minimum energy required for the process M- ? M + e where M- and M are in their ground rotational, vibrational and electronic states and the electron has zero kinetic energy.</summary>
    MS_electron_affinity = 1000213,

    /// <summary>EA: The electron affinity of M is the minimum energy required for the process M- ? M + e where M- and M are in their ground rotational, vibrational and electronic states and the electron has zero kinetic energy.</summary>
    MS_EA = MS_electron_affinity,

    /// <summary>electron energy obsolete: The potential difference through which electrons are accelerated before they are used to bring about electron ionization.</summary>
    MS_electron_energy_obsolete = 1000214,

    /// <summary>exact mass: The calculated mass of an ion or molecule containing a single isotope of each atom.</summary>
    MS_exact_mass = 1000215,

    /// <summary>field-free region: A section of a mass spectrometer in which there are no electric or magnetic fields.</summary>
    MS_field_free_region = 1000216,

    /// <summary>FFR: A section of a mass spectrometer in which there are no electric or magnetic fields.</summary>
    MS_FFR = MS_field_free_region,

    /// <summary>ionization cross section: A measure of the probability that a given ionization process will occur when an atom or molecule interacts with a photon, electron, atom or molecule.</summary>
    MS_ionization_cross_section = 1000217,

    /// <summary>ionization efficiency: The ratio of the number of ions formed to the number of electrons, molecules or photons used.</summary>
    MS_ionization_efficiency_OBSOLETE = 1000218,

    /// <summary>ionization energy: The minimum energy required to remove an electron from an atom or molecule to produce a positive ion.</summary>
    MS_ionization_energy = 1000219,

    /// <summary>IE: The minimum energy required to remove an electron from an atom or molecule to produce a positive ion.</summary>
    MS_IE = MS_ionization_energy,

    /// <summary>isotope dilution mass spectrometry: A quantitative mass spectrometry technique in which an isotopically enriched compound is used as an internal standard.</summary>
    MS_isotope_dilution_mass_spectrometry = 1000220,

    /// <summary>IDMS: A quantitative mass spectrometry technique in which an isotopically enriched compound is used as an internal standard.</summary>
    MS_IDMS = MS_isotope_dilution_mass_spectrometry,

    /// <summary>magnetic deflection: The deflection of charged particles in a magnetic field due to a force equal to qv B where q is the particle charge, v its velocity and B the magnetic field. Magnetic deflection of an ion beam is used for m/z separation in a magnetic sector mass spectrometer.</summary>
    MS_magnetic_deflection = 1000221,

    /// <summary>mass defect: The difference between the monoisotipic and nominal mass of a molecule or atom.</summary>
    MS_mass_defect = 1000222,

    /// <summary>mass number: The sum of the protons and neutrons in an atom, molecule or ion.</summary>
    MS_mass_number = 1000223,

    /// <summary>molecular mass: The mass of one mole of a molecular substance (6.022 1415(10) x 10^23 molecules).</summary>
    MS_molecular_mass = 1000224,

    /// <summary>monoisotopic mass: The mass of an ion or molecule calculated using the mass of the most abundant isotope of each element.</summary>
    MS_monoisotopic_mass = 1000225,

    /// <summary>molecular beam mass spectrometry: A mass spectrometry technique in which the sample is introduced into the mass spectrometer as a molecular beam.</summary>
    MS_molecular_beam_mass_spectrometry = 1000226,

    /// <summary>MBMS: A mass spectrometry technique in which the sample is introduced into the mass spectrometer as a molecular beam.</summary>
    MS_MBMS = MS_molecular_beam_mass_spectrometry,

    /// <summary>multiphoton ionization: Photoionization of an atom or molecule in which in two or more photons are absorbed.</summary>
    MS_multiphoton_ionization = 1000227,

    /// <summary>MPI: Photoionization of an atom or molecule in which in two or more photons are absorbed.</summary>
    MS_MPI = MS_multiphoton_ionization,

    /// <summary>nitrogen rule: An organic molecule containing the elements C, H, O, S, P, or halogen has an odd nominal mass if it contains an odd number of nitrogen atoms.</summary>
    MS_nitrogen_rule = 1000228,

    /// <summary>nominal mass: The mass of an ion or molecule calculated using the mass of the most abundant isotope of each element rounded to the nearest integer value.</summary>
    MS_nominal_mass = 1000229,

    /// <summary>odd-electron rule: Odd-electron ions may dissociate to form either odd or even-electron ions, whereas even-electron ions generally form even-electron fragment ions.</summary>
    MS_odd_electron_rule = 1000230,

    /// <summary>peak: A localized region of relatively large ion signal in a mass spectrum. Although peaks are often associated with particular ions, the terms peak and ion should not be used interchangeably.</summary>
    MS_peak = 1000231,

    /// <summary>peak intensity: The height or area of a peak in a mass spectrum.</summary>
    MS_peak_intensity_OBSOLETE = 1000232,

    /// <summary>proton affinity: The proton affinity of a species M is defined as the negative of the enthalpy change for the reaction M + H+ ->[M+H]+, where all species are in their ground rotational, vibrational and electronic states.</summary>
    MS_proton_affinity = 1000233,

    /// <summary>PA: The proton affinity of a species M is defined as the negative of the enthalpy change for the reaction M + H+ ->[M+H]+, where all species are in their ground rotational, vibrational and electronic states.</summary>
    MS_PA = MS_proton_affinity,

    /// <summary>mass resolving power: In a mass spectrum, the observed mass divided by the difference between two masses that can be separated. The method by which delta m was obtained and the mass at which the measurement was made should be reported.</summary>
    MS_mass_resolving_power_OBSOLETE = 1000234,

    /// <summary>total ion current chromatogram: Chromatogram obtained by plotting the total ion current detected in each of a series of mass spectra recorded as a function of retention time.</summary>
    MS_total_ion_current_chromatogram = 1000235,

    /// <summary>TIC chromatogram: Chromatogram obtained by plotting the total ion current detected in each of a series of mass spectra recorded as a function of retention time.</summary>
    MS_TIC_chromatogram = MS_total_ion_current_chromatogram,

    /// <summary>transmission: The ratio of the number of ions leaving a region of a mass spectrometer to the number entering that region.</summary>
    MS_transmission = 1000236,

    /// <summary>unified atomic mass unit: A non-SI unit of mass (u) defined as one twelfth of ^12 C in its ground state and equal to 1.660 538 86(28) x 10^-27 kg.</summary>
    MS_unified_atomic_mass_unit_OBSOLETE = 1000237,

    /// <summary>u: A non-SI unit of mass (u) defined as one twelfth of ^12 C in its ground state and equal to 1.660 538 86(28) x 10^-27 kg.</summary>
    MS_u_OBSOLETE = MS_unified_atomic_mass_unit_OBSOLETE,

    /// <summary>accelerator mass spectrometry: A mass spectrometry technique in which atoms extracted from a sample are ionized, accelerated to MeV energies and separated according to their momentum, charge and energy.</summary>
    MS_accelerator_mass_spectrometry = 1000238,

    /// <summary>AMS: A mass spectrometry technique in which atoms extracted from a sample are ionized, accelerated to MeV energies and separated according to their momentum, charge and energy.</summary>
    MS_AMS = MS_accelerator_mass_spectrometry,

    /// <summary>atmospheric pressure matrix-assisted laser desorption ionization: Matrix-assisted laser desorption ionization in which the sample target is at atmospheric pressure and the ions formed by the pulsed laser are sampled through a small aperture into the mass spectrometer.</summary>
    MS_atmospheric_pressure_matrix_assisted_laser_desorption_ionization = 1000239,

    /// <summary>AP MALDI: Matrix-assisted laser desorption ionization in which the sample target is at atmospheric pressure and the ions formed by the pulsed laser are sampled through a small aperture into the mass spectrometer.</summary>
    MS_AP_MALDI = MS_atmospheric_pressure_matrix_assisted_laser_desorption_ionization,

    /// <summary>atmospheric pressure ionization: Any ionization process in which ions are formed in the gas phase at atmospheric pressure.</summary>
    MS_atmospheric_pressure_ionization = 1000240,

    /// <summary>API: Any ionization process in which ions are formed in the gas phase at atmospheric pressure.</summary>
    MS_API = MS_atmospheric_pressure_ionization,

    /// <summary>Atmostpheric Pressure Photoionization: Atmospheric pressure chemical ionization in which the reactant ions are generated by photo-ionization.</summary>
    MS_Atmostpheric_Pressure_Photoionization_OBSOLETE = 1000241,

    /// <summary>APPI: Atmospheric pressure chemical ionization in which the reactant ions are generated by photo-ionization.</summary>
    MS_APPI_OBSOLETE = MS_Atmostpheric_Pressure_Photoionization_OBSOLETE,

    /// <summary>blackbody infrared radiative dissociation: A special case of infrared multiphoton dissociation wherein excitation of the reactant ion is caused by absorption of infrared photons radiating from heated blackbody surroundings, which are usually the walls of a vacuum chamber. See also infrared multiphoton dissociation.</summary>
    MS_blackbody_infrared_radiative_dissociation = 1000242,

    /// <summary>BIRD: A special case of infrared multiphoton dissociation wherein excitation of the reactant ion is caused by absorption of infrared photons radiating from heated blackbody surroundings, which are usually the walls of a vacuum chamber. See also infrared multiphoton dissociation.</summary>
    MS_BIRD = MS_blackbody_infrared_radiative_dissociation,

    /// <summary>charge-remote fragmentation: A fragmentation of an even-electron ion in which the cleaved bond is not adjacent to the apparent charge site.</summary>
    MS_charge_remote_fragmentation = 1000243,

    /// <summary>CRF: A fragmentation of an even-electron ion in which the cleaved bond is not adjacent to the apparent charge site.</summary>
    MS_CRF = MS_charge_remote_fragmentation,

    /// <summary>consecutive reaction monitoring: MSn experiment with three or more stages of m/z separation and in which a particular multi-step reaction path is monitored.</summary>
    MS_consecutive_reaction_monitoring = 1000244,

    /// <summary>CRM: MSn experiment with three or more stages of m/z separation and in which a particular multi-step reaction path is monitored.</summary>
    MS_CRM = MS_consecutive_reaction_monitoring,

    /// <summary>charge stripping: The reaction of a positive ion with an atom or molecule that results in the removal of one or more electrons from the ion.</summary>
    MS_charge_stripping = 1000245,

    /// <summary>CS: The reaction of a positive ion with an atom or molecule that results in the removal of one or more electrons from the ion.</summary>
    MS_CS = MS_charge_stripping,

    /// <summary>delayed extraction: The application of the accelerating voltage pulse after a time delay in desorption ionization from a surface. The extraction delay can produce energy focusing in a time-of-flight mass spectrometer.</summary>
    MS_delayed_extraction = 1000246,

    /// <summary>DE: The application of the accelerating voltage pulse after a time delay in desorption ionization from a surface. The extraction delay can produce energy focusing in a time-of-flight mass spectrometer.</summary>
    MS_DE = MS_delayed_extraction,

    /// <summary>desorption ionization: The formation of ions from a solid or liquid material after the rapid vaporization of that sample.</summary>
    MS_desorption_ionization = 1000247,

    /// <summary>DI: The formation of ions from a solid or liquid material after the rapid vaporization of that sample.</summary>
    MS_DI = MS_desorption_ionization,

    /// <summary>direct insertion probe: A device for introducing a solid or liquid sample into a mass spectrometer ion source for desorption ionization.</summary>
    MS_direct_insertion_probe = 1000248,

    /// <summary>DIP: A device for introducing a solid or liquid sample into a mass spectrometer ion source for desorption ionization.</summary>
    MS_DIP = MS_direct_insertion_probe,

    /// <summary>direct liquid introduction: The delivery of a liquid sample into a mass spectrometer for spray or desorption ionization.</summary>
    MS_direct_liquid_introduction = 1000249,

    /// <summary>DLI: The delivery of a liquid sample into a mass spectrometer for spray or desorption ionization.</summary>
    MS_DLI = MS_direct_liquid_introduction,

    /// <summary>electron capture dissociation: A process in which a multiply protonated molecules interacts with a low energy electrons. Capture of the electron leads the liberation of energy and a reduction in charge state of the ion with the production of the (M + nH) (n-1)+ odd electron ion, which readily fragments.</summary>
    MS_electron_capture_dissociation = 1000250,

    /// <summary>ECD: A process in which a multiply protonated molecules interacts with a low energy electrons. Capture of the electron leads the liberation of energy and a reduction in charge state of the ion with the production of the (M + nH) (n-1)+ odd electron ion, which readily fragments.</summary>
    MS_ECD = MS_electron_capture_dissociation,

    /// <summary>even-electron ion: An ion containing no unpaired electrons in its ground electronic state, e.g. CH3+ in its ground state.</summary>
    MS_even_electron_ion = 1000251,

    /// <summary>EE: An ion containing no unpaired electrons in its ground electronic state, e.g. CH3+ in its ground state.</summary>
    MS_EE = MS_even_electron_ion,

    /// <summary>electron-induced excitation in organics: The reaction of an ion with an electron in which the translational energy of the collision is converted into internal energy of the ion.</summary>
    MS_electron_induced_excitation_in_organics = 1000252,

    /// <summary>EIEIO: The reaction of an ion with an electron in which the translational energy of the collision is converted into internal energy of the ion.</summary>
    MS_EIEIO = MS_electron_induced_excitation_in_organics,

    /// <summary>electron multiplier: A device to amplify the current of a beam or packet of charged particles or photons by incidence upon the surface of an electrode to produce secondary electrons. The secondary electrons are then accelerated to other electrodes or parts of a continuous electrode to produce further secondary electrons.</summary>
    MS_electron_multiplier = 1000253,

    /// <summary>EM: A device to amplify the current of a beam or packet of charged particles or photons by incidence upon the surface of an electrode to produce secondary electrons. The secondary electrons are then accelerated to other electrodes or parts of a continuous electrode to produce further secondary electrons.</summary>
    MS_EM = MS_electron_multiplier,

    /// <summary>electrostatic energy analyzer: A device consisting of conducting parallel plates, concentric cylinders or concentric spheres that separates charged particles according to their kinetic energy by means of an electric field that is constant in time.</summary>
    MS_electrostatic_energy_analyzer = 1000254,

    /// <summary>ESA: A device consisting of conducting parallel plates, concentric cylinders or concentric spheres that separates charged particles according to their kinetic energy by means of an electric field that is constant in time.</summary>
    MS_ESA = MS_electrostatic_energy_analyzer,

    /// <summary>flowing afterglow: An ion source immersed in a flow of helium or other inert buffer gas that carries the ions through a meter-long reactor at pressures around 100 Pa.</summary>
    MS_flowing_afterglow = 1000255,

    /// <summary>FA: An ion source immersed in a flow of helium or other inert buffer gas that carries the ions through a meter-long reactor at pressures around 100 Pa.</summary>
    MS_FA = MS_flowing_afterglow,

    /// <summary>high-field asymmetric waveform ion mobility spectrometry: The separation of ions between two concentric cylindrical electrodes due to application of a high voltage asymmetric waveform whereby ions migrate towards one of the two electrodes depending on the ratio of the high- to low-field mobility of the ion.</summary>
    MS_high_field_asymmetric_waveform_ion_mobility_spectrometry = 1000256,

    /// <summary>FAIMS: The separation of ions between two concentric cylindrical electrodes due to application of a high voltage asymmetric waveform whereby ions migrate towards one of the two electrodes depending on the ratio of the high- to low-field mobility of the ion.</summary>
    MS_FAIMS = MS_high_field_asymmetric_waveform_ion_mobility_spectrometry,

    /// <summary>field desorption: The formation of gas-phase ions from a material deposited on a solid surface in the presence of a high electric field. Because this process may encompass ionization by field ionization or other mechanisms, it is not recommended as a synonym for field desorption ionization.</summary>
    MS_field_desorption = 1000257,

    /// <summary>FD: The formation of gas-phase ions from a material deposited on a solid surface in the presence of a high electric field. Because this process may encompass ionization by field ionization or other mechanisms, it is not recommended as a synonym for field desorption ionization.</summary>
    MS_FD = MS_field_desorption,

    /// <summary>field ionization: The removal of electrons from any species by interaction with a high electric field.</summary>
    MS_field_ionization = 1000258,

    /// <summary>FI: The removal of electrons from any species by interaction with a high electric field.</summary>
    MS_FI = MS_field_ionization,

    /// <summary>glow discharge ionization: The formation of ions in the gas phase and from solid samples at the cathode by application of a voltage to a low pressure gas.</summary>
    MS_glow_discharge_ionization = 1000259,

    /// <summary>GD-MS: The formation of ions in the gas phase and from solid samples at the cathode by application of a voltage to a low pressure gas.</summary>
    MS_GD_MS = MS_glow_discharge_ionization,

    /// <summary>ion kinetic energy spectrometry: A method of analysis in which a beam of ions is separated according to the ratio of its translational energy to charge.</summary>
    MS_ion_kinetic_energy_spectrometry = 1000260,

    /// <summary>IKES: A method of analysis in which a beam of ions is separated according to the ratio of its translational energy to charge.</summary>
    MS_IKES = MS_ion_kinetic_energy_spectrometry,

    /// <summary>ion mobility spectrometry: The separation of ions according to their velocity through a buffer gas under the influence of an electric field.</summary>
    MS_ion_mobility_spectrometry = 1000261,

    /// <summary>IMS: The separation of ions according to their velocity through a buffer gas under the influence of an electric field.</summary>
    MS_IMS = MS_ion_mobility_spectrometry,

    /// <summary>infrared multiphoton dissociation: Multiphoton ionization where the reactant ion dissociates as a result of the absorption of multiple infrared photons.</summary>
    MS_infrared_multiphoton_dissociation = 1000262,

    /// <summary>IRMPD: Multiphoton ionization where the reactant ion dissociates as a result of the absorption of multiple infrared photons.</summary>
    MS_IRMPD = MS_infrared_multiphoton_dissociation,

    /// <summary>isotope ratio mass spectrometry: The measurement of the relative quantity of the different isotopes of an element in a material with a mass spectrometer.</summary>
    MS_isotope_ratio_mass_spectrometry = 1000263,

    /// <summary>IRMS: The measurement of the relative quantity of the different isotopes of an element in a material with a mass spectrometer.</summary>
    MS_IRMS = MS_isotope_ratio_mass_spectrometry,

    /// <summary>ion trap: A device for spatially confining ions using electric and magnetic fields alone or in combination.</summary>
    MS_ion_trap = 1000264,

    /// <summary>IT: A device for spatially confining ions using electric and magnetic fields alone or in combination.</summary>
    MS_IT = MS_ion_trap,

    /// <summary>kinetic energy release distribution: Distribution of values of translational kinetic energy release for an ensemble of metastable ions undergoing a specific dissociation reaction.</summary>
    MS_kinetic_energy_release_distribution = 1000265,

    /// <summary>KERD: Distribution of values of translational kinetic energy release for an ensemble of metastable ions undergoing a specific dissociation reaction.</summary>
    MS_KERD = MS_kinetic_energy_release_distribution,

    /// <summary>Laser Desorption: The formation of ions through the interaction of a laser with a material or with gas-phase ions or molecules.</summary>
    MS_Laser_Desorption_OBSOLETE = 1000266,

    /// <summary>Laser Ionization MERGE: The formation of ions through the interaction of a laser with a material or with gas-phase ions or molecules.</summary>
    MS_Laser_Ionization_MERGE_OBSOLETE = MS_Laser_Desorption_OBSOLETE,

    /// <summary>LD: The formation of ions through the interaction of a laser with a material or with gas-phase ions or molecules.</summary>
    MS_LD_OBSOLETE = MS_Laser_Desorption_OBSOLETE,

    /// <summary>mass analyzed ion kinetic energy spectrometry: Spectra that are obtained from a sector mass spectrometer that incorporates at least one magnetic sector plus one electric sector in reverse geometry. The accelerating voltage, V, and the magnetic sector field, B, are set at fixed values to select the precursor ions, which are then allowed to dissociate or to react in a field free region between the two sectors. The kinetic energy product ions of m/z selected precursor ions is analyzed by scanning the electric sector field, E. The width of the product ion spectrum peaks is related to the kinetic energy release distribution (KERD) for the dissociation process.</summary>
    MS_mass_analyzed_ion_kinetic_energy_spectrometry = 1000267,

    /// <summary>MIKES: Spectra that are obtained from a sector mass spectrometer that incorporates at least one magnetic sector plus one electric sector in reverse geometry. The accelerating voltage, V, and the magnetic sector field, B, are set at fixed values to select the precursor ions, which are then allowed to dissociate or to react in a field free region between the two sectors. The kinetic energy product ions of m/z selected precursor ions is analyzed by scanning the electric sector field, E. The width of the product ion spectrum peaks is related to the kinetic energy release distribution (KERD) for the dissociation process.</summary>
    MS_MIKES = MS_mass_analyzed_ion_kinetic_energy_spectrometry,

    /// <summary>mass spectrometry: The branch of science that deals with all aspects of mass spectrometers and the results obtained with these instruments.</summary>
    MS_mass_spectrometry = 1000268,

    /// <summary>MS: The branch of science that deals with all aspects of mass spectrometers and the results obtained with these instruments.</summary>
    MS_MS = MS_mass_spectrometry,

    /// <summary>mass spectrometry/mass spectrometry: The acquisition, study and spectra of the electrically charged products or precursors of a m/z selected ion or ions.</summary>
    MS_mass_spectrometry_mass_spectrometry = 1000269,

    /// <summary>MS/MS: The acquisition, study and spectra of the electrically charged products or precursors of a m/z selected ion or ions.</summary>
    MS_MS_MS = MS_mass_spectrometry_mass_spectrometry,

    /// <summary>multiple stage mass spectrometry: Multiple stages of precursor ion m/z selection followed by product ion detection for successive progeny ions.</summary>
    MS_multiple_stage_mass_spectrometry = 1000270,

    /// <summary>MSn: Multiple stages of precursor ion m/z selection followed by product ion detection for successive progeny ions.</summary>
    MS_MSn = MS_multiple_stage_mass_spectrometry,

    /// <summary>Negative Ion chemical ionization: Chemical ionization that results in the formation of negative ions.</summary>
    MS_Negative_Ion_chemical_ionization = 1000271,

    /// <summary>NICI: Chemical ionization that results in the formation of negative ions.</summary>
    MS_NICI = MS_Negative_Ion_chemical_ionization,

    /// <summary>neutralization reionization mass spectrometry: With this technique, m/z selected ions form neutrals by charge transfer to a collision gas or by dissociation. The neutrals are separated from the remaining ions and ionized in collisions with a second gas. This method is used to investigate reaction intermediates and other unstable species.</summary>
    MS_neutralization_reionization_mass_spectrometry = 1000272,

    /// <summary>NRMS: With this technique, m/z selected ions form neutrals by charge transfer to a collision gas or by dissociation. The neutrals are separated from the remaining ions and ionized in collisions with a second gas. This method is used to investigate reaction intermediates and other unstable species.</summary>
    MS_NRMS = MS_neutralization_reionization_mass_spectrometry,

    /// <summary>photoionization: The ionization of an atom or molecule by a photon, written M + h? ? M^+ + e. The term photon impact is not recommended.</summary>
    MS_photoionization = 1000273,

    /// <summary>PI: The ionization of an atom or molecule by a photon, written M + h? ? M^+ + e. The term photon impact is not recommended.</summary>
    MS_PI = MS_photoionization,

    /// <summary>pyrolysis mass spectrometry: A mass spectrometry technique in which the sample is heated to the point of decomposition and the gaseous decomposition products are introduced into the ion source.</summary>
    MS_pyrolysis_mass_spectrometry = 1000274,

    /// <summary>PyMS: A mass spectrometry technique in which the sample is heated to the point of decomposition and the gaseous decomposition products are introduced into the ion source.</summary>
    MS_PyMS = MS_pyrolysis_mass_spectrometry,

    /// <summary>collision quadrupole: A transmission quadrupole to which an oscillating potential is applied so as to focus a beam of ions through a collision gas with no m/z separation.</summary>
    MS_collision_quadrupole = 1000275,

    /// <summary>q: A transmission quadrupole to which an oscillating potential is applied so as to focus a beam of ions through a collision gas with no m/z separation.</summary>
    MS_q = MS_collision_quadrupole,

    /// <summary>resonance enhanced multiphoton ionization: Multiphoton ionization in which the ionization cross section is significantly enhanced because the energy of the incident photons is resonant with an intermediate excited state of the neutral species.</summary>
    MS_resonance_enhanced_multiphoton_ionization = 1000276,

    /// <summary>REMPI: Multiphoton ionization in which the ionization cross section is significantly enhanced because the energy of the incident photons is resonant with an intermediate excited state of the neutral species.</summary>
    MS_REMPI = MS_resonance_enhanced_multiphoton_ionization,

    /// <summary>residual gas analyzer: A mass spectrometer used to measure the composition and pressure of gasses in an evacuated chamber.</summary>
    MS_residual_gas_analyzer = 1000277,

    /// <summary>RGA: A mass spectrometer used to measure the composition and pressure of gasses in an evacuated chamber.</summary>
    MS_RGA = MS_residual_gas_analyzer,

    /// <summary>surface enhanced laser desorption ionization: The formation of ionized species in the gas phase from analytes deposited on a particular surface substrate which is irradiated with a laser beam of which wavelength is absorbed by the surface. See also desorption/ionization on silicon and laser desorption/ionization.</summary>
    MS_surface_enhanced_laser_desorption_ionization = 1000278,

    /// <summary>SELDI: The formation of ionized species in the gas phase from analytes deposited on a particular surface substrate which is irradiated with a laser beam of which wavelength is absorbed by the surface. See also desorption/ionization on silicon and laser desorption/ionization.</summary>
    MS_SELDI = MS_surface_enhanced_laser_desorption_ionization,

    /// <summary>surface enhanced neat desorption: Matrix-assisted laser desorption ionization in which the matrix is covalently linked to the target surface.</summary>
    MS_surface_enhanced_neat_desorption = 1000279,

    /// <summary>SEND: Matrix-assisted laser desorption ionization in which the matrix is covalently linked to the target surface.</summary>
    MS_SEND = MS_surface_enhanced_neat_desorption,

    /// <summary>suface ionization: The ionization of a neutral species when it interacts with a solid surface with an appropriate work function and temperature.</summary>
    MS_suface_ionization_OBSOLETE = 1000280,

    /// <summary>SI: The ionization of a neutral species when it interacts with a solid surface with an appropriate work function and temperature.</summary>
    MS_SI_OBSOLETE = MS_suface_ionization_OBSOLETE,

    /// <summary>selected ion flow tube: A device in which m/z selected ions are entrained in an inert carrier gas and undergo ion-molecule reactions.</summary>
    MS_selected_ion_flow_tube = 1000281,

    /// <summary>SIFT: A device in which m/z selected ions are entrained in an inert carrier gas and undergo ion-molecule reactions.</summary>
    MS_SIFT = MS_selected_ion_flow_tube,

    /// <summary>sustained off-resonance irradiation: A technique associated with Fourier transform ion cyclotron resonance (FT-ICR) mass spectrometry to carry out ion/neutral reactions such as low-energy collision-induced dissociation. A radio-frequency electric field of slightly off-resonance to the cyclotron frequency of the reactant ion cyclically accelerates and decelerates the reactant ion that is confined in the Penning ion trap. The ion's orbit does not exceed the dimensions of ion trap while the ion undergoes an ion/neutral species process that produces a high average translational energy for an extended time.</summary>
    MS_sustained_off_resonance_irradiation = 1000282,

    /// <summary>SORI: A technique associated with Fourier transform ion cyclotron resonance (FT-ICR) mass spectrometry to carry out ion/neutral reactions such as low-energy collision-induced dissociation. A radio-frequency electric field of slightly off-resonance to the cyclotron frequency of the reactant ion cyclically accelerates and decelerates the reactant ion that is confined in the Penning ion trap. The ion's orbit does not exceed the dimensions of ion trap while the ion undergoes an ion/neutral species process that produces a high average translational energy for an extended time.</summary>
    MS_SORI = MS_sustained_off_resonance_irradiation,

    /// <summary>Spark Source Mass Spectrometry: Mass spectrometry using spark ionization.</summary>
    MS_Spark_Source_Mass_Spectrometry_OBSOLETE = 1000283,

    /// <summary>SSMS: Mass spectrometry using spark ionization.</summary>
    MS_SSMS_OBSOLETE = MS_Spark_Source_Mass_Spectrometry_OBSOLETE,

    /// <summary>stored waveform inverse fourier transform: A technique to create excitation waveforms for ions in FT-ICR mass spectrometer or Paul ion trap. An excitation waveform in the time-domain is generated by taking the inverse Fourier transform of an appropriate frequency-domain programmed excitation spectrum, in which the resonance frequencies of ions to be excited are included. This technique may be used for selection of precursor ions in MS/MS experiments.</summary>
    MS_stored_waveform_inverse_fourier_transform = 1000284,

    /// <summary>SWIFT: A technique to create excitation waveforms for ions in FT-ICR mass spectrometer or Paul ion trap. An excitation waveform in the time-domain is generated by taking the inverse Fourier transform of an appropriate frequency-domain programmed excitation spectrum, in which the resonance frequencies of ions to be excited are included. This technique may be used for selection of precursor ions in MS/MS experiments.</summary>
    MS_SWIFT = MS_stored_waveform_inverse_fourier_transform,

    /// <summary>total ion current: The sum of all the separate ion currents carried by the ions of different m/z contributing to a complete mass spectrum or in a specified m/z range of a mass spectrum.</summary>
    MS_total_ion_current = 1000285,

    /// <summary>TIC: The sum of all the separate ion currents carried by the ions of different m/z contributing to a complete mass spectrum or in a specified m/z range of a mass spectrum.</summary>
    MS_TIC = MS_total_ion_current,

    /// <summary>time lag focusing: Energy focusing in a time-of-flight mass spectrometer that is accomplished by introducing a time delay between the formation of the ions and the application of the accelerating voltage pulse.</summary>
    MS_time_lag_focusing = 1000286,

    /// <summary>TLF: Energy focusing in a time-of-flight mass spectrometer that is accomplished by introducing a time delay between the formation of the ions and the application of the accelerating voltage pulse.</summary>
    MS_TLF = MS_time_lag_focusing,

    /// <summary>time-of-flight mass spectrometer: An instrument that separates ions by m/z in a field-free region after acceleration to a fixed kinetic energy.</summary>
    MS_time_of_flight_mass_spectrometer = 1000287,

    /// <summary>TOF-MS: An instrument that separates ions by m/z in a field-free region after acceleration to a fixed kinetic energy.</summary>
    MS_TOF_MS = MS_time_of_flight_mass_spectrometer,

    /// <summary>cyclotron: A device that uses an oscillating electric field and magnetic field to accelerate charged particles.</summary>
    MS_cyclotron = 1000288,

    /// <summary>double-focusing mass spectrometer: A mass spectrometer that uses a magnetic sector for m/z focusing and an electric sector for energy focusing of an ion beam.</summary>
    MS_double_focusing_mass_spectrometer = 1000289,

    /// <summary>hybrid mass spectrometer: A mass spectrometer that combines m/z analyzers of different types to perform tandem mass spectrometry.</summary>
    MS_hybrid_mass_spectrometer = 1000290,

    /// <summary>linear ion trap: A two dimensional Paul ion trap in which ions are confined in the axial dimension by means of an electric field at the ends of the trap.</summary>
    MS_linear_ion_trap = 1000291,

    /// <summary>mass spectrograph obsolete: An instrument that separates a beam of ions according to their mass-to-charge ratio in which the ions are directed onto a focal plane detector such as a photographic plate.</summary>
    MS_mass_spectrograph_obsolete = 1000292,

    /// <summary>mass spectrometer: An instrument that measures the mass-to-charge ratio and relative abundances of ions.</summary>
    MS_mass_spectrometer = 1000293,

    /// <summary>mass spectrum: A plot of the relative abundance of a beam or other collection of ions as a function of the mass-to-charge ratio (m/z).</summary>
    MS_mass_spectrum = 1000294,

    /// <summary>mattauch-herzog geometry: An arrangement for a double-focusing mass spectrometer in which a deflection of ?/(4 ?(2)) radians in a radial electric field is followed by a magnetic deflection of ?/2 radians.</summary>
    MS_mattauch_herzog_geometry = 1000295,

    /// <summary>nier-johnson geometry: An arrangement for a double-focusing mass spectrometer in which a deflection of ?/2 radians in a radial electric field analyzer is followed by a magnetic deflection of ?/3 radians.</summary>
    MS_nier_johnson_geometry = 1000296,

    /// <summary>paul ion trap: A device that permits the trapping of ions by means of an alternating current voltage. The ejection of ions with a m/z less than a prescribed value and retention of those with higher mass depends on the application of radio frequency voltages between a ring electrode and two end-cap electrodes to confine the ions in a circular path. The choice of these voltages determines the m/z below which ions are ejected.</summary>
    MS_paul_ion_trap_OBSOLETE = 1000297,

    /// <summary>prolate traochoidal mass spectrometer: A mass spectrometer in which the ions of different m/z are separated by means of crossed electric and magnetic fields in such a way that the selected ions follow a prolate trochoidal path.</summary>
    MS_prolate_traochoidal_mass_spectrometer = 1000298,

    /// <summary>quistor: An abbreviation of quadrupole ion storage trap. This term is synonymous with Paul Ion Trap. If so then add a synonym to paul and obsolete this term.</summary>
    MS_quistor = 1000299,

    /// <summary>reflectron: A time-of-flight mass spectrometer that uses a static electric field to reverse the direction of travel of the ions entering it. A reflectron improves mass resolution by assuring that ions of the same m/z but different kinetic energy arrive at the detector at the same time.</summary>
    MS_reflectron = 1000300,

    /// <summary>sector mass spectrometer: A mass spectrometer consisting of one or more magnetic sectors for m/z selection in a beam of ions. Such instruments may also have one or more electric sectors for energy selection.</summary>
    MS_sector_mass_spectrometer = 1000301,

    /// <summary>tandem mass spectrometer: A mass spectrometer designed for mass spectrometry/mass spectrometry.</summary>
    MS_tandem_mass_spectrometer = 1000302,

    /// <summary>transmission quadrupole mass spectrometer: A mass spectrometer that consists of four parallel rods whose centers form the corners of a square and whose opposing poles are connected. The voltage applied to the rods is a superposition of a static potential and a sinusoidal radio frequency potential. The motion of an ion in the x and y dimensions is described by the Matthieu equation whose solutions show that ions in a particular m/z range can be transmitted along the z axis.</summary>
    MS_transmission_quadrupole_mass_spectrometer = 1000303,

    /// <summary>accelerating voltage: The electrical potential used to impart kinetic energy to ions in a mass spectrometer.</summary>
    MS_accelerating_voltage = 1000304,

    /// <summary>cyclotron motion: The circular motion of a charged particle moving at velocity v in a magnetic field B that results from the force qvB.</summary>
    MS_cyclotron_motion = 1000305,

    /// <summary>dynamic mass spectrometry: A mass spectrometer in which m/z separation using one or more electric fields that vary with time.</summary>
    MS_dynamic_mass_spectrometry = 1000306,

    /// <summary>einzel lens: Three element charged particle lens in which the first and third elements are held at the same voltage. Such a lens produces focusing without changing the translational energy of the particle.</summary>
    MS_einzel_lens = 1000307,

    /// <summary>electric field strength: The magnitude of the force per unit charge at a given point in space.</summary>
    MS_electric_field_strength = 1000308,

    /// <summary>first stability region: The region of a Mathieu stability diagram closest to the origin. Ions within this region can traverse the full length of a transmission quadrupole.</summary>
    MS_first_stability_region = 1000309,

    /// <summary>fringing field: The electric or magnetic field that extends from the edge of a sector, lens or other ion optics element.</summary>
    MS_fringing_field = 1000310,

    /// <summary>kinetic energy analyzer: A device for measuring the kinetic energy of charged particles using a retarding field, time-of-flight, or the extent of deflection in an electric or magnetic field.</summary>
    MS_kinetic_energy_analyzer = 1000311,

    /// <summary>mass limit: The m/z value above which ions cannot be detected in a mass spectrometer.</summary>
    MS_mass_limit = 1000312,

    /// <summary>scan m/z range?: The limit of m/z over which a mass spectrometer can detect ions.</summary>
    MS_scan_m_z_range_ = 1000313,

    /// <summary>mass selective axial ejection: The use of mass selective instability to eject ions of selected m/z values from an ion trap.</summary>
    MS_mass_selective_axial_ejection = 1000314,

    /// <summary>mass selective instability: A method for selective ejection of ions according to their m/z value in an ion trap.</summary>
    MS_mass_selective_instability = 1000315,

    /// <summary>mathieu stability diagram: A graphical representation expressed in terms of reduced coordinates that describes charged particle motion in a quadrupole mass filter or quadrupole ion trap mass spectrometer.</summary>
    MS_mathieu_stability_diagram = 1000316,

    /// <summary>orthogonal extraction: The pulsed acceleration of ions perpendicular to their direction of travel into a time-of-flight mass spectrometer. Ions may be extracted from a directional ion source, drift tube or m/z separation stage.</summary>
    MS_orthogonal_extraction = 1000317,

    /// <summary>resonance ion ejection: A mode of ion ejection in a quadrupole ion trap that relies on a auxiliary radio frequency voltage that is applied to the end-cap electrodes. The voltage is tuned to the secular frequency of a particular ion to eject it.</summary>
    MS_resonance_ion_ejection = 1000318,

    /// <summary>space charge effect: The mutual repulsion of particles of like charge that limits the current in a charged-particle beam and causes beams or packets of charged particles to expand radially over time.</summary>
    MS_space_charge_effect = 1000319,

    /// <summary>static field: An electric or magnetic field that does not change in time.</summary>
    MS_static_field = 1000320,

    /// <summary>2E Mass Spectrum: A mass spectrum obtained by setting the electric sector field E to twice the value required to transmit the main ion-beam thereby allowing ions with a kinetic energy-to-charge ratio twice that of the main ion-beam to be transmitted. Product ions resulting from partial charge transfer reactions such as m^2+ + N ? m^+ + N^+ that occur in a collision cell (containing a gas, N) located in a field-free region preceding a magnetic and electric sector combination are detected. When the magnetic sector field B is scanned, a mass spectrum of singly charged product ions of doubly charged precursor ions is obtained.</summary>
    MS_2E_Mass_Spectrum = 1000321,

    /// <summary>charge inversion mass spectrum: The measurement of the relative abundance of ions that result from a charge inversion reaction as a function of m/z.</summary>
    MS_charge_inversion_mass_spectrum = 1000322,

    /// <summary>constant neutral loss scan: Spectrum of all precursor ions that undergo a selected m/z decrement.</summary>
    MS_constant_neutral_loss_scan = 1000323,

    /// <summary>constant neutral gain scan: Spectrum of all precursor ions that undergo a selected m/z increment.</summary>
    MS_constant_neutral_gain_scan = 1000324,

    /// <summary>Constant Neutral Mass Gain Scan: Spectrum of all precursor ions that undergo a selected m/z increment.</summary>
    MS_Constant_Neutral_Mass_Gain_Scan = MS_constant_neutral_gain_scan,

    /// <summary>constant neutral gain spectrum: A spectrum formed of all product ions that have been produced by gain of a pre-selected neutral mass following the reaction with and addition of the gas in a collision cell.</summary>
    MS_constant_neutral_gain_spectrum = 1000325,

    /// <summary>constant neutral mass gain spectrum: A spectrum formed of all product ions that have been produced by gain of a pre-selected neutral mass following the reaction with and addition of the gas in a collision cell.</summary>
    MS_constant_neutral_mass_gain_spectrum = MS_constant_neutral_gain_spectrum,

    /// <summary>constant neutral loss spectrum: A spectrum formed of all product ions that have been produced with a selected m/z decrement from any precursor ions. The spectrum shown correlates to the precursor ion spectrum. See also neutral loss spectrum.</summary>
    MS_constant_neutral_loss_spectrum = 1000326,

    /// <summary>constant neutral mass loss spectrum: A spectrum formed of all product ions that have been produced with a selected m/z decrement from any precursor ions. The spectrum shown correlates to the precursor ion spectrum. See also neutral loss spectrum.</summary>
    MS_constant_neutral_mass_loss_spectrum = MS_constant_neutral_loss_spectrum,

    /// <summary>consecutive reaction monitoring: A type of MS/MS experiments with three or more stages of m/z separation and in which a particular multi-step reaction path is monitored.</summary>
    MS_consecutive_reaction_monitoring_OBSOLETE = 1000327,

    /// <summary>e/2 mass spectrum: A mass spectrum obtained using a sector mass spectrometer in which the electric sector field E is set to half the value required to transmit the main ion-beam. This spectrum records the signal from doubly charged product ions of charge-stripping reactions.</summary>
    MS_e_2_mass_spectrum = 1000328,

    /// <summary>linked scan: A scan in an instrument with two or more m/z analysers or in a sector mass spectrometer that incorporates at least one magnetic sector and one electric sector. Two or more of the analyzers are scanned simultaneously so as to preserve a predetermined relationship between scan parameters to produce a product ion, precursor ion or constant neutral loss spectrum.</summary>
    MS_linked_scan = 1000329,

    /// <summary>linked scan at constant b/e: A linked scan at constant B/E may be performed on a sector mass spectrometer that incorporates at least one magnetic sector plus one electric sector. The magnetic field B and the electric field E are scanned simultaneously while the accelerating voltage V is held constant, so as to maintain the ratio of the two fields constant. This linked scan may record a product ion spectrum of dissociation or other reactions occurring in a field free region preceding the two sectors.</summary>
    MS_linked_scan_at_constant_b_e = 1000330,

    /// <summary>Linked Scan at Constant E2/V: A linked scan performed on a sector instrument that incorporates at least one electric sector plus one magnetic sector. The electric sector field, E, and the accelerating voltage, V, are scanned simultaneously, so as to maintain the ratio E2/V at a constant value. This linked scan recordss a product ion spectrum of dissociation or other reactions occurring in a field free region (FFR) preceding the two sectors.</summary>
    MS_Linked_Scan_at_Constant_E2_V = 1000331,

    /// <summary>Linked Scan at Constant B2/E: A linked scan performed on a sector mass spectrometer that incorporates at least one electric sector plus one magnetic sector in either order. The accelerating voltage is fixed and the magnetic field, B, and the electric field, E, are scanned simultaneously so as to maintain the ratio B2/E at a constant value. This linked scan records a precursor ion spectrum of dissociation or other reactions occurring in the field free region preceding the two sectors. The term B2/E linked scan is not recommended.</summary>
    MS_Linked_Scan_at_Constant_B2_E = 1000332,

    /// <summary>Linked Scan at Constant B[1-(E/E0)]^1/2 / E: A linked scan performed on a sector instrument that incorporates at least one electric sector plus one magnetic sector placed in either order. The accelerating voltage is fixed while scanning the magnetic field, B, and electric field, E, simultaneously, so as to maintain the quantity B[1-(E/E0)]1/2/E at a constant value. This linked scan records a constant neutral mass loss (or gain) spectrum of dissociation or other reactions occurring in a field free region preceding the two sectors. E0 is the electric field required to transmit the singly charged analog of the desired neutral fragment. The term B[1-(E/E0)]1/2/E linked scan.</summary>
    MS_Linked_Scan_at_Constant_B_1__E_E0___1_2___E = 1000333,

    /// <summary>MS/MS in Time: A tandem mass spectrometry method in which product ion spectra are recorded in a single m/z analyzer (such as a Paul Ion Trap or FTMS) in discreet steps over time. Ions in a specific m/z range are selected, dissociated, and the product ions analyzed sequentially in time.</summary>
    MS_MS_MS_in_Time = 1000334,

    /// <summary>MS/MS in Space: A tandem mass spectrometry method in which product ion spectra are recorded in m/z analyzers separated in space. Specific m/z separation functions are designed such that in one section of the instrument ions are selected, dissociated in an intermediate region, and the product ions are then transmitted to another analyser for m/z separation and data acquisition.</summary>
    MS_MS_MS_in_Space = 1000335,

    /// <summary>neutral loss: The loss of an uncharged species during a rearrangement process.</summary>
    MS_neutral_loss = 1000336,

    /// <summary>nth generation product ion: Serial product ions from dissociation of selected precursor ions where n refers to the number of stages of dissociation. The term granddaughter ion is deprecated.</summary>
    MS_nth_generation_product_ion = 1000337,

    /// <summary>nth generation product ion scan: The specific scan functions or processes that record the appropriate generation of product ion or ions of any m/z selected precursor ions.</summary>
    MS_nth_generation_product_ion_scan = 1000338,

    /// <summary>nth generation product ion spectrum: The mass spectrum recorded from any mass spectrometer in which the appropriate scan function can be set to record the appropriate generation product ion or ions of m/z selected precursor ions.</summary>
    MS_nth_generation_product_ion_spectrum_OBSOLETE = 1000339,

    /// <summary>precursor ion: An ion that reacts to form particular product ions. The reaction can be unimolecular dissociation, ion/molecule reaction, isomerization, or change in charge state. The term parent ion is not recommended.</summary>
    MS_precursor_ion = 1000340,

    /// <summary>precursor ion spectrum: Spectrum generated by scanning precursor m/z while monitoring a fixed product m/z.</summary>
    MS_precursor_ion_spectrum = 1000341,

    /// <summary>product ion: An ion formed as the product of a reaction involving a particular precursor ion. The reaction can be unimolecular dissociation to form fragment ions, an ion/molecule reaction, or simply involve a change in the number of charges. The term fragment ion is deprecated. The term daughter ion is deprecated.</summary>
    MS_product_ion = 1000342,

    /// <summary>product ion spectrum: A mass spectrum recorded from any spectrometer in which the appropriate m/z separation scan function is set to record the product ion or ions of selected precursor ions.</summary>
    MS_product_ion_spectrum_OBSOLETE = 1000343,

    /// <summary>progeny ion: A charged product of a series of consecutive reactions that includes product ions, 1st generation product ions, 2nd generation product ions, etc. Given the sequential fragmentation scheme: M1+ -> M2+ -> M3+ -> M4+ -> M5+. M4+ is the precursor ion of M5+, a 1st generation product ion of M3+, a 2nd generation product ion of M2+ and a 3rd generation product ion of M1+.</summary>
    MS_progeny_ion = 1000344,

    /// <summary>Progeny Fragment Ion: A charged product of a series of consecutive reactions that includes product ions, 1st generation product ions, 2nd generation product ions, etc. Given the sequential fragmentation scheme: M1+ -> M2+ -> M3+ -> M4+ -> M5+. M4+ is the precursor ion of M5+, a 1st generation product ion of M3+, a 2nd generation product ion of M2+ and a 3rd generation product ion of M1+.</summary>
    MS_Progeny_Fragment_Ion = MS_progeny_ion,

    /// <summary>array detector: Detector comprising several ion collection elements, arranged in a line or grid where each element is an individual detector.</summary>
    MS_array_detector = 1000345,

    /// <summary>conversion dynode: A surface that is held at high potential such that ions striking the surface produce electrons that are subsequently detected.</summary>
    MS_conversion_dynode = 1000346,

    /// <summary>dynode: One of a series of electrodes in a photomultiplier tube. Such an arrangement is able to amplify the current emitted by the photocathode.</summary>
    MS_dynode = 1000347,

    /// <summary>focal plane collector: A detector for spatially disperse ion beams in which all ions simultaneously impinge on the detector plane.</summary>
    MS_focal_plane_collector = 1000348,

    /// <summary>ion-to-photon detector: A detector in which ions strike a conversion dynode to produce electrons that in turn strike a phosphor and the resulting photons are detected by a photomultiplier.</summary>
    MS_ion_to_photon_detector = 1000349,

    /// <summary>point collector: A detector in which the ion beam is focused onto a point and the individual ions arrive sequentially.</summary>
    MS_point_collector = 1000350,

    /// <summary>postacceleration detector: A detector in which the charged particles are accelerated to a high velocity and impinge on a conversion dynode, emitting secondary electrons. The electrons are accelerated onto a phosphor screen, which emits photons that are in turn detected using a photomultiplier or other photon detector.</summary>
    MS_postacceleration_detector = 1000351,

    /// <summary>secondary electron: Electrons that are ejected from a sample surface as a result of bombardment by a primary beam of atoms, ions or photons. WAS IN DETECTOR TYPE. Where should it go?</summary>
    MS_secondary_electron = 1000352,

    /// <summary>adduct ion: Ion formed by the interaction of an ion with one or more atoms or molecules to form an ion containing all the constituent atoms of the precursor ion as well as the additional atoms from the associated atoms or molecules.</summary>
    MS_adduct_ion = 1000353,

    /// <summary>aromatic ion: A planar cyclic ion that obeys the Hckel (4n + 2) rule where n is a positive integer representing the number of conjugated Pi electrons. Charge delocalization leads to greater stability compared to a hypothetical localized structure.</summary>
    MS_aromatic_ion = 1000354,

    /// <summary>analog ion: Ions that have similar chemical valence, for example the acetyl cation CH3-CO+ and the thioacetyl cation CH3-CS+.</summary>
    MS_analog_ion = 1000355,

    /// <summary>anti-aromatic ion: A planar cyclic ion with 4n ? electrons and is therefore not aromatic.</summary>
    MS_anti_aromatic_ion = 1000356,

    /// <summary>cationized molecule: An ion formed by the association of a cation with a neutral molecule, M, for example [M+ Na]+ and [M + K]+. The terms quasi-molecular ion and pseudo-molecular ion should not be used.</summary>
    MS_cationized_molecule = 1000357,

    /// <summary>cluster ion: An ion formed by a multi-component atomic or molecular assembly of one or more ions with atoms or molecules, such as [(H20)nH]+, [(NaCl)nNa]+ and [(H3PO3)nHPO3]-.</summary>
    MS_cluster_ion = 1000358,

    /// <summary>Conventional ion: A radical cation or anion in which the charge site and the unpaired electron spin are both formally located in the same atom or group of atoms, as opposed to the spatially separate electronic configuration of distonic ions. The radical cation of methanol, CH3OH+, in which the charge and spin sites are formally located at the O atom is an example of a conventional ion, whereas .CH2-OH2+ is a distonic ion.</summary>
    MS_Conventional_ion = 1000359,

    /// <summary>diagnostic ion: A product ion whose formation reveals structural or compositional information of its precursor. For instance, the phenyl cation in an electron ionization mass spectrum is a diagnostic ion for benzene and derivatives.</summary>
    MS_diagnostic_ion = 1000360,

    /// <summary>dimeric ion: An ion formed by ionization of a dimer or by the association of an ion with its neutral counterpart such as [M2]+ or [M-H-M]+.</summary>
    MS_dimeric_ion = 1000361,

    /// <summary>distonic ion: A radical cation or anion in which the charge site and the unpaired electron spin cannot be both formally located in the same atom or group of atoms as it can be with a conventional ion. For example, CH2-OH2+ is a distonic ion whereas the radical cation of methanol, CH3OH+ is a conventional ion.</summary>
    MS_distonic_ion = 1000362,

    /// <summary>enium ion: A positively charged lower-valency ion of the nonmetallic elements. The methenium ion is CH3+. Other examples are the oxenium, sulfenium, nitrenium, phosphenium, and halenium ions.</summary>
    MS_enium_ion = 1000363,

    /// <summary>fragment ion: A product ion that results from the dissociation of a precursor ion.</summary>
    MS_fragment_ion_OBSOLETE = 1000364,

    /// <summary>ion?: An atomic or molecular species having a net positive or negative electric charge.</summary>
    MS_ion_ = 1000365,

    /// <summary>Isotopologue ion: An ion that differs only in the isotopic composition of one or more of its constituent atoms. For example CH4+ and CH3D+ or 10BF3 and 11BF3. The term isotopologue is a contraction of isotopic homologue.</summary>
    MS_Isotopologue_ion = 1000366,

    /// <summary>Isotopomeric ion: Isomeric ion having the same numbers of each isotopic atom but differing in their positions. Isotopomeric ions can be either configurational isomers in which two atomic isotopes exchange positions or isotopic stereoisomers. The term isotopomer is a shortening of isotopic isomer.</summary>
    MS_Isotopomeric_ion = 1000367,

    /// <summary>metastable ion: An ion that is formed with internal energy higher than the threshold for dissociation but with a lifetime great enough to allow it to exit the ion source and enter the mass spectrometer where it dissociates before detection.</summary>
    MS_metastable_ion = 1000368,

    /// <summary>molecular ion: An ion formed by the removal of one or more electrons to form a positive ion or the addition off one or more electrons to form a negative ion.</summary>
    MS_molecular_ion = 1000369,

    /// <summary>negative ion: An atomic or molecular species having a net negative electric charge.</summary>
    MS_negative_ion = 1000370,

    /// <summary>non-classical ion: Hyper-coordinated carbonium ion such as the penta-coordinated norbornyl cation. Note: Tri-coordinated carbenium ions are termed classical ions.</summary>
    MS_non_classical_ion = 1000371,

    /// <summary>onium ion: A positively charged hypervalent ion of the nonmetallic elements. Examples are the methonium ion CH5+, the hydrogenonium ion H3+ and the hydronium ion H3O+. Other examples are the carbonium, oxonium, sulfonium, nitronium, diazonium, phosphonium, and halonium ions. Onium ions are not limited to monopositive ions; multiply-charged onium ions exist such as the gitonic (proximal) oxonium dication H4O2+ and the distonic oxonium dication H2O+-CH2-CH2-OH2+.</summary>
    MS_onium_ion = 1000372,

    /// <summary>principal ion: Most abundant ion of an isotope cluster, such as the 11B79Br2 81Br+ ion of m/z 250 of the cluster of isotopologue molecular ions of BBr3. The term principal ion has also been used to describe ions that have been artificially isotopically enriched in one or more positions such as CH3 13CH3+ or CH2D2 +, but those are best defined as isotopologue ions.</summary>
    MS_principal_ion = 1000373,

    /// <summary>positive ion: An atomic or molecular species having a net positive electric charge.</summary>
    MS_positive_ion = 1000374,

    /// <summary>protonated molecule: An ion formed by interaction of a neutral molecule with a proton and represented by the symbol [M + H]+, where M is the neutral molecule. The term 'protonated molecular ion,' 'quasi-molecular ion' and 'pseudo-molecular ion' are not recommended.</summary>
    MS_protonated_molecule = 1000375,

    /// <summary>radical ion: An ion, either a cation or anion, containing unpaired electrons in its ground state. The unpaired electron is denoted by a superscript dot alongside the superscript symbol for charge, such as for the molecular ion of a molecule M, that is, M+. Radical ions with more than one charge and/or more than one unpaired electron are denoted such as M(2+)(2). Unless the positions of the unpaired electron and charge can be associated with specific atoms, superscript charge designation should be placed before the superscript dot designation.</summary>
    MS_radical_ion = 1000376,

    /// <summary>reference ion: A stable ion whose structure is known with certainty. These ions are usually formed by direct ionization of a neutral molecule of known structure and are used to verify by comparison the structure of an unknown ion.</summary>
    MS_reference_ion = 1000377,

    /// <summary>stable ion: An ion with internal energy sufficiently low that it does not rearrange or dissociate prior to detection in a mass spectrometer.</summary>
    MS_stable_ion = 1000378,

    /// <summary>unstable ion: An ion with sufficient enerrgy to dissociate within the ion source.</summary>
    MS_unstable_ion = 1000379,

    /// <summary>adiabatic ionization: A process whereby an electron is removed from an atom, ion, or molecule to produce an ion in its lowest energy state.</summary>
    MS_adiabatic_ionization = 1000380,

    /// <summary>associative ionization: An ionization process in which two excited atoms or molecules react to form a single positive ion and an electron.</summary>
    MS_associative_ionization = 1000381,

    /// <summary>atmospheric pressure photoionization: Atmospheric pressure chemical ionization in which the reactant ions are generated by photo-ionization.</summary>
    MS_atmospheric_pressure_photoionization = 1000382,

    /// <summary>autodetachment: The formation of a neutral when a negative ion in a disrtete state with an energy greater than the detachment threshold loses an electron spontaneously without further interaction with an energy source.</summary>
    MS_autodetachment = 1000383,

    /// <summary>autoionization: The formation of an ion when an atom or molecule in a discrete state with an energy greater than the ionization threshold loses an electron spontaneously without further interaction with an energy source.</summary>
    MS_autoionization = 1000384,

    /// <summary>charge exchange ionization: The interaction of an ion with an atom or molecule in which the charge on the ion is transferred to the neutral without the dissociation of either. Synonymous with charge transfer ionization.</summary>
    MS_charge_exchange_ionization = 1000385,

    /// <summary>chemi-ionization: The reaction of a neutral molecule with an internally excited molecule to form an ion. Note that this term is not synonymous with chemical ionization.</summary>
    MS_chemi_ionization = 1000386,

    /// <summary>desorption/ionization on silicon: The formation of ions by laser desorption ionization of a sample deposited on a porous silicon surface.</summary>
    MS_desorption_ionization_on_silicon = 1000387,

    /// <summary>dissociative ionization: The reaction of a gas-phase molecule that results in its decomposition to form products, one of which is an ion.</summary>
    MS_dissociative_ionization = 1000388,

    /// <summary>electron ionization: The ionization of an atom or molecule by electrons that are typically accelerated to energies between 50 and 150 eV. Usually 70 eV electrons are used to produce positive ions. The term 'electron impact' is not recommended.</summary>
    MS_electron_ionization = 1000389,

    /// <summary>ion desolvation: The removal of solvent molecules clustered around a gas-phase ion by means of heating and/or collisions with gas molecules.</summary>
    MS_ion_desolvation = 1000390,

    /// <summary>ion-pair formation: The reaction of a molecule to form both a positive ion and negative ion fragment among the products.</summary>
    MS_ion_pair_formation = 1000391,

    /// <summary>ionization efficiency: The ratio of the number of ions formed to the number of electrons, molecules or photons used.</summary>
    MS_ionization_efficiency = 1000392,

    /// <summary>laser desorption ionization: The formation of gas-phase ions by the interaction of a pulsed laser with a solid or liquid material.</summary>
    MS_laser_desorption_ionization = 1000393,

    /// <summary>liquid secondary ionization: The ionization of any species by the interaction of a focused beam of ions with a sample that is dissolved in a solvent matrix. See also fast atom bombardment and secondary ionization.</summary>
    MS_liquid_secondary_ionization = 1000395,

    /// <summary>membrane inlet: A semi-permeable membrane separator that permits the passage of gas sample directly to the mass spectrometer ion source.</summary>
    MS_membrane_inlet = 1000396,

    /// <summary>microelectrospray: Electrospray ionization at a solvent flow rate of 300-800 nL/min where the flow is a result of a mechanical pump. See nanaoelectrospray.</summary>
    MS_microelectrospray = 1000397,

    /// <summary>nanoelectrospray: Electrospray ionization at a flow rate less than ~25 nL/min. Nanoelectrospray is synonymous with nanospray. The flow is dependent on the potenial on the tip of the electrospray needle and/or a gas presure to push the sample through the needle. See also electrospray ionization and microelectrospray.</summary>
    MS_nanoelectrospray = 1000398,

    /// <summary>nanospray: Electrospray ionization at a flow rate less than ~25 nL/min. Nanoelectrospray is synonymous with nanospray. The flow is dependent on the potenial on the tip of the electrospray needle and/or a gas presure to push the sample through the needle. See also electrospray ionization and microelectrospray.</summary>
    MS_nanospray = MS_nanoelectrospray,

    /// <summary>penning ionization: Ionization that occurs through the interaction of two or more neutral gaseous species, at least one of which is internally excited.</summary>
    MS_penning_ionization = 1000399,

    /// <summary>plasma desorption ionization: The ionization of material in a solid sample by bombarding it with ionic or neutral atoms formed as a result of the fission of a suitable nuclide, typically 252Cf. Synonymous with fission fragment ionization.</summary>
    MS_plasma_desorption_ionization = 1000400,

    /// <summary>pre-ionization state: An electronic state capable of undergoing auto-Ionization.</summary>
    MS_pre_ionization_state = 1000401,

    /// <summary>secondary ionization: The process in which ions are ejected from a sample surface as a result of bombardment by a primary beam of atoms or ions.</summary>
    MS_secondary_ionization = 1000402,

    /// <summary>soft ionization: The formation of gas-phase ions without extensive fragmentation.</summary>
    MS_soft_ionization = 1000403,

    /// <summary>spark ionization: The formation of ions from a solid material by an intermittent electrical discharge.</summary>
    MS_spark_ionization = 1000404,

    /// <summary>surface-assisted laser desorption ionization: The formation of gas-phase ions from molecules that are deposited on a particular surface substrate that is irradiated with a pulsed laser. See also matrix-assisted laser desorption ionization.</summary>
    MS_surface_assisted_laser_desorption_ionization = 1000405,

    /// <summary>surface ionization: The ionization of a neutral species when it interacts with a solid surface with an appropriate work function and temperature.</summary>
    MS_surface_ionization = 1000406,

    /// <summary>thermal ionization: The ionization of a neutral species through contact with a high temperature surface.</summary>
    MS_thermal_ionization = 1000407,

    /// <summary>vertical ionization: A process in which an electron is removed from or added to a molecule without a change in the positions of the atoms. The resulting ion is typically in an excited vibrational state.</summary>
    MS_vertical_ionization = 1000408,

    /// <summary>association reaction: The reaction of an ion with a neutral species in which the reactants combine to form a single ion.</summary>
    MS_association_reaction = 1000409,

    /// <summary>alpha-cleavage: A homolytic cleavage where the bond fission occurs between at the atom adjacent to the atom at the apparent charge site and an atom removed from the aparent charge site by two bonds.</summary>
    MS_alpha_cleavage = 1000410,

    /// <summary>beta-cleavage: A homolytic cleavage where the bond fission occurs between at an atom removed from the apparent charge site atom by two bonds and an atom adjacent to that atom and removed from the aparent charge site by three bonds.</summary>
    MS_beta_cleavage = 1000411,

    /// <summary>buffer gas: An inert gas used for collisional deactivation of internally excited ions.</summary>
    MS_buffer_gas = 1000412,

    /// <summary>charge-induced fragmentation: Fragmentation of an odd electron ion in which the cleaved bond is adjacent to the apparent charge site. Synonymous with charge mediated fragmentation.</summary>
    MS_charge_induced_fragmentation = 1000413,

    /// <summary>charge inversion reaction: Reaction of an ion with a neutral species in which the charge on the product ion is reversed in sign with respect to the reactant ion.</summary>
    MS_charge_inversion_reaction = 1000414,

    /// <summary>charge permutation reaction: The reaction of an ion with a neutral species with a resulting change in the magnitude or sign of the charge on the reactant ion.</summary>
    MS_charge_permutation_reaction = 1000415,

    /// <summary>charge stripping reaction: Reaction of a positive ion with a neutral species in which the positive charge on the product ion is greater than that on the reactant ion.</summary>
    MS_charge_stripping_reaction = 1000416,

    /// <summary>charge transfer reaction: The reaction of an ion with a neutral species in which some or all of the charge of the reactant ion is transferred to the neutral species.</summary>
    MS_charge_transfer_reaction = 1000417,

    /// <summary>collisional excitation: The reaction of an ion with a neutral species in which the translational energy of the collision is converted into internal energy of the ion.</summary>
    MS_collisional_excitation = 1000418,

    /// <summary>collision gas: An inert gas used for collisional excitation. The term target gas is not recommended.</summary>
    MS_collision_gas = 1000419,

    /// <summary>heterolytic cleavage: Fragmentation of a molecule or ion in which both electrons forming the single bond that is broken remain on one of the atoms that were originally bonded. This term is synonymous with heterolysis.</summary>
    MS_heterolytic_cleavage = 1000420,

    /// <summary>high energy collision: Collision-induced dissociation process wherein the projectile ion has laboratory-frame translational energy higher than 1 keV.</summary>
    MS_high_energy_collision = 1000421,

    /// <summary>high-energy collision-induced dissociation: A collision-induced dissociation process wherein the projectile ion has the translational energy higher than approximately 1000 eV.</summary>
    MS_high_energy_collision_induced_dissociation = 1000422,

    /// <summary>HCD: A collision-induced dissociation process wherein the projectile ion has the translational energy higher than approximately 1000 eV.</summary>
    MS_HCD = MS_high_energy_collision_induced_dissociation,

    /// <summary>homolytic cleavage: Fragmentation of an odd electron ion that results from one of a pair of electrons that form a bond between two atoms moving to form a pair with the odd electron on the atom at the apparent charge site. Fragmentation results in the formation of an even electron ion and a radical. This reaction involves the movement of a single electron and is symbolized by a single-barbed arrow. Synonymous with Homolysis.</summary>
    MS_homolytic_cleavage = 1000423,

    /// <summary>hydrogen/deuterium exchange: Exchange of hydrogen atoms with deuterium atoms in a molecule or pre-formed ion in solution prior to introduction into a mass spectrometer, or by reaction of an ion with a deuterated collision gas inside a mass spectrometer.</summary>
    MS_hydrogen_deuterium_exchange = 1000424,

    /// <summary>ion energy loss spectrum: A plot of the relative abundance of a beam or other collection of ions as a function their loss of translational energy in reactions with neutral species.</summary>
    MS_ion_energy_loss_spectrum = 1000425,

    /// <summary>ionizing collision: The reaction of an ion with a neutral species in which one or more electrons are removed from either the ion or neutral.</summary>
    MS_ionizing_collision = 1000426,

    /// <summary>ion/molecule reaction: The reaction of an ion with a neutral molecule. The term ion-molecule reaction is not recommended because the hyphen suggests a single species that is that is both an ion and a molecule.</summary>
    MS_ion_molecule_reaction = 1000427,

    /// <summary>ion/neutral complex: A particular type of transition state that lies between precursor and product ions on the reaction coordinate of some ion reactions.</summary>
    MS_ion_neutral_complex = 1000428,

    /// <summary>ion/neutral species reaction: A process wherein a charged species interacts with a neutral reactant to produce either chemically different species or changes in the internal energy of one or both of the reactants.</summary>
    MS_ion_neutral_species_reaction = 1000429,

    /// <summary>ion/neutral species exchange reaction: In this reaction an association reaction is accompanied by the subsequent or simultaneous liberation of a different neutral species as a product.</summary>
    MS_ion_neutral_species_exchange_reaction = 1000430,

    /// <summary>kinetic method: An approach to determination of ion thermodynamic quantities by a bracketing procedure in which the relative probabilities of competing ion fragmentations are measured via the relative abundances of the reaction products. The extended kinetic method takes the associated entropy changes into account.</summary>
    MS_kinetic_method = 1000431,

    /// <summary>low energy collisions: A collision between an ion and neutral species with translational energy approximately 1000 eV or lower.</summary>
    MS_low_energy_collisions = 1000432,

    /// <summary>low-energy collision-induced dissociation: A collision-induced dissociation process wherein the precursor ion has the translational energy lower than approximately 1000 eV. This process typically requires multiple collisions and the collisional excitation is cumulative.</summary>
    MS_low_energy_collision_induced_dissociation = 1000433,

    /// <summary>McLafferty Rearrangement: A dissociation reaction triggered by transfer of a hydrogen atom via a 6-member transition state to the formal radical/charge site from a carbon atom four atoms removed from the charge/radical site (the gamma-carbon); subsequent rearrangement of electron density leads to expulsion of an olefin molecule. This term was originally applied to ketone ions where the charge/radical site is the carbonyl oxygen, but it is now more widely applied.</summary>
    MS_McLafferty_Rearrangement = 1000434,

    /// <summary>photodissociation: A process wherein the reactant ion is dissociated as a result of absorption of one or more photons.</summary>
    MS_photodissociation = 1000435,

    /// <summary>multiphoton dissociation: A process wherein the reactant ion is dissociated as a result of absorption of one or more photons.</summary>
    MS_multiphoton_dissociation = MS_photodissociation,

    /// <summary>MPD: A process wherein the reactant ion is dissociated as a result of absorption of one or more photons.</summary>
    MS_MPD = MS_photodissociation,

    /// <summary>partial charge transfer reaction: Reaction of an ion with a neutral species in which some but not all of the ion charge is transferred to the neutral.</summary>
    MS_partial_charge_transfer_reaction = 1000436,

    /// <summary>ion reaction: Chemical transformation involving an ion.</summary>
    MS_ion_reaction = 1000437,

    /// <summary>superelastic collision: Collision in which the translational energy of the fast-moving collision partner is increased at the expense of internal energy of one or both collision partners.</summary>
    MS_superelastic_collision = 1000438,

    /// <summary>surface-induced reaction: A process wherein a reactant ion interacts with a surface to produce either chemically different species or a change in the internal energy of the reactant ion.</summary>
    MS_surface_induced_reaction = 1000439,

    /// <summary>unimolecular dissociation: Fragmentation reaction in which the molecularity is treated as one, irrespective of whether the dissociative state is that of a metastable ion produced in the ion source or results from collisional excitation of a stable ion.</summary>
    MS_unimolecular_dissociation = 1000440,

    /// <summary>scan: Function or process of the mass spectrometer where it records a spectrum.</summary>
    MS_scan = 1000441,

    /// <summary>spectrum: A mass spectrum is an intensity vs m/z (mass-to-charge ratio) plot representing a chemical analysis.</summary>
    MS_spectrum = 1000442,

    /// <summary>mass analyzer type: Mass analyzer separates the ions according to their mass-to-charge ratio.</summary>
    MS_mass_analyzer_type = 1000443,

    /// <summary>m/z Separation Method: m/z Separation Method.</summary>
    MS_m_z_Separation_Method = 1000444,

    /// <summary>sequential m/z separation method: Sequential m/z separation method.</summary>
    MS_sequential_m_z_separation_method = 1000445,

    /// <summary>fast ion bombardment: The ionization of any species by the interaction of a focused beam of ions having a translational energy of several thousand eV with a solid sample.</summary>
    MS_fast_ion_bombardment = 1000446,

    /// <summary>FIB: The ionization of any species by the interaction of a focused beam of ions having a translational energy of several thousand eV with a solid sample.</summary>
    MS_FIB = MS_fast_ion_bombardment,

    /// <summary>LTQ: Finnigan LTQ MS.</summary>
    MS_LTQ = 1000447,

    /// <summary>LTQ FT: Finnigan LTQ FT MS.</summary>
    MS_LTQ_FT = 1000448,

    /// <summary>LTQ Orbitrap: Finnigan LTQ Orbitrap MS.</summary>
    MS_LTQ_Orbitrap = 1000449,

    /// <summary>LXQ: Finnigan LXQ MS.</summary>
    MS_LXQ = 1000450,

    /// <summary>mass analyzer: Terms used to describe the Analyzer.</summary>
    MS_mass_analyzer = 1000451,

    /// <summary>analyzer: Terms used to describe the Analyzer.</summary>
    MS_analyzer = MS_mass_analyzer,

    /// <summary>data transformation: Terms used to describe types of data processing.</summary>
    MS_data_transformation = 1000452,

    /// <summary>data processing: Terms used to describe types of data processing.</summary>
    MS_data_processing = MS_data_transformation,

    /// <summary>detector: The device that detects ions.</summary>
    MS_detector = 1000453,

    /// <summary>instrument additional description: Additional terms to describe the instrument as outlined in the mass spec doc, Appendix 1, section 1.5.</summary>
    MS_instrument_additional_description = 1000454,

    /// <summary>ion selection attribute: Ion selection properties that are associated with a value.</summary>
    MS_ion_selection_attribute = 1000455,

    /// <summary>precursor activation: Terms to describe the precursor activation.</summary>
    MS_precursor_activation = 1000456,

    /// <summary>activation: Terms to describe the precursor activation.</summary>
    MS_activation = MS_precursor_activation,

    /// <summary>sample: Terms to describe the sample.</summary>
    MS_sample = 1000457,

    /// <summary>source: Terms to describe the source.</summary>
    MS_source = 1000458,

    /// <summary>spectrum instrument description: Terms used to describe the spectrum.</summary>
    MS_spectrum_instrument_description = 1000459,

    /// <summary>unit: Terms to describe units.</summary>
    MS_unit = 1000460,

    /// <summary>additional description: Terms to describe Additional.</summary>
    MS_additional_description = 1000461,

    /// <summary>ion optics: Device used in the construction of a mass spectrometer to focus, contain or otherwise manipulate ions.</summary>
    MS_ion_optics = 1000462,

    /// <summary>instrument: Description of the instrument or the mass spectrometer.</summary>
    MS_instrument = 1000463,

    /// <summary>instrument configuration: Description of the instrument or the mass spectrometer.</summary>
    MS_instrument_configuration = MS_instrument,

    /// <summary>mass unit: A unit of measurement for mass.</summary>
    MS_mass_unit_OBSOLETE = 1000464,

    /// <summary>scan polarity: An acquisition mode to which specifies weather polarity is negative, positive or alternating.</summary>
    MS_scan_polarity = 1000465,

    /// <summary>alternating: Alternating.</summary>
    MS_alternating_OBSOLETE = 1000466,

    /// <summary>1200 series LC/MSD SL: The 1200 Series LC/MSD SL ion trap belongs to the Agilent LC/MSD ion trap family. It provides fast polarity switching and multisignal data acquisition capabilities in a single run while also providing 5 stages of automated data dependent MS/MS and 11 stages of manual MS/MS.</summary>
    MS_1200_series_LC_MSD_SL = 1000467,

    /// <summary>6110 Quadrupole LC/MS: The 6110 Quadrupole LC/MS system is a Agilent liquid chromatography instrument combined with an entry level single quadrupole mass spectrometer from the 6100 Series of Agilent quadrupole mass spectrometers. 6110 Quadrupole mass spectrometer has m/z range of 10-1500 and 2500 u/s scan speed. It proves useful for wide range of SIM quantitative applications.</summary>
    MS_6110_Quadrupole_LC_MS = 1000468,

    /// <summary>6120 Quadrupole LC/MS: The 6120 Quadrupole LC/MS system is a Agilent liquid chromatography instrument combined with a single quadrupole mass spectrometer from the 6100 Series of Agilent mass spectrometers. 6120 quadrupole mass spectrometer has m/z range of 10-1500, 2500 u/s scan speed and utilizes multiple signal acquisition.</summary>
    MS_6120_Quadrupole_LC_MS = 1000469,

    /// <summary>6130 Quadrupole LC/MS: The 6130 Quadrupole LC/MS system is a Agilent liquid chromatography instrument combined with a single quadrupole mass spectrometer from the 6100 series of Agilent mass spectrometers. The 6130 quadrupole mass spectrometer has m/z range of 2-3000, 2500 u/s scan speed in standard mode and 5250 u/s speed in fast-scan mode. It also uses multiple signal acquisition.</summary>
    MS_6130_Quadrupole_LC_MS = 1000470,

    /// <summary>6140 Quadrupole LC/MS: The 6140 Quadrupole LC/MS system is a Agilent liquid chromatography instrument combined with a single quadrupole mass spectrometer from the 6100 Series of Agilent quadrupole mass spectrometers. 6140 Quadrupole mass spectrometer has m/z range of 10-1350, 2500 u/s scan speed in standard mode and 10000 u/s speed in fast-scan mode. It also uses multiple signal acquisition.</summary>
    MS_6140_Quadrupole_LC_MS = 1000471,

    /// <summary>6210 Time-of-Flight LC/MS: The 6210 Time-of-Flight LC/MS is a Agilent liquid chromatography instrument combined with a Agilent time of flight mass spectrometer. This time of flight mass spectrometer has a m/z range of 50-12000, mass accuracy of less than 2 ppm and resolution greater than 13,000 at m/z 2722. It has multiple ion sources and can be used with multimode ion sources.</summary>
    MS_6210_Time_of_Flight_LC_MS = 1000472,

    /// <summary>6310 Ion Trap LC/MS: The 6310 Ion Trap LC/MS is a Agilent liquid chromatography instrument combined with a 6300 series Agilent ion trap. It has a mass range of 50-2200 between 0.6 to 0.35 resolution and mass range of 200-4000 with resolution of 3-4. The scan speed varies from 1650-27000 for the respective mass ranges.</summary>
    MS_6310_Ion_Trap_LC_MS = 1000473,

    /// <summary>6320 Ion Trap LC/MS: The 6320 Ion Trap LC/MS is a Agilent liquid chromatography instrument combined with a 6300 series Agilent ion trap. It has a mass range of 50-2200 between 0.6 to 0.25 resolution and mass range of 200-4000 with resolution of less than 3. The scan speed varies from 1650-27000 for the respective mass ranges.</summary>
    MS_6320_Ion_Trap_LC_MS = 1000474,

    /// <summary>6330 Ion Trap LC/MS: The 6330 Ion Trap LC/MS is a Agilent liquid chromatography instrument combined with a 6300 series Agilent ion trap. It has a mass range of 50-2200 between 0.6 to 0.25 resolution and mass range of 200-4000 with resolution of less than 3. The scan speed varies from 1650-27000 for the respective mass ranges.</summary>
    MS_6330_Ion_Trap_LC_MS = 1000475,

    /// <summary>6340 Ion Trap LC/MS: The 6340 Ion Trap LC/MS is a Agilent liquid chromatography instrument combined with a 6300 series Agilent ion trap. It has a mass range of 50-2200 between 0.6 to 0.25 resolution and mass range of 200-4000 with resolution of less than 3. The scan speed varies from 1650-27000 for the respective mass ranges.</summary>
    MS_6340_Ion_Trap_LC_MS = 1000476,

    /// <summary>6410 Triple Quadrupole LC/MS: The 6410 Quadrupole LC/MS system is a Agilent liquid chromatography instrument combined with a Agilent triple quadrupole mass spectrometer. Mass range of the mass spectrometer is 15-1650 m/z, resolution is at three settings of 0.7 u (unit), 1.2 u (wide) and 2.5 u (widest). The mass accuracy for 6410 mass spectrometer is 0.1 across the mass range. The collision cell is a hexapole with linear acceleration.</summary>
    MS_6410_Triple_Quadrupole_LC_MS = 1000477,

    /// <summary>6410 Triple Quad LC/MS: The 6410 Quadrupole LC/MS system is a Agilent liquid chromatography instrument combined with a Agilent triple quadrupole mass spectrometer. Mass range of the mass spectrometer is 15-1650 m/z, resolution is at three settings of 0.7 u (unit), 1.2 u (wide) and 2.5 u (widest). The mass accuracy for 6410 mass spectrometer is 0.1 across the mass range. The collision cell is a hexapole with linear acceleration.</summary>
    MS_6410_Triple_Quad_LC_MS = MS_6410_Triple_Quadrupole_LC_MS,

    /// <summary>1200 series LC/MSD VL: The LC/MSD VL ion trap is part of the family of Agilent ion trap mass spectrometers. It has ESI, APCI and APPI ion sources and is a useful ion trap when the amount of sample is not the limiting factor.</summary>
    MS_1200_series_LC_MSD_VL = 1000478,

    /// <summary>purgatory: Terms that will likely become obsolete unless there are wails of dissent.</summary>
    MS_purgatory = 1000479,

    /// <summary>mass analyzer attribute: Analyzer properties that are associated with a value.</summary>
    MS_mass_analyzer_attribute = 1000480,

    /// <summary>detector attribute: Detector attribute recognized as a value.</summary>
    MS_detector_attribute = 1000481,

    /// <summary>source attribute: Property of a source device that need a value.</summary>
    MS_source_attribute = 1000482,

    /// <summary>Thermo Fisher Scientific instrument model: Thermo Fisher Scientific instrument model. The company has gone through several names including Thermo Finnigan, Thermo Scientific.</summary>
    MS_Thermo_Fisher_Scientific_instrument_model = 1000483,

    /// <summary>orbitrap: An ion trapping device that consists of an outer barrel-like electrode and a coaxial inner spindle-like electrode that form an electrostatic field with quadro-logarithmic potential distribution. The frequency of harmonic oscillations of the orbitally trapped ions along the axis of the electrostatic field is independent of the ion velocity and is inversely proportional to the square root of m/z so that the trap can be used as a mass analyzer.</summary>
    MS_orbitrap = 1000484,

    /// <summary>nanospray inlet: Nanospray Inlet.</summary>
    MS_nanospray_inlet = 1000485,

    /// <summary>source potential: Potential difference at the MS source in volts.</summary>
    MS_source_potential = 1000486,

    /// <summary>ion optics attribute: Ion optics involves components that help focus ion streams in mass spectrometry.</summary>
    MS_ion_optics_attribute = 1000487,

    /// <summary>Hitachi instrument model: Hitachi instrument model.</summary>
    MS_Hitachi_instrument_model = 1000488,

    /// <summary>Varian instrument model: Varian instrument model.</summary>
    MS_Varian_instrument_model = 1000489,

    /// <summary>Agilent instrument model: Agilent instrument model.</summary>
    MS_Agilent_instrument_model = 1000490,

    /// <summary>Dionex instrument model: Dionex instrument model.</summary>
    MS_Dionex_instrument_model = 1000491,

    /// <summary>Thermo Electron instrument model: Thermo Electron Corporation instrument model.</summary>
    MS_Thermo_Electron_instrument_model = 1000492,

    /// <summary>Finnigan MAT instrument model: Finnigan MAT instrument model.</summary>
    MS_Finnigan_MAT_instrument_model = 1000493,

    /// <summary>Thermo Scientific instrument model: Thermo Scientific instrument model.</summary>
    MS_Thermo_Scientific_instrument_model = 1000494,

    /// <summary>Applied Biosystems instrument model: Applied Biosystems instrument model.</summary>
    MS_Applied_Biosystems_instrument_model = 1000495,

    /// <summary>ABI: Applied Biosystems instrument model.</summary>
    MS_ABI = MS_Applied_Biosystems_instrument_model,

    /// <summary>instrument attribute: Instrument properties that are associated with a value.</summary>
    MS_instrument_attribute = 1000496,

    /// <summary>zoom scan: Special scan mode, where data with improved resoltuion is acquired. This is typically achieved by scanning a more narrow m/z window or scanning with a lower scan rate.</summary>
    MS_zoom_scan = 1000497,

    /// <summary>enhanced resolution scan: Special scan mode, where data with improved resoltuion is acquired. This is typically achieved by scanning a more narrow m/z window or scanning with a lower scan rate.</summary>
    MS_enhanced_resolution_scan = MS_zoom_scan,

    /// <summary>full scan: Feature of the ion trap mass spectrometer where MS data is acquired over a mass range.</summary>
    MS_full_scan = 1000498,

    /// <summary>spectrum attribute: Spectrum properties that are associated with a value.</summary>
    MS_spectrum_attribute = 1000499,

    /// <summary>scan window upper limit: The lower m/z bound of a mass spectrometer scan window.</summary>
    MS_scan_window_upper_limit = 1000500,

    /// <summary>scan window lower limit: The upper m/z bound of a mass spectrometer scan window.</summary>
    MS_scan_window_lower_limit = 1000501,

    /// <summary>dwell time: The time spent gathering data across a peak.</summary>
    MS_dwell_time = 1000502,

    /// <summary>scan attribute: Scan properties that are associated with a value.</summary>
    MS_scan_attribute = 1000503,

    /// <summary>base peak m/z: M/z value of the signal of highest intensity in the mass spectrum.</summary>
    MS_base_peak_m_z = 1000504,

    /// <summary>base peak intensity: The intensity of the greatest peak in the mass spectrum.</summary>
    MS_base_peak_intensity = 1000505,

    /// <summary>ion role: Ion Role.</summary>
    MS_ion_role = 1000506,

    /// <summary>ion attribute: Ion properties that are associated with a value.</summary>
    MS_ion_attribute = 1000507,

    /// <summary>ion chemical type: Ion Type.</summary>
    MS_ion_chemical_type = 1000508,

    /// <summary>activation energy: Activation Energy.</summary>
    MS_activation_energy = 1000509,

    /// <summary>precursor activation attribute: Precursor Activation Attribute.</summary>
    MS_precursor_activation_attribute = 1000510,

    /// <summary>ms level: Stages of ms achieved in a multi stage mass spectrometry experiment.</summary>
    MS_ms_level = 1000511,

    /// <summary>filter string: A string unique to Thermo instrument describing instrument settings for the scan.</summary>
    MS_filter_string = 1000512,

    /// <summary>binary data array: A data array of values.</summary>
    MS_binary_data_array = 1000513,

    /// <summary>m/z array: A data array of m/z values.</summary>
    MS_m_z_array = 1000514,

    /// <summary>intensity array: A data array of intensity values.</summary>
    MS_intensity_array = 1000515,

    /// <summary>charge array: A data array of charge values.</summary>
    MS_charge_array = 1000516,

    /// <summary>signal to noise array: A data array of signal-to-noise values.</summary>
    MS_signal_to_noise_array = 1000517,

    /// <summary>binary data type: Encoding type of binary data specifying the binary representation and precision, e.g. 64-bit float.</summary>
    MS_binary_data_type = 1000518,

    /// <summary>32-bit integer: Signed 32-bit little-endian integer.</summary>
    MS_32_bit_integer = 1000519,

    /// <summary>16-bit float: Signed 16-bit float.</summary>
    MS_16_bit_float_OBSOLETE = 1000520,

    /// <summary>32-bit float: 32-bit precision little-endian floating point conforming to IEEE-754.</summary>
    MS_32_bit_float = 1000521,

    /// <summary>64-bit integer: Signed 64-bit little-endian integer.</summary>
    MS_64_bit_integer = 1000522,

    /// <summary>64-bit float: 64-bit precision little-endian floating point conforming to IEEE-754.</summary>
    MS_64_bit_float = 1000523,

    /// <summary>data file content: Describes the data content on the file.</summary>
    MS_data_file_content = 1000524,

    /// <summary>spectrum representation: Way in which the spectrum is represented, either with regularly spaced data points or with a list of centroided peaks.</summary>
    MS_spectrum_representation = 1000525,

    /// <summary>Waters raw file: Waters data file found in a Waters RAW directory, generated from an MS acquisition.</summary>
    MS_Waters_raw_file = 1000526,

    /// <summary>highest observed m/z: Highest m/z value observed in the m/z array.</summary>
    MS_highest_observed_m_z = 1000527,

    /// <summary>lowest observed m/z: Lowest m/z value observed in the m/z array.</summary>
    MS_lowest_observed_m_z = 1000528,

    /// <summary>instrument serial number: Serial Number of the instrument.</summary>
    MS_instrument_serial_number = 1000529,

    /// <summary>file format conversion: Conversion of one file format to another.</summary>
    MS_file_format_conversion = 1000530,

    /// <summary>software: Software related to the recording or transformation of spectra.</summary>
    MS_software = 1000531,

    /// <summary>Xcalibur: Thermo Finnigan software for data acquisition and analysis.</summary>
    MS_Xcalibur = 1000532,

    /// <summary>Bioworks: Thermo Finnigan software for data analysis of peptides and proteins.</summary>
    MS_Bioworks = 1000533,

    /// <summary>MassLynx: Micromass software for data acquisition and analysis.</summary>
    MS_MassLynx = 1000534,

    /// <summary>FlexAnalysis: Bruker software for data analysis.</summary>
    MS_FlexAnalysis = 1000535,

    /// <summary>Data Explorer: Applied Biosystems software for data acquisition and analysis.</summary>
    MS_Data_Explorer = 1000536,

    /// <summary>4700 Explorer: Applied Biosystems software for data acquisition and analysis.</summary>
    MS_4700_Explorer = 1000537,

    /// <summary>massWolf: A software for converting Waters raw directory format to mzXML or mzML. massWolf was originally developed at the Institute for Systems Biology.</summary>
    MS_massWolf = 1000538,

    /// <summary>wolf: A software for converting Waters raw directory format to mzXML or mzML. massWolf was originally developed at the Institute for Systems Biology.</summary>
    MS_wolf = MS_massWolf,

    /// <summary>Voyager Biospectrometry Workstation System: Applied Biosystems MALDI-TOF data acquisition and analysis system.</summary>
    MS_Voyager_Biospectrometry_Workstation_System = 1000539,

    /// <summary>FlexControl: Bruker software for data acquisition.</summary>
    MS_FlexControl = 1000540,

    /// <summary>ReAdW: A software program for converting Thermo Finnigan RAW file format to mzXML or mzML. ReAdW was originally developed at the Institute for Systems Biology. Its whimsical interleaved spelling and capitalization is pronounced \"readraw\".</summary>
    MS_ReAdW = 1000541,

    /// <summary>MzStar: A software program for converting Applied Biosystems wiff file format to mzXML format. MzStar was originally developed at the Institute for Systems Biology. It is now obsoleted by the MzWiff program.</summary>
    MS_MzStar = 1000542,

    /// <summary>data processing action: Data processing attribute used to describe the type of data processing performed on the data file.</summary>
    MS_data_processing_action = 1000543,

    /// <summary>Conversion to mzML: Conversion of a file format to Proteomics Standards Initiative mzData file format.</summary>
    MS_Conversion_to_mzML = 1000544,

    /// <summary>Conversion to mzXML: Conversion of a file format to Institute of Systems Biology mzXML file format.</summary>
    MS_Conversion_to_mzXML = 1000545,

    /// <summary>Conversion to mzData: Conversion of a file format to Proteomics Standards Initiative mzData file format.</summary>
    MS_Conversion_to_mzData = 1000546,

    /// <summary>object attribute: Object Attribute.</summary>
    MS_object_attribute = 1000547,

    /// <summary>sample attribute: Sample properties that are associated with a value.</summary>
    MS_sample_attribute = 1000548,

    /// <summary>selection window attribute: Selection window properties that are associated with a value.</summary>
    MS_selection_window_attribute = 1000549,

    /// <summary>time unit: Time Unit.</summary>
    MS_time_unit_OBSOLETE = 1000550,

    /// <summary>Analyst: AB SCIEX or Applied Biosystems|MDS SCIEX software for data acquisition.</summary>
    MS_Analyst = 1000551,

    /// <summary>maldi spot identifier: Maldi Spot Identifier.</summary>
    MS_maldi_spot_identifier = 1000552,

    /// <summary>Trapper: A software program for converting Agilent MassHunter format to mzXML or mzML. Trapper was originally developed at the Institute for Systems Biology.</summary>
    MS_Trapper = 1000553,

    /// <summary>LCQ Deca: ThermoFinnigan LCQ Deca.</summary>
    MS_LCQ_Deca = 1000554,

    /// <summary>LTQ Orbitrap Discovery: LTQ Orbitrap Discovery.</summary>
    MS_LTQ_Orbitrap_Discovery = 1000555,

    /// <summary>LTQ Orbitrap XL: LTQ Orbitrap XL.</summary>
    MS_LTQ_Orbitrap_XL = 1000556,

    /// <summary>LTQ FT Ultra: LTQ FT Ultra.</summary>
    MS_LTQ_FT_Ultra = 1000557,

    /// <summary>GC Quantum: GC Quantum.</summary>
    MS_GC_Quantum = 1000558,

    /// <summary>spectrum type: Spectrum type.</summary>
    MS_spectrum_type = 1000559,

    /// <summary>mass spectrometer file format: The format of the file being used. This could be a instrument or vendor specific proprietary file format or a converted open file format.</summary>
    MS_mass_spectrometer_file_format = 1000560,

    /// <summary>data file checksum type: Checksum is a form of redundancy check, a simple way to protect the integrity of data by detecting errors in data.</summary>
    MS_data_file_checksum_type = 1000561,

    /// <summary>ABI WIFF file: Applied Biosystems WIFF file format.</summary>
    MS_ABI_WIFF_file = 1000562,

    /// <summary>Thermo RAW file: Thermo Scientific RAW file format.</summary>
    MS_Thermo_RAW_file = 1000563,

    /// <summary>PSI mzData file: Proteomics Standards Inititative mzData file format.</summary>
    MS_PSI_mzData_file = 1000564,

    /// <summary>Micromass PKL file: Micromass PKL file format.</summary>
    MS_Micromass_PKL_file = 1000565,

    /// <summary>ISB mzXML file: Institute of Systems Biology mzXML file format.</summary>
    MS_ISB_mzXML_file = 1000566,

    /// <summary>Bruker/Agilent YEP file: Bruker/Agilent YEP file format.</summary>
    MS_Bruker_Agilent_YEP_file = 1000567,

    /// <summary>MD5: MD5 (Message-Digest algorithm 5) is a cryptographic hash function with a 128-bit hash value used to check the integrity of files.</summary>
    MS_MD5 = 1000568,

    /// <summary>SHA-1: SHA-1 (Secure Hash Algorithm-1) is a cryptographic hash function designed by the National Security Agency (NSA) and published by the NIST as a U. S. government standard. It is also used to verify file integrity.</summary>
    MS_SHA_1 = 1000569,

    /// <summary>spectra combination: Method used to combine the mass spectra.</summary>
    MS_spectra_combination = 1000570,

    /// <summary>sum of spectra: Spectra Sum.</summary>
    MS_sum_of_spectra = 1000571,

    /// <summary>binary data compression type: Compression Type.</summary>
    MS_binary_data_compression_type = 1000572,

    /// <summary>median of spectra: Spectra is combined by calculating the median of the spectra.</summary>
    MS_median_of_spectra = 1000573,

    /// <summary>zlib compression: Zlib.</summary>
    MS_zlib_compression = 1000574,

    /// <summary>mean of spectra: Spectra is combined by calculating the mean of the spectra.</summary>
    MS_mean_of_spectra = 1000575,

    /// <summary>no compression: No Compression.</summary>
    MS_no_compression = 1000576,

    /// <summary>raw data file: Describes the type of file and its content.</summary>
    MS_raw_data_file = 1000577,

    /// <summary>source file: Describes the type of file and its content.</summary>
    MS_source_file = MS_raw_data_file,

    /// <summary>LCQ Fleet: LCQ Fleet.</summary>
    MS_LCQ_Fleet = 1000578,

    /// <summary>MS1 spectrum: Mass spectrum created by a single-stage MS experiment or the first stage of a multi-stage experiment.</summary>
    MS_MS1_spectrum = 1000579,

    /// <summary>full spectrum: Mass spectrum created by a single-stage MS experiment or the first stage of a multi-stage experiment.</summary>
    MS_full_spectrum = MS_MS1_spectrum,

    /// <summary>Q1 spectrum: Mass spectrum created by a single-stage MS experiment or the first stage of a multi-stage experiment.</summary>
    MS_Q1_spectrum = MS_MS1_spectrum,

    /// <summary>Q3 spectrum: Mass spectrum created by a single-stage MS experiment or the first stage of a multi-stage experiment.</summary>
    MS_Q3_spectrum = MS_MS1_spectrum,

    /// <summary>Single-Stage Mass Spectrometry: Mass spectrum created by a single-stage MS experiment or the first stage of a multi-stage experiment.</summary>
    MS_Single_Stage_Mass_Spectrometry = MS_MS1_spectrum,

    /// <summary>MSn spectrum: MSn refers to multi-stage MS/MS experiments designed to record product ion spectra where n is the number of product ion stages (progeny ions). For ion traps, sequential MS/MS experiments can be undertaken where n > 2 whereas for a simple triple quadrupole system n= 2.</summary>
    MS_MSn_spectrum = 1000580,

    /// <summary>multiple-stage mass spectrometry spectrum: MSn refers to multi-stage MS/MS experiments designed to record product ion spectra where n is the number of product ion stages (progeny ions). For ion traps, sequential MS/MS experiments can be undertaken where n > 2 whereas for a simple triple quadrupole system n= 2.</summary>
    MS_multiple_stage_mass_spectrometry_spectrum = MS_MSn_spectrum,

    /// <summary>nth generation product ion spectrum: MSn refers to multi-stage MS/MS experiments designed to record product ion spectra where n is the number of product ion stages (progeny ions). For ion traps, sequential MS/MS experiments can be undertaken where n > 2 whereas for a simple triple quadrupole system n= 2.</summary>
    MS_nth_generation_product_ion_spectrum = MS_MSn_spectrum,

    /// <summary>product ion spectrum: MSn refers to multi-stage MS/MS experiments designed to record product ion spectra where n is the number of product ion stages (progeny ions). For ion traps, sequential MS/MS experiments can be undertaken where n > 2 whereas for a simple triple quadrupole system n= 2.</summary>
    MS_product_ion_spectrum = MS_MSn_spectrum,

    /// <summary>CRM spectrum: Spectrum generated from MSn experiment with three or more stages of m/z separation and in which a particular multi-step reaction path is monitored.</summary>
    MS_CRM_spectrum = 1000581,

    /// <summary>SIM spectrum: Spectrum obtained with the operation of a mass spectrometer in which the abundances of one ion or several ions of specific m/z values are recorded rather than the entire mass spectrum (Selected Ion Monitoring).</summary>
    MS_SIM_spectrum = 1000582,

    /// <summary>MIM spectrum: Spectrum obtained with the operation of a mass spectrometer in which the abundances of one ion or several ions of specific m/z values are recorded rather than the entire mass spectrum (Selected Ion Monitoring).</summary>
    MS_MIM_spectrum = MS_SIM_spectrum,

    /// <summary>multiple ion monitoring spectrum: Spectrum obtained with the operation of a mass spectrometer in which the abundances of one ion or several ions of specific m/z values are recorded rather than the entire mass spectrum (Selected Ion Monitoring).</summary>
    MS_multiple_ion_monitoring_spectrum = MS_SIM_spectrum,

    /// <summary>selected ion monitoring spectrum: Spectrum obtained with the operation of a mass spectrometer in which the abundances of one ion or several ions of specific m/z values are recorded rather than the entire mass spectrum (Selected Ion Monitoring).</summary>
    MS_selected_ion_monitoring_spectrum = MS_SIM_spectrum,

    /// <summary>SRM spectrum: Spectrum obtained when data are acquired from specific product ions corresponding to m/z values of selected precursor ions a recorded via two or more stages of mass spectrometry. The precursor/product ion pair is called a transition pair.  Data can be obtained for a single transition pair or multiple transition pairs.  Multiple time segments of different transition pairs can exist in a single file. Single precursor ions can have multiple product ions consitituting multiple transition pairs.  Selected reaction monitoring can be performed as tandem mass spectrometry in time or tandem mass spectrometry in space.</summary>
    MS_SRM_spectrum = 1000583,

    /// <summary>MRM spectrum: Spectrum obtained when data are acquired from specific product ions corresponding to m/z values of selected precursor ions a recorded via two or more stages of mass spectrometry. The precursor/product ion pair is called a transition pair.  Data can be obtained for a single transition pair or multiple transition pairs.  Multiple time segments of different transition pairs can exist in a single file. Single precursor ions can have multiple product ions consitituting multiple transition pairs.  Selected reaction monitoring can be performed as tandem mass spectrometry in time or tandem mass spectrometry in space.</summary>
    MS_MRM_spectrum = MS_SRM_spectrum,

    /// <summary>multiple reaction monitoring spectrum: Spectrum obtained when data are acquired from specific product ions corresponding to m/z values of selected precursor ions a recorded via two or more stages of mass spectrometry. The precursor/product ion pair is called a transition pair.  Data can be obtained for a single transition pair or multiple transition pairs.  Multiple time segments of different transition pairs can exist in a single file. Single precursor ions can have multiple product ions consitituting multiple transition pairs.  Selected reaction monitoring can be performed as tandem mass spectrometry in time or tandem mass spectrometry in space.</summary>
    MS_multiple_reaction_monitoring_spectrum = MS_SRM_spectrum,

    /// <summary>selected reaction monitoring spectrum: Spectrum obtained when data are acquired from specific product ions corresponding to m/z values of selected precursor ions a recorded via two or more stages of mass spectrometry. The precursor/product ion pair is called a transition pair.  Data can be obtained for a single transition pair or multiple transition pairs.  Multiple time segments of different transition pairs can exist in a single file. Single precursor ions can have multiple product ions consitituting multiple transition pairs.  Selected reaction monitoring can be performed as tandem mass spectrometry in time or tandem mass spectrometry in space.</summary>
    MS_selected_reaction_monitoring_spectrum = MS_SRM_spectrum,

    /// <summary>mzML file: Proteomics Standards Inititative mzML file format.</summary>
    MS_mzML_file = 1000584,

    /// <summary>contact attribute: Details about a person or organization to contact in case of concern or discussion about the file.</summary>
    MS_contact_attribute = 1000585,

    /// <summary>contact name: Name of the contact person or organization.</summary>
    MS_contact_name = 1000586,

    /// <summary>contact address: Postal address of the contact person or organization.</summary>
    MS_contact_address = 1000587,

    /// <summary>contact URL: Uniform Resource Locator related to the contact person or organization.</summary>
    MS_contact_URL = 1000588,

    /// <summary>contact email: Email address of the contact person or organization.</summary>
    MS_contact_email = 1000589,

    /// <summary>contact organization: Home institution of the contact person.</summary>
    MS_contact_organization = 1000590,

    /// <summary>MzWiff: A software program for converting Applied Biosystems wiff file format to the mzXML or mzML format. MzWiff is currently maintained at the Institute for Systems Biology. It replaces the slower mzStar program.</summary>
    MS_MzWiff = 1000591,

    /// <summary>smoothing: A process of reducing spikes of intensity in order to reduce noise while preserving real peak signal. Many algorithms can be applied for this process.</summary>
    MS_smoothing = 1000592,

    /// <summary>baseline reduction: A process of removal of varying intensities generated due to variable energy absorption before further processing can take place. Baseline reduction facilitates meaningful comparision between intensities of m/z values.</summary>
    MS_baseline_reduction = 1000593,

    /// <summary>low intensity data point removal: The removal of very low intensity data points that are likely to be spurious noise rather than real signal.</summary>
    MS_low_intensity_data_point_removal = 1000594,

    /// <summary>thresholding: The removal of very low intensity data points that are likely to be spurious noise rather than real signal.</summary>
    MS_thresholding = MS_low_intensity_data_point_removal,

    /// <summary>time array: A data array of relative time offset values from a reference time.</summary>
    MS_time_array = 1000595,

    /// <summary>measurement method: An attribute of resolution when recording the detector response in absence of the analyte.</summary>
    MS_measurement_method = 1000596,

    /// <summary>ion optics type: The electrical potential used to impart kinetic energy to ions in a mass spectrometer.</summary>
    MS_ion_optics_type = 1000597,

    /// <summary>electron transfer dissociation: A process to fragment ions in a mass spectrometer by inducing fragmentation of cations (e.g. peptides or proteins) by transferring electrons to them.</summary>
    MS_electron_transfer_dissociation = 1000598,

    /// <summary>ETD: A process to fragment ions in a mass spectrometer by inducing fragmentation of cations (e.g. peptides or proteins) by transferring electrons to them.</summary>
    MS_ETD = MS_electron_transfer_dissociation,

    /// <summary>pulsed q dissociation: A process that involves precursor ion activation at high Q, a time delay to allow the precursor to fragment, then a rapid pulse to low Q where all fragment ions are trapped. The product ions can then be scanned out of the ion trap and detected.</summary>
    MS_pulsed_q_dissociation = 1000599,

    /// <summary>PQD: A process that involves precursor ion activation at high Q, a time delay to allow the precursor to fragment, then a rapid pulse to low Q where all fragment ions are trapped. The product ions can then be scanned out of the ion trap and detected.</summary>
    MS_PQD = MS_pulsed_q_dissociation,

    /// <summary>Proteios: Database application and analysis platform for proteomics.</summary>
    MS_Proteios = 1000600,

    /// <summary>ProteinLynx Global Server: Waters software for data analysis.</summary>
    MS_ProteinLynx_Global_Server = 1000601,

    /// <summary>Shimadzu Biotech instrument model: Shimadzu Biotech instrument model.</summary>
    MS_Shimadzu_Biotech_instrument_model = 1000602,

    /// <summary>Shimadzu Scientific Instruments instrument model: Shimadzu Scientific Instruments instrument model.</summary>
    MS_Shimadzu_Scientific_Instruments_instrument_model = 1000603,

    /// <summary>LCMS-IT-TOF: Shimadzu Scientific Instruments LCMS-IT-TOF MS.</summary>
    MS_LCMS_IT_TOF = 1000604,

    /// <summary>LCMS-2010EV: Shimadzu Scientific Instruments LCMS-2010EV MS.</summary>
    MS_LCMS_2010EV = 1000605,

    /// <summary>LCMS-2010A: Shimadzu Scientific Instruments LCMS-2010A MS.</summary>
    MS_LCMS_2010A = 1000606,

    /// <summary>AXIMA CFR MALDI-TOF: Shimadzu Biotech AXIMA CFR MALDI-TOF MS.</summary>
    MS_AXIMA_CFR_MALDI_TOF = 1000607,

    /// <summary>AXIMA-QIT: Shimadzu Biotech AXIMA-QIT MS.</summary>
    MS_AXIMA_QIT = 1000608,

    /// <summary>AXIMA-CFR plus: Shimadzu Biotech AXIMA-CFR plus MS.</summary>
    MS_AXIMA_CFR_plus = 1000609,

    /// <summary>AXIMA Performance MALDI-TOF/TOF: Shimadzu Biotech AXIMA Performance MALDI-TOF/TOF MS.</summary>
    MS_AXIMA_Performance_MALDI_TOF_TOF = 1000610,

    /// <summary>AXIMA Confidence MALDI-TOF: Shimadzu Biotech AXIMA Confidence MALDI-TOF (curved field reflectron) MS.</summary>
    MS_AXIMA_Confidence_MALDI_TOF = 1000611,

    /// <summary>AXIMA Assurance Linear MALDI-TOF: Shimadzu Biotech AXIMA Assurance Linear MALDI-TOF MS.</summary>
    MS_AXIMA_Assurance_Linear_MALDI_TOF = 1000612,

    /// <summary>DTA file: Sequest DTA file format.</summary>
    MS_DTA_file = 1000613,

    /// <summary>ProteinLynx Global Server mass spectrum XML file: Peak list file format used by ProteinLynx Global Server.</summary>
    MS_ProteinLynx_Global_Server_mass_spectrum_XML_file = 1000614,

    /// <summary>ProteoWizard: ProteoWizard software for data processing and analysis. Primarily developed by the labs of P. Malick and D. Tabb.</summary>
    MS_ProteoWizard = 1000615,

    /// <summary>pwiz: ProteoWizard software for data processing and analysis. Primarily developed by the labs of P. Malick and D. Tabb.</summary>
    MS_pwiz = MS_ProteoWizard,

    /// <summary>preset scan configuration: A user-defined scan configuration that specifies the instrumental settings in which a spectrum is acquired. An instrument may cycle through a list of preset scan configurations to acquire data. This is a more generic term for the Thermo \"scan event\", which is defined in the Thermo Xcalibur glossary as: a mass spectrometer scan that is defined by choosing the necessary scan parameter settings. Multiple scan events can be defined for each segment of time.</summary>
    MS_preset_scan_configuration = 1000616,

    /// <summary>wavelength array: A data array of electromagnetic radiation wavelength values.</summary>
    MS_wavelength_array = 1000617,

    /// <summary>highest observed wavelength: Highest wavelength observed in an EMR spectrum.</summary>
    MS_highest_observed_wavelength = 1000618,

    /// <summary>lowest observed wavelength: Lowest wavelength observed in an EMR spectrum.</summary>
    MS_lowest_observed_wavelength = 1000619,

    /// <summary>PDA spectrum: Spectrum generated from a photodiode array detector (ultraviolet/visible spectrum).</summary>
    MS_PDA_spectrum_OBSOLETE = 1000620,

    /// <summary>photodiode array detector: An array detector used to record spectra in the ultraviolet and visable region of light.</summary>
    MS_photodiode_array_detector = 1000621,

    /// <summary>PDA: An array detector used to record spectra in the ultraviolet and visable region of light.</summary>
    MS_PDA = MS_photodiode_array_detector,

    /// <summary>Surveyor PDA: Surveyor PDA.</summary>
    MS_Surveyor_PDA = 1000622,

    /// <summary>Accela PDA: Accela PDA.</summary>
    MS_Accela_PDA = 1000623,

    /// <summary>inductive detector: Inductive detector.</summary>
    MS_inductive_detector = 1000624,

    /// <summary>image current detector: Inductive detector.</summary>
    MS_image_current_detector = MS_inductive_detector,

    /// <summary>chromatogram: The representation of detector response versus time.</summary>
    MS_chromatogram = 1000625,

    /// <summary>chromatogram type: Broad category or type of a chromatogram.</summary>
    MS_chromatogram_type = 1000626,

    /// <summary>selected ion current chromatogram: Chromatogram created by creating an array of the measurements of a specific single ion current at each time point.</summary>
    MS_selected_ion_current_chromatogram = 1000627,

    /// <summary>SIC chromatogram: Chromatogram created by creating an array of the measurements of a specific single ion current at each time point.</summary>
    MS_SIC_chromatogram = MS_selected_ion_current_chromatogram,

    /// <summary>basepeak chromatogram: Chromatogram created by creating an array of the most intense peaks at each time point.</summary>
    MS_basepeak_chromatogram = 1000628,

    /// <summary>low intensity threshold: Threshold below which some action is taken.</summary>
    MS_low_intensity_threshold = 1000629,

    /// <summary>data processing parameter: Data processing parameter used in the data processing performed on the data file.</summary>
    MS_data_processing_parameter = 1000630,

    /// <summary>high intensity threshold: Threshold above which some action is taken.</summary>
    MS_high_intensity_threshold = 1000631,

    /// <summary>Q-Tof Premier: Waters Q-Tof Premier MS.</summary>
    MS_Q_Tof_Premier = 1000632,

    /// <summary>possible charge state: A possible charge state of the ion in a situation where the charge of an ion is known to be one of several possible values rather than a completely unknown value or determined to be a specific charge with reasonable certainty.</summary>
    MS_possible_charge_state = 1000633,

    /// <summary>DSQ: ThermoFinnigan DSQ GC-MS.</summary>
    MS_DSQ = 1000634,

    /// <summary>ITQ 700: Thermo Scientific ITQ 700 GC-MS.</summary>
    MS_ITQ_700 = 1000635,

    /// <summary>ITQ 900: Thermo Scientific ITQ 900 GC-MS.</summary>
    MS_ITQ_900 = 1000636,

    /// <summary>ITQ 1100: Thermo Scientific ITQ 1100 GC-MS.</summary>
    MS_ITQ_1100 = 1000637,

    /// <summary>LTQ XL ETD: Thermo Scientific LTQ XL MS with ETD.</summary>
    MS_LTQ_XL_ETD = 1000638,

    /// <summary>LTQ Orbitrap XL ETD: Thermo Scientific LTQ Orbitrap XL MS with ETD.</summary>
    MS_LTQ_Orbitrap_XL_ETD = 1000639,

    /// <summary>DFS: Thermo Scientific DFS HR GC-MS.</summary>
    MS_DFS = 1000640,

    /// <summary>DSQ II: Thermo Scientific DSQ II GC-MS.</summary>
    MS_DSQ_II = 1000641,

    /// <summary>MALDI LTQ XL: Thermo Scientific MALDI LTQ XL MS.</summary>
    MS_MALDI_LTQ_XL = 1000642,

    /// <summary>MALDI LTQ Orbitrap: Thermo Scientific MALDI LTQ Orbitrap MS.</summary>
    MS_MALDI_LTQ_Orbitrap = 1000643,

    /// <summary>TSQ Quantum Access: Thermo Scientific TSQ Quantum Access MS.</summary>
    MS_TSQ_Quantum_Access = 1000644,

    /// <summary>Element XR: Thermo Scientific Element XR HR-ICP-MS.</summary>
    MS_Element_XR = 1000645,

    /// <summary>Element 2: Thermo Scientific Element 2 HR-ICP-MS.</summary>
    MS_Element_2 = 1000646,

    /// <summary>Element GD: Thermo Scientific Element GD Glow Discharge MS.</summary>
    MS_Element_GD = 1000647,

    /// <summary>GC IsoLink: Thermo Scientific GC IsoLink Isotope Ratio MS.</summary>
    MS_GC_IsoLink = 1000648,

    /// <summary>Exactive: Thermo Scientific Exactive MS.</summary>
    MS_Exactive = 1000649,

    /// <summary>Proteome Discoverer: Thermo Scientific software for data analysis of peptides and proteins.</summary>
    MS_Proteome_Discoverer = 1000650,

    /// <summary>3200 QTRAP: AB SCIEX or Applied Biosystems|MDS SCIEX QTRAP 3200.</summary>
    MS_3200_QTRAP = 1000651,

    /// <summary>4800 Plus MALDI TOF/TOF: AB SCIEX or Applied Biosystems|MDS SCIEX 4800 Plus MALDI TOF-TOF Analyzer.</summary>
    MS_4800_Plus_MALDI_TOF_TOF = 1000652,

    /// <summary>API 3200: AB SCIEX or Applied Biosystems|MDS SCIEX API 3200 MS.</summary>
    MS_API_3200 = 1000653,

    /// <summary>API 5000: AB SCIEX or Applied Biosystems|MDS SCIEX API 5000 MS.</summary>
    MS_API_5000 = 1000654,

    /// <summary>QSTAR Elite: AB SCIEX or Applied Biosystems|MDS SCIEX QSTAR Elite.</summary>
    MS_QSTAR_Elite = 1000655,

    /// <summary>QSTAR Pulsar: Applied Biosystems|MDS SCIEX QSTAR Pulsar.</summary>
    MS_QSTAR_Pulsar = 1000656,

    /// <summary>QSTAR XL: Applied Biosystems|MDS SCIEX QSTAR XL.</summary>
    MS_QSTAR_XL = 1000657,

    /// <summary>4800 Proteomics Analyzer: Applied Biosystems|MDS SCIEX 4800 Proteomics Analyzer.</summary>
    MS_4800_Proteomics_Analyzer = 1000658,

    /// <summary>4000 Series Explorer Software: AB SCIEX or Applied Biosystems software for data acquisition and analysis.</summary>
    MS_4000_Series_Explorer_Software = 1000659,

    /// <summary>GPS Explorer: AB SCIEX or Applied Biosystems software for data acquisition and analysis.</summary>
    MS_GPS_Explorer = 1000661,

    /// <summary>LightSight Software: AB SCIEX or Applied Biosystems|MDS SCIEX software metabolite identification.</summary>
    MS_LightSight_Software = 1000662,

    /// <summary>ProteinPilot Software: AB SCIEX or Applied Biosystems|MDS SCIEX software for protein ID and quant.</summary>
    MS_ProteinPilot_Software = 1000663,

    /// <summary>TissueView Software: Applied Biosystems|MDS SCIEX software for tissue imaging.</summary>
    MS_TissueView_Software = 1000664,

    /// <summary>MarkerView Software: Applied Biosystems|MDS SCIEX software for metabolomics and biomarker profiling.</summary>
    MS_MarkerView_Software = 1000665,

    /// <summary>MRMPilot Software: Applied Biosystems|MDS SCIEX software for MRM assay development.</summary>
    MS_MRMPilot_Software = 1000666,

    /// <summary>BioAnalyst: Applied Biosystems|MDS SCIEX software for bio-related data exploration.</summary>
    MS_BioAnalyst = 1000667,

    /// <summary>Pro ID: Applied Biosystems|MDS SCIEX software for protein identification.</summary>
    MS_Pro_ID = 1000668,

    /// <summary>Pro ICAT: Applied Biosystems|MDS SCIEX software for protein ID and quant by ICAT.</summary>
    MS_Pro_ICAT = 1000669,

    /// <summary>Pro Quant: Applied Biosystems|MDS SCIEX software for protein ID and quant by iTRAQ.</summary>
    MS_Pro_Quant = 1000670,

    /// <summary>Pro BLAST: Applied Biosystems|MDS SCIEX software for MS-BLAST identification.</summary>
    MS_Pro_BLAST = 1000671,

    /// <summary>Cliquid: AB SCIEX Cliquid software for data analysis and quantitation.</summary>
    MS_Cliquid = 1000672,

    /// <summary>MIDAS Workflow Designer: Applied Biosystems|MDS SCIEX software for MRM assay development.</summary>
    MS_MIDAS_Workflow_Designer = 1000673,

    /// <summary>MultiQuant: Applied Biosystems|MDS SCIEX software for MRM-based quantitation.</summary>
    MS_MultiQuant = 1000674,

    /// <summary>6220 Time-of-Flight LC/MS: The 6220 Time-of-Flight LC/MS is a Agilent liquid chromatography instrument combined with a Agilent time of flight mass spectrometer. This time of flight mass spectrometer has a m/z range of 50-12000, mass accuracy of less than 2 ppm and resolution greater than 13,000 at m/z 2722. It has multiple ion sources and can be used with multimode ion sources.</summary>
    MS_6220_Time_of_Flight_LC_MS = 1000675,

    /// <summary>6510 Quadrupole Time-of-Flight LC/MS: The 6510 Quadrupole Time-of-Flight LC/MS is a Agilent liquid chromatography instrument combined with a Agilent time of flight mass spectrometer. This time of flight mass spectrometer has a m/z range of 50-12000, mass accuracy of less than 2 ppm and resolution greater than 13,000 at m/z 2722. It has multiple ion sources and can be used with multimode ion sources.</summary>
    MS_6510_Quadrupole_Time_of_Flight_LC_MS = 1000676,

    /// <summary>6520 Quadrupole Time-of-Flight LC/MS: The 6520 Quadrupole Time-of-Flight LC/MS is a Agilent liquid chromatography instrument combined with a Agilent time of flight mass spectrometer. This time of flight mass spectrometer has a m/z range of 50-12000, mass accuracy of less than 2 ppm and resolution greater than 26,000 at m/z 2722. It has multiple ion sources and can be used with multimode ion sources.</summary>
    MS_6520_Quadrupole_Time_of_Flight_LC_MS = 1000677,

    /// <summary>MassHunter Data Acquisition: Software for data acquisition of 6000 series instruments.</summary>
    MS_MassHunter_Data_Acquisition = 1000678,

    /// <summary>MassHunter Easy Access: Software for open access data acquisition.</summary>
    MS_MassHunter_Easy_Access = 1000679,

    /// <summary>MassHunter Qualitative Analysis: Software for data analysis of data from 6000 series instruments.</summary>
    MS_MassHunter_Qualitative_Analysis = 1000680,

    /// <summary>MassHunter Quantitative Analysis: Software for quantitation of Triple Quadruople and Quadrupole Time-of-Flight data.</summary>
    MS_MassHunter_Quantitative_Analysis = 1000681,

    /// <summary>MassHunter Metabolite ID: Software for identification of metabolites.</summary>
    MS_MassHunter_Metabolite_ID = 1000682,

    /// <summary>MassHunter BioConfirm: Software for protein characterization.</summary>
    MS_MassHunter_BioConfirm = 1000683,

    /// <summary>Genespring MS: Software for quantitation and statistical analysis of TOF and Q-TOF LC/MS data.</summary>
    MS_Genespring_MS = 1000684,

    /// <summary>MassHunter Mass Profiler: Software for quantitation and statistical analysis of TOF and Q-TOF LC/MS data.</summary>
    MS_MassHunter_Mass_Profiler = 1000685,

    /// <summary>METLIN: Personal Metabolite Database for MassHunter Workstation. Software for identification of human metabolites.</summary>
    MS_METLIN = 1000686,

    /// <summary>Spectrum Mill for MassHunter Workstation: Software for protein identification and characterization of complex protein digest mixtures.</summary>
    MS_Spectrum_Mill_for_MassHunter_Workstation = 1000687,

    /// <summary>6300 Series Ion Trap Data Analysis Software: Software for data analysis of 6300 series ion trap mass spectrometers.</summary>
    MS_6300_Series_Ion_Trap_Data_Analysis_Software = 1000688,

    /// <summary>Agilent software: Agilent software for data acquisition and analysis.</summary>
    MS_Agilent_software = 1000689,

    /// <summary>AB SCIEX software: AB SCIEX or Applied Biosystems software for data acquisition and analysis.</summary>
    MS_AB_SCIEX_software = 1000690,

    /// <summary>Applied Biosystems software: Applied Biosystems|MDS SCIEX software for data acquisition and analysis.</summary>
    MS_Applied_Biosystems_software = 1000691,

    /// <summary>Bruker software: Bruker software for data acquisition and analysis.</summary>
    MS_Bruker_software = 1000692,

    /// <summary>Thermo Finnigan software: Thermo Finnigan software for data acquisition and analysis.</summary>
    MS_Thermo_Finnigan_software = 1000693,

    /// <summary>Waters software: Waters software for data acquisition and analysis.</summary>
    MS_Waters_software = 1000694,

    /// <summary>apex ultra: Bruker Daltonics' apex ultra: ESI, MALDI, Nanospray, APCI, APPI, Qh-FT_ICR.</summary>
    MS_apex_ultra = 1000695,

    /// <summary>autoflex III smartbeam: Bruker Daltonics' autoflex III smartbeam: MALDI TOF.</summary>
    MS_autoflex_III_smartbeam = 1000696,

    /// <summary>Bruker Daltonics HCT Series: Bruker Daltonics HCT Series.</summary>
    MS_Bruker_Daltonics_HCT_Series = 1000697,

    /// <summary>HCTultra: Bruker Daltonics' HCTultra: ESI TOF, Nanospray, APCI, APPI.</summary>
    MS_HCTultra = 1000698,

    /// <summary>HCTultra PTM: Bruker Daltonics' HCTultra PTM: ESI TOF, Nanospray, APCI, APPI, PTR.</summary>
    MS_HCTultra_PTM = 1000699,

    /// <summary>HCTultra ETD II: Bruker Daltonics' HCTultra ETD II: ESI Q-TOF, Nanospray, APCI, APPI, ETD.</summary>
    MS_HCTultra_ETD_II = 1000700,

    /// <summary>microflex LT: Bruker Daltonics' microflex LT: MALDI TOF.</summary>
    MS_microflex_LT = 1000701,

    /// <summary>micrOTOF: Bruker Daltonics' micrOTOF: ESI TOF, APCI, APPI.</summary>
    MS_micrOTOF = 1000702,

    /// <summary>micrOTOF-Q: Bruker Daltonics' micrOTOF-Q: ESI Q-TOF, Nanospray, APCI, APPI.</summary>
    MS_micrOTOF_Q = 1000703,

    /// <summary>micrOTOF-Q II: Bruker Daltonics' micrOTOF-Q II: ESI Q-TOF, Nanospray, APCI, APPI.</summary>
    MS_micrOTOF_Q_II = 1000704,

    /// <summary>ultraflex III TOF/TOF: Bruker Daltonics' ultraflex III TOF/TOF: MALDI TOF.</summary>
    MS_ultraflex_III_TOF_TOF = 1000705,

    /// <summary>apexControl: Bruker software for data acquisition.</summary>
    MS_apexControl = 1000706,

    /// <summary>BioTools: Bruker software for data analysis.</summary>
    MS_BioTools = 1000707,

    /// <summary>CLINPROT: Bruker CLINPROT software.</summary>
    MS_CLINPROT = 1000708,

    /// <summary>CLINPROT micro: Bruker CLINPROT micro software.</summary>
    MS_CLINPROT_micro = 1000709,

    /// <summary>CLINPROT robot: Bruker CLINPROT robot software.</summary>
    MS_CLINPROT_robot = 1000710,

    /// <summary>ClinProTools: Bruker ClinProTools software.</summary>
    MS_ClinProTools = 1000711,

    /// <summary>Compass: Bruker Compass software.</summary>
    MS_Compass = 1000712,

    /// <summary>Compass for HCT/esquire: Bruker Compass for HCT/esquire software.</summary>
    MS_Compass_for_HCT_esquire = 1000713,

    /// <summary>Compass for micrOTOF: Bruker Compass for micrOTOF software.</summary>
    MS_Compass_for_micrOTOF = 1000714,

    /// <summary>Compass OpenAccess: Bruker compass OpenAccess software.</summary>
    MS_Compass_OpenAccess = 1000715,

    /// <summary>Compass Security Pack: Bruker compass Security Pack software.</summary>
    MS_Compass_Security_Pack = 1000716,

    /// <summary>CompassXport: Bruker stand-alone software for data conversion.</summary>
    MS_CompassXport = 1000717,

    /// <summary>CompassXtract: Bruker software library for data access.</summary>
    MS_CompassXtract = 1000718,

    /// <summary>DataAnalysis: Bruker software for data analysis.</summary>
    MS_DataAnalysis = 1000719,

    /// <summary>dpControl: Bruker software for data acquisition.</summary>
    MS_dpControl = 1000720,

    /// <summary>esquireControl: Bruker software for data acquisition.</summary>
    MS_esquireControl = 1000721,

    /// <summary>flexImaging: Bruker software for data analysis.</summary>
    MS_flexImaging = 1000722,

    /// <summary>GENOLINK: Bruker GENOLINK software.</summary>
    MS_GENOLINK = 1000723,

    /// <summary>GenoTools: Bruker GenoTools software.</summary>
    MS_GenoTools = 1000724,

    /// <summary>HCTcontrol: Bruker software for data acquisition.</summary>
    MS_HCTcontrol = 1000725,

    /// <summary>micrOTOFcontrol: Bruker software for data acquisition.</summary>
    MS_micrOTOFcontrol = 1000726,

    /// <summary>PolyTools: Bruker PolyTools software.</summary>
    MS_PolyTools = 1000727,

    /// <summary>ProfileAnalysis: Bruker software for data analysis.</summary>
    MS_ProfileAnalysis = 1000728,

    /// <summary>PROTEINEER: Bruker PROTEINEER software.</summary>
    MS_PROTEINEER = 1000729,

    /// <summary>PROTEINEER dp: Bruker PROTEINEER dp software.</summary>
    MS_PROTEINEER_dp = 1000730,

    /// <summary>PROTEINEER fc: Bruker PROTEINEER fc software.</summary>
    MS_PROTEINEER_fc = 1000731,

    /// <summary>PROTEINEER spII: Bruker PROTEINEER spII software.</summary>
    MS_PROTEINEER_spII = 1000732,

    /// <summary>PROTEINEER-LC: Bruker PROTEINEER-LC software.</summary>
    MS_PROTEINEER_LC = 1000733,

    /// <summary>ProteinScape: Bruker ProteinScape software.</summary>
    MS_ProteinScape = 1000734,

    /// <summary>PureDisk: BrukerPureDisk software.</summary>
    MS_PureDisk = 1000735,

    /// <summary>QuantAnalysis: Bruker software for data analysis.</summary>
    MS_QuantAnalysis = 1000736,

    /// <summary>spControl: Bruker software for data acquisition.</summary>
    MS_spControl = 1000737,

    /// <summary>TargetAnalysis: Bruker TargetAnalysis software.</summary>
    MS_TargetAnalysis = 1000738,

    /// <summary>WARP-LC: Bruker WARP-LC software.</summary>
    MS_WARP_LC = 1000739,

    /// <summary>parameter file: Parameter file used to configure the acquisition of raw data on the instrument.</summary>
    MS_parameter_file = 1000740,

    /// <summary>Conversion to dta: Conversion to dta format.</summary>
    MS_Conversion_to_dta = 1000741,

    /// <summary>Bioworks SRF file: Thermo Finnigan SRF file format.</summary>
    MS_Bioworks_SRF_file = 1000742,

    /// <summary>TSQ Quantum Ultra AM: Thermo Scientific TSQ Quantum Ultra AM.</summary>
    MS_TSQ_Quantum_Ultra_AM = 1000743,

    /// <summary>selected ion m/z: Mass-to-charge ratio of the ion selected for fragmentation.</summary>
    MS_selected_ion_m_z = 1000744,

    /// <summary>retention time alignment: The correction of the spectrum scan times, as used e.g. in label-free proteomics.</summary>
    MS_retention_time_alignment = 1000745,

    /// <summary>high intensity data point removal: The removal of very high intensity data points.</summary>
    MS_high_intensity_data_point_removal = 1000746,

    /// <summary>completion time: The time that a data processing action was finished.</summary>
    MS_completion_time = 1000747,

    /// <summary>SSQ 7000: ThermoFinnigan SSQ 7000 MS.</summary>
    MS_SSQ_7000 = 1000748,

    /// <summary>TSQ 7000: ThermoFinnigan TSQ 7000 MS.</summary>
    MS_TSQ_7000 = 1000749,

    /// <summary>TSQ: ThermoFinnigan TSQ MS.</summary>
    MS_TSQ = 1000750,

    /// <summary>TSQ Quantum Ultra: Thermo Scientific TSQ Quantum Ultra.</summary>
    MS_TSQ_Quantum_Ultra = 1000751,

    /// <summary>TOPP software: TOPP (The OpenMS proteomics pipeline) software.</summary>
    MS_TOPP_software = 1000752,

    /// <summary>BaselineFilter: Removes the baseline from profile spectra using a top-hat filter.</summary>
    MS_BaselineFilter = 1000753,

    /// <summary>DBExporter: Exports data from an OpenMS database to a file.</summary>
    MS_DBExporter = 1000754,

    /// <summary>DBImporter: Imports data to an OpenMS database.</summary>
    MS_DBImporter = 1000755,

    /// <summary>FileConverter: Converts between different MS file formats.</summary>
    MS_FileConverter = 1000756,

    /// <summary>FileFilter: Extracts or manipulates portions of data from peak, feature or consensus feature files.</summary>
    MS_FileFilter = 1000757,

    /// <summary>FileMerger: Merges several MS files into one file.</summary>
    MS_FileMerger = 1000758,

    /// <summary>InternalCalibration: Applies an internal calibration.</summary>
    MS_InternalCalibration = 1000759,

    /// <summary>MapAligner: Corrects retention time distortions between maps.</summary>
    MS_MapAligner = 1000760,

    /// <summary>MapNormalizer: Normalizes peak intensities in an MS run.</summary>
    MS_MapNormalizer = 1000761,

    /// <summary>NoiseFilter: Removes noise from profile spectra by using different smoothing techniques.</summary>
    MS_NoiseFilter = 1000762,

    /// <summary>PeakPicker: Finds mass spectrometric peaks in profile mass spectra.</summary>
    MS_PeakPicker = 1000763,

    /// <summary>Resampler: Transforms an LC/MS map into a resampled map or a png image.</summary>
    MS_Resampler = 1000764,

    /// <summary>SpectraFilter: Applies a filter to peak spectra.</summary>
    MS_SpectraFilter = 1000765,

    /// <summary>TOFCalibration: Applies time of flight calibration.</summary>
    MS_TOFCalibration = 1000766,

    /// <summary>native spectrum identifier format: Describes how the native spectrum identifiers are formated.</summary>
    MS_native_spectrum_identifier_format = 1000767,

    /// <summary>nativeID format: Describes how the native spectrum identifiers are formated.</summary>
    MS_nativeID_format = MS_native_spectrum_identifier_format,

    /// <summary>Thermo nativeID format: controllerType=xsd:nonNegativeInteger controllerNumber=xsd:positiveInteger scan=xsd:positiveInteger.</summary>
    MS_Thermo_nativeID_format = 1000768,

    /// <summary>Waters nativeID format: function=xsd:positiveInteger process=xsd:nonNegativeInteger scan=xsd:nonNegativeInteger.</summary>
    MS_Waters_nativeID_format = 1000769,

    /// <summary>WIFF nativeID format: sample=xsd:nonNegativeInteger period=xsd:nonNegativeInteger cycle=xsd:nonNegativeInteger experiment=xsd:nonNegativeInteger.</summary>
    MS_WIFF_nativeID_format = 1000770,

    /// <summary>Bruker/Agilent YEP nativeID format: scan=xsd:nonNegativeInteger.</summary>
    MS_Bruker_Agilent_YEP_nativeID_format = 1000771,

    /// <summary>Bruker BAF nativeID format: scan=xsd:nonNegativeInteger.</summary>
    MS_Bruker_BAF_nativeID_format = 1000772,

    /// <summary>Bruker FID nativeID format: file=xsd:IDREF.</summary>
    MS_Bruker_FID_nativeID_format = 1000773,

    /// <summary>multiple peak list nativeID format: index=xsd:nonNegativeInteger.</summary>
    MS_multiple_peak_list_nativeID_format = 1000774,

    /// <summary>single peak list nativeID format: file=xsd:IDREF.</summary>
    MS_single_peak_list_nativeID_format = 1000775,

    /// <summary>scan number only nativeID format: scan=xsd:nonNegativeInteger.</summary>
    MS_scan_number_only_nativeID_format = 1000776,

    /// <summary>spectrum identifier nativeID format: spectrum=xsd:nonNegativeInteger.</summary>
    MS_spectrum_identifier_nativeID_format = 1000777,

    /// <summary>charge state calculation: A process that infers the charge state of an MSn spectrum's precursor(s) by the application of some algorithm.</summary>
    MS_charge_state_calculation = 1000778,

    /// <summary>below precursor intensity dominance charge state calculation: Infers charge state as single or amibiguously multiple by determining the fraction of intensity below the precursor m/z.</summary>
    MS_below_precursor_intensity_dominance_charge_state_calculation = 1000779,

    /// <summary>precursor recalculation: A process that recalculates existing precursor selected ions with one or more algorithmically determined precursor selected ions.</summary>
    MS_precursor_recalculation = 1000780,

    /// <summary>msPrefix precursor recalculation: Recalculates one or more precursor selected ions by peak detection in the isolation windows of high accuracy MS precursor scans.</summary>
    MS_msPrefix_precursor_recalculation = 1000781,

    /// <summary>Savitzky-Golay smoothing: Reduces intensity spikes by applying local polynomial regression (of degree k) on a distribution (of at least k+1 equally spaced points) to determine the smoothed value for each point. It tends to preserve features of the distribution such as relative maxima, minima and width, which are usually 'flattened' by other adjacent averaging techniques.</summary>
    MS_Savitzky_Golay_smoothing = 1000782,

    /// <summary>LOWESS smoothing: Reduces intensity spikes by applying a modeling method known as locally weighted polynomial regression. At each point in the data set a low-degree polynomial is fit to a subset of the data, with explanatory variable values near the point whose response is being estimated. The polynomial is fit using weighted least squares, giving more weight to points near the point whose response is being estimated and less weight to points further away. The value of the regression function for the point is then obtained by evaluating the local polynomial using the explanatory variable values for that data point. The LOESS fit is complete after regression function values have been computed for each of the n data points. Many of the details of this method, such as the degree of the polynomial model and the weights, are flexible.</summary>
    MS_LOWESS_smoothing = 1000783,

    /// <summary>Gaussian smoothing: Reduces intensity spikes by convolving the data with a one-dimensional Gaussian function.</summary>
    MS_Gaussian_smoothing = 1000784,

    /// <summary>binomial smoothing: Reduces intensity spikes by convolving the data with a one-dimensional Gaussian function.</summary>
    MS_binomial_smoothing = MS_Gaussian_smoothing,

    /// <summary>Weierstrass transform: Reduces intensity spikes by convolving the data with a one-dimensional Gaussian function.</summary>
    MS_Weierstrass_transform = MS_Gaussian_smoothing,

    /// <summary>moving average smoothing: Reduces intensity spikes by averaging each point with two or more adjacent points. The more adjacent points that used, the stronger the smoothing effect.</summary>
    MS_moving_average_smoothing = 1000785,

    /// <summary>box smoothing: Reduces intensity spikes by averaging each point with two or more adjacent points. The more adjacent points that used, the stronger the smoothing effect.</summary>
    MS_box_smoothing = MS_moving_average_smoothing,

    /// <summary>boxcar smoothing: Reduces intensity spikes by averaging each point with two or more adjacent points. The more adjacent points that used, the stronger the smoothing effect.</summary>
    MS_boxcar_smoothing = MS_moving_average_smoothing,

    /// <summary>sliding average smoothing: Reduces intensity spikes by averaging each point with two or more adjacent points. The more adjacent points that used, the stronger the smoothing effect.</summary>
    MS_sliding_average_smoothing = MS_moving_average_smoothing,

    /// <summary>non-standard data array: A data array that contains data not covered by any other term in this group. Please do not use this term, if the binary data array type might be commonly used - contact the PSI-MS working group in order to have another CV term added.</summary>
    MS_non_standard_data_array = 1000786,

    /// <summary>inclusive low intensity threshold: Threshold at or below which some action is taken.</summary>
    MS_inclusive_low_intensity_threshold = 1000787,

    /// <summary>inclusive high intensity threshold: Threshold at or above which some action is taken.</summary>
    MS_inclusive_high_intensity_threshold = 1000788,

    /// <summary>enhanced multiply charged spectrum: MS1 spectrum that is enriched in multiply-charged ions compared to singly-charged ions.</summary>
    MS_enhanced_multiply_charged_spectrum = 1000789,

    /// <summary>time-delayed fragmentation spectrum: MSn spectrum in which the product ions are collected after a time delay, which allows the observation of lower energy fragmentation processes after precursor ion activation.</summary>
    MS_time_delayed_fragmentation_spectrum = 1000790,

    /// <summary>enhanced resolution scan: Scan with enhanced resolution.</summary>
    MS_enhanced_resolution_scan_OBSOLETE = 1000791,

    /// <summary>isolation window attribute: Isolation window parameter.</summary>
    MS_isolation_window_attribute = 1000792,

    /// <summary>isolation window upper limit: The highest m/z being isolated in an isolation window.</summary>
    MS_isolation_window_upper_limit_OBSOLETE = 1000793,

    /// <summary>isolation window lower limit: The lowest m/z being isolated in an isolation window.</summary>
    MS_isolation_window_lower_limit_OBSOLETE = 1000794,

    /// <summary>no combination: Use this term if only one scan was recorded or there is no information about scans available.</summary>
    MS_no_combination = 1000795,

    /// <summary>spectrum title: A free-form text title describing a spectrum.</summary>
    MS_spectrum_title = 1000796,

    /// <summary>peak list scans: A list of scan numbers and or scan ranges associated with a peak list. If possible the list of scans should be converted to native spectrum identifiers instead of using this term.</summary>
    MS_peak_list_scans = 1000797,

    /// <summary>peak list raw scans: A list of raw scans and or scan ranges used to generate a peak list. If possible the list of scans should be converted to native spectrum identifiers instead of using this term.</summary>
    MS_peak_list_raw_scans = 1000798,

    /// <summary>custom unreleased software tool: A software tool that has not yet been released. The value should describe the software. Please do not use this term for publicly available software - contact the PSI-MS working group in order to have another CV term added.</summary>
    MS_custom_unreleased_software_tool = 1000799,

    /// <summary>mass resolving power: The observed mass divided by the difference between two masses that can be separated: m/dm. The procedure by which dm was obtained and the mass at which the measurement was made should be reported.</summary>
    MS_mass_resolving_power = 1000800,

    /// <summary>area peak picking: Spectral peak processing conducted on the acquired data to convert profile data to centroided data. The area defined by all raw data points that belong to the peak is reported.</summary>
    MS_area_peak_picking = 1000801,

    /// <summary>sum peak picking: Spectral peak processing conducted on the acquired data to convert profile data to centroided data. The area defined by all raw data points that belong to the peak is reported.</summary>
    MS_sum_peak_picking = MS_area_peak_picking,

    /// <summary>height peak picking: Spectral peak processing conducted on the acquired data to convert profile data to centroided data. The maximum intensity of all raw data points that belong to the peak is reported.</summary>
    MS_height_peak_picking = 1000802,

    /// <summary>max peak picking: Spectral peak processing conducted on the acquired data to convert profile data to centroided data. The maximum intensity of all raw data points that belong to the peak is reported.</summary>
    MS_max_peak_picking = MS_height_peak_picking,

    /// <summary>analyzer scan offset: Offset between two analyzers in a constant neutral loss or neutral gain scan. The value corresponds to the neutral loss or neutral gain value.</summary>
    MS_analyzer_scan_offset = 1000803,

    /// <summary>electromagnetic radiation spectrum: A plot of the relative intensity of electromagnetic radiation as a function of the wavelength.</summary>
    MS_electromagnetic_radiation_spectrum = 1000804,

    /// <summary>EMR spectrum: A plot of the relative intensity of electromagnetic radiation as a function of the wavelength.</summary>
    MS_EMR_spectrum = MS_electromagnetic_radiation_spectrum,

    /// <summary>emission spectrum: A plot of the relative intensity of electromagnetic radiation emitted by atoms or molecules when excited.</summary>
    MS_emission_spectrum = 1000805,

    /// <summary>absorption spectrum: A plot of the relative intensity of electromagnetic radiation absorbed by atoms or molecules when excited.</summary>
    MS_absorption_spectrum = 1000806,

    /// <summary>Th/s: Unit describing the scan rate of a spectrum in Thomson per second.</summary>
    MS_Th_s = 1000807,

    /// <summary>chromatogram attribute: Chromatogram properties that are associated with a value.</summary>
    MS_chromatogram_attribute = 1000808,

    /// <summary>chromatogram title: A free-form text title describing a chromatogram.</summary>
    MS_chromatogram_title = 1000809,

    /// <summary>mass chromatogram: A plot of the relative abundance of a beam or other collection of ions as a function of the retention time.</summary>
    MS_mass_chromatogram = 1000810,

    /// <summary>electromagnetic radiation chromatogram: The measurement of electromagnetic properties as a function of the retention time.</summary>
    MS_electromagnetic_radiation_chromatogram = 1000811,

    /// <summary>EMR radiation chromatogram: The measurement of electromagnetic properties as a function of the retention time.</summary>
    MS_EMR_radiation_chromatogram = MS_electromagnetic_radiation_chromatogram,

    /// <summary>absorption chromatogram: The measurement of light absorbed by the sample as a function of the retention time.</summary>
    MS_absorption_chromatogram = 1000812,

    /// <summary>emission chromatogram: The measurement of light emitted by the sample as a function of the retention time.</summary>
    MS_emission_chromatogram = 1000813,

    /// <summary>counts per second: The number of counted events observed per second in one or a group of elements of a detector.</summary>
    MS_counts_per_second = 1000814,

    /// <summary>Bruker BAF file: Bruker BAF raw file format.</summary>
    MS_Bruker_BAF_file = 1000815,

    /// <summary>Bruker U2 file: Bruker HyStar U2 file format.</summary>
    MS_Bruker_U2_file = 1000816,

    /// <summary>HyStar: Bruker software for hyphenated experiments.</summary>
    MS_HyStar = 1000817,

    /// <summary>Acquity UPLC PDA: Acquity UPLC Photodiode Array Detector.</summary>
    MS_Acquity_UPLC_PDA = 1000818,

    /// <summary>Acquity UPLC FLR: Acquity UPLC Fluorescence Detector.</summary>
    MS_Acquity_UPLC_FLR = 1000819,

    /// <summary>flow rate array: A data array of flow rate measurements.</summary>
    MS_flow_rate_array = 1000820,

    /// <summary>pressure array: A data array of pressure measurements.</summary>
    MS_pressure_array = 1000821,

    /// <summary>temperature array: A data array of temperature measurements.</summary>
    MS_temperature_array = 1000822,

    /// <summary>Bruker U2 nativeID format: declaration=xsd:nonNegativeInteger collection=xsd:nonNegativeInteger scan=xsd:nonNegativeInteger.</summary>
    MS_Bruker_U2_nativeID_format = 1000823,

    /// <summary>no nativeID format: No nativeID format indicates that the file tagged with this term does not contain spectra that can have a nativeID format.</summary>
    MS_no_nativeID_format = 1000824,

    /// <summary>Bruker FID file: Bruker FID file format.</summary>
    MS_Bruker_FID_file = 1000825,

    /// <summary>elution time: The time of elution from a chromatographic column, relative to the start of the chromatography.</summary>
    MS_elution_time = 1000826,

    /// <summary>isolation window target m/z: The primary or reference m/z about which the isolation window is defined.</summary>
    MS_isolation_window_target_m_z = 1000827,

    /// <summary>isolation window lower offset: The extent of the isolation window in m/z below the isolation window target m/z. The lower and upper offsets may be asymmetric about the target m/z.</summary>
    MS_isolation_window_lower_offset = 1000828,

    /// <summary>isolation window upper offset: The extent of the isolation window in m/z above the isolation window target m/z. The lower and upper offsets may be asymmetric about the target m/z.</summary>
    MS_isolation_window_upper_offset = 1000829,

    /// <summary>sample preparation: Properties of the preparation steps which took place before the measurement was performed.</summary>
    MS_sample_preparation = 1000831,

    /// <summary>MALDI matrix application: Attributes to describe the technique how the sample is prepared with the matrix solution.</summary>
    MS_MALDI_matrix_application = 1000832,

    /// <summary>matrix application type: Describes the technique how the matrix is put on the sample target.</summary>
    MS_matrix_application_type = 1000833,

    /// <summary>matrix solution: Describes the chemical solution used as matrix.</summary>
    MS_matrix_solution = 1000834,

    /// <summary>matrix solution concentration: Concentration of the chemical solution used as matrix.</summary>
    MS_matrix_solution_concentration = 1000835,

    /// <summary>dried droplet MALDI matrix preparation: Dried droplet in MALDI matrix preparation.</summary>
    MS_dried_droplet_MALDI_matrix_preparation = 1000836,

    /// <summary>printed MALDI matrix preparation: Printed MALDI matrix preparation.</summary>
    MS_printed_MALDI_matrix_preparation = 1000837,

    /// <summary>sprayed MALDI matrix preparation: Sprayed MALDI matrix preparation.</summary>
    MS_sprayed_MALDI_matrix_preparation = 1000838,

    /// <summary>precoated MALDI sample plate: Precoated MALDI sample plate.</summary>
    MS_precoated_MALDI_sample_plate = 1000839,

    /// <summary>laser: Device that emits light (electromagnetic radiation) through a process called stimulated emission. The term is an acronym for Light Amplification by Stimulated Emission of Radiation.</summary>
    MS_laser = 1000840,

    /// <summary>laser attribute: Laser properties that are associated with a value.</summary>
    MS_laser_attribute = 1000841,

    /// <summary>laser type: Type of laser used used for desorption purpose.</summary>
    MS_laser_type = 1000842,

    /// <summary>wavelength: The distance between two peaks of the emitted laser beam.</summary>
    MS_wavelength = 1000843,

    /// <summary>focus diameter x: Describes the diameter of the laser beam in x direction.</summary>
    MS_focus_diameter_x = 1000844,

    /// <summary>focus diameter y: Describes the diameter of the laser beam in y direction.</summary>
    MS_focus_diameter_y = 1000845,

    /// <summary>pulse energy: Describes output energy of the laser system. May be attenuated by filters or other means.</summary>
    MS_pulse_energy = 1000846,

    /// <summary>pulse duration: Describes how long the laser beam was emitted from the laser device.</summary>
    MS_pulse_duration = 1000847,

    /// <summary>attenuation: Describes the reduction of the intensity of the laser beam energy.</summary>
    MS_attenuation = 1000848,

    /// <summary>impact angle: Describes the angle between the laser beam and the sample target.</summary>
    MS_impact_angle = 1000849,

    /// <summary>gas laser: Laser which is powered by a gaseous medium.</summary>
    MS_gas_laser = 1000850,

    /// <summary>solid-state laser: Solid state laser materials are commonly made by doping a crystalline solid host with ions that provide the required energy states.</summary>
    MS_solid_state_laser = 1000851,

    /// <summary>dye-laser: Dye lasers use an organic dye as the gain medium.</summary>
    MS_dye_laser = 1000852,

    /// <summary>free electron laser: Free electron laser uses a relativistic electron beam as the lasing medium which move freely through a magnetic structure, hence the term.</summary>
    MS_free_electron_laser = 1000853,

    /// <summary>LTQ XL: Thermo Scientific LTQ XL MS.</summary>
    MS_LTQ_XL = 1000854,

    /// <summary>LTQ Velos: Thermo Scientific LTQ Velos MS.</summary>
    MS_LTQ_Velos = 1000855,

    /// <summary>LTQ Velos ETD: Thermo Scientific LTQ Velos MS with ETD.</summary>
    MS_LTQ_Velos_ETD = 1000856,

    /// <summary>run attribute: Properties of the described run.</summary>
    MS_run_attribute = 1000857,

    /// <summary>fraction identifier: Identier string that describes the sample fraction. This identifer should contain the fraction number(s) or similar information.</summary>
    MS_fraction_identifier = 1000858,

    /// <summary>molecule: A molecules is a fundamental component of a chemical compound that is the smallest part of the compound that can participate in a chemical reaction.</summary>
    MS_molecule = 1000859,

    /// <summary>peptide: A compound of low molecular weight that is composed of two or more amino acids.</summary>
    MS_peptide = 1000860,

    /// <summary>chemical compound attribute: A describable property of a chemical compound.</summary>
    MS_chemical_compound_attribute = 1000861,

    /// <summary>isoelectric point: The pH of a solution at which a charged molecule does not migrate in an electric field.</summary>
    MS_isoelectric_point = 1000862,

    /// <summary>pI: The pH of a solution at which a charged molecule does not migrate in an electric field.</summary>
    MS_pI = MS_isoelectric_point,

    /// <summary>predicted isoelectric point: The pH of a solution at which a charged molecule would not migrate in an electric field, as predicted by a software algorithm.</summary>
    MS_predicted_isoelectric_point = 1000863,

    /// <summary>predicted pI: The pH of a solution at which a charged molecule would not migrate in an electric field, as predicted by a software algorithm.</summary>
    MS_predicted_pI = MS_predicted_isoelectric_point,

    /// <summary>chemical compound formula: A combination of symbols used to express the chemical composition of a compound.</summary>
    MS_chemical_compound_formula = 1000864,

    /// <summary>empirical formula: A chemical formula which expresses the proportions of the elements present in a substance.</summary>
    MS_empirical_formula = 1000865,

    /// <summary>molecular formula: A chemical compound formula expressing the number of atoms of each element present in a compound, without indicating how they are linked.</summary>
    MS_molecular_formula = 1000866,

    /// <summary>structural formula: A chemical formula showing the number of atoms of each element in a molecule, their spatial arrangement, and their linkage to each other.</summary>
    MS_structural_formula = 1000867,

    /// <summary>SMILES string: The simplified molecular input line entry specification or SMILES is a specification for unambiguously describing the structure of a chemical compound using a short ASCII string.</summary>
    MS_SMILES_string = 1000868,

    /// <summary>collision gas pressure: The gas pressure of the collision gas used for collisional excitation.</summary>
    MS_collision_gas_pressure = 1000869,

    /// <summary>4000 QTRAP: AB SCIEX or Applied Biosystems|MDS SCIEX QTRAP 4000.</summary>
    MS_4000_QTRAP_OBSOLETE = 1000870,

    /// <summary>SRM software: Software used to predict, select, or optimize transitions or analyze the results of selected reaction monitoring runs.</summary>
    MS_SRM_software = 1000871,

    /// <summary>MaRiMba: Software used to predict transitions for selected reaction monitoring experiments based on observed spectrum libraries developed and distributed by the Institute for Systems Biology.</summary>
    MS_MaRiMba = 1000872,

    /// <summary>peptide attribute calculation software: Software used to predict or calculate numberical attributes of peptides.</summary>
    MS_peptide_attribute_calculation_software = 1000873,

    /// <summary>SSRCalc: Sequence Specific Retention Calculator esimates the retention time of peptides based on their sequence.</summary>
    MS_SSRCalc = 1000874,

    /// <summary>declustering potential: Potential difference between the orifice and the skimmer in volts.</summary>
    MS_declustering_potential = 1000875,

    /// <summary>cone voltage: Potential difference between the sampling cone/orifice in volts.</summary>
    MS_cone_voltage = 1000876,

    /// <summary>tube lens voltage: Potential difference setting of the tube lens in volts.</summary>
    MS_tube_lens_voltage = 1000877,

    /// <summary>external reference identifier: An identifier/accession number to an external reference database.</summary>
    MS_external_reference_identifier = 1000878,

    /// <summary>PubMed identifier: A unique identifier for a publication in the PubMed database (MIR:00000015).</summary>
    MS_PubMed_identifier = 1000879,

    /// <summary>interchannel delay: The duration of intervals between scanning, during which the instrument configuration is switched.</summary>
    MS_interchannel_delay = 1000880,

    /// <summary>chemical compound: A substance formed by chemical union of two or more elements or ingredients in definite proportion by weight.</summary>
    MS_chemical_compound = 1000881,

    /// <summary>protein: A compound composed of one or more chains of amino acids in a specific order determined by the base sequence of nucleotides in the DNA coding for the protein.</summary>
    MS_protein = 1000882,

    /// <summary>protein short name: A short name or symbol of a protein (e.g., HSF 1 or HSF1_HUMAN).</summary>
    MS_protein_short_name = 1000883,

    /// <summary>protein attribute: An nonphysical attribute describing a specific protein.</summary>
    MS_protein_attribute = 1000884,

    /// <summary>protein accession: Accession number for a specific protein in a database.</summary>
    MS_protein_accession = 1000885,

    /// <summary>protein name: A long name describing the function of the protein.</summary>
    MS_protein_name = 1000886,

    /// <summary>peptide attribute: An nonphysical attribute that can be used to describe a peptide.</summary>
    MS_peptide_attribute = 1000887,

    /// <summary>unmodified peptide sequence: A sequence of letter symbols denoting the order of amino acids that compose the peptide, without encoding any amino acid mass modifications that might be present.</summary>
    MS_unmodified_peptide_sequence = 1000888,

    /// <summary>modified peptide sequence: A sequence of letter symbols denoting the order of amino acids that compose the peptide plus the encoding any amino acid modifications that are present.</summary>
    MS_modified_peptide_sequence = 1000889,

    /// <summary>peptide labeling state: A state description of how a peptide might be isotopically or isobarically labelled.</summary>
    MS_peptide_labeling_state = 1000890,

    /// <summary>heavy labeled peptide: A peptide that has been created or labeled with some heavier-than-usual isotopes.</summary>
    MS_heavy_labeled_peptide = 1000891,

    /// <summary>unlabeled peptide: A peptide that has not been labeled with heavier-than-usual isotopes. This is often referred to as \"light\" to distinguish from \"heavy\".</summary>
    MS_unlabeled_peptide = 1000892,

    /// <summary>light labeled peptide: A peptide that has not been labeled with heavier-than-usual isotopes. This is often referred to as \"light\" to distinguish from \"heavy\".</summary>
    MS_light_labeled_peptide = MS_unlabeled_peptide,

    /// <summary>peptide group label: An arbitrary string label used to mark a set of peptides that belong together in a set, whereby the members are differentiated by different isotopic labels. For example, the heavy and light forms of the same peptide will both be assigned the same peptide group label.</summary>
    MS_peptide_group_label = 1000893,

    /// <summary>retention time: A measure of the interval relative to the beginning of a mass spectrometric run when a peptide will exit the chromatographic column.</summary>
    MS_retention_time = 1000894,

    /// <summary>local retention time: A measure of the interval relative to the beginning of a mass spectrometric run when a peptide will exit an unspecified local chromatographic column and instrumental setup.</summary>
    MS_local_retention_time = 1000895,

    /// <summary>normalized retention time: A measure of the interval relative to the beginning of a mass spectrometric run when a peptide will exit a standardized reference chromatographic column and instrumental setup.</summary>
    MS_normalized_retention_time = 1000896,

    /// <summary>predicted retention time: A time interval relative to the beginning of a mass spectrometric run when a peptide will exit a chromatographic column as predicted by a referenced software application.</summary>
    MS_predicted_retention_time = 1000897,

    /// <summary>standard: Something, such as a practice or a product, that is widely recognized or employed, especially because of its excellence.</summary>
    MS_standard = 1000898,

    /// <summary>de facto standard: A practice or product that has become a standard not because it has been approved by a standards organization but because it is widely used and recognized by the industry as being standard.</summary>
    MS_de_facto_standard = 1000899,

    /// <summary>minimum information standard: A specification of a minimum amount of information needed to reproduce or fully interpret a scientific result.</summary>
    MS_minimum_information_standard = 1000900,

    /// <summary>retention time normalization standard: A standard providing the retention times at which a set of reference compounds exit the reference chromatographic column.</summary>
    MS_retention_time_normalization_standard = 1000901,

    /// <summary>H-PINS retention time normalization standard: The de facto standard providing the retention times at which a set of halogenated reference peptides exit the reference chromatographic column.</summary>
    MS_H_PINS_retention_time_normalization_standard = 1000902,

    /// <summary>product ion series ordinal: The ordinal of the fragment within a specified ion series. (e.g. 8 for a y8 ion).</summary>
    MS_product_ion_series_ordinal = 1000903,

    /// <summary>product ion m/z delta: The difference in m/z of the predicted m/z based on the assigned product ion minus the actual observed peak m/z.</summary>
    MS_product_ion_m_z_delta = 1000904,

    /// <summary>percent of base peak times 100: The magnitude of a peak expressed in terms of the percentage of the magnitude of the base peak intensity multiplied by 100. The base peak is therefore 10000. This unit is common in normalized spectrum libraries.</summary>
    MS_percent_of_base_peak_times_100 = 1000905,

    /// <summary>peak intensity rank: Ordinal specifying the rank in intensity of a peak in a spectrum. Base peak is 1. The next most intense peak is 2, etc.</summary>
    MS_peak_intensity_rank = 1000906,

    /// <summary>peak targeting suitability rank: Ordinal specifying the rank of a peak in a spectrum in terms of suitability for targeting. The most suitable peak is 1. The next most suitability peak is 2, etc. Suitability is algorithm and context dependant.</summary>
    MS_peak_targeting_suitability_rank = 1000907,

    /// <summary>transition: A set of two m/z values corresponding to the precursor m/z and a fragment m/z that in combination can be used to identify or quantify a specific ion, although not necessarily uniquely.</summary>
    MS_transition = 1000908,

    /// <summary>reaction: A set of two m/z values corresponding to the precursor m/z and a fragment m/z that in combination can be used to identify or quantify a specific ion, although not necessarily uniquely.</summary>
    MS_reaction = MS_transition,

    /// <summary>transition validation method: The strategy used to validate that a transition is effective.</summary>
    MS_transition_validation_method = 1000909,

    /// <summary>transition optimized on specified instrument: The transition has been optimized by direct injection of the peptide into an instrument specified in a separate term, and the optimum voltages and fragmentation energies have been determined.</summary>
    MS_transition_optimized_on_specified_instrument = 1000910,

    /// <summary>transition validated with an MS/MS spectrum on specified instrument: The transition has been validated by obtaining an MS/MS spectrum and demonstrating that the peak is detectable on the instrument specified with a separate term.</summary>
    MS_transition_validated_with_an_MS_MS_spectrum_on_specified_instrument = 1000911,

    /// <summary>transition purported from an MS/MS spectrum on a different, specified instrument: The transition has been purported by obtaining an MS/MS spectrum and demonstrating that the peak is detectable on the instrument specified with a separate term. However, the detecting instrument is of a different type (e.g. ion trap) than the instrument that the transition will eventually be used on (e.g. triple quad).</summary>
    MS_transition_purported_from_an_MS_MS_spectrum_on_a_different__specified_instrument = 1000912,

    /// <summary>transition predicted by informatic analysis: The transition has been predicted by informatics software without any direct spectral evidence.</summary>
    MS_transition_predicted_by_informatic_analysis = 1000913,

    /// <summary>tab delimited text file: A file that has two or more columns of tabular data where each column is separated by a TAB character.</summary>
    MS_tab_delimited_text_file = 1000914,

    /// <summary>retention time window attribute: An attribute of a window in time about which a peptide might elute from the column.</summary>
    MS_retention_time_window_attribute = 1000915,

    /// <summary>retention time window lower offset: The extent of the retention time window in time units below the target retention time. The lower and upper offsets may be asymetric about the target time.</summary>
    MS_retention_time_window_lower_offset = 1000916,

    /// <summary>retention time window upper offset: The extent of the retention time window in time units above the target retention time. The lower and upper offsets may be asymetric about the target time.</summary>
    MS_retention_time_window_upper_offset = 1000917,

    /// <summary>target list: A list of peptides or compounds and their expected m/z coordinates that can be used to cause a mass spectrometry to obtain spectra of those molecules specifically.</summary>
    MS_target_list = 1000918,

    /// <summary>target inclusion exclusion priority: A priority setting specifying whether included or excluded targets have priority over the other.</summary>
    MS_target_inclusion_exclusion_priority = 1000919,

    /// <summary>includes supersede excludes: A priority setting specifying that included targets have priority over the excluded targets if there is a conflict.</summary>
    MS_includes_supersede_excludes = 1000920,

    /// <summary>excludes supersede includes: A priority setting specifying that excluded targets have priority over the included targets if there is a conflict.</summary>
    MS_excludes_supersede_includes = 1000921,

    /// <summary>Skyline: Software used to predict, select, and optimize transitions as well as analyze the results of selected reaction monitoring runs developed and distributed by the MacCoss lab at the University of Washington. https://brendanx-uw1.gs.washington.edu/labkey/wiki/home/software/Skyline/page.view?name=default.</summary>
    MS_Skyline = 1000922,

    /// <summary>TIQAM: Software used to predict, select, and optimize transitions for selected reaction monitoring experiments developed and distributed by the Institute for Systems Biology.</summary>
    MS_TIQAM = 1000923,

    /// <summary>MaRiMba: Software used to predict transitions for selected reaction monitoring experiments based on observed spectrum libraries developed and distributed by the Institute for Systems Biology.</summary>
    MS_MaRiMba_OBSOLETE = 1000924,

    /// <summary>ATAQS: Software suite used to predict, select, and optimize transitions as well as analyze the results of selected reaction monitoring runs developed and distributed by the Institute for Systems Biology.</summary>
    MS_ATAQS = 1000925,

    /// <summary>product interpretation rank: The integer rank given an interpretation of an observed product ion. For example, if y8 is selected as the most likely interpretation of a peak, then it is assigned a rank of 1.</summary>
    MS_product_interpretation_rank = 1000926,

    /// <summary>ion injection time: The length of time spent filling an ion trapping device.</summary>
    MS_ion_injection_time = 1000927,

    /// <summary>calibration spectrum: A spectrum derived from a special calibration source, rather than from the primary injected sample. A calibration spectrum is typically derived from a substance that can be used to correct systematic shift in m/z for spectra of the primary inject sample.</summary>
    MS_calibration_spectrum = 1000928,

    /// <summary>Shimadzu Biotech nativeID format: source=xsd:string start=xsd:nonNegativeInteger end=xsd:nonNegativeInteger.</summary>
    MS_Shimadzu_Biotech_nativeID_format = 1000929,

    /// <summary>Shimadzu Biotech database entity: Shimadzu Biotech format.</summary>
    MS_Shimadzu_Biotech_database_entity = 1000930,

    /// <summary>QTRAP 5500: Applied Biosystems|MDS SCIEX QTRAP 5500.</summary>
    MS_QTRAP_5500 = 1000931,

    /// <summary>TripleTOF 5600: AB SCIEX TripleTOF 5600, a quadrupole - quadrupole - time-of-flight mass spectrometer.</summary>
    MS_TripleTOF_5600 = 1000932,

    /// <summary>protein modifications: Encoding of modifications of the protein sequence from the specified accession, written in PEFF notation.</summary>
    MS_protein_modifications = 1000933,

    /// <summary>gene name: Name of the gene from which the protein is translated.</summary>
    MS_gene_name = 1000934,

    /// <summary>spectrum interpretation: Collection of terms from the PSI Proteome Informatics standards describing the interpretation of spectra.</summary>
    MS_spectrum_interpretation = 1001000,

    /// <summary>Sequest:CleavesAt: </summary>
    MS_Sequest_CleavesAt = 1001005,

    /// <summary>Sequest:ViewCV: Branch containing the CV terms for Sequest View Input Parameters.</summary>
    MS_Sequest_ViewCV = 1001006,

    /// <summary>Sequest:OutputLines: </summary>
    MS_Sequest_OutputLines = 1001007,

    /// <summary>Sequest:DescriptionLines: </summary>
    MS_Sequest_DescriptionLines = 1001009,

    /// <summary>de novo search: A de novo sequencing search (without database).</summary>
    MS_de_novo_search = 1001010,

    /// <summary>search database details: Details about the database searched.</summary>
    MS_search_database_details = 1001011,

    /// <summary>database source: The organisation, project or laboratory from where the database is obtained (uniprot, ncbi, ebi, other).</summary>
    MS_database_source = 1001012,

    /// <summary>database name: The name of the search database (nr, SwissProt or est_human).</summary>
    MS_database_name = 1001013,

    /// <summary>database local file path: OBSOLETE: Use attribute in mzIdentML instead. Local file path of the search database from the search engine's point of view.</summary>
    MS_database_local_file_path_OBSOLETE = 1001014,

    /// <summary>database original uri: URI, from where the search database was originally downloaded.</summary>
    MS_database_original_uri = 1001015,

    /// <summary>database version: OBSOLETE: Use attribute in mzIdentML instead. Version of the search database .</summary>
    MS_database_version_OBSOLETE = 1001016,

    /// <summary>database release date: OBSOLETE: Use attribute in mzIdentML instead. Release date of the search database.</summary>
    MS_database_release_date_OBSOLETE = 1001017,

    /// <summary>database type: Database containing amino acid or nucleic acid sequences.</summary>
    MS_database_type = 1001018,

    /// <summary>database filtering: Was there filtering used on the database.</summary>
    MS_database_filtering = 1001019,

    /// <summary>DB filter taxonomy: A taxonomy filter was to the database search.</summary>
    MS_DB_filter_taxonomy = 1001020,

    /// <summary>DB filter on accession numbers: Filtering applied specifically by accession number pattern.</summary>
    MS_DB_filter_on_accession_numbers = 1001021,

    /// <summary>DB MW filter: Filtering applied specifically by protein molecular weight, specified as either a range or above/below a threshold value.</summary>
    MS_DB_MW_filter = 1001022,

    /// <summary>DB PI filter: Filtering applied specifically by predicted protein isoelectric focussing point (pI), specified as either a range or above/below a threshold value.</summary>
    MS_DB_PI_filter = 1001023,

    /// <summary>translation frame: OSBOLETE: use attribute in mzIdentML instead. The translated open reading frames from a nucleotide database considered in the search (range: 1-6).</summary>
    MS_translation_frame_OBSOLETE = 1001024,

    /// <summary>translation table: The translation table used to translate the nucleotides to amino acids.</summary>
    MS_translation_table = 1001025,

    /// <summary>Sequest:NormalizeXCorrValues: </summary>
    MS_Sequest_NormalizeXCorrValues = 1001026,

    /// <summary>DB filter on sequence pattern: Filtering applied specifically by amino acid sequence pattern.</summary>
    MS_DB_filter_on_sequence_pattern = 1001027,

    /// <summary>Sequest:SequenceHeaderFilter: </summary>
    MS_Sequest_SequenceHeaderFilter = 1001028,

    /// <summary>number of sequences searched: The number of sequences (proteins / nucleotides) from the database search after filtering.</summary>
    MS_number_of_sequences_searched = 1001029,

    /// <summary>number of peptide seqs compared to each spectrum: Number of peptide seqs compared to each spectrum.</summary>
    MS_number_of_peptide_seqs_compared_to_each_spectrum = 1001030,

    /// <summary>spectral library search: A search using a library of spectra.</summary>
    MS_spectral_library_search = 1001031,

    /// <summary>Sequest:SequencePartialFilter: </summary>
    MS_Sequest_SequencePartialFilter = 1001032,

    /// <summary>date / time search performed: OBSOLETE: use attribute in mzIdentML instead. Date and time of the actual search run.</summary>
    MS_date___time_search_performed_OBSOLETE = 1001035,

    /// <summary>search time taken: The time taken to complete the search in seconds.</summary>
    MS_search_time_taken = 1001036,

    /// <summary>Sequest:ShowFragmentIons: </summary>
    MS_Sequest_ShowFragmentIons = 1001037,

    /// <summary>Sequest:Consensus: Specify depth as value of the CVParam!</summary>
    MS_Sequest_Consensus = 1001038,

    /// <summary>intermediate analysis format: Type of the source file, the  mzIdentML was created from.</summary>
    MS_intermediate_analysis_format = 1001040,

    /// <summary>Sequest:sortCV: Branch containing the CV terms for Sequest View / Sort Input Parameters.</summary>
    MS_Sequest_sortCV = 1001041,

    /// <summary>Sequest:LimitTo: Specify \"number of dtas shown\" as value of the CVParam!</summary>
    MS_Sequest_LimitTo = 1001042,

    /// <summary>cleavage agent details: Details of cleavage agent (enzyme).</summary>
    MS_cleavage_agent_details = 1001044,

    /// <summary>cleavage agent name: The name of the cleavage agent.</summary>
    MS_cleavage_agent_name = 1001045,

    /// <summary>Sequest:sort_by_dCn: </summary>
    MS_Sequest_sort_by_dCn = 1001046,

    /// <summary>Sequest:sort_by_dM: </summary>
    MS_Sequest_sort_by_dM = 1001047,

    /// <summary>Sequest:sort_by_Ions: </summary>
    MS_Sequest_sort_by_Ions = 1001048,

    /// <summary>Sequest:sort_by_MH+: </summary>
    MS_Sequest_sort_by_MH_ = 1001049,

    /// <summary>Sequest:sort_by_P: </summary>
    MS_Sequest_sort_by_P = 1001050,

    /// <summary>multiple enzyme combination rules: OBSOLETE: use attribute independent in mzIdentML instead. Description of multiple enzyme digestion protocol, if any.</summary>
    MS_multiple_enzyme_combination_rules_OBSOLETE = 1001051,

    /// <summary>Sequest:sort_by_PreviousAminoAcid: </summary>
    MS_Sequest_sort_by_PreviousAminoAcid = 1001052,

    /// <summary>Sequest:sort_by_Ref: </summary>
    MS_Sequest_sort_by_Ref = 1001053,

    /// <summary>modification parameters: Modification parameters for the search engine run.</summary>
    MS_modification_parameters = 1001055,

    /// <summary>modification specificity rule: The specificity rules for the modifications applied by the search engine.</summary>
    MS_modification_specificity_rule = 1001056,

    /// <summary>tolerance on types: OBSOLETE: Tolerance on types.</summary>
    MS_tolerance_on_types_OBSOLETE = 1001057,

    /// <summary>quality estimation by manual validation: The quality estimation was done manually.</summary>
    MS_quality_estimation_by_manual_validation = 1001058,

    /// <summary>Sequest:sort_by_RSp: </summary>
    MS_Sequest_sort_by_RSp = 1001059,

    /// <summary>quality estimation method details: Method for quality estimation (manually or with decoy database).</summary>
    MS_quality_estimation_method_details = 1001060,

    /// <summary>neutral loss: OBSOLETE: replaced by MS:1000336 (neutral loss): Leave this to PSI-MOD?</summary>
    MS_neutral_loss_OBSOLETE = 1001061,

    /// <summary>Mascot MGF file: </summary>
    MS_Mascot_MGF_file = 1001062,

    /// <summary>TODOscoring model: OBSOLETE: There is Phenyx:ScoringModel for Phenyx! Scoring model (more detailed granularity). TODO: add some child terms!</summary>
    MS_TODOscoring_model_OBSOLETE = 1001065,

    /// <summary>ions series considered in search: The description of the ion fragment series (including charges and neutral losses) that are considered by the search engine.</summary>
    MS_ions_series_considered_in_search = 1001066,

    /// <summary>Sequest:sort_by_Sp: </summary>
    MS_Sequest_sort_by_Sp = 1001068,

    /// <summary>Sequest:sort_by_TIC: </summary>
    MS_Sequest_sort_by_TIC = 1001069,

    /// <summary>Sequest:sort_by_Scan: </summary>
    MS_Sequest_sort_by_Scan = 1001070,

    /// <summary>Sequest:sort_by_Sequence: </summary>
    MS_Sequest_sort_by_Sequence = 1001071,

    /// <summary>Sequest:sort_by_Sf: </summary>
    MS_Sequest_sort_by_Sf = 1001072,

    /// <summary>database type amino acid: Database contains amino acid sequences.</summary>
    MS_database_type_amino_acid = 1001073,

    /// <summary>database type nucleotide: Database contains nucleid acid sequences.</summary>
    MS_database_type_nucleotide = 1001079,

    /// <summary>search type: Enumeration of type of search value (i.e. from PMF, sequence tag, MS-MS).</summary>
    MS_search_type = 1001080,

    /// <summary>pmf search: A peptide mass fingerprint search.</summary>
    MS_pmf_search = 1001081,

    /// <summary>tag search: A sequence tag search.</summary>
    MS_tag_search = 1001082,

    /// <summary>ms-ms search: An ms/ms search (with fragment ions).</summary>
    MS_ms_ms_search = 1001083,

    /// <summary>database nr: </summary>
    MS_database_nr = 1001084,

    /// <summary>protein result details: Protein level information.</summary>
    MS_protein_result_details = 1001085,

    /// <summary>Sequest:sort_by_XCorr: </summary>
    MS_Sequest_sort_by_XCorr = 1001086,

    /// <summary>Sequest:ProcessCV: Branch containing the CV terms for Sequest View / Process Input Parameters.</summary>
    MS_Sequest_ProcessCV = 1001087,

    /// <summary>protein description: The protein description line from the sequence entry in the source database FASTA file.</summary>
    MS_protein_description = 1001088,

    /// <summary>molecule taxonomy: The taxonomy of the resultant molecule from the search.</summary>
    MS_molecule_taxonomy = 1001089,

    /// <summary>taxonomy nomenclature: OBSOLETE: The system used to indicate taxonomy. There should be an enumerated list of options: latin name, NCBI TaxID, common name, Swiss-Prot species ID (ex. RABIT from the full protein ID ALBU_RABIT).</summary>
    MS_taxonomy_nomenclature_OBSOLETE = 1001090,

    /// <summary>NoEnzyme: </summary>
    MS_NoEnzyme_OBSOLETE = 1001091,

    /// <summary>peptide identification confidence metric: Peptide identification confidence metric for a peptide (p-value, Expect value, confidence, local FDR).</summary>
    MS_peptide_identification_confidence_metric = 1001092,

    /// <summary>sequence coverage: The percent coverage for the protein based upon the matched peptide sequences (can be calculated).</summary>
    MS_sequence_coverage = 1001093,

    /// <summary>Sequest:sort_by_z: </summary>
    MS_Sequest_sort_by_z = 1001094,

    /// <summary>Sequest:ProcessAll: </summary>
    MS_Sequest_ProcessAll = 1001095,

    /// <summary>Sequest:TopPercentMostIntense: Specify \"percentage\" as value of the CVParam!</summary>
    MS_Sequest_TopPercentMostIntense = 1001096,

    /// <summary>distinct peptide sequences: This counts distinct sequences hitting the protein without regard to a minimal confidence threshold.</summary>
    MS_distinct_peptide_sequences = 1001097,

    /// <summary>confident distinct peptide sequences: This counts the number of distinct peptide sequences. Multiple charge states and multiple modification states do NOT count as multiple sequences. The definition of 'confident' must be qualified elsewhere.</summary>
    MS_confident_distinct_peptide_sequences = 1001098,

    /// <summary>confident peptide qualification: The point of this entry is to define what is meant by confident for the term Confident distinct peptide sequence and/or Confident peptides. Example 1 - metric=Paragon:Confidence value=95 sense=greater than Example 2 - metric=Mascot:Eval value=0.05 sense=less than.</summary>
    MS_confident_peptide_qualification = 1001099,

    /// <summary>confident peptide: This counts the number of peptide sequences without regard to whether they are distinct. Multiple charges states and multiple modification states DO count as multiple peptides. The definition of 'confident' must be qualified elsewhere.</summary>
    MS_confident_peptide = 1001100,

    /// <summary>protein group/subset relationship: Branch containing protein group/subset relationships.</summary>
    MS_protein_group_subset_relationship = 1001101,

    /// <summary>Sequest:Chromatogram: </summary>
    MS_Sequest_Chromatogram = 1001102,

    /// <summary>Sequest:InfoAndLog: </summary>
    MS_Sequest_InfoAndLog = 1001103,

    /// <summary>database SwissProt: </summary>
    MS_database_SwissProt = 1001104,

    /// <summary>peptide result details: Peptide level information.</summary>
    MS_peptide_result_details = 1001105,

    /// <summary>Sequest:TopNumber: Specify \"number\" as value of the CVParam!</summary>
    MS_Sequest_TopNumber = 1001106,

    /// <summary>data stored in database: Source file for this mzIdentML was a data set in a database.</summary>
    MS_data_stored_in_database = 1001107,

    /// <summary>param: a ion: </summary>
    MS_param__a_ion = 1001108,

    /// <summary>Sequest:CullTo: Specify cull string as value of the CVParam!</summary>
    MS_Sequest_CullTo = 1001109,

    /// <summary>Sequest:modeCV: Branch containing the CV terms for Sequest Mode Input Parameters.</summary>
    MS_Sequest_modeCV = 1001110,

    /// <summary>Sequest:Full: </summary>
    MS_Sequest_Full = 1001111,

    /// <summary>n-terminal flanking residue: The residue preceeding the first amino acid in the peptide sequence as it occurs in the protein. Use 'N-term' to denote if the peptide starts at the N terminus of the protein.</summary>
    MS_n_terminal_flanking_residue = 1001112,

    /// <summary>c-terminal flanking residue: The residue following the last amino acid in the peptide sequence as it occurs in the protein. Use 'C-term' to denote if the peptide ends at the C terminus of the protein.</summary>
    MS_c_terminal_flanking_residue = 1001113,

    /// <summary>retention time(s): Retention time of the spectrum from the source file.</summary>
    MS_retention_time_s_ = 1001114,

    /// <summary>scan number(s): OBSOLETE: use spectrumID attribute of SpectrumIdentificationResult. Take from mzData.</summary>
    MS_scan_number_s__OBSOLETE = 1001115,

    /// <summary>single protein result details: Results specific for one protein as part of a protein ambiguity group (a result not valid for all the other proteins in the protein ambiguity group).</summary>
    MS_single_protein_result_details = 1001116,

    /// <summary>theoretical mass: The theoretical mass of the molecule (e.g. the peptide sequence and its modifications).</summary>
    MS_theoretical_mass = 1001117,

    /// <summary>param: b ion: </summary>
    MS_param__b_ion = 1001118,

    /// <summary>param: c ion: </summary>
    MS_param__c_ion = 1001119,

    /// <summary>Sequest:FormatAndLinks: </summary>
    MS_Sequest_FormatAndLinks = 1001120,

    /// <summary>number of matched peaks: The number of peaks that were matched as qualified by the ion series considered field. If a peak matches multiple ions then only 1 would be added the count.</summary>
    MS_number_of_matched_peaks = 1001121,

    /// <summary>ions series considered: The ion series that were used during the calculation of the count (e.g. b, y, a, b, c, y, z, b, b-NH3, b-H20, b+, z, z-, z+, y-H3PO4, immonium).</summary>
    MS_ions_series_considered = 1001122,

    /// <summary>number of peaks used: The number of peaks from the original peak list that are used to calculate the scores for a particular search engine. All ions that have the opportunity to match or be counted even if they don't.</summary>
    MS_number_of_peaks_used = 1001123,

    /// <summary>number of peaks submitted: The number of peaks from the original peaks listed that were submitted to the search engine.</summary>
    MS_number_of_peaks_submitted = 1001124,

    /// <summary>manual validation: Result of quality estimation: decision of a manual validation.</summary>
    MS_manual_validation = 1001125,

    /// <summary>Sequest:Fast: </summary>
    MS_Sequest_Fast = 1001126,

    /// <summary>peptide sharing details: Accessions Containing Sequence - Accessions for each protein containing this peptide.</summary>
    MS_peptide_sharing_details = 1001127,

    /// <summary>Sequest:selectCV: Branch containing the CV terms for Sequest Select Input Parameters.</summary>
    MS_Sequest_selectCV = 1001128,

    /// <summary>quantification information: Quantification information.</summary>
    MS_quantification_information = 1001129,

    /// <summary>peptide raw area: Peptide raw area.</summary>
    MS_peptide_raw_area = 1001130,

    /// <summary>error on peptide area: Error on peptide area.</summary>
    MS_error_on_peptide_area = 1001131,

    /// <summary>peptide ratio: Peptide ratio.</summary>
    MS_peptide_ratio = 1001132,

    /// <summary>error on peptide ratio: Error on peptide ratio.</summary>
    MS_error_on_peptide_ratio = 1001133,

    /// <summary>protein ratio: Protein ratio.</summary>
    MS_protein_ratio = 1001134,

    /// <summary>error on protein ratio: Error on protein ratio.</summary>
    MS_error_on_protein_ratio = 1001135,

    /// <summary>p-value (protein diff from 1 randomly): P-value (protein diff from 1 randomly).</summary>
    MS_p_value__protein_diff_from_1_randomly_ = 1001136,

    /// <summary>absolute quantity: Absolute quantity in terms of real concentration or molecule copy number in sample.</summary>
    MS_absolute_quantity = 1001137,

    /// <summary>error on absolute quantity: Error on absolute quantity.</summary>
    MS_error_on_absolute_quantity = 1001138,

    /// <summary>quantitation software name: Quantitation software name.</summary>
    MS_quantitation_software_name = 1001139,

    /// <summary>quantitation software version: Quantitation software version.</summary>
    MS_quantitation_software_version_OBSOLETE = 1001140,

    /// <summary>intensity of precursor ion: The intensity of the precursor ion.</summary>
    MS_intensity_of_precursor_ion = 1001141,

    /// <summary>database IPI_human: </summary>
    MS_database_IPI_human = 1001142,

    /// <summary>search engine specific score for peptides: </summary>
    MS_search_engine_specific_score_for_peptides = 1001143,

    /// <summary>Sequest:SelectDefault: </summary>
    MS_Sequest_SelectDefault = 1001144,

    /// <summary>Sequest:SelectAdvancedCV: Branch containing the CV terms for Sequest Select Advanced Input Parameters.</summary>
    MS_Sequest_SelectAdvancedCV = 1001145,

    /// <summary>param: a ion-NH3: </summary>
    MS_param__a_ion_NH3 = 1001146,

    /// <summary>protein ambiguity group result details: </summary>
    MS_protein_ambiguity_group_result_details = 1001147,

    /// <summary>param: a ion-H2O: Ion a - H2O if a significant and fragment includes STED.</summary>
    MS_param__a_ion_H2O = 1001148,

    /// <summary>param: b ion-NH3: </summary>
    MS_param__b_ion_NH3 = 1001149,

    /// <summary>param: b ion-H2O: Ion b - H2O if b significant and fragment includes STED.</summary>
    MS_param__b_ion_H2O = 1001150,

    /// <summary>param: y ion-NH3: </summary>
    MS_param__y_ion_NH3 = 1001151,

    /// <summary>param: y ion-H2O: Ion y - H2O if y significant and fragment includes STED.</summary>
    MS_param__y_ion_H2O = 1001152,

    /// <summary>search engine specific score: Search engine specific scores.</summary>
    MS_search_engine_specific_score = 1001153,

    /// <summary>Sequest:probability: The SEQUEST result 'Probability'.</summary>
    MS_Sequest_probability = 1001154,

    /// <summary>Sequest:xcorr: The SEQUEST result 'XCorr'.</summary>
    MS_Sequest_xcorr = 1001155,

    /// <summary>Sequest:deltacn: The SEQUEST result 'DeltaCn'.</summary>
    MS_Sequest_deltacn = 1001156,

    /// <summary>Sequest:sp: The SEQUEST result 'Sp' (protein).</summary>
    MS_Sequest_sp = 1001157,

    /// <summary>Sequest:Uniq: </summary>
    MS_Sequest_Uniq = 1001158,

    /// <summary>Sequest:expectation value: The SEQUEST result 'Expectation value'.</summary>
    MS_Sequest_expectation_value = 1001159,

    /// <summary>Sequest:sf: The SEQUEST result 'Sf'.</summary>
    MS_Sequest_sf = 1001160,

    /// <summary>Sequest:matched ions: The SEQUEST result 'Matched Ions'.</summary>
    MS_Sequest_matched_ions = 1001161,

    /// <summary>Sequest:total ions: The SEQUEST result 'Total Ions'.</summary>
    MS_Sequest_total_ions = 1001162,

    /// <summary>Sequest:consensus score: The SEQUEST result 'Consensus Score'.</summary>
    MS_Sequest_consensus_score = 1001163,

    /// <summary>Paragon:unused protscore: The Paragon result 'Unused ProtScore'.</summary>
    MS_Paragon_unused_protscore = 1001164,

    /// <summary>Paragon:total protscore: The Paragon result 'Total ProtScore'.</summary>
    MS_Paragon_total_protscore = 1001165,

    /// <summary>Paragon:score: The Paragon result 'Score'.</summary>
    MS_Paragon_score = 1001166,

    /// <summary>Paragon:confidence: The Paragon result 'Confidence'.</summary>
    MS_Paragon_confidence = 1001167,

    /// <summary>Paragon:expression error factor: The Paragon result 'Expression Error Factor'.</summary>
    MS_Paragon_expression_error_factor = 1001168,

    /// <summary>Paragon:expression change p-value: The Paragon result 'Expression change P-value'.</summary>
    MS_Paragon_expression_change_p_value = 1001169,

    /// <summary>Paragon:contrib: The Paragon result 'Contrib'.</summary>
    MS_Paragon_contrib = 1001170,

    /// <summary>Mascot:score: The Mascot result 'Score'.</summary>
    MS_Mascot_score = 1001171,

    /// <summary>Mascot:expectation value: The Mascot result 'expectation value'.</summary>
    MS_Mascot_expectation_value = 1001172,

    /// <summary>Mascot:matched ions: The Mascot result 'Matched ions'.</summary>
    MS_Mascot_matched_ions = 1001173,

    /// <summary>Mascot:total ions: The Mascot result 'Total ions'.</summary>
    MS_Mascot_total_ions = 1001174,

    /// <summary>peptide shared in multiple proteins: </summary>
    MS_peptide_shared_in_multiple_proteins = 1001175,

    /// <summary>(?<=[KR])(?!P): </summary>
    MS______KR_____P_ = 1001176,

    /// <summary>number of molecular hypothesis considered: Number of Molecular Hypothesis Considered - This is the number of molecules (e.g. peptides for proteomics) considered for a particular search.</summary>
    MS_number_of_molecular_hypothesis_considered = 1001177,

    /// <summary>database EST: </summary>
    MS_database_EST = 1001178,

    /// <summary>Cleavage agent regular expression: Branch containing regular expressions for cleavage enzymes.</summary>
    MS_Cleavage_agent_regular_expression = 1001180,

    /// <summary>search statistics: The details of the actual run of the search.</summary>
    MS_search_statistics = 1001184,

    /// <summary>modification specificity N-term: As parameter for search engine: apply the modification only N-terminal.</summary>
    MS_modification_specificity_N_term = 1001189,

    /// <summary>modification specificity C-term: As parameter for search engine: apply the modification only C-terminal.</summary>
    MS_modification_specificity_C_term = 1001190,

    /// <summary>p-value: Quality estimation by p-value.</summary>
    MS_p_value_OBSOLETE = 1001191,

    /// <summary>Expect value: Result of quality estimation: Expect value.</summary>
    MS_Expect_value = 1001192,

    /// <summary>confidence score: Result of quality estimation: confidence score.</summary>
    MS_confidence_score = 1001193,

    /// <summary>quality estimation with decoy database: Quality estimation by decoy database.</summary>
    MS_quality_estimation_with_decoy_database = 1001194,

    /// <summary>decoy DB type reverse: Decoy type: Amino acids of protein sequences are used in reverse order.</summary>
    MS_decoy_DB_type_reverse = 1001195,

    /// <summary>decoy DB type randomized: Decoy type: Amino acids of protein sequences are randomized (keeping the original protein mass).</summary>
    MS_decoy_DB_type_randomized = 1001196,

    /// <summary>DB composition target+decoy: Decoy database composition: database contains original (target) and decoy entries.</summary>
    MS_DB_composition_target_decoy = 1001197,

    /// <summary>protein identification confidence metric: Protein identification confidence metric (p-value, Expect value, confidence, local FDR).</summary>
    MS_protein_identification_confidence_metric = 1001198,

    /// <summary>Mascot DAT file: Source file for this mzIdentML was a Mascot DAT file.</summary>
    MS_Mascot_DAT_file = 1001199,

    /// <summary>Sequest out file: Source file for this mzIdentML was ONE Sequest out file.</summary>
    MS_Sequest_out_file = 1001200,

    /// <summary>DB MW filter maximum: </summary>
    MS_DB_MW_filter_maximum = 1001201,

    /// <summary>DB MW filter minimum: </summary>
    MS_DB_MW_filter_minimum = 1001202,

    /// <summary>DB PI filter maximum: </summary>
    MS_DB_PI_filter_maximum = 1001203,

    /// <summary>DB PI filter minimum: </summary>
    MS_DB_PI_filter_minimum = 1001204,

    /// <summary>Mascot: </summary>
    MS_Mascot = 1001207,

    /// <summary>Sequest: </summary>
    MS_Sequest = 1001208,

    /// <summary>Phenyx: </summary>
    MS_Phenyx = 1001209,

    /// <summary>mass type settings: The type of mass difference value to be considered by the search engine (monoisotopic or average).</summary>
    MS_mass_type_settings = 1001210,

    /// <summary>parent mass type mono: Mass type setting for parent mass was monoisotopic.</summary>
    MS_parent_mass_type_mono = 1001211,

    /// <summary>parent mass type average: Mass type setting for parent mass was average isotopic.</summary>
    MS_parent_mass_type_average = 1001212,

    /// <summary>search result details: OBSOLETE: Scores and global result characteristics.</summary>
    MS_search_result_details_OBSOLETE = 1001213,

    /// <summary>prot:global FDR: Estimation of global false discovery rate of the proteins.</summary>
    MS_prot_global_FDR = 1001214,

    /// <summary>Sequest:PeptideSp: The SEQUEST result 'Sp' in out file (peptide).</summary>
    MS_Sequest_PeptideSp = 1001215,

    /// <summary>Sequest:PeptideRankSp: The SEQUEST result 'Sp' of 'Rank/Sp' in out file (peptide). Also called 'rsp'.</summary>
    MS_Sequest_PeptideRankSp = 1001217,

    /// <summary>Sequest:PeptideNumber: The SEQUEST result '#' in out file (peptide).</summary>
    MS_Sequest_PeptideNumber = 1001218,

    /// <summary>Sequest:PeptideIdnumber: The SEQUEST result 'Id#' in out file (peptide).</summary>
    MS_Sequest_PeptideIdnumber = 1001219,

    /// <summary>frag: y ion: Fragmentation information, type of product: y ion.</summary>
    MS_frag__y_ion = 1001220,

    /// <summary>fragmentation information: Fragmentation information like ion types.</summary>
    MS_fragmentation_information = 1001221,

    /// <summary>frag: b ion - H2O: Fragmentation information, type of product: b ion without water.</summary>
    MS_frag__b_ion___H2O = 1001222,

    /// <summary>frag: y ion - H2O: Fragmentation information, type of product: y ion without water.</summary>
    MS_frag__y_ion___H2O = 1001223,

    /// <summary>frag: b ion: Fragmentation information, type of product: b ion.</summary>
    MS_frag__b_ion = 1001224,

    /// <summary>product ion m/z: The m/z of the product ion.</summary>
    MS_product_ion_m_z = 1001225,

    /// <summary>fragment ion m/z: The m/z of the product ion.</summary>
    MS_fragment_ion_m_z = MS_product_ion_m_z,

    /// <summary>product ion intensity: The intensity of the product ion.</summary>
    MS_product_ion_intensity = 1001226,

    /// <summary>fragment ion intensity: The intensity of the product ion.</summary>
    MS_fragment_ion_intensity = MS_product_ion_intensity,

    /// <summary>product ion m/z error: The product ion m/z error.</summary>
    MS_product_ion_m_z_error = 1001227,

    /// <summary>frag: x ion: Fragmentation information, type of product: x ion.</summary>
    MS_frag__x_ion = 1001228,

    /// <summary>frag: a ion: Fragmentation information, type of product: a ion.</summary>
    MS_frag__a_ion = 1001229,

    /// <summary>frag: z ion: Fragmentation information, type of product: z ion.</summary>
    MS_frag__z_ion = 1001230,

    /// <summary>frag: c ion: Fragmentation information, type of product: c ion.</summary>
    MS_frag__c_ion = 1001231,

    /// <summary>frag: b ion - NH3: Fragmentation information, type of product: b ion without ammonium ion.</summary>
    MS_frag__b_ion___NH3 = 1001232,

    /// <summary>frag: y ion - NH3: Fragmentation information, type of product: y ion without ammonium ion.</summary>
    MS_frag__y_ion___NH3 = 1001233,

    /// <summary>frag: a ion - H2O: Fragmentation information, type of product: a ion without water.</summary>
    MS_frag__a_ion___H2O = 1001234,

    /// <summary>frag: a ion - NH3: Fragmentation information, type of product: a ion without ammonium.</summary>
    MS_frag__a_ion___NH3 = 1001235,

    /// <summary>frag: d ion: Fragmentation information, type of product: d ion.</summary>
    MS_frag__d_ion = 1001236,

    /// <summary>frag: v ion: Fragmentation information, type of product: v ion.</summary>
    MS_frag__v_ion = 1001237,

    /// <summary>frag: w ion: Fragmentation information, type of product: w ion.</summary>
    MS_frag__w_ion = 1001238,

    /// <summary>frag: immonium ion: Fragmentation information, type of product: immonium ion.</summary>
    MS_frag__immonium_ion = 1001239,

    /// <summary>non-identified ion: Non-identified ion.</summary>
    MS_non_identified_ion = 1001240,

    /// <summary>co-eluting ion: Co-eluting ion.</summary>
    MS_co_eluting_ion = 1001241,

    /// <summary>Sequest out folder: Source file for this mzIdentML was a Sequest folder with its out files.</summary>
    MS_Sequest_out_folder = 1001242,

    /// <summary>Sequest summary: Source file for this mzIdentML was a Sequest summary page (proteins).</summary>
    MS_Sequest_summary = 1001243,

    /// <summary>PerSeptive PKS file: </summary>
    MS_PerSeptive_PKS_file = 1001245,

    /// <summary>Sciex API III file: </summary>
    MS_Sciex_API_III_file = 1001246,

    /// <summary>Bruker XML file: Bruker data exchange xml.</summary>
    MS_Bruker_XML_file = 1001247,

    /// <summary>search input details: </summary>
    MS_search_input_details = 1001249,

    /// <summary>local FDR: Result of quality estimation: the local FDR at the current position of a sorted list.</summary>
    MS_local_FDR = 1001250,

    /// <summary>Trypsin: </summary>
    MS_Trypsin = 1001251,

    /// <summary>DB source EBI: </summary>
    MS_DB_source_EBI = 1001252,

    /// <summary>DB source NCBI: </summary>
    MS_DB_source_NCBI = 1001253,

    /// <summary>DB source UniProt: </summary>
    MS_DB_source_UniProt = 1001254,

    /// <summary>fragment mass type average: Mass type setting for fragment mass was average isotopic.</summary>
    MS_fragment_mass_type_average = 1001255,

    /// <summary>fragment mass type mono: Mass type setting for fragment mass was monoisotopic.</summary>
    MS_fragment_mass_type_mono = 1001256,

    /// <summary>param: v ion: </summary>
    MS_param__v_ion = 1001257,

    /// <summary>param: d ion: </summary>
    MS_param__d_ion = 1001258,

    /// <summary>param: immonium ion: </summary>
    MS_param__immonium_ion = 1001259,

    /// <summary>param: w ion: </summary>
    MS_param__w_ion = 1001260,

    /// <summary>param: x ion: </summary>
    MS_param__x_ion = 1001261,

    /// <summary>param: y ion: </summary>
    MS_param__y_ion = 1001262,

    /// <summary>param: z ion: </summary>
    MS_param__z_ion = 1001263,

    /// <summary>role type: Role of a Person or Organization.</summary>
    MS_role_type = 1001266,

    /// <summary>software vendor: </summary>
    MS_software_vendor = 1001267,

    /// <summary>programmer: </summary>
    MS_programmer = 1001268,

    /// <summary>instrument vendor: </summary>
    MS_instrument_vendor = 1001269,

    /// <summary>lab personnel: </summary>
    MS_lab_personnel = 1001270,

    /// <summary>researcher: </summary>
    MS_researcher = 1001271,

    /// <summary>(?<=R)(?!P): </summary>
    MS_____R____P_ = 1001272,

    /// <summary>(?=[BD]): </summary>
    MS_____BD__ = 1001273,

    /// <summary>(?=[DE]): </summary>
    MS_____DE__ = 1001274,

    /// <summary>ProteinScape SearchEvent: Source data for this mzIdentML was a ProteinScape SearchEvent.</summary>
    MS_ProteinScape_SearchEvent = 1001275,

    /// <summary>ProteinScape Gel: Source data for this mzIdentML was a ProteinScape Gel.</summary>
    MS_ProteinScape_Gel = 1001276,

    /// <summary>decoy DB accession regexp: Specify the regular expression for decoy accession numbers.</summary>
    MS_decoy_DB_accession_regexp = 1001283,

    /// <summary>decoy DB derived from: The name of the database, the search database was derived from.</summary>
    MS_decoy_DB_derived_from = 1001284,

    /// <summary>database IPI_mouse: </summary>
    MS_database_IPI_mouse = 1001285,

    /// <summary>database IPI_rat: </summary>
    MS_database_IPI_rat = 1001286,

    /// <summary>database IPI_zebrafish: </summary>
    MS_database_IPI_zebrafish = 1001287,

    /// <summary>database IPI_chicken: </summary>
    MS_database_IPI_chicken = 1001288,

    /// <summary>database IPI_cow: </summary>
    MS_database_IPI_cow = 1001289,

    /// <summary>database IPI_arabidopsis: </summary>
    MS_database_IPI_arabidopsis = 1001290,

    /// <summary>decoy DB from nr: </summary>
    MS_decoy_DB_from_nr = 1001291,

    /// <summary>decoy DB from IPI_rat: </summary>
    MS_decoy_DB_from_IPI_rat = 1001292,

    /// <summary>decoy DB from IPI_mouse: </summary>
    MS_decoy_DB_from_IPI_mouse = 1001293,

    /// <summary>decoy DB from IPI_arabidopsis: </summary>
    MS_decoy_DB_from_IPI_arabidopsis = 1001294,

    /// <summary>decoy DB from EST: </summary>
    MS_decoy_DB_from_EST = 1001295,

    /// <summary>decoy DB from IPI_zebrafish: </summary>
    MS_decoy_DB_from_IPI_zebrafish = 1001296,

    /// <summary>decoy DB from SwissProt: </summary>
    MS_decoy_DB_from_SwissProt = 1001297,

    /// <summary>decoy DB from IPI_chicken: </summary>
    MS_decoy_DB_from_IPI_chicken = 1001298,

    /// <summary>decoy DB from IPI_cow: </summary>
    MS_decoy_DB_from_IPI_cow = 1001299,

    /// <summary>decoy DB from IPI_human: </summary>
    MS_decoy_DB_from_IPI_human = 1001300,

    /// <summary>protein rank: The rank of the protein in a list sorted by the search engine.</summary>
    MS_protein_rank = 1001301,

    /// <summary>search engine specific input parameter: This branch contains search engine specific input parameters (e.g. Mascot Sig Threshold).</summary>
    MS_search_engine_specific_input_parameter = 1001302,

    /// <summary>Arg-C: Endoproteinase Arg-C.</summary>
    MS_Arg_C = 1001303,

    /// <summary>Clostripain: Endoproteinase Arg-C.</summary>
    MS_Clostripain = MS_Arg_C,

    /// <summary>Asp-N: </summary>
    MS_Asp_N = 1001304,

    /// <summary>Asp-N_ambic: </summary>
    MS_Asp_N_ambic = 1001305,

    /// <summary>Chymotrypsin: </summary>
    MS_Chymotrypsin = 1001306,

    /// <summary>CNBr: Cyanogen bromide.</summary>
    MS_CNBr = 1001307,

    /// <summary>Formic_acid: </summary>
    MS_Formic_acid = 1001308,

    /// <summary>Lys-C: </summary>
    MS_Lys_C = 1001309,

    /// <summary>Lys-C/P: </summary>
    MS_Lys_C_P = 1001310,

    /// <summary>PepsinA: </summary>
    MS_PepsinA = 1001311,

    /// <summary>TrypChymo: </summary>
    MS_TrypChymo = 1001312,

    /// <summary>Trypsin/P: </summary>
    MS_Trypsin_P = 1001313,

    /// <summary>V8-DE: </summary>
    MS_V8_DE = 1001314,

    /// <summary>V8-E: </summary>
    MS_V8_E = 1001315,

    /// <summary>Mascot:SigThreshold: </summary>
    MS_Mascot_SigThreshold = 1001316,

    /// <summary>Mascot:MaxProteinHits: </summary>
    MS_Mascot_MaxProteinHits = 1001317,

    /// <summary>Mascot:ProteinScoringMethod: </summary>
    MS_Mascot_ProteinScoringMethod = 1001318,

    /// <summary>Mascot:MinMSMSThreshold: </summary>
    MS_Mascot_MinMSMSThreshold = 1001319,

    /// <summary>Mascot:ShowHomologousProteinsWithSamePeptides: </summary>
    MS_Mascot_ShowHomologousProteinsWithSamePeptides = 1001320,

    /// <summary>Mascot:ShowHomologousProteinsWithSubsetOfPeptides: </summary>
    MS_Mascot_ShowHomologousProteinsWithSubsetOfPeptides = 1001321,

    /// <summary>Mascot:RequireBoldRed: </summary>
    MS_Mascot_RequireBoldRed = 1001322,

    /// <summary>Mascot:UseUnigeneClustering: </summary>
    MS_Mascot_UseUnigeneClustering = 1001323,

    /// <summary>Mascot:IncludeErrorTolerantMatches: </summary>
    MS_Mascot_IncludeErrorTolerantMatches = 1001324,

    /// <summary>Mascot:ShowDecoyMatches: </summary>
    MS_Mascot_ShowDecoyMatches = 1001325,

    /// <summary>add_others: </summary>
    MS_add_others_OBSOLETE = 1001326,

    /// <summary>OMSSA:evalue: OMSSA E-value.</summary>
    MS_OMSSA_evalue = 1001328,

    /// <summary>OMSSA:pvalue: OMSSA p-value.</summary>
    MS_OMSSA_pvalue = 1001329,

    /// <summary>X!Tandem:expect: The X!Tandem expectation value.</summary>
    MS_X_Tandem_expect = 1001330,

    /// <summary>X!Tandem:hyperscore: The X!Tandem hyperscore.</summary>
    MS_X_Tandem_hyperscore = 1001331,

    /// <summary>(?<=[FYWL])(?!P): </summary>
    MS______FYWL_____P_ = 1001332,

    /// <summary>(?<=M): </summary>
    MS_____M_ = 1001333,

    /// <summary>((?<=D))|((?=D)): </summary>
    MS______D_______D__ = 1001334,

    /// <summary>(?<=K)(?!P): </summary>
    MS_____K____P_ = 1001335,

    /// <summary>(?<=K): </summary>
    MS_____K_ = 1001336,

    /// <summary>(?<=[FL]): </summary>
    MS______FL__ = 1001337,

    /// <summary>(?<=[FYWLKR])(?!P): </summary>
    MS______FYWLKR_____P_ = 1001338,

    /// <summary>(?<=[KR]): </summary>
    MS______KR__ = 1001339,

    /// <summary>(?<=[BDEZ])(?!P): </summary>
    MS______BDEZ_____P_ = 1001340,

    /// <summary>(?<=[EZ])(?!P): </summary>
    MS______EZ_____P_ = 1001341,

    /// <summary>database sequence details: Details about a single database sequence.</summary>
    MS_database_sequence_details = 1001342,

    /// <summary>NA sequence: The sequence is a nucleic acid sequence.</summary>
    MS_NA_sequence = 1001343,

    /// <summary>AA sequence: The sequence is a amino acid sequence.</summary>
    MS_AA_sequence = 1001344,

    /// <summary>mass table source: Children of this term specify the source of the mass table used.</summary>
    MS_mass_table_source = 1001345,

    /// <summary>AAIndex mass table: The masses used in the mass table are taken from AAIndex.</summary>
    MS_AAIndex_mass_table = 1001346,

    /// <summary>database file formats: The children of this term define file formats of the sequence database used.</summary>
    MS_database_file_formats = 1001347,

    /// <summary>FASTA format: The sequence database was stored in the FASTA format.</summary>
    MS_FASTA_format = 1001348,

    /// <summary>ASN.1: The sequence database was stored in the ASN.1 format.</summary>
    MS_ASN_1 = 1001349,

    /// <summary>NCBI *.p*: The sequence database was stored in the NCBI formatdb (*.p*) format.</summary>
    MS_NCBI___p_ = 1001350,

    /// <summary>clustal aln: ClustalW ALN (multiple alignment) format.</summary>
    MS_clustal_aln = 1001351,

    /// <summary>embl em: EMBL entry format.</summary>
    MS_embl_em = 1001352,

    /// <summary>NBRF PIR: The NBRF PIR was used as format.</summary>
    MS_NBRF_PIR = 1001353,

    /// <summary>mass table options: Root node for options for the mass table used.</summary>
    MS_mass_table_options = 1001354,

    /// <summary>peptide descriptions: Descriptions of peptides.</summary>
    MS_peptide_descriptions = 1001355,

    /// <summary>spectrum descriptions: Descriptions of the input spectra.</summary>
    MS_spectrum_descriptions = 1001356,

    /// <summary>spectrum quality descriptions: Description of the quality of the input spectrum.</summary>
    MS_spectrum_quality_descriptions = 1001357,

    /// <summary>msmsEval quality: This term reports the quality of the spectrum assigned by msmsEval.</summary>
    MS_msmsEval_quality = 1001358,

    /// <summary>ambiguous residues: Children of this term describe abiguous residues.</summary>
    MS_ambiguous_residues = 1001359,

    /// <summary>alternate single letter codes: List of standard residue one letter codes which are used to replace a non-standard.</summary>
    MS_alternate_single_letter_codes = 1001360,

    /// <summary>alternate mass: List of masses a non-standard letter code is replaced with.</summary>
    MS_alternate_mass = 1001361,

    /// <summary>number of unmatched peaks: The number of unmatched peaks.</summary>
    MS_number_of_unmatched_peaks = 1001362,

    /// <summary>peptide unique to one protein: </summary>
    MS_peptide_unique_to_one_protein = 1001363,

    /// <summary>pep:global FDR: Estimation of global false discovery rate of the peptides.</summary>
    MS_pep_global_FDR = 1001364,

    /// <summary>frag: internal yb ion: </summary>
    MS_frag__internal_yb_ion = 1001365,

    /// <summary>frag: internal ya ion: </summary>
    MS_frag__internal_ya_ion = 1001366,

    /// <summary>frag: z+1 ion: </summary>
    MS_frag__z_1_ion = 1001367,

    /// <summary>frag: z+2 ion: </summary>
    MS_frag__z_2_ion = 1001368,

    /// <summary>text file: Simple text file of \"m/z [intensity]\" values for a PMF (or single MS-MS) search.</summary>
    MS_text_file = 1001369,

    /// <summary>Mascot:homology threshold: The Mascot result 'homology threshold'.</summary>
    MS_Mascot_homology_threshold = 1001370,

    /// <summary>Mascot:identity threshold: The Mascot result 'identity threshold'.</summary>
    MS_Mascot_identity_threshold = 1001371,

    /// <summary>Sequest:Sequences: </summary>
    MS_Sequest_Sequences = 1001372,

    /// <summary>Sequest:TIC: </summary>
    MS_Sequest_TIC = 1001373,

    /// <summary>Sequest:Sum: </summary>
    MS_Sequest_Sum = 1001374,

    /// <summary>Phenyx:Instrument Type: </summary>
    MS_Phenyx_Instrument_Type = 1001375,

    /// <summary>Phenyx:Scoring Model: </summary>
    MS_Phenyx_Scoring_Model = 1001376,

    /// <summary>Phenyx:Default Parent Charge: </summary>
    MS_Phenyx_Default_Parent_Charge = 1001377,

    /// <summary>Phenyx:Trust Parent Charge: </summary>
    MS_Phenyx_Trust_Parent_Charge = 1001378,

    /// <summary>Phenyx:Turbo: </summary>
    MS_Phenyx_Turbo = 1001379,

    /// <summary>Phenyx:Turbo:ErrorTol: </summary>
    MS_Phenyx_Turbo_ErrorTol = 1001380,

    /// <summary>Phenyx:Turbo:Coverage: </summary>
    MS_Phenyx_Turbo_Coverage = 1001381,

    /// <summary>Phenyx:Turbo:Series: </summary>
    MS_Phenyx_Turbo_Series = 1001382,

    /// <summary>Phenyx:MinPepLength: </summary>
    MS_Phenyx_MinPepLength = 1001383,

    /// <summary>Phenyx:MinPepzscore: </summary>
    MS_Phenyx_MinPepzscore = 1001384,

    /// <summary>Phenyx:MaxPepPvalue: </summary>
    MS_Phenyx_MaxPepPvalue = 1001385,

    /// <summary>Phenyx:AC Score: </summary>
    MS_Phenyx_AC_Score = 1001386,

    /// <summary>Phenyx:Conflict Resolution: </summary>
    MS_Phenyx_Conflict_Resolution = 1001387,

    /// <summary>Phenyx:AC: </summary>
    MS_Phenyx_AC = 1001388,

    /// <summary>Phenyx:ID: </summary>
    MS_Phenyx_ID = 1001389,

    /// <summary>Phenyx:Score: </summary>
    MS_Phenyx_Score = 1001390,

    /// <summary>Phenyx:Peptides1: First number of phenyx result \"#Peptides\".</summary>
    MS_Phenyx_Peptides1 = 1001391,

    /// <summary>Phenyx:Peptides2: Second number of phenyx result \"#Peptides\".</summary>
    MS_Phenyx_Peptides2 = 1001392,

    /// <summary>Phenyx:Auto: </summary>
    MS_Phenyx_Auto = 1001393,

    /// <summary>Phenyx:User: </summary>
    MS_Phenyx_User = 1001394,

    /// <summary>Phenyx:Pepzscore: </summary>
    MS_Phenyx_Pepzscore = 1001395,

    /// <summary>Phenyx:PepPvalue: </summary>
    MS_Phenyx_PepPvalue = 1001396,

    /// <summary>Phenyx:NumberOfMC: </summary>
    MS_Phenyx_NumberOfMC = 1001397,

    /// <summary>Phenyx:Modif: </summary>
    MS_Phenyx_Modif = 1001398,

    /// <summary>OMSSA csv file: Source file for this mzIdentML was an OMSSA csv file.</summary>
    MS_OMSSA_csv_file = 1001399,

    /// <summary>OMSSA xml file: Source file for this mzIdentML was an OMSSA xml file.</summary>
    MS_OMSSA_xml_file = 1001400,

    /// <summary>X!Tandem xml file: Source file for this mzIdentML was an X!Tandem xml file.</summary>
    MS_X_Tandem_xml_file = 1001401,

    /// <summary>spectrum identification result details: This subsection describes terms which can describe details of spectrum identification results.</summary>
    MS_spectrum_identification_result_details = 1001405,

    /// <summary>param: internal yb ion: </summary>
    MS_param__internal_yb_ion = 1001406,

    /// <summary>param: internal ya ion: </summary>
    MS_param__internal_ya_ion = 1001407,

    /// <summary>param: z+1 ion: </summary>
    MS_param__z_1_ion = 1001408,

    /// <summary>param: z+2 ion: </summary>
    MS_param__z_2_ion = 1001409,

    /// <summary>translation start codons: The translation start codons used to translate the nucleotides to amino acids.</summary>
    MS_translation_start_codons = 1001410,

    /// <summary>search tolerance specification: </summary>
    MS_search_tolerance_specification = 1001411,

    /// <summary>search tolerance plus value: </summary>
    MS_search_tolerance_plus_value = 1001412,

    /// <summary>search tolerance minus value: </summary>
    MS_search_tolerance_minus_value = 1001413,

    /// <summary>MGF scans: OBSOLETE: replaced by MS:1000797 (peak list scans): This term can hold the scans attribute from an MGF input file.</summary>
    MS_MGF_scans_OBSOLETE = 1001414,

    /// <summary>MGF raw scans: OBSOLETE: replaced by MS:1000798 (peak list raw scans): This term can hold the raw scans attribute from an MGF input file.</summary>
    MS_MGF_raw_scans_OBSOLETE = 1001415,

    /// <summary>spectrum title: OBSOLETE: replaced by MS:1000796 (spectrum title): Holds the spectrum title from different input file formats, e.g. MGF TITLE.</summary>
    MS_spectrum_title_OBSOLETE = 1001416,

    /// <summary>SpectraST:dot: SpectraST dot product of two spectra, measuring spectral similarity.</summary>
    MS_SpectraST_dot = 1001417,

    /// <summary>SpectraST:dot_bias: SpectraST measure of how much of the dot product is dominated by a few peaks.</summary>
    MS_SpectraST_dot_bias = 1001418,

    /// <summary>SpectraST:discriminant score F: SpectraST spectrum score.</summary>
    MS_SpectraST_discriminant_score_F = 1001419,

    /// <summary>SpectraST:delta: SpectraST normalised difference between dot product of top hit and runner-up.</summary>
    MS_SpectraST_delta = 1001420,

    /// <summary>pepXML file: Source file for this mzIdentML was a pepXML file.</summary>
    MS_pepXML_file = 1001421,

    /// <summary>protXML file: Source file for this mzIdentML was a protXML file.</summary>
    MS_protXML_file = 1001422,

    /// <summary>translation table description: A URL that describes the translation table used to translate the nucleotides to amino acids.</summary>
    MS_translation_table_description = 1001423,

    /// <summary>ProteinExtractor:Methodname: </summary>
    MS_ProteinExtractor_Methodname = 1001424,

    /// <summary>ProteinExtractor:GenerateNonRedundant: </summary>
    MS_ProteinExtractor_GenerateNonRedundant = 1001425,

    /// <summary>ProteinExtractor:IncludeIdentified: </summary>
    MS_ProteinExtractor_IncludeIdentified = 1001426,

    /// <summary>ProteinExtractor:MaxNumberOfProteins: </summary>
    MS_ProteinExtractor_MaxNumberOfProteins = 1001427,

    /// <summary>ProteinExtractor:MaxProteinMass: </summary>
    MS_ProteinExtractor_MaxProteinMass = 1001428,

    /// <summary>ProteinExtractor:MinNumberOfPeptides: </summary>
    MS_ProteinExtractor_MinNumberOfPeptides = 1001429,

    /// <summary>ProteinExtractor:UseMascot: </summary>
    MS_ProteinExtractor_UseMascot = 1001430,

    /// <summary>ProteinExtractor:MascotPeptideScoreThreshold: </summary>
    MS_ProteinExtractor_MascotPeptideScoreThreshold = 1001431,

    /// <summary>ProteinExtractor:MascotUniqueScore: </summary>
    MS_ProteinExtractor_MascotUniqueScore = 1001432,

    /// <summary>ProteinExtractor:MascotUseIdentityScore: </summary>
    MS_ProteinExtractor_MascotUseIdentityScore = 1001433,

    /// <summary>ProteinExtractor:MascotWeighting: </summary>
    MS_ProteinExtractor_MascotWeighting = 1001434,

    /// <summary>ProteinExtractor:UseSequest: </summary>
    MS_ProteinExtractor_UseSequest = 1001435,

    /// <summary>ProteinExtractor:SequestPeptideScoreThreshold: </summary>
    MS_ProteinExtractor_SequestPeptideScoreThreshold = 1001436,

    /// <summary>ProteinExtractor:SequestUniqueScore: </summary>
    MS_ProteinExtractor_SequestUniqueScore = 1001437,

    /// <summary>ProteinExtractor:SequestWeighting: </summary>
    MS_ProteinExtractor_SequestWeighting = 1001438,

    /// <summary>ProteinExtractor:UseProteinSolver: </summary>
    MS_ProteinExtractor_UseProteinSolver = 1001439,

    /// <summary>ProteinExtractor:ProteinSolverPeptideScoreThreshold: </summary>
    MS_ProteinExtractor_ProteinSolverPeptideScoreThreshold = 1001440,

    /// <summary>ProteinExtractor:ProteinSolverUniqueScore: </summary>
    MS_ProteinExtractor_ProteinSolverUniqueScore = 1001441,

    /// <summary>ProteinExtractor:ProteinSolverWeighting: </summary>
    MS_ProteinExtractor_ProteinSolverWeighting = 1001442,

    /// <summary>ProteinExtractor:UsePhenyx: </summary>
    MS_ProteinExtractor_UsePhenyx = 1001443,

    /// <summary>ProteinExtractor:PhenyxPeptideScoreThreshold: </summary>
    MS_ProteinExtractor_PhenyxPeptideScoreThreshold = 1001444,

    /// <summary>ProteinExtractor:PhenyxUniqueScore: </summary>
    MS_ProteinExtractor_PhenyxUniqueScore = 1001445,

    /// <summary>ProteinExtractor:PhenyxWeighting: </summary>
    MS_ProteinExtractor_PhenyxWeighting = 1001446,

    /// <summary>prot:FDR threshold: False-discovery rate threshold for proteins.</summary>
    MS_prot_FDR_threshold = 1001447,

    /// <summary>pep:FDR threshold: False-discovery rate threshold for peptides.</summary>
    MS_pep_FDR_threshold = 1001448,

    /// <summary>OMSSA e-value threshold: Threshold for OMSSA e-value for quality estimation.</summary>
    MS_OMSSA_e_value_threshold = 1001449,

    /// <summary>decoy DB details: Details of decoy generation and database structure.</summary>
    MS_decoy_DB_details = 1001450,

    /// <summary>decoy DB generation algorithm: Name of algorithm used for decoy generation.</summary>
    MS_decoy_DB_generation_algorithm = 1001451,

    /// <summary>decoy DB type shuffle: Decoy type: Amino acids of protein sequences are used in a random order.</summary>
    MS_decoy_DB_type_shuffle = 1001452,

    /// <summary>DB composition only decoy: Decoy database composition: database contains only decoy entries.</summary>
    MS_DB_composition_only_decoy = 1001453,

    /// <summary>quality estimation with implicite decoy sequences: Decoy entries are generated during the search, not explicitly stored in a database (like Mascot Decoy).</summary>
    MS_quality_estimation_with_implicite_decoy_sequences = 1001454,

    /// <summary>acquisition software: Acquisition software.</summary>
    MS_acquisition_software = 1001455,

    /// <summary>analysis software: Analysis software.</summary>
    MS_analysis_software = 1001456,

    /// <summary>data processing software: Conversion software.</summary>
    MS_data_processing_software = 1001457,

    /// <summary>spectrum generation information: Vocabularies describing the spectrum generation information.</summary>
    MS_spectrum_generation_information = 1001458,

    /// <summary>file format: Format of data files.</summary>
    MS_file_format = 1001459,

    /// <summary>unknown modification: This term should be given if the modification was unknown.</summary>
    MS_unknown_modification = 1001460,

    /// <summary>greylag: Greylag identification software, see greylag.org.</summary>
    MS_greylag = 1001461,

    /// <summary>PEFF format: The sequence database was stored in the PEFF (PSI enhanced FastA file) format.</summary>
    MS_PEFF_format = 1001462,

    /// <summary>Phenyx XML format: </summary>
    MS_Phenyx_XML_format = 1001463,

    /// <summary>DTASelect file: </summary>
    MS_DTASelect_file = 1001464,

    /// <summary>MS2 file: MS2 file.</summary>
    MS_MS2_file = 1001466,

    /// <summary>taxonomy: NCBI TaxID: This term is used if a NCBI TaxID is specified, e.g. 9606 for Homo sapiens.</summary>
    MS_taxonomy__NCBI_TaxID = 1001467,

    /// <summary>taxonomy: common name: This term is used if a common name is specified, e.g. human. Recommend using MS:1001467 (taxonomy: NCBI TaxID) where possible.</summary>
    MS_taxonomy__common_name = 1001468,

    /// <summary>taxonomy: scientific name: This term is used if a scientific name is specified, e.g. Homo sapiens. Recommend using MS:1001467 (taxonomy: NCBI TaxID) where possible.</summary>
    MS_taxonomy__scientific_name = 1001469,

    /// <summary>taxonomy: Swiss-Prot ID: This term is used if a swiss prot taxonomy id is specified, e.g. Human. Recommend using MS:1001467 (taxonomy: NCBI TaxID) where possible.</summary>
    MS_taxonomy__Swiss_Prot_ID = 1001470,

    /// <summary>peptide modification details: The children of this term can be used to describe modifications.</summary>
    MS_peptide_modification_details = 1001471,

    /// <summary>selected ion monitoring chromatogram: Chromatogram created by creating an array of the measurements of a selectively monitored ion at each time point.</summary>
    MS_selected_ion_monitoring_chromatogram = 1001472,

    /// <summary>SIM chromatogram: Chromatogram created by creating an array of the measurements of a selectively monitored ion at each time point.</summary>
    MS_SIM_chromatogram = MS_selected_ion_monitoring_chromatogram,

    /// <summary>selected reaction monitoring chromatogram: Chromatogram created by creating an array of the measurements of a selectively monitored reaction at each time point.</summary>
    MS_selected_reaction_monitoring_chromatogram = 1001473,

    /// <summary>SRM chromatogram: Chromatogram created by creating an array of the measurements of a selectively monitored reaction at each time point.</summary>
    MS_SRM_chromatogram = MS_selected_reaction_monitoring_chromatogram,

    /// <summary>consecutive reaction monitoring chromatogram: Chromatogram created by creating an array of the measurements of a series of monitored reactions at each time point.</summary>
    MS_consecutive_reaction_monitoring_chromatogram_OBSOLETE = 1001474,

    /// <summary>CRM chromatogram: Chromatogram created by creating an array of the measurements of a series of monitored reactions at each time point.</summary>
    MS_CRM_chromatogram_OBSOLETE = MS_consecutive_reaction_monitoring_chromatogram_OBSOLETE,

    /// <summary>OMSSA: Open Mass Spectrometry Search Algorithm was used to analyze the spectra.</summary>
    MS_OMSSA = 1001475,

    /// <summary>X!Tandem: X!Tandem was used to analyze the spectra.</summary>
    MS_X_Tandem = 1001476,

    /// <summary>SpectraST: SpectraST was used to analyze the spectra.</summary>
    MS_SpectraST = 1001477,

    /// <summary>Mascot Parser: Mascot Parser was used to analyze the spectra.</summary>
    MS_Mascot_Parser = 1001478,

    /// <summary>null-terminated ASCII string: Sequence of zero or more non-zero ASCII characters terminated by a single null (0) byte.</summary>
    MS_null_terminated_ASCII_string = 1001479,

    /// <summary>AB SCIEX TOF/TOF nativeID format: jobRun=xsd:nonNegativeInteger spotLabel=xsd:string spectrum=xsd:nonNegativeInteger.</summary>
    MS_AB_SCIEX_TOF_TOF_nativeID_format = 1001480,

    /// <summary>AB SCIEX TOF/TOF database: Applied Biosystems/MDS Analytical Technologies TOF/TOF instrument database.</summary>
    MS_AB_SCIEX_TOF_TOF_database = 1001481,

    /// <summary>TOF/TOF 5800: AB SCIEX or Applied Biosystems|MDS Analytical Technologies AB SCIEX TOF/TOF 5800 Analyzer.</summary>
    MS_TOF_TOF_5800 = 1001482,

    /// <summary>AB SCIEX TOF/TOF Series Explorer Software: AB SCIEX or Applied Biosystems software for TOF/TOF data acquisition and analysis.</summary>
    MS_AB_SCIEX_TOF_TOF_Series_Explorer_Software = 1001483,

    /// <summary>intensity normalization: Normalization of data point intensities.</summary>
    MS_intensity_normalization = 1001484,

    /// <summary>m/z calibration: Calibration of data point m/z positions.</summary>
    MS_m_z_calibration = 1001485,

    /// <summary>data filtering: Filtering out part of the data.</summary>
    MS_data_filtering = 1001486,

    /// <summary>ProteinExtractor: An algorithm for protein determination/assembly integrated into Bruker's ProteinScape.</summary>
    MS_ProteinExtractor = 1001487,

    /// <summary>Mascot Distiller: Mascot Distiller.</summary>
    MS_Mascot_Distiller = 1001488,

    /// <summary>Mascot Integra: Mascot Integra.</summary>
    MS_Mascot_Integra = 1001489,

    /// <summary>Percolator: Percolator.</summary>
    MS_Percolator = 1001490,

    /// <summary>percolator:Q value: Percolator:Q value.</summary>
    MS_percolator_Q_value = 1001491,

    /// <summary>percolator:score: Percolator:score.</summary>
    MS_percolator_score = 1001492,

    /// <summary>percolaror:PEP: Posterior error probability.</summary>
    MS_percolaror_PEP = 1001493,

    /// <summary>no threshold: In case no threshold was used.</summary>
    MS_no_threshold = 1001494,

    /// <summary>ProteinScape:SearchResultId: The SearchResultId of this peptide as SearchResult in the ProteinScape database.</summary>
    MS_ProteinScape_SearchResultId = 1001495,

    /// <summary>ProteinScape:SearchEventId: The SearchEventId of the SearchEvent in the ProteinScape database.</summary>
    MS_ProteinScape_SearchEventId = 1001496,

    /// <summary>ProteinScape:ProfoundProbability: The Profound probability score stored by ProteinScape.</summary>
    MS_ProteinScape_ProfoundProbability = 1001497,

    /// <summary>Profound:z value: The Profound z value.</summary>
    MS_Profound_z_value = 1001498,

    /// <summary>Profound:Cluster: The Profound cluster score.</summary>
    MS_Profound_Cluster = 1001499,

    /// <summary>Profound:ClusterRank: The Profound cluster rank.</summary>
    MS_Profound_ClusterRank = 1001500,

    /// <summary>MSFit:Mowse score: The MSFit Mowse score.</summary>
    MS_MSFit_Mowse_score = 1001501,

    /// <summary>Sonar:Score: The Sonar score.</summary>
    MS_Sonar_Score = 1001502,

    /// <summary>ProteinScape:PFFSolverExp: The ProteinSolver exp value stored by ProteinScape.</summary>
    MS_ProteinScape_PFFSolverExp = 1001503,

    /// <summary>ProteinScape:PFFSolverScore: The ProteinSolver score stored by ProteinScape.</summary>
    MS_ProteinScape_PFFSolverScore = 1001504,

    /// <summary>ProteinScape:IntensityCoverage: The intensity coverage of the identified peaks in the spectrum calculated by ProteinScape.</summary>
    MS_ProteinScape_IntensityCoverage = 1001505,

    /// <summary>ProteinScape:SequestMetaScore: The Sequest meta score calculated by ProteinScape from the original Sequest scores.</summary>
    MS_ProteinScape_SequestMetaScore = 1001506,

    /// <summary>ProteinExtractor:Score: The score calculated by ProteinExtractor.</summary>
    MS_ProteinExtractor_Score = 1001507,

    /// <summary>Agilent MassHunter nativeID format: scanId=xsd:nonNegativeInteger.</summary>
    MS_Agilent_MassHunter_nativeID_format = 1001508,

    /// <summary>Agilent MassHunter file: A data file found in an Agilent MassHunter directory which contains raw data acquired by an Agilent mass spectrometer.</summary>
    MS_Agilent_MassHunter_file = 1001509,

    /// <summary>TSQ Vantage: TSQ Vantage.</summary>
    MS_TSQ_Vantage = 1001510,

    /// <summary>Sequence database filter types: Filter types which are used to filter a sequence database.</summary>
    MS_Sequence_database_filter_types = 1001511,

    /// <summary>Sequence database filters: Sequence database filters which actually can contains values, e.g. to limit PI value of the sequences used to search.</summary>
    MS_Sequence_database_filters = 1001512,

    /// <summary>DB sequence filter pattern: </summary>
    MS_DB_sequence_filter_pattern = 1001513,

    /// <summary>DB accession filter string: </summary>
    MS_DB_accession_filter_string = 1001514,

    /// <summary>frag: c ion - H2O: Fragmentation information, type of product: c ion without water.</summary>
    MS_frag__c_ion___H2O = 1001515,

    /// <summary>frag: c ion - NH3: Fragmentation information, type of product: c ion without ammonia.</summary>
    MS_frag__c_ion___NH3 = 1001516,

    /// <summary>frag: z ion - H2O: Fragmentation information, type of product: z ion without water.</summary>
    MS_frag__z_ion___H2O = 1001517,

    /// <summary>frag: z ion - NH3: Fragmentation information, type of product: z ion without ammonia.</summary>
    MS_frag__z_ion___NH3 = 1001518,

    /// <summary>frag: x ion - H2O: Fragmentation information, type of product: x ion without water.</summary>
    MS_frag__x_ion___H2O = 1001519,

    /// <summary>frag: x ion - NH3: Fragmentation information, type of product: x ion without ammonia.</summary>
    MS_frag__x_ion___NH3 = 1001520,

    /// <summary>frag: precursor ion - H2O: Fragmentation information, type of product: precursor ion without water.</summary>
    MS_frag__precursor_ion___H2O = 1001521,

    /// <summary>frag: precursor ion - NH3: Fragmentation information, type of product: precursor ion without ammonia.</summary>
    MS_frag__precursor_ion___NH3 = 1001522,

    /// <summary>frag: precursor ion: Fragmentation information, type of product: precursor ion.</summary>
    MS_frag__precursor_ion = 1001523,

    /// <summary>fragment neutral loss: This term can describe a neutral loss m/z value that is lost from an ion.</summary>
    MS_fragment_neutral_loss = 1001524,

    /// <summary>precursor neutral loss: This term can describe a neutral loss m/z value that is lost from an ion.</summary>
    MS_precursor_neutral_loss = 1001525,

    /// <summary>spectrum from database integer nativeID format: databasekey=xsd:long.</summary>
    MS_spectrum_from_database_integer_nativeID_format = 1001526,

    /// <summary>Proteinscape spectra: Spectra from Bruker/Protagen Proteinscape database.</summary>
    MS_Proteinscape_spectra = 1001527,

    /// <summary>Mascot query number: query=xsd:nonNegativeInteger.</summary>
    MS_Mascot_query_number = 1001528,

    /// <summary>spectra data details: Child-terms contain information to map the results back to spectra.</summary>
    MS_spectra_data_details = 1001529,

    /// <summary>mzML unique identifier: mzMLid=xsd:IDREF.</summary>
    MS_mzML_unique_identifier = 1001530,

    /// <summary>spectrum from ProteinScape database nativeID format: databasekey=xsd:long.</summary>
    MS_spectrum_from_ProteinScape_database_nativeID_format = 1001531,

    /// <summary>spectrum from database string nativeID format: databasekey=xsd:string.</summary>
    MS_spectrum_from_database_string_nativeID_format = 1001532,

    /// <summary>Bruker Daltonics esquire series: </summary>
    MS_Bruker_Daltonics_esquire_series = 1001533,

    /// <summary>Bruker Daltonics flex series: </summary>
    MS_Bruker_Daltonics_flex_series = 1001534,

    /// <summary>Bruker Daltonics BioTOF series: </summary>
    MS_Bruker_Daltonics_BioTOF_series = 1001535,

    /// <summary>Bruker Daltonics micrOTOF series: </summary>
    MS_Bruker_Daltonics_micrOTOF_series = 1001536,

    /// <summary>BioTOF: Bruker Daltonics' BioTOF: ESI TOF.</summary>
    MS_BioTOF = 1001537,

    /// <summary>BioTOF III: Bruker Daltonics' BioTOF III: ESI TOF.</summary>
    MS_BioTOF_III = 1001538,

    /// <summary>UltroTOF-Q: Bruker Daltonics' UltroTOF-Q: ESI Q-TOF (MALDI optional).</summary>
    MS_UltroTOF_Q = 1001539,

    /// <summary>micrOTOF II: Bruker Daltonics' micrOTOF II: ESI TOF, Nanospray, APCI, APPI.</summary>
    MS_micrOTOF_II = 1001540,

    /// <summary>maXis: Bruker Daltonics' maXis: ESI Q-TOF, Nanospray, APCI, APPI.</summary>
    MS_maXis = 1001541,

    /// <summary>amaZon ETD: Bruker Daltonics' amaZon ETD: ESI quadrupole ion trap, Nanospray, APCI, APPI, ETD, PTR.</summary>
    MS_amaZon_ETD = 1001542,

    /// <summary>microflex LRF: Bruker Daltonics' microflex LRF: MALDI TOF.</summary>
    MS_microflex_LRF = 1001543,

    /// <summary>ultrafleXtreme: Bruker Daltonics' ultrafleXtreme: MALDI TOF.</summary>
    MS_ultrafleXtreme = 1001544,

    /// <summary>Bruker Daltonics amaZon series: </summary>
    MS_Bruker_Daltonics_amaZon_series = 1001545,

    /// <summary>amaZon X: Bruker Daltonics' amaZon X: ESI quadrupole ion trap, APCI, APPI, ETD, PTR.</summary>
    MS_amaZon_X = 1001546,

    /// <summary>Bruker Daltonics maXis series: </summary>
    MS_Bruker_Daltonics_maXis_series = 1001547,

    /// <summary>Bruker Daltonics solarix series: Bruker Daltonics' solarix: ESI quadrupole ion trap, APCI, APPI, ETD, PTR.</summary>
    MS_Bruker_Daltonics_solarix_series = 1001548,

    /// <summary>solariX: Bruker Daltonics' solariX: ESI, MALDI, Qh-FT_ICR.</summary>
    MS_solariX = 1001549,

    /// <summary>microflex II: Bruker Daltonics' microflex II: MALDI TOF.</summary>
    MS_microflex_II = 1001550,

    /// <summary>autoflex II TOF/TOF: Bruker Daltonics' autoflex II TOF/TOF: MALDI TOF.</summary>
    MS_autoflex_II_TOF_TOF = 1001553,

    /// <summary>autoflex III TOF/TOF smartbeam: Bruker Daltonics' autoflex III TOF/TOF smartbeam: MALDI TOF.</summary>
    MS_autoflex_III_TOF_TOF_smartbeam = 1001554,

    /// <summary>autoflex: Bruker Daltonics' autoflex: MALDI TOF.</summary>
    MS_autoflex = 1001555,

    /// <summary>Bruker Daltonics apex series: </summary>
    MS_Bruker_Daltonics_apex_series = 1001556,

    /// <summary>Shimadzu Biotech software: </summary>
    MS_Shimadzu_Biotech_software = 1001557,

    /// <summary>MALDI Solutions: Shimadzu Biotech software for data acquisition, processing, and analysis.</summary>
    MS_MALDI_Solutions = 1001558,

    /// <summary>AB SCIEX TOF/TOF T2D nativeID format: file=xsd:IDREF.</summary>
    MS_AB_SCIEX_TOF_TOF_T2D_nativeID_format = 1001559,

    /// <summary>AB SCIEX TOF/TOF T2D file: Applied Biosystems/MDS Analytical Technologies TOF/TOF instrument export format.</summary>
    MS_AB_SCIEX_TOF_TOF_T2D_file = 1001560,

    /// <summary>Scaffold: Scaffold analysis software, see www.proteomesoftware.com.</summary>
    MS_Scaffold = 1001561,

    /// <summary>Scaffold nativeID format: Scaffold native ID format.</summary>
    MS_Scaffold_nativeID_format = 1001562,

    /// <summary>Sequest SQT: Source file for this mzIdentML was a Sequest SQT.</summary>
    MS_Sequest_SQT = 1001563,

    /// <summary>Discoverer MSF: Source file for this mzIdentML was in Thermo Scientific Discoverer MSF format.</summary>
    MS_Discoverer_MSF = 1001564,

    /// <summary>IdentityE XML: Source file for this mzIdentML was in Waters IdentityE XML format.</summary>
    MS_IdentityE_XML = 1001565,

    /// <summary>ProteinLynx XML: Source file for this mzIdentML was in Waters ProteinLynx XML format.</summary>
    MS_ProteinLynx_XML = 1001566,

    /// <summary>SpectrumMill directories: Source file for this mzIdentML was in Agilent SpectrumMill directory format.</summary>
    MS_SpectrumMill_directories = 1001567,

    /// <summary>Scaffold:Peptide Probability: Scaffold peptide probability score.</summary>
    MS_Scaffold_Peptide_Probability = 1001568,

    /// <summary>IdentityE Score: Waters IdentityE peptide score.</summary>
    MS_IdentityE_Score = 1001569,

    /// <summary>ProteinLynx:Log Likelihood: ProteinLynx log likelihood score.</summary>
    MS_ProteinLynx_Log_Likelihood = 1001570,

    /// <summary>ProteinLynx:Ladder Score: Waters ProteinLynx Ladder score.</summary>
    MS_ProteinLynx_Ladder_Score = 1001571,

    /// <summary>SpectrumMill:Score: Spectrum mill peptide score.</summary>
    MS_SpectrumMill_Score = 1001572,

    /// <summary>SpectrumMill:SPI: SpectrumMill SPI score (%).</summary>
    MS_SpectrumMill_SPI = 1001573,

    /// <summary>report only spectra assigned to identified proteins: .</summary>
    MS_report_only_spectra_assigned_to_identified_proteins = 1001574,

    /// <summary>Scaffold: Minimum Peptide Count: Minimum number of peptides a protein must have to be accepted.</summary>
    MS_Scaffold__Minimum_Peptide_Count = 1001575,

    /// <summary>Scaffold: Minimum Protein Probability: Minimum protein probability a protein must have to be accepted.</summary>
    MS_Scaffold__Minimum_Protein_Probability = 1001576,

    /// <summary>Scaffold: Minimum Peptide Probability: Minimum probability a peptide must have to be accepted for protein scoring.</summary>
    MS_Scaffold__Minimum_Peptide_Probability = 1001577,

    /// <summary>minimum number of enzymatic termini: Minimum number of enzymatic termini a pepide must have to be accepted.</summary>
    MS_minimum_number_of_enzymatic_termini = 1001578,

    /// <summary>Scaffold:Protein Probability: Scaffold protein probability score.</summary>
    MS_Scaffold_Protein_Probability = 1001579,

    /// <summary>SpectrumMill:Discriminant Score: Discriminant score from Agilent SpectrumMill software.</summary>
    MS_SpectrumMill_Discriminant_Score = 1001580,

    /// <summary>FAIMS compensation voltage: The DC potential applied to the asymmetric waveform in FAIMS that compensates for the difference between high and low field mobility of an ion.</summary>
    MS_FAIMS_compensation_voltage = 1001581,

    /// <summary>FAIMS CV: The DC potential applied to the asymmetric waveform in FAIMS that compensates for the difference between high and low field mobility of an ion.</summary>
    MS_FAIMS_CV = MS_FAIMS_compensation_voltage,

    /// <summary>XCMS: Bioconductor package XCMS for preprocessing high-throughput, untargeted analyte profiling data.</summary>
    MS_XCMS = 1001582,

    /// <summary>MaxQuant: MaxQuant is a quantitative proteomics software package designed for analyzing large mass spectrometric data sets. It is specifically aimed at high resolution MS data.</summary>
    MS_MaxQuant = 1001583,

    /// <summary>combined pmf + ms-ms search: Search that includes data from Peptide Mass Fingerprint (PMF) and MS/MS (aka Peptide Fragment Fingerprint - PFF).</summary>
    MS_combined_pmf___ms_ms_search = 1001584,

    /// <summary>MyriMatch: Tabb Lab software for directly comparing peptides in a database to tandem mass spectra.</summary>
    MS_MyriMatch = 1001585,

    /// <summary>DirecTag: Tabb Lab software for generating sequence tags from tandem mass spectra.</summary>
    MS_DirecTag = 1001586,

    /// <summary>TagRecon: Tabb Lab software for reconciling sequence tags to a protein database.</summary>
    MS_TagRecon = 1001587,

    /// <summary>Pepitome: Tabb Lab software for spectral library searches on tandem mass spectra.</summary>
    MS_Pepitome = 1001588,

    /// <summary>MyriMatch:MVH: Using the multivariate hypergeometric distribution and a peak list divided into several intensity classes, this score is the negative natural log probability that the predicted peaks matched to experimental peaks by random chance.</summary>
    MS_MyriMatch_MVH = 1001589,

    /// <summary>Pepitome:MVH: Using the multivariate hypergeometric distribution and a peak list divided into several intensity classes, this score is the negative natural log probability that the predicted peaks matched to experimental peaks by random chance.</summary>
    MS_Pepitome_MVH = MS_MyriMatch_MVH,

    /// <summary>TagRecon:MVH: Using the multivariate hypergeometric distribution and a peak list divided into several intensity classes, this score is the negative natural log probability that the predicted peaks matched to experimental peaks by random chance.</summary>
    MS_TagRecon_MVH = MS_MyriMatch_MVH,

    /// <summary>MyriMatch:mzFidelity: The negative natural log probability that predicted peaks match to experimental peaks by random chance by scoring the m/z delta of the matches in a multinomial distribution.</summary>
    MS_MyriMatch_mzFidelity = 1001590,

    /// <summary>Pepitome:mzFidelity: The negative natural log probability that predicted peaks match to experimental peaks by random chance by scoring the m/z delta of the matches in a multinomial distribution.</summary>
    MS_Pepitome_mzFidelity = MS_MyriMatch_mzFidelity,

    /// <summary>TagRecon:mzFidelity: The negative natural log probability that predicted peaks match to experimental peaks by random chance by scoring the m/z delta of the matches in a multinomial distribution.</summary>
    MS_TagRecon_mzFidelity = MS_MyriMatch_mzFidelity,

    /// <summary>anchor protein: A representative protein selected from a set of sequence same-set or spectrum same-set proteins.</summary>
    MS_anchor_protein = 1001591,

    /// <summary>family member protein: A protein with significant homology to another protein, but some distinguishing peptide matches.</summary>
    MS_family_member_protein = 1001592,

    /// <summary>group member with undefined relationship OR ortholog protein: TO ENDETAIL: a really generic relationship OR ortholog protein.</summary>
    MS_group_member_with_undefined_relationship_OR_ortholog_protein = 1001593,

    /// <summary>sequence same-set protein: A protein which is indistinguishable or equivalent to another protein, having matches to an identical set of peptide sequences.</summary>
    MS_sequence_same_set_protein = 1001594,

    /// <summary>spectrum same-set protein: A protein which is indistinguishable or equivalent to another protein, having matches to a set of peptide sequences that cannot be distinguished using the evidence in the mass spectra.</summary>
    MS_spectrum_same_set_protein = 1001595,

    /// <summary>sequence sub-set protein: A protein with a sub-set of the peptide sequence matches for another protein, and no distinguishing peptide matches.</summary>
    MS_sequence_sub_set_protein = 1001596,

    /// <summary>spectrum sub-set protein: A protein with a sub-set of the matched spectra for another protein, where the matches cannot be distinguished using the evidence in the mass spectra, and no distinguishing peptide matches.</summary>
    MS_spectrum_sub_set_protein = 1001597,

    /// <summary>sequence subsumable protein: A sequence same-set or sequence sub-set protein where the matches are distributed across two or more proteins.</summary>
    MS_sequence_subsumable_protein = 1001598,

    /// <summary>spectrum subsumable protein: A spectrum same-set or spectrum sub-set protein where the matches are distributed across two or more proteins.</summary>
    MS_spectrum_subsumable_protein = 1001599,

    /// <summary>Protein Inference Confidence Category: Confidence category of inferred protein (conclusive, non conclusive, ambiguous group or indistinguishable).</summary>
    MS_Protein_Inference_Confidence_Category = 1001600,

    /// <summary>ProteomeDiscoverer:Spectrum Files:Raw File names: Name and location of the .raw file or files.</summary>
    MS_ProteomeDiscoverer_Spectrum_Files_Raw_File_names_OBSOLETE = 1001601,

    /// <summary>ProteomeDiscoverer:SRF File Selector:SRF File Path: Path and name of the .srf (Sequest Result Format) file.</summary>
    MS_ProteomeDiscoverer_SRF_File_Selector_SRF_File_Path_OBSOLETE = 1001602,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:Ionization Source: Ionization source (electro-, nano-, thermospray, electron impact, APCI, MALDI, FAB etc).</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_Ionization_Source_OBSOLETE = 1001603,

    /// <summary>ProteomeDiscoverer:Activation Type: Fragmentation method used (CID, MPD, ECD, PQD, ETD, HCD, Any).</summary>
    MS_ProteomeDiscoverer_Activation_Type_OBSOLETE = 1001604,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:Lower RT Limit: Lower retention-time limit.</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_Lower_RT_Limit = 1001605,

    /// <summary>ProteomeDiscoverer:Mass Analyzer: Type of mass spectrometer used (ITMS, FTMS, TOFMS, SQMS, TQMS, SectorMS).</summary>
    MS_ProteomeDiscoverer_Mass_Analyzer_OBSOLETE = 1001606,

    /// <summary>ProteomeDiscoverer:Max Precursor Mass: Maximum mass limit of a singly charged precursor ion.</summary>
    MS_ProteomeDiscoverer_Max_Precursor_Mass = 1001607,

    /// <summary>ProteomeDiscoverer:Min Precursor Mass: Minimum mass limit of a singly charged precursor ion.</summary>
    MS_ProteomeDiscoverer_Min_Precursor_Mass = 1001608,

    /// <summary>ProteomeDiscoverer:Minimum Peak Count: Minimum number of peaks in a tandem mass spectrum that is allowed to pass the filter and to be subjected to further processing in the workflow.</summary>
    MS_ProteomeDiscoverer_Minimum_Peak_Count = 1001609,

    /// <summary>ProteomeDiscoverer:MS Order: Level of the mass spectrum (MS/MS=MS2 ... MS10).</summary>
    MS_ProteomeDiscoverer_MS_Order_OBSOLETE = 1001610,

    /// <summary>ProteomeDiscoverer:Polarity Mode: Polarity mode (positive or negative).</summary>
    MS_ProteomeDiscoverer_Polarity_Mode_OBSOLETE = 1001611,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:Precursor Selection: Determines which precursor mass to use for a given MSn scan. This option applies only to higher-order MSn scans (n >= 3).</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_Precursor_Selection = 1001612,

    /// <summary>ProteomeDiscoverer:SN Threshold: Signal-to-Noise ratio below which peaks are removed.</summary>
    MS_ProteomeDiscoverer_SN_Threshold = 1001613,

    /// <summary>ProteomeDiscoverer:Scan Type: Scan type for the precursor ion (full, Single Ion Monitoring (SIM), Single Reaction Monitoring (SRM)).</summary>
    MS_ProteomeDiscoverer_Scan_Type_OBSOLETE = 1001614,

    /// <summary>ProteomeDiscoverer:Total Intensity Threshold: Used to filter out tandem mass spectra that have a total intensity current(sum of the intensities of all peaks in a spectrum) below the specified value.</summary>
    MS_ProteomeDiscoverer_Total_Intensity_Threshold = 1001615,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:Unrecognized Activation Type Replacements: Specifies the fragmentation method to use in the search algorithm if it is not included in the scan header.</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_Unrecognized_Activation_Type_Replacements = 1001616,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:Unrecognized Charge Replacements: Specifies the charge state of the precursor ions, if it is not defined in the scan header.</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_Unrecognized_Charge_Replacements = 1001617,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:Unrecognized Mass Analyzer Replacements: Specifies the mass spectrometer to use to produce the spectra, if it is not included in the scan header.</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_Unrecognized_Mass_Analyzer_Replacements = 1001618,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:Unrecognized MS Order Replacements: Specifies the MS scan order used to produce the product spectra, if it is not included in the scan header.</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_Unrecognized_MS_Order_Replacements = 1001619,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:Unrecognized Polarity Replacements: Specifies the polarity of the ions monitored if it is not included in the scan header.</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_Unrecognized_Polarity_Replacements = 1001620,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:Upper RT Limit: Upper retention-time limit.</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_Upper_RT_Limit = 1001621,

    /// <summary>ProteomeDiscoverer:Non-Fragment Filter:Mass Window Offset: Specifies the size of the mass-to-charge ratio (m/z) window in daltons used to remove precursors.</summary>
    MS_ProteomeDiscoverer_Non_Fragment_Filter_Mass_Window_Offset = 1001622,

    /// <summary>ProteomeDiscoverer:Non-Fragment Filter:Maximum Neutral Loss Mass: Maximum allowed mass of a neutral loss.</summary>
    MS_ProteomeDiscoverer_Non_Fragment_Filter_Maximum_Neutral_Loss_Mass = 1001623,

    /// <summary>ProteomeDiscoverer:Non-Fragment Filter:Remove Charge Reduced Precursor: Determines whether the charge-reduced precursor peaks found in an ETD or ECD spectrum are removed.</summary>
    MS_ProteomeDiscoverer_Non_Fragment_Filter_Remove_Charge_Reduced_Precursor = 1001624,

    /// <summary>ProteomeDiscoverer:Non-Fragment Filter:Remove Neutral Loss Peaks: Determines whether neutral loss peaks are removed from ETD and ECD spectra.</summary>
    MS_ProteomeDiscoverer_Non_Fragment_Filter_Remove_Neutral_Loss_Peaks = 1001625,

    /// <summary>ProteomeDiscoverer:Non-Fragment Filter:Remove Only Known Masses: Determines whether overtone peaks are removed from LTQ FT or LTQ FT Ultra ECD spectra.</summary>
    MS_ProteomeDiscoverer_Non_Fragment_Filter_Remove_Only_Known_Masses = 1001626,

    /// <summary>ProteomeDiscoverer:Non-Fragment Filter:Remove Precursor Overtones: Determines whether precursor overtone peaks in the spectrum are removed from the input spectrum.</summary>
    MS_ProteomeDiscoverer_Non_Fragment_Filter_Remove_Precursor_Overtones = 1001627,

    /// <summary>ProteomeDiscoverer:Non-Fragment Filter:Remove Precursor Peak: Determines whether precursor artifact peaks from the MS/MS input spectra are removed.</summary>
    MS_ProteomeDiscoverer_Non_Fragment_Filter_Remove_Precursor_Peak = 1001628,

    /// <summary>ProteomeDiscoverer:Spectrum Grouper:Allow Mass Analyzer Mismatch: Determines whether the fragment spectrum for scans with the same precursor mass is grouped, regardless of mass analyzer and activation type.</summary>
    MS_ProteomeDiscoverer_Spectrum_Grouper_Allow_Mass_Analyzer_Mismatch = 1001629,

    /// <summary>ProteomeDiscoverer:Spectrum Grouper:Allow MS Order Mismatch: Determines whether spectra from different MS order scans can be grouped together.</summary>
    MS_ProteomeDiscoverer_Spectrum_Grouper_Allow_MS_Order_Mismatch = 1001630,

    /// <summary>ProteomeDiscoverer:Spectrum Grouper:Max RT Difference: Chromatographic window where precursors to be grouped must reside to be considered the same species.</summary>
    MS_ProteomeDiscoverer_Spectrum_Grouper_Max_RT_Difference_OBSOLETE = 1001631,

    /// <summary>ProteomeDiscoverer:Spectrum Grouper:Precursor Mass Criterion: Groups spectra measured within the given mass and retention-time tolerances into a single spectrum for analysis.</summary>
    MS_ProteomeDiscoverer_Spectrum_Grouper_Precursor_Mass_Criterion = 1001632,

    /// <summary>ProteomeDiscoverer:Xtract:Highest Charge: Highest charge state that is allowed for the deconvolution of multiply charged data.</summary>
    MS_ProteomeDiscoverer_Xtract_Highest_Charge = 1001633,

    /// <summary>ProteomeDiscoverer:Xtract:Highest MZ: Highest mass-to-charge (mz) value for spectral peaks in the measured spectrum that are considered for Xtract.</summary>
    MS_ProteomeDiscoverer_Xtract_Highest_MZ_OBSOLETE = 1001634,

    /// <summary>ProteomeDiscoverer:Xtract:Lowest Charge: Lowest charge state that is allowed for the deconvolution of multiply charged data.</summary>
    MS_ProteomeDiscoverer_Xtract_Lowest_Charge = 1001635,

    /// <summary>ProteomeDiscoverer:Xtract:Lowest MZ: Lowest mass-to-charge (mz) value for spectral peaks in the measured spectrum that are considered for Xtract.</summary>
    MS_ProteomeDiscoverer_Xtract_Lowest_MZ_OBSOLETE = 1001636,

    /// <summary>ProteomeDiscoverer:Xtract:Monoisotopic Mass Only: Determines whether the isotopic pattern, i.e. all isotopes of a mass are removed from the spectrum.</summary>
    MS_ProteomeDiscoverer_Xtract_Monoisotopic_Mass_Only = 1001637,

    /// <summary>ProteomeDiscoverer:Xtract:Overlapping Remainder: Fraction of the more abundant peak that an overlapping multiplet must exceed in order to be processed (deconvoluted).</summary>
    MS_ProteomeDiscoverer_Xtract_Overlapping_Remainder = 1001638,

    /// <summary>ProteomeDiscoverer:Xtract:Required Fitting Accuracy: Accuracy required for a pattern fit to be considered valid.</summary>
    MS_ProteomeDiscoverer_Xtract_Required_Fitting_Accuracy = 1001639,

    /// <summary>ProteomeDiscoverer:Xtract:Resolution At 400: Resolution at mass 400.</summary>
    MS_ProteomeDiscoverer_Xtract_Resolution_At_400 = 1001640,

    /// <summary>ProteomeDiscoverer:Lowest Charge State: Minimum charge state below which peptides are filtered out.</summary>
    MS_ProteomeDiscoverer_Lowest_Charge_State = 1001641,

    /// <summary>ProteomeDiscoverer:Highest Charge State: Maximum charge above which peptides are filtered out.</summary>
    MS_ProteomeDiscoverer_Highest_Charge_State = 1001642,

    /// <summary>ProteomeDiscoverer:Spectrum Score Filter:Let Pass Above Scores: Determines whether spectra with scores above the threshold score are retained rather than filtered out.</summary>
    MS_ProteomeDiscoverer_Spectrum_Score_Filter_Let_Pass_Above_Scores = 1001643,

    /// <summary>ProteomeDiscoverer:Dynamic Modifications: Determine dynamic post-translational modifications (PTMs).</summary>
    MS_ProteomeDiscoverer_Dynamic_Modifications = 1001644,

    /// <summary>ProteomeDiscoverer:Static Modifications: Static Modification to all occurrences of a named amino acid.</summary>
    MS_ProteomeDiscoverer_Static_Modifications = 1001645,

    /// <summary>ProteomeDiscoverer:Mascot:Decoy Search: Determines whether the Proteome Discoverer application searches an additional decoy database.</summary>
    MS_ProteomeDiscoverer_Mascot_Decoy_Search_OBSOLETE = 1001646,

    /// <summary>ProteomeDiscoverer:Mascot:Error tolerant Search: Determines whether to search error-tolerant.</summary>
    MS_ProteomeDiscoverer_Mascot_Error_tolerant_Search = 1001647,

    /// <summary>ProteomeDiscoverer:Mascot:Max MGF File Size: Maximum size of the .mgf (Mascot Generic Format) file in MByte.</summary>
    MS_ProteomeDiscoverer_Mascot_Max_MGF_File_Size = 1001648,

    /// <summary>ProteomeDiscoverer:Mascot:Mascot Server URL: URL (Uniform resource Locator) of the Mascot server.</summary>
    MS_ProteomeDiscoverer_Mascot_Mascot_Server_URL = 1001649,

    /// <summary>ProteomeDiscoverer:Mascot:Number of attempts to submit the search: Number of attempts to submit the Mascot search.</summary>
    MS_ProteomeDiscoverer_Mascot_Number_of_attempts_to_submit_the_search = 1001650,

    /// <summary>ProteomeDiscoverer:Mascot:X Static Modification: Number of attempts to submit the Mascot search.</summary>
    MS_ProteomeDiscoverer_Mascot_X_Static_Modification = 1001651,

    /// <summary>ProteomeDiscoverer:Mascot:User Name: Name of the user submitting the Mascot search.</summary>
    MS_ProteomeDiscoverer_Mascot_User_Name_OBSOLETE = 1001652,

    /// <summary>ProteomeDiscoverer:Mascot:Time interval between attempts to submit a search: Time interval between attempts to submit a search in seconds.</summary>
    MS_ProteomeDiscoverer_Mascot_Time_interval_between_attempts_to_submit_a_search = 1001653,

    /// <summary>ProteomeDiscoverer:Enzyme Name: Specifies the enzyme reagent used for protein digestion.</summary>
    MS_ProteomeDiscoverer_Enzyme_Name_OBSOLETE = 1001654,

    /// <summary>ProteomeDiscoverer:Fragment Mass Tolerance: Mass tolerance used for matching fragment peaks in Da or mmu.</summary>
    MS_ProteomeDiscoverer_Fragment_Mass_Tolerance_OBSOLETE = 1001655,

    /// <summary>Mascot:Instrument: Type of instrument used to acquire the data in the raw file.</summary>
    MS_Mascot_Instrument = 1001656,

    /// <summary>ProteomeDiscoverer:Maximum Missed Cleavage Sites: Maximum number of missed cleavage sites to consider during the digest.</summary>
    MS_ProteomeDiscoverer_Maximum_Missed_Cleavage_Sites = 1001657,

    /// <summary>ProteomeDiscoverer:Mascot:Peptide CutOff Score: Minimum score in the IonScore column that each peptide must exceed in order to be reported.</summary>
    MS_ProteomeDiscoverer_Mascot_Peptide_CutOff_Score = 1001658,

    /// <summary>ProteomeDiscoverer:Precursor Mass Tolerance: Mass window for which precursor ions are considered to be the same species.</summary>
    MS_ProteomeDiscoverer_Precursor_Mass_Tolerance_OBSOLETE = 1001659,

    /// <summary>ProteomeDiscoverer:Mascot:Protein CutOff Score: Minimum protein score in the IonScore column that each protein must exceed in order to be reported.</summary>
    MS_ProteomeDiscoverer_Mascot_Protein_CutOff_Score = 1001660,

    /// <summary>ProteomeDiscoverer:Protein Database: Database to use in the search (configured on the Mascot server).</summary>
    MS_ProteomeDiscoverer_Protein_Database_OBSOLETE = 1001661,

    /// <summary>ProteomeDiscoverer:Mascot:Protein Relevance Factor: Specifies a factor that is used in calculating a threshold that determines whether a protein appears in the results report.</summary>
    MS_ProteomeDiscoverer_Mascot_Protein_Relevance_Factor = 1001662,

    /// <summary>ProteomeDiscoverer:Target FDR Relaxed: Specifies the relaxed target false discovery rate (FDR, 0.0 - 1.0) for peptide hits with moderate confidence.</summary>
    MS_ProteomeDiscoverer_Target_FDR_Relaxed = 1001663,

    /// <summary>ProteomeDiscoverer:Target FDR Strict: Specifies the strict target false discovery rate (FDR, 0.0 - 1.0) for peptide hits with high confidence.</summary>
    MS_ProteomeDiscoverer_Target_FDR_Strict = 1001664,

    /// <summary>ProteomeDiscoverer:Mascot:Taxonomy: Limits searches to entries from a particular species or group of species.</summary>
    MS_ProteomeDiscoverer_Mascot_Taxonomy_OBSOLETE = 1001665,

    /// <summary>ProteomeDiscoverer:Use Average Precursor Mass: Use average mass for the precursor.</summary>
    MS_ProteomeDiscoverer_Use_Average_Precursor_Mass_OBSOLETE = 1001666,

    /// <summary>Mascot:use MudPIT scoring: Determines whether to use MudPIT or normal scoring.</summary>
    MS_Mascot_use_MudPIT_scoring_OBSOLETE = 1001667,

    /// <summary>ProteomeDiscoverer:Absolute XCorr Threshold: Minimum cross-correlation threshold that determines whether peptides in an .srf file are imported.</summary>
    MS_ProteomeDiscoverer_Absolute_XCorr_Threshold = 1001668,

    /// <summary>ProteomeDiscoverer:SEQUEST:Calculate Probability Score: Determines whether to calculate a probability score for every peptide match.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Calculate_Probability_Score = 1001669,

    /// <summary>ProteomeDiscoverer:SEQUEST:CTerminal Modification: Dynamic C-terminal modification that is used during the search.</summary>
    MS_ProteomeDiscoverer_SEQUEST_CTerminal_Modification = 1001670,

    /// <summary>ProteomeDiscoverer:SEQUEST:Fragment Ion Cutoff Percentage: Percentage of the theoretical ions that must be found in order for a peptide to be scored and retained.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Fragment_Ion_Cutoff_Percentage = 1001671,

    /// <summary>ProteomeDiscoverer:SEQUEST:Max Identical Modifications Per Peptide: Maximum number of identical modifications that a single peptide can have.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Max_Identical_Modifications_Per_Peptide = 1001672,

    /// <summary>ProteomeDiscoverer:Max Modifications Per Peptide: Maximum number of different modifications that a peptide can have, e.g. because of steric hindrance.</summary>
    MS_ProteomeDiscoverer_Max_Modifications_Per_Peptide = 1001673,

    /// <summary>ProteomeDiscoverer:SEQUEST:Maximum Peptides Considered: Maximum number of peptides that are searched and scored per spectrum.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Maximum_Peptides_Considered = 1001674,

    /// <summary>ProteomeDiscoverer:Maximum Peptides Output: Maximum number of peptide matches reported per spectrum.</summary>
    MS_ProteomeDiscoverer_Maximum_Peptides_Output = 1001675,

    /// <summary>ProteomeDiscoverer:Maximum Protein References Per Peptide: Maximum number of proteins that a single identified peptide can be associated with during protein assembly.</summary>
    MS_ProteomeDiscoverer_Maximum_Protein_References_Per_Peptide = 1001676,

    /// <summary>ProteomeDiscoverer:SEQUEST:NTerminal Modification: Dynamic N-terminal modification that is used during the search.</summary>
    MS_ProteomeDiscoverer_SEQUEST_NTerminal_Modification = 1001677,

    /// <summary>ProteomeDiscoverer:Peptide CTerminus: Static modification for the C terminal of the peptide used during the search.</summary>
    MS_ProteomeDiscoverer_Peptide_CTerminus = 1001678,

    /// <summary>ProteomeDiscoverer:Peptide NTerminus: Static modification for the N terminal of the peptide used during the search.</summary>
    MS_ProteomeDiscoverer_Peptide_NTerminus = 1001679,

    /// <summary>ProteomeDiscoverer:SEQUEST:Peptide Relevance Factor: Specifies a factor to apply to the protein score.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Peptide_Relevance_Factor = 1001680,

    /// <summary>ProteomeDiscoverer:Protein Relevance Threshold: Specifies a peptide threshold that determines whether the protein that it is a part of is scored and retained in the report.</summary>
    MS_ProteomeDiscoverer_Protein_Relevance_Threshold = 1001681,

    /// <summary>ProteomeDiscoverer:Search Against Decoy Database: Determines whether the Proteome Discoverer application searches against a decoy database.</summary>
    MS_ProteomeDiscoverer_Search_Against_Decoy_Database_OBSOLETE = 1001682,

    /// <summary>ProteomeDiscoverer:SEQUEST:Use Average Fragment Masses: Use average masses for the fragments.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Use_Average_Fragment_Masses = 1001683,

    /// <summary>ProteomeDiscoverer:Use Neutral Loss a Ions: Determines whether a ions with neutral loss are used for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Use_Neutral_Loss_a_Ions = 1001684,

    /// <summary>ProteomeDiscoverer:Use Neutral Loss b Ions: Determines whether b ions with neutral loss are used for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Use_Neutral_Loss_b_Ions = 1001685,

    /// <summary>ProteomeDiscoverer:Use Neutral Loss y Ions: Determines whether y ions with neutral loss are used for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Use_Neutral_Loss_y_Ions = 1001686,

    /// <summary>ProteomeDiscoverer:Use Neutral Loss z Ions: Determines whether z ions with neutral loss are used for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Use_Neutral_Loss_z_Ions = 1001687,

    /// <summary>ProteomeDiscoverer:SEQUEST:Weight of a Ions: Uses a ions for spectrum matching with this relative factor.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Weight_of_a_Ions = 1001688,

    /// <summary>ProteomeDiscoverer:SEQUEST:Weight of b Ions: Uses b ions for spectrum matching with this relative factor.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Weight_of_b_Ions = 1001689,

    /// <summary>ProteomeDiscoverer:SEQUEST:Weight of c Ions: Uses c ions for spectrum matching with this relative factor.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Weight_of_c_Ions = 1001690,

    /// <summary>ProteomeDiscoverer:SEQUEST:Weight of d Ions: Uses c ions for spectrum matching with this relative factor.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Weight_of_d_Ions = 1001691,

    /// <summary>ProteomeDiscoverer:SEQUEST:Weight of v Ions: Uses c ions for spectrum matching with this relative factor.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Weight_of_v_Ions = 1001692,

    /// <summary>ProteomeDiscoverer:SEQUEST:Weight of w Ions: Uses c ions for spectrum matching with this relative factor.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Weight_of_w_Ions = 1001693,

    /// <summary>ProteomeDiscoverer:SEQUEST:Weight of x Ions: Uses x ions for spectrum matching with this relative factor.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Weight_of_x_Ions = 1001694,

    /// <summary>ProteomeDiscoverer:SEQUEST:Weight of y Ions: Uses y ions for spectrum matching with this relative factor.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Weight_of_y_Ions = 1001695,

    /// <summary>ProteomeDiscoverer:SEQUEST:Weight of z Ions: Uses z ions for spectrum matching with this relative factor.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Weight_of_z_Ions = 1001696,

    /// <summary>ProteomeDiscoverer:ZCore:Protein Score Cutoff: Sets a minimum protein score that each protein must exceed in order to be reported.</summary>
    MS_ProteomeDiscoverer_ZCore_Protein_Score_Cutoff = 1001697,

    /// <summary>ProteomeDiscoverer:Reporter Ions Quantizer:Integration Method: Specifies which peak to select if more than one peak is found inside the integration window.</summary>
    MS_ProteomeDiscoverer_Reporter_Ions_Quantizer_Integration_Method = 1001698,

    /// <summary>ProteomeDiscoverer:Reporter Ions Quantizer:Integration Window Tolerance: Specifies the mass-to-charge window that enables one to look for the reporter peaks.</summary>
    MS_ProteomeDiscoverer_Reporter_Ions_Quantizer_Integration_Window_Tolerance = 1001699,

    /// <summary>ProteomeDiscoverer:Reporter Ions Quantizer:Quantitation Method: Quantitation method for isobarically labeled quantitation.</summary>
    MS_ProteomeDiscoverer_Reporter_Ions_Quantizer_Quantitation_Method = 1001700,

    /// <summary>ProteomeDiscoverer:Spectrum Exporter:Export Format: Format of the exported spectra (dta, mgf or mzData).</summary>
    MS_ProteomeDiscoverer_Spectrum_Exporter_Export_Format_OBSOLETE = 1001701,

    /// <summary>ProteomeDiscoverer:Spectrum Exporter:File name: Name of the output file that contains the exported data.</summary>
    MS_ProteomeDiscoverer_Spectrum_Exporter_File_name = 1001702,

    /// <summary>ProteomeDiscoverer:Search Modifications Only For Identified Proteins: Influences the modifications search.</summary>
    MS_ProteomeDiscoverer_Search_Modifications_Only_For_Identified_Proteins = 1001703,

    /// <summary>ProteomeDiscoverer:SEQUEST:Std High Confidence XCorr Charge1: Standard high confidence XCorr parameter for charge = 1.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Std_High_Confidence_XCorr_Charge1 = 1001704,

    /// <summary>ProteomeDiscoverer:SEQUEST:Std High Confidence XCorr Charge2: Standard high confidence XCorr parameter for charge = 2.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Std_High_Confidence_XCorr_Charge2 = 1001705,

    /// <summary>ProteomeDiscoverer:SEQUEST:Std High Confidence XCorr Charge3: Standard high confidence XCorr parameter for charge = 3.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Std_High_Confidence_XCorr_Charge3 = 1001706,

    /// <summary>ProteomeDiscoverer:SEQUEST:Std High Confidence XCorr Charge4: Standard high confidence XCorr parameter for charge >= 4.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Std_High_Confidence_XCorr_Charge4 = 1001707,

    /// <summary>ProteomeDiscoverer:SEQUEST:Std Medium Confidence XCorr Charge1: Standard medium confidence XCorr parameter for charge = 1.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Std_Medium_Confidence_XCorr_Charge1 = 1001708,

    /// <summary>ProteomeDiscoverer:SEQUEST:Std Medium Confidence XCorr Charge2: Standard medium confidence XCorr parameter for charge = 2.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Std_Medium_Confidence_XCorr_Charge2 = 1001709,

    /// <summary>ProteomeDiscoverer:SEQUEST:Std Medium Confidence XCorr Charge3: Standard medium confidence XCorr parameter for charge = 3.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Std_Medium_Confidence_XCorr_Charge3 = 1001710,

    /// <summary>ProteomeDiscoverer:SEQUEST:Std Medium Confidence XCorr Charge4: Standard medium confidence XCorr parameter for charge >= 4.</summary>
    MS_ProteomeDiscoverer_SEQUEST_Std_Medium_Confidence_XCorr_Charge4 = 1001711,

    /// <summary>ProteomeDiscoverer:SEQUEST:FT High Confidence XCorr Charge1: FT high confidence XCorr parameter for charge = 1.</summary>
    MS_ProteomeDiscoverer_SEQUEST_FT_High_Confidence_XCorr_Charge1 = 1001712,

    /// <summary>ProteomeDiscoverer:SEQUEST:FT High Confidence XCorr Charge2: FT high confidence XCorr parameter for charge = 2.</summary>
    MS_ProteomeDiscoverer_SEQUEST_FT_High_Confidence_XCorr_Charge2 = 1001713,

    /// <summary>ProteomeDiscoverer:SEQUEST:FT High Confidence XCorr Charge3: FT high confidence XCorr parameter for charge = 3.</summary>
    MS_ProteomeDiscoverer_SEQUEST_FT_High_Confidence_XCorr_Charge3 = 1001714,

    /// <summary>ProteomeDiscoverer:SEQUEST:FT High Confidence XCorr Charge4: FT high confidence XCorr parameter for charge >= 4.</summary>
    MS_ProteomeDiscoverer_SEQUEST_FT_High_Confidence_XCorr_Charge4 = 1001715,

    /// <summary>ProteomeDiscoverer:SEQUEST:FT Medium Confidence XCorr Charge1: FT medium confidence XCorr parameter for charge = 1.</summary>
    MS_ProteomeDiscoverer_SEQUEST_FT_Medium_Confidence_XCorr_Charge1 = 1001716,

    /// <summary>ProteomeDiscoverer:SEQUEST:FT Medium Confidence XCorr Charge2: FT medium confidence XCorr parameter for charge = 2.</summary>
    MS_ProteomeDiscoverer_SEQUEST_FT_Medium_Confidence_XCorr_Charge2 = 1001717,

    /// <summary>ProteomeDiscoverer:SEQUEST:FT Medium Confidence XCorr Charge3: FT medium confidence XCorr parameter for charge = 3.</summary>
    MS_ProteomeDiscoverer_SEQUEST_FT_Medium_Confidence_XCorr_Charge3 = 1001718,

    /// <summary>ProteomeDiscoverer:SEQUEST:FT Medium Confidence XCorr Charge4: FT medium confidence XCorr parameter for charge >= 4.</summary>
    MS_ProteomeDiscoverer_SEQUEST_FT_Medium_Confidence_XCorr_Charge4 = 1001719,

    /// <summary>ProteomeDiscoverer:1. Dynamic Modification: Determine 1st dynamic post-translational modifications (PTMs).</summary>
    MS_ProteomeDiscoverer_1__Dynamic_Modification = 1001720,

    /// <summary>ProteomeDiscoverer:2. Dynamic Modification: Determine 2nd dynamic post-translational modifications (PTMs).</summary>
    MS_ProteomeDiscoverer_2__Dynamic_Modification = 1001721,

    /// <summary>ProteomeDiscoverer:3. Dynamic Modification: Determine 3rd dynamic post-translational modifications (PTMs).</summary>
    MS_ProteomeDiscoverer_3__Dynamic_Modification = 1001722,

    /// <summary>ProteomeDiscoverer:4. Dynamic Modification: Determine 4th dynamic post-translational modifications (PTMs).</summary>
    MS_ProteomeDiscoverer_4__Dynamic_Modification = 1001723,

    /// <summary>ProteomeDiscoverer:Static Modification for X: Static Modification for X.</summary>
    MS_ProteomeDiscoverer_Static_Modification_for_X = 1001724,

    /// <summary>ProteomeDiscoverer:Initial minimal peptide probability: Minimal initial peptide probability to contribute to analysis.</summary>
    MS_ProteomeDiscoverer_Initial_minimal_peptide_probability = 1001725,

    /// <summary>ProteomeDiscoverer:Minimal peptide probability: Minimum adjusted peptide probability contributing to protein probability.</summary>
    MS_ProteomeDiscoverer_Minimal_peptide_probability = 1001726,

    /// <summary>ProteomeDiscoverer:Minimal peptide weight: Minimum peptide weight contributing to protein probability.</summary>
    MS_ProteomeDiscoverer_Minimal_peptide_weight = 1001727,

    /// <summary>ProteomeDiscoverer:Number of input1 spectra: Number of spectra from 1+ precursor ions.</summary>
    MS_ProteomeDiscoverer_Number_of_input1_spectra = 1001728,

    /// <summary>ProteomeDiscoverer:Number of input2 spectra: Number of spectra from 2+ precursor ions.</summary>
    MS_ProteomeDiscoverer_Number_of_input2_spectra = 1001729,

    /// <summary>ProteomeDiscoverer:Number of input3 spectra: Number of spectra from 3+ precursor ions.</summary>
    MS_ProteomeDiscoverer_Number_of_input3_spectra = 1001730,

    /// <summary>ProteomeDiscoverer:Number of input4 spectra: Number of spectra from 4+ precursor ions.</summary>
    MS_ProteomeDiscoverer_Number_of_input4_spectra = 1001731,

    /// <summary>ProteomeDiscoverer:Number of input5 spectra: Number of spectra from 5+ precursor ions.</summary>
    MS_ProteomeDiscoverer_Number_of_input5_spectra = 1001732,

    /// <summary>ProteomeDiscoverer:Number of predicted correct proteins: Total number of predicted correct protein ids (sum of probabilities).</summary>
    MS_ProteomeDiscoverer_Number_of_predicted_correct_proteins = 1001733,

    /// <summary>ProteomeDiscoverer:Organism: Sample organism (used for annotation purposes).</summary>
    MS_ProteomeDiscoverer_Organism_OBSOLETE = 1001734,

    /// <summary>ProteomeDiscoverer:Reference Database: Full path database name.</summary>
    MS_ProteomeDiscoverer_Reference_Database_OBSOLETE = 1001735,

    /// <summary>ProteomeDiscoverer:Residue substitution list: Residues considered equivalent when comparing peptides.</summary>
    MS_ProteomeDiscoverer_Residue_substitution_list = 1001736,

    /// <summary>ProteomeDiscoverer:Source file extension: File type (if not pepXML).</summary>
    MS_ProteomeDiscoverer_Source_file_extension_OBSOLETE = 1001737,

    /// <summary>ProteomeDiscoverer:Source Files: Input pepXML files.</summary>
    MS_ProteomeDiscoverer_Source_Files_OBSOLETE = 1001738,

    /// <summary>ProteomeDiscoverer:Source Files old: Input pepXML files (old).</summary>
    MS_ProteomeDiscoverer_Source_Files_old_OBSOLETE = 1001739,

    /// <summary>ProteomeDiscoverer:WinCyg reference database: Windows full path for database.</summary>
    MS_ProteomeDiscoverer_WinCyg_reference_database = 1001740,

    /// <summary>ProteomeDiscoverer:WinCyg source files: Windows pepXML file names.</summary>
    MS_ProteomeDiscoverer_WinCyg_source_files = 1001741,

    /// <summary>LTQ Orbitrap Velos: Finnigan LTQ Orbitrap Velos MS.</summary>
    MS_LTQ_Orbitrap_Velos = 1001742,

    /// <summary>ProteomeDiscoverer:Mascot:Weight of A Ions: Determines if to use A ions for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Mascot_Weight_of_A_Ions = 1001743,

    /// <summary>ProteomeDiscoverer:Mascot:Weight of B Ions: Determines if to use B ions for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Mascot_Weight_of_B_Ions = 1001744,

    /// <summary>ProteomeDiscoverer:Mascot:Weight of C Ions: Determines if to use C ions for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Mascot_Weight_of_C_Ions = 1001745,

    /// <summary>ProteomeDiscoverer:Mascot:Weight of D Ions: Determines if to use D ions for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Mascot_Weight_of_D_Ions = 1001746,

    /// <summary>ProteomeDiscoverer:Mascot:Weight of V Ions: Determines if to use V ions for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Mascot_Weight_of_V_Ions = 1001747,

    /// <summary>ProteomeDiscoverer:Mascot:Weight of W Ions: Determines if to use W ions for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Mascot_Weight_of_W_Ions = 1001748,

    /// <summary>ProteomeDiscoverer:Mascot:Weight of X Ions: Determines if to use X ions for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Mascot_Weight_of_X_Ions = 1001749,

    /// <summary>ProteomeDiscoverer:Mascot:Weight of Y Ions: Determines if to use Y ions for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Mascot_Weight_of_Y_Ions = 1001750,

    /// <summary>ProteomeDiscoverer:Mascot:Weight of Z Ions: Determines if to use z ions for spectrum matching.</summary>
    MS_ProteomeDiscoverer_Mascot_Weight_of_Z_Ions = 1001751,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:Use New Precursor Reevaluation: Determines if to use precursor reevaluation.</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_Use_New_Precursor_Reevaluation = 1001752,

    /// <summary>ProteomeDiscoverer:Spectrum Selector:SN Threshold FTonly: Signal-to-Noise ratio below which peaks are removed (in FT mode only).</summary>
    MS_ProteomeDiscoverer_Spectrum_Selector_SN_Threshold_FTonly = 1001753,

    /// <summary>ProteomeDiscoverer:Mascot:Please Do not Touch this: Unknown Mascot parameter which ProteomeDiscoverer uses for mascot searches.</summary>
    MS_ProteomeDiscoverer_Mascot_Please_Do_not_Touch_this = 1001754,

    /// <summary>contact phone number: Phone number of the contact person or organization.</summary>
    MS_contact_phone_number = 1001755,

    /// <summary>contact fax number: Fax number for the contact person or organization.</summary>
    MS_contact_fax_number = 1001756,

    /// <summary>contact toll-free phone number: Toll-free phone number of the contact person or organization.</summary>
    MS_contact_toll_free_phone_number = 1001757,

    /// <summary>Mascot:SigThresholdType: Significance threshold type used in Mascot reporting (either 'identity' or 'homology').</summary>
    MS_Mascot_SigThresholdType = 1001758,

    /// <summary>Mascot:ProteinGrouping: Strategy used by Mascot to group proteins with same peptide matches (one of 'none', 'Occam's razor' or 'family clustering').</summary>
    MS_Mascot_ProteinGrouping = 1001759,

    /// <summary>Percolator:features: List of Percolator features that were used in processing the peptide matches. Typical Percolator features are 'retentionTime', 'dM', 'mScore', 'lgDScore', 'mrCalc', 'charge' and 'dMppm'.</summary>
    MS_Percolator_features = 1001760,

    /// <summary>ACQUITY UPLC: Waters LC-system.</summary>
    MS_ACQUITY_UPLC = 1001761,

    /// <summary>ACQUITY UPLC H-Class: Waters LC-system.</summary>
    MS_ACQUITY_UPLC_H_Class = 1001762,

    /// <summary>ACQUITY UPLC H-Class Bio: Waters LC-system.</summary>
    MS_ACQUITY_UPLC_H_Class_Bio = 1001763,

    /// <summary>ACQUITY UPLC I-Class: Waters LC-system.</summary>
    MS_ACQUITY_UPLC_I_Class = 1001764,

    /// <summary>ACQUITY UPLC Systems with 2D Technology: Waters LC-system.</summary>
    MS_ACQUITY_UPLC_Systems_with_2D_Technology = 1001765,

    /// <summary>nanoACQUITY UPLC: Waters LC-system.</summary>
    MS_nanoACQUITY_UPLC = 1001766,

    /// <summary>nanoACQUITY UPLC System with Technology: Waters LC-system.</summary>
    MS_nanoACQUITY_UPLC_System_with_Technology = 1001767,

    /// <summary>nanoACQUITY UPLC with HDX Technology: Waters LC-system.</summary>
    MS_nanoACQUITY_UPLC_with_HDX_Technology = 1001768,

    /// <summary>TRIZAIC UPLC nanoTile: Waters LC-system.</summary>
    MS_TRIZAIC_UPLC_nanoTile = 1001769,

    /// <summary>GCT Premier: Waters oa-ToF based.</summary>
    MS_GCT_Premier = 1001770,

    /// <summary>MALDI Synapt G2 HDMS: Waters oa-ToF based.</summary>
    MS_MALDI_Synapt_G2_HDMS = 1001771,

    /// <summary>MALDI Synapt G2 MS: Waters oa-ToF based.</summary>
    MS_MALDI_Synapt_G2_MS = 1001772,

    /// <summary>MALDI Synapt G2-S HDMS: Waters oa-ToF based.</summary>
    MS_MALDI_Synapt_G2_S_HDMS = 1001773,

    /// <summary>MALDI Synapt G2-S MS: Waters oa-ToF based.</summary>
    MS_MALDI_Synapt_G2_S_MS = 1001774,

    /// <summary>MALDI Synapt HDMS: Waters oa-ToF based.</summary>
    MS_MALDI_Synapt_HDMS = 1001775,

    /// <summary>MALDI Synapt MS: Waters oa-ToF based.</summary>
    MS_MALDI_Synapt_MS = 1001776,

    /// <summary>Synapt G2 HDMS: Waters oa-ToF based.</summary>
    MS_Synapt_G2_HDMS = 1001777,

    /// <summary>Synapt G2 MS: Waters oa-ToF based.</summary>
    MS_Synapt_G2_MS = 1001778,

    /// <summary>Synapt G2-S HDMS: Waters oa-ToF based.</summary>
    MS_Synapt_G2_S_HDMS = 1001779,

    /// <summary>Synapt G2-S MS: Waters oa-ToF based.</summary>
    MS_Synapt_G2_S_MS = 1001780,

    /// <summary>Synapt HDMS: Waters oa-ToF based.</summary>
    MS_Synapt_HDMS = 1001781,

    /// <summary>Synapt MS: Waters oa-ToF based.</summary>
    MS_Synapt_MS = 1001782,

    /// <summary>Xevo G2 Q-Tof: Waters oa-ToF based.</summary>
    MS_Xevo_G2_Q_Tof = 1001783,

    /// <summary>Xevo G2 Tof: Waters oa-ToF based.</summary>
    MS_Xevo_G2_Tof = 1001784,

    /// <summary>Xevo Q-Tof: Waters oa-ToF based.</summary>
    MS_Xevo_Q_Tof = 1001785,

    /// <summary>3100: Waters quadrupole based.</summary>
    MS_3100 = 1001786,

    /// <summary>Acquity SQD: Waters quadrupole based.</summary>
    MS_Acquity_SQD = 1001787,

    /// <summary>Acquity TQD: Waters quadrupole based.</summary>
    MS_Acquity_TQD = 1001788,

    /// <summary>Quattro micro GC: Waters quadrupole based.</summary>
    MS_Quattro_micro_GC = 1001789,

    /// <summary>Xevo TQ MS: Waters quadrupole based.</summary>
    MS_Xevo_TQ_MS = 1001790,

    /// <summary>Xevo TQD: Waters quadrupole based.</summary>
    MS_Xevo_TQD = 1001791,

    /// <summary>Xevo TQ-S: Waters quadrupole based.</summary>
    MS_Xevo_TQ_S = 1001792,

    /// <summary>Mascot:PreferredTaxonomy: NCBI TaxID taxonomy ID to prefer when two or more proteins match the same set of peptides or when protein entry in database represents multiple sequences.</summary>
    MS_Mascot_PreferredTaxonomy = 1001793,

    /// <summary>Empower: Waters software for liquid chromatography acquisition, processing, data analysis and data management.</summary>
    MS_Empower = 1001795,

    /// <summary>Unify: Waters software for liquid chromatography and mass spectrometry acquisition, processing, data analysis and data management.</summary>
    MS_Unify = 1001796,

    /// <summary>travelling wave ion mobility mass spectrometer: An ion mobility mass spectrometry technique based on the superimposition of travelling voltage waves on a radially-confining RF voltage in a gas-filled, stacked-ring ion guide.</summary>
    MS_travelling_wave_ion_mobility_mass_spectrometer = 1001797,

    /// <summary>TWIMS: An ion mobility mass spectrometry technique based on the superimposition of travelling voltage waves on a radially-confining RF voltage in a gas-filled, stacked-ring ion guide.</summary>
    MS_TWIMS = MS_travelling_wave_ion_mobility_mass_spectrometer,

    /// <summary>LECO software: LECO software for data acquisition and analysis.</summary>
    MS_LECO_software = 1001798,

    /// <summary>ChromaTOF software: Software for acquisition, processing and analysis of data for LECO instruments.</summary>
    MS_ChromaTOF_software = 1001799,

    /// <summary>LECO instrument model: LECO instrument model.</summary>
    MS_LECO_instrument_model = 1001800,

    /// <summary>Pegasus HRT: LECO high resolution time-of-flight GC mass spectrometer.</summary>
    MS_Pegasus_HRT = 1001801,

    /// <summary>Citius HRT: LECO high resolution time-of-flight LC mass spectrometer.</summary>
    MS_Citius_HRT = 1001802,

    /// <summary>Pegasus: LECO GC time-of-flight mass spectrometer.</summary>
    MS_Pegasus = 1001803,

    /// <summary>TruTOF: LECO bench-top GC time-of-flight mass spectrometer.</summary>
    MS_TruTOF = 1001804,

    /// <summary>quantification datatype: The data type of the value reported in a QuantLayer for a feature, matchedfeature, peptide, protein, protein group.</summary>
    MS_quantification_datatype = 1001805,

    /// <summary>quantification object attribute: Attributes describing the details of an object relevant for reporting quantification workflows or values.</summary>
    MS_quantification_object_attribute = 1001806,

    /// <summary>StudyVariable attribute: Attribute describing a study variable.</summary>
    MS_StudyVariable_attribute = 1001807,

    /// <summary>technical replicate: The study variable is 'technical replicate'. The string value denotes the category of technical replicate, e.g. 'run generated from same sample'.</summary>
    MS_technical_replicate = 1001808,

    /// <summary>biological replicate: The study variable is 'biological replicate'. This means, the run was generated from another individuum or sample.</summary>
    MS_biological_replicate = 1001809,

    /// <summary>experimental condition 'case': The experimental condition is 'case' in contrast to 'control'.</summary>
    MS_experimental_condition__case_ = 1001810,

    /// <summary>experimental condition 'control': The experimental condition is 'control' in contrast to 'case'.</summary>
    MS_experimental_condition__control_ = 1001811,

    /// <summary>experimental condition 'disease': The experimental condition is 'disease' in contrast to 'healthy'.</summary>
    MS_experimental_condition__disease_ = 1001812,

    /// <summary>experimental condition 'healthy': The experimental condition is 'healthy' in contrast to 'disease'.</summary>
    MS_experimental_condition__healthy_ = 1001813,

    /// <summary>generic experimental condition: The experimental condition is given in the value of this term.</summary>
    MS_generic_experimental_condition = 1001814,

    /// <summary>time series, time point X: The experimental design followed a time series design. The time point of this run is given in the value of this term.</summary>
    MS_time_series__time_point_X = 1001815,

    /// <summary>dilution series, concentration X: The experimental design followed a dilution series design. The concentration of this run is given in the value of this term.</summary>
    MS_dilution_series__concentration_X = 1001816,

    /// <summary>RawFile attribute: Attribute describing a raw file.</summary>
    MS_RawFile_attribute = 1001817,

    /// <summary>one sample run: The raw file contains the run of one sample (e.g. spectral counting, LC-MS label-free).</summary>
    MS_one_sample_run = 1001818,

    /// <summary>two sample run: The raw file contains the run of two samples (e.g. SILAC, metabolic labeling).</summary>
    MS_two_sample_run = 1001819,

    /// <summary>three sample run: The raw file contains the run of three samples (e.g. 3-plex SILAC).</summary>
    MS_three_sample_run = 1001820,

    /// <summary>four sample run: The raw file contains the run of four samples (e.g. 4-plex iTraq).</summary>
    MS_four_sample_run = 1001821,

    /// <summary>eight sample run: The raw file contains the run of eight samples (e.g. 8-plex iTraq).</summary>
    MS_eight_sample_run = 1001822,

    /// <summary>RawFilesGroup attribute: Attribute describing, how raw files build a raw file group.</summary>
    MS_RawFilesGroup_attribute = 1001823,

    /// <summary>merge of runs of 1D gel bands: Attribute describing, how raw files build a raw file group.</summary>
    MS_merge_of_runs_of_1D_gel_bands = 1001824,

    /// <summary>FeatureList attribute: Attribute describing a FeatureList.</summary>
    MS_FeatureList_attribute = 1001825,

    /// <summary>mass trace reporting: rectangles: The mass trace of the features of this feature list specifies rectangles. Each mass trace has the syntax (RT_start,MZ_start,RT_end,MZ_end), i.e. opposite corners are given.</summary>
    MS_mass_trace_reporting__rectangles = 1001826,

    /// <summary>mass trace reporting: polygons: The mass trace of the features of this feature list specifies polygons. Each mass trace has the syntax (RT_1,MZ_1,RT_2,MZ_2,...,RT_i,MZ_i,...,RT_n,MZ_n), where the line (RT_n,MZ_n)->(RT_1,MZ_1) is implicite.</summary>
    MS_mass_trace_reporting__polygons = 1001827,

    /// <summary>Feature attribute: Attribute describing a Feature.</summary>
    MS_Feature_attribute = 1001828,

    /// <summary>SRM transition ID: SRM transition ID.</summary>
    MS_SRM_transition_ID = 1001829,

    /// <summary>Progenesis LC-MS: Software from Nonlinear Dynamics for LC-MS label-free workflow.</summary>
    MS_Progenesis_LC_MS = 1001830,

    /// <summary>SILACAnalyzer: Software for SILAC workflow.</summary>
    MS_SILACAnalyzer = 1001831,

    /// <summary>quantitation software comment / customizations: Quantitation software comment or any customizations to the default setup of the software.</summary>
    MS_quantitation_software_comment___customizations = 1001832,

    /// <summary>quantitation analysis summary: The overall workflow of this quantitation report.</summary>
    MS_quantitation_analysis_summary = 1001833,

    /// <summary>LC-MS label-free quantitation analysis: LC-MS label-free workflow (RT m/z map).</summary>
    MS_LC_MS_label_free_quantitation_analysis = 1001834,

    /// <summary>SILAC quantitation analysis: SILAC workflow (heavy, light, and sometimes medium peak).</summary>
    MS_SILAC_quantitation_analysis = 1001835,

    /// <summary>spectral counting quantitation analysis: Spectral counting workflow (number of identified MS/MS spectra as approximation of peptide / protein quant).</summary>
    MS_spectral_counting_quantitation_analysis = 1001836,

    /// <summary>iTRAQ quantitation analysis: Quantification analysis using the AB SCIEX iTRAQ isobaric labeling workflow, wherein 2-8 reporter ions are measured in MS/MS spectra near 114 m/z.</summary>
    MS_iTRAQ_quantitation_analysis = 1001837,

    /// <summary>SRM quantitation analysis: Selected Reaction Monitoring workflow (XIC quant. of precursor / fragment mass pair).</summary>
    MS_SRM_quantitation_analysis = 1001838,

    /// <summary>metabolic labeling 14N / 15N quantitation analysis: Metabolic labeling workflow (heavy and light versions of peptides, depending on number of nitrogens).</summary>
    MS_metabolic_labeling_14N___15N_quantitation_analysis = 1001839,

    /// <summary>LC-MS feature intensity: Maximum peak intensity of the LC-MS feature.</summary>
    MS_LC_MS_feature_intensity = 1001840,

    /// <summary>LC-MS feature volume: Real (intensity times area) volume of the LC-MS feature.</summary>
    MS_LC_MS_feature_volume = 1001841,

    /// <summary>peptide PSM count: The number of MS/MS spectra identified for this peptide in spectral counting.</summary>
    MS_peptide_PSM_count = 1001842,

    /// <summary>MS1 feature maximum intensity: Maximum intensity of MS1 feature.</summary>
    MS_MS1_feature_maximum_intensity = 1001843,

    /// <summary>MS1 feature area: Area of MS1 feature.</summary>
    MS_MS1_feature_area = 1001844,

    /// <summary>peak area: Area of MS1 peak (e.g. SILAC, 15N).</summary>
    MS_peak_area_OBSOLETE = 1001845,

    /// <summary>isotopic pattern area: Area of all peaks belonging to the isotopic pattern of light or heavy peak (e.g. 15N).</summary>
    MS_isotopic_pattern_area = 1001846,

    /// <summary>reporter ion intensity: Intensity of MS/MS reporter ion (e.g. iTraq).</summary>
    MS_reporter_ion_intensity = 1001847,

    /// <summary>simple ratio of two value: Simple ratio of two values (enumerator and denominator).</summary>
    MS_simple_ratio_of_two_value = 1001848,

    /// <summary>sum of MatchedFeature values: Peptide quantification value calculated as sum of MatchedFeature quantification values.</summary>
    MS_sum_of_MatchedFeature_values = 1001849,

    /// <summary>normalized peptide value: Normalized peptide value.</summary>
    MS_normalized_peptide_value = 1001850,

    /// <summary>protein value: sum of peptide values: Protein quantification value calculated as sum of peptide values.</summary>
    MS_protein_value__sum_of_peptide_values = 1001851,

    /// <summary>normalized protein value: Normalized protein value.</summary>
    MS_normalized_protein_value = 1001852,

    /// <summary>max fold change: Global datatype: Maximum of all pair-wise fold changes of group means (e.g. Progenesis).</summary>
    MS_max_fold_change = 1001853,

    /// <summary>ANOVA p-value: Global datatype: p-value of ANOVA of group means (e.g. Progenesis).</summary>
    MS_ANOVA_p_value = 1001854,

    /// <summary>t-test p-value: P-value of t-Test of two groups.</summary>
    MS_t_test_p_value = 1001855,

    /// <summary>reporter ion raw value: Intensity (or area) of MS/MS reporter ion (e.g. iTraq).</summary>
    MS_reporter_ion_raw_value = 1001856,

    /// <summary>reporter ion normalized value: Normalized value of MS/MS reporter ion (e.g. iTraq).</summary>
    MS_reporter_ion_normalized_value = 1001857,

    /// <summary>XIC area: Area of the extracted ion chromatogram (e.g. of a transition in SRM).</summary>
    MS_XIC_area = 1001858,

    /// <summary>normalized XIC area: Normalized area of the extracted ion chromatogram (e.g. of a transition in SRM).</summary>
    MS_normalized_XIC_area = 1001859,

    /// <summary>protein value: mean of peptide ratios: Protein quantification value calculated as mean of peptide ratios.</summary>
    MS_protein_value__mean_of_peptide_ratios = 1001860,

    /// <summary>quantification data processing: Terms used to describe types of quantification data processing.</summary>
    MS_quantification_data_processing = 1001861,

    /// <summary>normalization to mean of sum of all proteins: Normalization of protein values to the mean of the sum of all protein PSM counts (e.g. spectral counting).</summary>
    MS_normalization_to_mean_of_sum_of_all_proteins = 1001862,

    /// <summary>quantile normalization, proteins: Normalization of protein values to approach the same distribution.</summary>
    MS_quantile_normalization__proteins = 1001863,

    /// <summary>quantile normalization, peptides: Normalization of peptide values to approach the same distribution.</summary>
    MS_quantile_normalization__peptides = 1001864,

    /// <summary>Progenesis automatic alignment: Automatic RT alignment of Progenesis software.</summary>
    MS_Progenesis_automatic_alignment = 1001865,

    /// <summary>Progenesis manual alignment: RT alignment of Progenesis software using automatic and manual vectors.</summary>
    MS_Progenesis_manual_alignment = 1001866,

    /// <summary>Progenesis normalization: Normalization as performed by Progenesis LC-MS.</summary>
    MS_Progenesis_normalization = 1001867,

    /// <summary>q-value for peptides: Peptide identification confidence metric q-value.</summary>
    MS_q_value_for_peptides = 1001868,

    /// <summary>q-value for proteins: Protein identification confidence metric q-value.</summary>
    MS_q_value_for_proteins = 1001869,

    /// <summary>p-value for peptides: Peptide identification confidence metric p-value.</summary>
    MS_p_value_for_peptides = 1001870,

    /// <summary>p-value for proteins: Protein identification confidence metric p-value.</summary>
    MS_p_value_for_proteins = 1001871,

    /// <summary>E-value for peptides: Peptide identification confidence metric E-value.</summary>
    MS_E_value_for_peptides = 1001872,

    /// <summary>E-value for proteins: Protein identification confidence metric E-value.</summary>
    MS_E_value_for_proteins = 1001873,

    /// <summary>FDRScore: Quality measurement score.</summary>
    MS_FDRScore = 1001874,

    /// <summary>modification motif: The regular expression describing the sequence motif for a modification.</summary>
    MS_modification_motif = 1001875,

    /// <summary>modification probability: The a priori probability of a modification.</summary>
    MS_modification_probability = 1001876,

    /// <summary>ChromaTOF HRT software: Software for acquisition, processing and analysis of data for LECO instruments.</summary>
    MS_ChromaTOF_HRT_software = 1001877,

    /// <summary>MALDI Solutions Microbial Identification: Shimadzu Biotech software for data acquisition, processing, and analysis.</summary>
    MS_MALDI_Solutions_Microbial_Identification = 1001878,

    /// <summary>offset voltage: The potential difference between two adjacent interface voltages affecting in-source collision induced dissociation.</summary>
    MS_offset_voltage = 1001879,

    /// <summary>in-source collision-induced dissociation: The dissociation of an ion as a result of collisional excitation during ion transfer from an atmospheric pressure ion source and the mass spectrometer vacuum.</summary>
    MS_in_source_collision_induced_dissociation = 1001880,

    /// <summary>mz5 file: mz5 file format, modeled after mzML.</summary>
    MS_mz5_file = 1001881,

    /// <summary>transition validation attribute: Attributes of the quality of a transition that affect its selection as appropriate.</summary>
    MS_transition_validation_attribute = 1001882,

    /// <summary>coefficient of variation: Variation of a set of signal measurements calculated as the standard deviation relative to the mean.</summary>
    MS_coefficient_of_variation = 1001883,

    /// <summary>signal-to-noise ratio: Unitless number providing the ratio of the total measured intensity of a signal relative to the estimated noise level for that signal.</summary>
    MS_signal_to_noise_ratio = 1001884,

    /// <summary>command-line parameters: Parameters string passed to a command-line interface software application, omitting the executable name.</summary>
    MS_command_line_parameters = 1001885,

    /// <summary>SQID: Software for data analysis of peptides and proteins.</summary>
    MS_SQID = 1001886,

    /// <summary>SQID:score: The SQID result 'Score'.</summary>
    MS_SQID_score = 1001887,

    /// <summary>SQID:deltaScore: The SQID result 'deltaScore'.</summary>
    MS_SQID_deltaScore = 1001888,

    /// <summary>SQID:protein score: The SQID result 'protein score'.</summary>
    MS_SQID_protein_score = 1001889,

    /// <summary>Progenesis:protein normalised abundance: The data type normalised abundance for proteins produced by Progenesis LC-MS.</summary>
    MS_Progenesis_protein_normalised_abundance = 1001890,

    /// <summary>Progenesis:peptide normalised abundance: The data type normalised abundance for peptides produced by Progenesis LC-MS.</summary>
    MS_Progenesis_peptide_normalised_abundance = 1001891,

    /// <summary>Progenesis:protein raw abundance: The data type raw abundance for proteins produced by Progenesis LC-MS.</summary>
    MS_Progenesis_protein_raw_abundance = 1001892,

    /// <summary>Progenesis:peptide raw abundance: The data type raw abundance for peptide produced by Progenesis LC-MS.</summary>
    MS_Progenesis_peptide_raw_abundance = 1001893,

    /// <summary>Progenesis:confidence score: The data type confidence score  produced by Progenesis LC-MS.</summary>
    MS_Progenesis_confidence_score = 1001894,

    /// <summary>Progenesis:peptide count: The data type peptide count produced by Progenesis LC-MS.</summary>
    MS_Progenesis_peptide_count = 1001895,

    /// <summary>Progenesis:feature intensity: The data type feature intensity produced by Progenesis LC-MS.</summary>
    MS_Progenesis_feature_intensity = 1001896,

    /// <summary>MaxQuant:peptide counts (unique): The data type peptide counts (unique) produced by MaxQuant.</summary>
    MS_MaxQuant_peptide_counts__unique_ = 1001897,

    /// <summary>MaxQuant:peptide counts (all): The data type peptide counts (all) produced by MaxQuant.</summary>
    MS_MaxQuant_peptide_counts__all_ = 1001898,

    /// <summary>MaxQuant:peptide counts (razor+unique): The data type peptide counts (razor+unique)  produced by MaxQuant.</summary>
    MS_MaxQuant_peptide_counts__razor_unique_ = 1001899,

    /// <summary>MaxQuant:sequence length: The data type sequence length produced by MaxQuant.</summary>
    MS_MaxQuant_sequence_length = 1001900,

    /// <summary>MaxQuant:PEP: The data type PEP (posterior error probability) produced by MaxQuant.</summary>
    MS_MaxQuant_PEP = 1001901,

    /// <summary>MaxQuant:LFQ intensity: The data type LFQ intensity produced by MaxQuant.</summary>
    MS_MaxQuant_LFQ_intensity = 1001902,

    /// <summary>MaxQuant:feature intensity: The data type feature intensity produced by MaxQuant.</summary>
    MS_MaxQuant_feature_intensity = 1001903,

    /// <summary>MaxQuant:MS/MS count: The data type MS/MS count produced by MaxQuant.</summary>
    MS_MaxQuant_MS_MS_count = 1001904,

    /// <summary>emPAI value: The emPAI value of protein abundance, produced from the emPAI algorithm.</summary>
    MS_emPAI_value = 1001905,

    /// <summary>APEX value: The APEX value of protein abundance, produced from the APEX software.</summary>
    MS_APEX_value = 1001906,

    /// <summary>retention time window width: The full width of a retention time window for a chromatographic peak.</summary>
    MS_retention_time_window_width = 1001907,

    /// <summary>ISQ: Thermo Scientific ISQ single quadrupole MS with the ExtractraBrite source.</summary>
    MS_ISQ = 1001908,

    /// <summary>Velos Plus: Thermo Scientific second generation Velos.</summary>
    MS_Velos_Plus = 1001909,

    /// <summary>LTQ Orbitrap Elite: Thermo Scientific second generation Velos and Orbitrap.</summary>
    MS_LTQ_Orbitrap_Elite = 1001910,

    /// <summary>Q Exactive: Thermo Scientific Q Exactive.</summary>
    MS_Q_Exactive = 1001911,

    /// <summary>PinPoint: Thermo Scientific PinPoint SRM analysis software.</summary>
    MS_PinPoint = 1001912,

    /// <summary>S-lens voltage: Potential difference setting of the Thermo Scientific S-lens stacked-ring ion guide in volts.</summary>
    MS_S_lens_voltage = 1001913,

    /// <summary>pymzML: Python module to interface mzML Data.</summary>
    MS_pymzML = 1001914,

    /// <summary>leukocyte elastase: Enzyme leukocyte elastase (EC 3.4.21.37).</summary>
    MS_leukocyte_elastase = 1001915,

    /// <summary>proline endopeptidase: Enzyme proline endopeptidase (EC 3.4.21.26).</summary>
    MS_proline_endopeptidase = 1001916,

    /// <summary>glutamyl endopeptidase: Enzyme glutamyl endopeptidase (EC 3.4.21.19).</summary>
    MS_glutamyl_endopeptidase = 1001917,

    /// <summary>staphylococcal protease: Enzyme glutamyl endopeptidase (EC 3.4.21.19).</summary>
    MS_staphylococcal_protease = MS_glutamyl_endopeptidase,

    /// <summary>Glu-C: Enzyme glutamyl endopeptidase (EC 3.4.21.19).</summary>
    MS_Glu_C = MS_glutamyl_endopeptidase,

    /// <summary>2-iodobenzoate: Chemical iodobenzoate. Cleaves after W.</summary>
    MS_2_iodobenzoate = 1001918,

    /// <summary>ProteomeXchange accession number: Main identifier of a ProteomeXchange dataset.</summary>
    MS_ProteomeXchange_accession_number = 1001919,

    /// <summary>ProteomeXchange accession number version number: Version number of a ProteomeXchange accession number.</summary>
    MS_ProteomeXchange_accession_number_version_number = 1001921,

    /// <summary>Digital Object Identifier (DOI): DOI unique identifier.</summary>
    MS_Digital_Object_Identifier__DOI_ = 1001922,

    /// <summary>external reference keyword: Free text attribute that can enrich the information about an entity.</summary>
    MS_external_reference_keyword = 1001923,

    /// <summary>journal article keyword: Keyword present in a scientific publication.</summary>
    MS_journal_article_keyword = 1001924,

    /// <summary>submitter keyword: Keyword assigned by the data submitter.</summary>
    MS_submitter_keyword = 1001925,

    /// <summary>curator keyword: Keyword assigned by a data curator.</summary>
    MS_curator_keyword = 1001926,

    /// <summary>Tranche file hash: Hash assigned by the Tranche resource to an individual file.</summary>
    MS_Tranche_file_hash = 1001927,

    /// <summary>Tranche project hash: Hash assigned by the Tranche resource to a whole project.</summary>
    MS_Tranche_project_hash = 1001928,

    /// <summary>PRIDE experiment URI: URI that allows the access to one experiment in the PRIDE database.</summary>
    MS_PRIDE_experiment_URI = 1001929,

    /// <summary>PRIDE project URI: URI that allows the access to one project in the PRIDE database.</summary>
    MS_PRIDE_project_URI = 1001930,

    /// <summary>source interface: The source interface.</summary>
    MS_source_interface = 1001931,

    /// <summary>source interface model: The source interface model.</summary>
    MS_source_interface_model = 1001932,

    /// <summary>source sprayer: The source sprayer.</summary>
    MS_source_sprayer = 1001933,

    /// <summary>source sprayer type: The source sprayer type.</summary>
    MS_source_sprayer_type = 1001934,

    /// <summary>source sprayer manufacturer: The source sprayer manufacturer.</summary>
    MS_source_sprayer_manufacturer = 1001935,

    /// <summary>source sprayer model: The source sprayer model.</summary>
    MS_source_sprayer_model = 1001936,

    /// <summary>sample plate: Plate where the sample solution is spotted in a MALDI or similar instrument.</summary>
    MS_sample_plate = 1001937,

    /// <summary>sample plate type: The sample plate type.</summary>
    MS_sample_plate_type = 1001938,

    /// <summary>stainless steel plate: Stainless steel plate.</summary>
    MS_stainless_steel_plate = 1001939,

    /// <summary>coated glass plate: Coated glass plate.</summary>
    MS_coated_glass_plate = 1001940,

    /// <summary>electrospray supply type: Whether the sprayer is fed or is loaded with sample once.</summary>
    MS_electrospray_supply_type = 1001941,

    /// <summary>static supply electrospray: The sprayer is loaded with sample once.</summary>
    MS_static_supply_electrospray = 1001942,

    /// <summary>fed supply electrospray: The sprayer is continuously fed with sample.</summary>
    MS_fed_supply_electrospray = 1001943,

    /// <summary>Collision cell exit potential: Potential difference between Q2 and Q3 in a triple quadrupole instrument in volts.</summary>
    MS_Collision_cell_exit_potential = 1001944,

    /// <summary>CXP: Potential difference between Q2 and Q3 in a triple quadrupole instrument in volts.</summary>
    MS_CXP = MS_Collision_cell_exit_potential,

    /// <summary>Pegasus 4D: LECO nominal mass resolution time-of-flight GCxGC mass spectrometer.</summary>
    MS_Pegasus_4D = 1001945,

    /// <summary>PEAKS Studio: PEAKS Studio software for data analysis.</summary>
    MS_PEAKS_Studio = 1001946,

    /// <summary>PEAKS Online: PEAKS Online software for high throughput data analysis.</summary>
    MS_PEAKS_Online = 1001947,

    /// <summary>: PEAKS Node software for high throughput data analysis.</summary>
    MS_ = 1001948,

    /// <summary>BSI software: Bioinformatics Solutions Inc. Software for data processing and analysis.</summary>
    MS_BSI_software = 1001949,

    /// <summary>PEAKS:peptideScore: The PEAKS peptide '-10lgP Score'.</summary>
    MS_PEAKS_peptideScore = 1001950,

    /// <summary>PEAKS:proteinScore: The PEAKS protein '-10lgP Score'.</summary>
    MS_PEAKS_proteinScore = 1001951,

    /// <summary>ZCore:probScore: The ZCore probability score.</summary>
    MS_ZCore_probScore = 1001952,

    /// <summary>source interface manufacturer: The source interface manufacturer.</summary>
    MS_source_interface_manufacturer = 1001953,

    /// <summary>acquisition parameter: Parameters used in the mass spectrometry acquisition.</summary>
    MS_acquisition_parameter = 1001954,

    /// <summary>no cleavage: No cleavage.</summary>
    MS_no_cleavage = 1001955,

    /// <summary>unspecific cleavage: Unspecific cleavage.</summary>
    MS_unspecific_cleavage = 1001956,

    /// <summary>(?<=[ALIV])(?!P): </summary>
    MS______ALIV_____P_ = 1001957,

    /// <summary>(?<=[HKR]P)(?!P): </summary>
    MS______HKR_P____P_ = 1001958,

    /// <summary>(?<=[^E]E): </summary>
    MS_______E_E_ = 1001959,

    /// <summary>(?<=W): </summary>
    MS_____W_ = 1001960,

    /// <summary>peptide spectrum match scoring algorithm: Algorithm used to score the match between a spectrum and a peptide ion.</summary>
    MS_peptide_spectrum_match_scoring_algorithm = 1001961,

    /// <summary>Mascot:C13 counts: C13 peaks to use in peak detection.</summary>
    MS_Mascot_C13_counts = 1001962,

    /// <summary>ProteinExtractor:Weighting: Weighting factor for protein list compilation by ProteinExtractor.</summary>
    MS_ProteinExtractor_Weighting = 1001963,

    /// <summary>ProteinScape:second round Mascot: Flag indicating a second round search with Mascot.</summary>
    MS_ProteinScape_second_round_Mascot = 1001964,

    /// <summary>ProteinScape:second round Phenyx: Flag indicating a second round search with Phenyx.</summary>
    MS_ProteinScape_second_round_Phenyx = 1001965,

    /// <summary>product ion mobility: The mobility of an MS2 product ion, as measured by ion mobility mass spectrometry.</summary>
    MS_product_ion_mobility = 1001966,

    /// <summary>product ion drift time: The ion drift time of an MS2 product ion.</summary>
    MS_product_ion_drift_time = 1001967,

    /// <summary>PTM localization score: A score that assign confidence to the localization of an amino acid modification on a peptide sequence.</summary>
    MS_PTM_localization_score = 1001968,

    /// <summary>ProteomeDiscoverer:phosphoRS score: Peptide score based on the cumulative binomial probability that the observed match is a random event.</summary>
    MS_ProteomeDiscoverer_phosphoRS_score = 1001969,

    /// <summary>ProteomeDiscoverer:phosphoRS sequence probability: Probability that the respective isoform is correct.</summary>
    MS_ProteomeDiscoverer_phosphoRS_sequence_probability = 1001970,

    /// <summary>ProteomeDiscoverer:phosphoRS site probability: Estimate of the probability that the respective site is truly phosphorylated.</summary>
    MS_ProteomeDiscoverer_phosphoRS_site_probability = 1001971,

    /// <summary>PTM scoring algorithm version: Version of the post-translational modification scoring algorithm.</summary>
    MS_PTM_scoring_algorithm_version = 1001972,

    /// <summary>DeBunker: DeBunker software.</summary>
    MS_DeBunker = 1001973,

    /// <summary>DeBunker:score: Score specific to DeBunker.</summary>
    MS_DeBunker_score = 1001974,

    /// <summary>delta m/z: The difference between a theoretically calculated m/z and the corresponding experimentally measured m/z. It can be expressed as absolute or relative value.</summary>
    MS_delta_m_z = 1001975,

    /// <summary>m/z difference: The difference between a theoretically calculated m/z and the corresponding experimentally measured m/z. It can be expressed as absolute or relative value.</summary>
    MS_m_z_difference = MS_delta_m_z,

    /// <summary>delta M: The difference between a theoretically calculated molecular mass M and the corresponding experimentally measured M. It can be expressed as absolute or relative value.</summary>
    MS_delta_M = 1001976,

    /// <summary>mass difference: The difference between a theoretically calculated molecular mass M and the corresponding experimentally measured M. It can be expressed as absolute or relative value.</summary>
    MS_mass_difference = MS_delta_M,

    /// <summary>MSQuant: MSQuant software.</summary>
    MS_MSQuant = 1001977,

    /// <summary>MSQuant:PTM-score: The PTM score from MSQuant software.</summary>
    MS_MSQuant_PTM_score = 1001978,

    /// <summary>MaxQuant:PTM Score: The PTM score from MaxQuant software.</summary>
    MS_MaxQuant_PTM_Score = 1001979,

    /// <summary>MaxQuant:Phospho (STY) Probabilities: The Phospho (STY) Probabilities from MaxQuant software.</summary>
    MS_MaxQuant_Phospho__STY__Probabilities = 1001980,

    /// <summary>MaxQuant:Phospho (STY) Score Diffs: The Phospho (STY) Score Diffs from MaxQuant software.</summary>
    MS_MaxQuant_Phospho__STY__Score_Diffs = 1001981,

    /// <summary>MaxQuant:P-site localization probability: The P-site localization probability value from MaxQuant software.</summary>
    MS_MaxQuant_P_site_localization_probability = 1001982,

    /// <summary>MaxQuant:PTM Delta Score: The PTM Delta Score value from MaxQuant software (Difference between highest scoring site and second highest).</summary>
    MS_MaxQuant_PTM_Delta_Score = 1001983,

    /// <summary>Ascore: Ascore software.</summary>
    MS_Ascore = 1001984,

    /// <summary>Ascore:Ascore: The Ascore score value from Ascore software.</summary>
    MS_Ascore_Ascore = 1001985,

    /// <summary>H-Score: H-Score for peptide phosphorylation site location.</summary>
    MS_H_Score = 1001986,

    /// <summary>vacuum drying MALDI sample preparation: Vacuum-drying MALDI sample preparation crystallization method.</summary>
    MS_vacuum_drying_MALDI_sample_preparation = 1001987,

    /// <summary>crushed crystal MALDI sample preparation: Crushed-crystal MALDI sample preparation method.</summary>
    MS_crushed_crystal_MALDI_sample_preparation = 1001988,

    /// <summary>fast evaporation MALDI sample preparation: Fast-evaporation MALDI sample preparation method.</summary>
    MS_fast_evaporation_MALDI_sample_preparation = 1001989,

    /// <summary>overlayer MALDI sample preparation: Overlayer method combining features of the crushed-crystal method and the fast-evaporation method.</summary>
    MS_overlayer_MALDI_sample_preparation = 1001990,

    /// <summary>sandwich MALDI sample preparation: Sandwich MALDI sample preparation method.</summary>
    MS_sandwich_MALDI_sample_preparation = 1001991,

    /// <summary>spin coating MALDI sample preparation: Spin coating MALDI sample preparation method.</summary>
    MS_spin_coating_MALDI_sample_preparation = 1001992,

    /// <summary>quick and dirty MALDI sample preparation: Quick & dirty (Q&D) sample preparation separating matrix handling from sample handling.</summary>
    MS_quick_and_dirty_MALDI_sample_preparation = 1001993,

    /// <summary>top hat baseline reduction: Top-hat morphological filter based on the basic morphological operations 'erosion' and 'dilatation'.</summary>
    MS_top_hat_baseline_reduction = 1001994,

    /// <summary>convex hull baseline reduction: Constructs the baseline by fitting multiple parabolas to the spectrum starting with the large scale structures.</summary>
    MS_convex_hull_baseline_reduction = 1001995,

    /// <summary>median baseline reduction: The spectrum that will be baseline subtracted is divided into a number of segments.</summary>
    MS_median_baseline_reduction = 1001996,

    /// <summary>wavelet transformation smoothing: The random noise is removed by using the undecimated wavelet transform.</summary>
    MS_wavelet_transformation_smoothing = 1001997,

    /// <summary>sophisticated numerical annotation procedure: It searches for known patterns in the measured spectrum.</summary>
    MS_sophisticated_numerical_annotation_procedure = 1001998,

    /// <summary>SNAP: It searches for known patterns in the measured spectrum.</summary>
    MS_SNAP = MS_sophisticated_numerical_annotation_procedure,

    /// <summary>area normalization: Normalization of areas below the curves.</summary>
    MS_area_normalization = 1001999,

    /// <summary>LIFT: A Bruker's proprietary technique where molecular ions are initially accelerated at lower energy, then collide with inert gas in a collision cell that is then 'lifted' to high potential. The use of inert gas is optional, as it could lift also fragments provided by LID.</summary>
    MS_LIFT = 1002000,

    /// <summary>MS1 Label-based raw feature quantitation: MS1 Label-based raw feature quantitation.</summary>
    MS_MS1_Label_based_raw_feature_quantitation = 1002001,

    /// <summary>MS1 Label-based peptide level quantitation: MS1 Label-based peptide level quantitation.</summary>
    MS_MS1_Label_based_peptide_level_quantitation = 1002002,

    /// <summary>MS1 Label-based protein level quantitation: MS1 Label-based protein level quantitation.</summary>
    MS_MS1_Label_based_protein_level_quantitation = 1002003,

    /// <summary>MS1 Label-based proteinGroup level quantitation: MS1 Label-based proteinGroup level quantitation.</summary>
    MS_MS1_Label_based_proteinGroup_level_quantitation = 1002004,

    /// <summary>iRT retention time normalization standard: A de facto standard providing the retention times at which a specific set of 10 reference peptides exit the reference chromatographic column. The kit may be obtain from Biognosys.</summary>
    MS_iRT_retention_time_normalization_standard = 1002005,

    /// <summary>SRM transition type: The type of the transitions, e.g. target or decoy.</summary>
    MS_SRM_transition_type = 1002006,

    /// <summary>MRM transition type: The type of the transitions, e.g. target or decoy.</summary>
    MS_MRM_transition_type = MS_SRM_transition_type,

    /// <summary>target SRM transition: A transition used to target a specific compound that may be in the sample.</summary>
    MS_target_SRM_transition = 1002007,

    /// <summary>target MRM transition: A transition used to target a specific compound that may be in the sample.</summary>
    MS_target_MRM_transition = MS_target_SRM_transition,

    /// <summary>decoy SRM transition: A transition not expected to be present in the sample and used to calculate statistical confidence of target transition detections in some workflows.</summary>
    MS_decoy_SRM_transition = 1002008,

    /// <summary>decoy MRM transition: A transition not expected to be present in the sample and used to calculate statistical confidence of target transition detections in some workflows.</summary>
    MS_decoy_MRM_transition = MS_decoy_SRM_transition,

    /// <summary>isobaric label quantification analysis: Quantification analysis using an isobaric labeling workflow.</summary>
    MS_isobaric_label_quantification_analysis = 1002009,

    /// <summary>TMT quantification analysis: Quantification analysis using the Thermo Fisher tandem mass tag (TMT) labeling workflow.</summary>
    MS_TMT_quantification_analysis = 1002010,

    /// <summary>desorption electrospray ionization: Combination of electrospray and desorption ionization method that ionizes gases, liquids and solids in open air under atmospheric pressure.</summary>
    MS_desorption_electrospray_ionization = 1002011,

    /// <summary>DESI: Combination of electrospray and desorption ionization method that ionizes gases, liquids and solids in open air under atmospheric pressure.</summary>
    MS_DESI = MS_desorption_electrospray_ionization,

    /// <summary>Mascot:PTM site assignment confidence: Relative probability that PTM site assignment is correct, derived from the Mascot score difference between matches to the same spectrum (Mascot Delta Score).</summary>
    MS_Mascot_PTM_site_assignment_confidence = 1002012,

    /// <summary>collision energy ramp start: Collision energy at the start of the collision energy ramp.</summary>
    MS_collision_energy_ramp_start = 1002013,

    /// <summary>collision energy ramp end: Collision energy at the end of the collision energy ramp.</summary>
    MS_collision_energy_ramp_end = 1002014,

    /// <summary>unimod root node: The root node of the unimod modifications ontology.</summary>
    UNIMOD_unimod_root_node = 100000000,

    /// <summary>Acetyl: Acetylation.</summary>
    UNIMOD_Acetyl = 100000001,

    /// <summary>Amidated: Amidation.</summary>
    UNIMOD_Amidated = 100000002,

    /// <summary>Biotin: Biotinylation.</summary>
    UNIMOD_Biotin = 100000003,

    /// <summary>Carbamidomethyl: Iodoacetamide derivative.</summary>
    UNIMOD_Carbamidomethyl = 100000004,

    /// <summary>Carbamyl: Carbamylation.</summary>
    UNIMOD_Carbamyl = 100000005,

    /// <summary>Carboxymethyl: Iodoacetic acid derivative.</summary>
    UNIMOD_Carboxymethyl = 100000006,

    /// <summary>Deamidated: Deamidation.</summary>
    UNIMOD_Deamidated = 100000007,

    /// <summary>ICAT-G: Gygi ICAT(TM) d0.</summary>
    UNIMOD_ICAT_G = 100000008,

    /// <summary>ICAT-G:2H(8): Gygi ICAT(TM) d8.</summary>
    UNIMOD_ICAT_G_2H_8_ = 100000009,

    /// <summary>Met->Hse: Homoserine.</summary>
    UNIMOD_Met__Hse = 100000010,

    /// <summary>Met->Hsl: Homoserine lactone.</summary>
    UNIMOD_Met__Hsl = 100000011,

    /// <summary>ICAT-D:2H(8): Applied Biosystems original ICAT(TM) d8.</summary>
    UNIMOD_ICAT_D_2H_8_ = 100000012,

    /// <summary>ICAT-D: Applied Biosystems original ICAT(TM) d0.</summary>
    UNIMOD_ICAT_D = 100000013,

    /// <summary>NIPCAM: N-isopropylcarboxamidomethyl.</summary>
    UNIMOD_NIPCAM = 100000017,

    /// <summary>PEO-Iodoacetyl-LC-Biotin: Biotinyl-iodoacetamidyl-3,6-dioxaoctanediamine.</summary>
    UNIMOD_PEO_Iodoacetyl_LC_Biotin = 100000020,

    /// <summary>Phospho: Phosphorylation.</summary>
    UNIMOD_Phospho = 100000021,

    /// <summary>Dehydrated: Dehydration.</summary>
    UNIMOD_Dehydrated = 100000023,

    /// <summary>Propionamide: Acrylamide adduct.</summary>
    UNIMOD_Propionamide = 100000024,

    /// <summary>Pyridylacetyl: Pyridylacetyl.</summary>
    UNIMOD_Pyridylacetyl = 100000025,

    /// <summary>Pyro-carbamidomethyl: S-carbamoylmethylcysteine cyclization (N-terminus).</summary>
    UNIMOD_Pyro_carbamidomethyl = 100000026,

    /// <summary>Glu->pyro-Glu: Pyro-glu from E.</summary>
    UNIMOD_Glu__pyro_Glu = 100000027,

    /// <summary>Gln->pyro-Glu: Pyro-glu from Q.</summary>
    UNIMOD_Gln__pyro_Glu = 100000028,

    /// <summary>SMA: N-Succinimidyl-2-morpholine acetate.</summary>
    UNIMOD_SMA = 100000029,

    /// <summary>Cation:Na: Sodium adduct.</summary>
    UNIMOD_Cation_Na = 100000030,

    /// <summary>Pyridylethyl: S-pyridylethylation.</summary>
    UNIMOD_Pyridylethyl = 100000031,

    /// <summary>Methyl: Methylation.</summary>
    UNIMOD_Methyl = 100000034,

    /// <summary>Oxidation: Oxidation or Hydroxylation.</summary>
    UNIMOD_Oxidation = 100000035,

    /// <summary>Dimethyl: Di-Methylation.</summary>
    UNIMOD_Dimethyl = 100000036,

    /// <summary>Trimethyl: Tri-Methylation.</summary>
    UNIMOD_Trimethyl = 100000037,

    /// <summary>Methylthio: Beta-methylthiolation.</summary>
    UNIMOD_Methylthio = 100000039,

    /// <summary>Sulfo: O-Sulfonation.</summary>
    UNIMOD_Sulfo = 100000040,

    /// <summary>Hex: Hexose.</summary>
    UNIMOD_Hex = 100000041,

    /// <summary>Lipoyl: Lipoyl.</summary>
    UNIMOD_Lipoyl = 100000042,

    /// <summary>HexNAc: N-Acetylhexosamine.</summary>
    UNIMOD_HexNAc = 100000043,

    /// <summary>Farnesyl: Farnesylation.</summary>
    UNIMOD_Farnesyl = 100000044,

    /// <summary>Myristoyl: Myristoylation.</summary>
    UNIMOD_Myristoyl = 100000045,

    /// <summary>PyridoxalPhosphate: Pyridoxal phosphate.</summary>
    UNIMOD_PyridoxalPhosphate = 100000046,

    /// <summary>Palmitoyl: Palmitoylation.</summary>
    UNIMOD_Palmitoyl = 100000047,

    /// <summary>GeranylGeranyl: Geranyl-geranyl.</summary>
    UNIMOD_GeranylGeranyl = 100000048,

    /// <summary>Phosphopantetheine: Phosphopantetheine.</summary>
    UNIMOD_Phosphopantetheine = 100000049,

    /// <summary>FAD: Flavin adenine dinucleotide.</summary>
    UNIMOD_FAD = 100000050,

    /// <summary>Tripalmitate: N-acyl diglyceride cysteine.</summary>
    UNIMOD_Tripalmitate = 100000051,

    /// <summary>Guanidinyl: Guanidination.</summary>
    UNIMOD_Guanidinyl = 100000052,

    /// <summary>HNE: 4-hydroxynonenal (HNE).</summary>
    UNIMOD_HNE = 100000053,

    /// <summary>Glucuronyl: N-glucuronylation.</summary>
    UNIMOD_Glucuronyl = 100000054,

    /// <summary>Glutathione: Glutathione disulfide.</summary>
    UNIMOD_Glutathione = 100000055,

    /// <summary>Acetyl:2H(3): Acetate labeling reagent (N-term & K) (heavy form, +3amu).</summary>
    UNIMOD_Acetyl_2H_3_ = 100000056,

    /// <summary>Propionyl: Propionate labeling reagent light form (N-term & K).</summary>
    UNIMOD_Propionyl = 100000058,

    /// <summary>Propionyl:13C(3): Propionate labeling reagent heavy form (+3amu), N-term & K.</summary>
    UNIMOD_Propionyl_13C_3_ = 100000059,

    /// <summary>GIST-Quat: Quaternary amine labeling reagent light form (N-term & K).</summary>
    UNIMOD_GIST_Quat = 100000060,

    /// <summary>GIST-Quat:2H(3): Quaternary amine labeling reagent heavy (+3amu) form, N-term & K.</summary>
    UNIMOD_GIST_Quat_2H_3_ = 100000061,

    /// <summary>GIST-Quat:2H(6): Quaternary amine labeling reagent heavy form (+6amu), N-term & K.</summary>
    UNIMOD_GIST_Quat_2H_6_ = 100000062,

    /// <summary>GIST-Quat:2H(9): Quaternary amine labeling reagent heavy form (+9amu), N-term & K.</summary>
    UNIMOD_GIST_Quat_2H_9_ = 100000063,

    /// <summary>Succinyl: Succinic anhydride labeling reagent light form (N-term & K).</summary>
    UNIMOD_Succinyl = 100000064,

    /// <summary>Succinyl:2H(4): Succinic anhydride labeling reagent, heavy form (+4amu, 4H2), N-term & K.</summary>
    UNIMOD_Succinyl_2H_4_ = 100000065,

    /// <summary>Succinyl:13C(4): Succinic anhydride labeling reagent, heavy form (+4amu, 4C13), N-term & K.</summary>
    UNIMOD_Succinyl_13C_4_ = 100000066,

    /// <summary>Iminobiotin: Iminobiotinylation.</summary>
    UNIMOD_Iminobiotin = 100000089,

    /// <summary>ESP: ESP-Tag light d0.</summary>
    UNIMOD_ESP = 100000090,

    /// <summary>ESP:2H(10): ESP-Tag heavy d10.</summary>
    UNIMOD_ESP_2H_10_ = 100000091,

    /// <summary>NHS-LC-Biotin: NHS-LC-Biotin.</summary>
    UNIMOD_NHS_LC_Biotin = 100000092,

    /// <summary>EDT-maleimide-PEO-biotin: EDT-maleimide-PEO-biotin.</summary>
    UNIMOD_EDT_maleimide_PEO_biotin = 100000093,

    /// <summary>IMID: IMID d0.</summary>
    UNIMOD_IMID = 100000094,

    /// <summary>IMID:2H(4): IMID d4.</summary>
    UNIMOD_IMID_2H_4_ = 100000095,

    /// <summary>Propionamide:2H(3): Acrylamide d3.</summary>
    UNIMOD_Propionamide_2H_3_ = 100000097,

    /// <summary>ICAT-C: Applied Biosystems cleavable ICAT(TM) light.</summary>
    UNIMOD_ICAT_C = 100000105,

    /// <summary>ICAT-C:13C(9): Applied Biosystems cleavable ICAT(TM) heavy.</summary>
    UNIMOD_ICAT_C_13C_9_ = 100000106,

    /// <summary>FormylMet: Addition of N-formyl met.</summary>
    UNIMOD_FormylMet = 100000107,

    /// <summary>Nethylmaleimide: N-ethylmaleimide on cysteines.</summary>
    UNIMOD_Nethylmaleimide = 100000108,

    /// <summary>OxLysBiotinRed: Oxidized lysine biotinylated with biotin-LC-hydrazide, reduced.</summary>
    UNIMOD_OxLysBiotinRed = 100000112,

    /// <summary>OxLysBiotin: Oxidized lysine biotinylated with biotin-LC-hydrazide.</summary>
    UNIMOD_OxLysBiotin = 100000113,

    /// <summary>OxProBiotinRed: Oxidized proline biotinylated with biotin-LC-hydrazide, reduced.</summary>
    UNIMOD_OxProBiotinRed = 100000114,

    /// <summary>OxProBiotin: Oxidized Proline biotinylated with biotin-LC-hydrazide.</summary>
    UNIMOD_OxProBiotin = 100000115,

    /// <summary>OxArgBiotin: Oxidized arginine biotinylated with biotin-LC-hydrazide.</summary>
    UNIMOD_OxArgBiotin = 100000116,

    /// <summary>OxArgBiotinRed: Oxidized arginine biotinylated with biotin-LC-hydrazide, reduced.</summary>
    UNIMOD_OxArgBiotinRed = 100000117,

    /// <summary>EDT-iodoacetyl-PEO-biotin: EDT-iodo-PEO-biotin.</summary>
    UNIMOD_EDT_iodoacetyl_PEO_biotin = 100000118,

    /// <summary>IBTP: Thio Ether Formation - BTP Adduct.</summary>
    UNIMOD_IBTP = 100000119,

    /// <summary>GlyGly: Ubiquitinylation residue.</summary>
    UNIMOD_GlyGly = 100000121,

    /// <summary>Formyl: Formylation.</summary>
    UNIMOD_Formyl = 100000122,

    /// <summary>ICAT-H: N-iodoacetyl, p-chlorobenzyl-12C6-glucamine.</summary>
    UNIMOD_ICAT_H = 100000123,

    /// <summary>ICAT-H:13C(6): N-iodoacetyl, p-chlorobenzyl-13C6-glucamine.</summary>
    UNIMOD_ICAT_H_13C_6_ = 100000124,

    /// <summary>Thioacyl: 3-sulfanylpropanoyl.</summary>
    UNIMOD_Thioacyl = 100000126,

    /// <summary>Fluoro: Fluorophenylalanine replacement of phenylalanine.</summary>
    UNIMOD_Fluoro = 100000127,

    /// <summary>Fluorescein: 5-Iodoacetamidofluorescein (Molecular Probe, Eugene, OR).</summary>
    UNIMOD_Fluorescein = 100000128,

    /// <summary>Iodo: Iodination.</summary>
    UNIMOD_Iodo = 100000129,

    /// <summary>Diiodo: Di-Iodination.</summary>
    UNIMOD_Diiodo = 100000130,

    /// <summary>Triiodo: Tri-Iodination.</summary>
    UNIMOD_Triiodo = 100000131,

    /// <summary>Myristoleyl: (cis-delta 5)-tetradecaenoyl.</summary>
    UNIMOD_Myristoleyl = 100000134,

    /// <summary>Myristoyl+Delta:H(-4): (cis,cis-delta 5, delta 8)-tetradecadienoyl.</summary>
    UNIMOD_Myristoyl_Delta_H__4_ = 100000135,

    /// <summary>Benzoyl: Labeling reagent light form (N-term & K).</summary>
    UNIMOD_Benzoyl = 100000136,

    /// <summary>Hex(5)HexNAc(2): N-linked glycan core.</summary>
    UNIMOD_Hex_5_HexNAc_2_ = 100000137,

    /// <summary>Dansyl: 5-dimethylaminonaphthalene-1-sulfonyl.</summary>
    UNIMOD_Dansyl = 100000139,

    /// <summary>a-type-ion: ISD a-series (C-Term).</summary>
    UNIMOD_a_type_ion = 100000140,

    /// <summary>Amidine: Amidination of lysines or N-terminal amines with methyl acetimidate.</summary>
    UNIMOD_Amidine = 100000141,

    /// <summary>HexNAc(1)dHex(1): HexNAc1dHex1.</summary>
    UNIMOD_HexNAc_1_dHex_1_ = 100000142,

    /// <summary>HexNAc(2): HexNAc2.</summary>
    UNIMOD_HexNAc_2_ = 100000143,

    /// <summary>Hex(3): Hex3.</summary>
    UNIMOD_Hex_3_ = 100000144,

    /// <summary>HexNAc(1)dHex(2): HexNAc1dHex2.</summary>
    UNIMOD_HexNAc_1_dHex_2_ = 100000145,

    /// <summary>Hex(1)HexNAc(1)dHex(1): Hex1HexNAc1dHex1.</summary>
    UNIMOD_Hex_1_HexNAc_1_dHex_1_ = 100000146,

    /// <summary>HexNAc(2)dHex(1): HexNAc2dHex1.</summary>
    UNIMOD_HexNAc_2_dHex_1_ = 100000147,

    /// <summary>Hex(1)HexNAc(2): Hex1HexNAc2.</summary>
    UNIMOD_Hex_1_HexNAc_2_ = 100000148,

    /// <summary>Hex(1)HexNAc(1)NeuAc(1): Hex1HexNAc1NeuAc1.</summary>
    UNIMOD_Hex_1_HexNAc_1_NeuAc_1_ = 100000149,

    /// <summary>HexNAc(2)dHex(2): HexNAc2dHex2.</summary>
    UNIMOD_HexNAc_2_dHex_2_ = 100000150,

    /// <summary>Hex(1)HexNAc(2)Pent(1): Hex1HexNAc2Pent1.</summary>
    UNIMOD_Hex_1_HexNAc_2_Pent_1_ = 100000151,

    /// <summary>Hex(1)HexNAc(2)dHex(1): Hex1HexNAc2dHex1.</summary>
    UNIMOD_Hex_1_HexNAc_2_dHex_1_ = 100000152,

    /// <summary>Hex(2)HexNAc(2): Hex2HexNAc2.</summary>
    UNIMOD_Hex_2_HexNAc_2_ = 100000153,

    /// <summary>Hex(3)HexNAc(1)Pent(1): Hex3HexNAc1Pent1.</summary>
    UNIMOD_Hex_3_HexNAc_1_Pent_1_ = 100000154,

    /// <summary>Hex(1)HexNAc(2)dHex(1)Pent(1): Hex1HexNAc2dHex1Pent1.</summary>
    UNIMOD_Hex_1_HexNAc_2_dHex_1_Pent_1_ = 100000155,

    /// <summary>Hex(1)HexNAc(2)dHex(2): Hex1HexNAc2dHex2.</summary>
    UNIMOD_Hex_1_HexNAc_2_dHex_2_ = 100000156,

    /// <summary>Hex(2)HexNAc(2)Pent(1): Hex2HexNAc2Pent1.</summary>
    UNIMOD_Hex_2_HexNAc_2_Pent_1_ = 100000157,

    /// <summary>Hex(2)HexNAc(2)dHex(1): Hex2HexNAc2dHex1.</summary>
    UNIMOD_Hex_2_HexNAc_2_dHex_1_ = 100000158,

    /// <summary>Hex(3)HexNAc(2): Hex3HexNAc2.</summary>
    UNIMOD_Hex_3_HexNAc_2_ = 100000159,

    /// <summary>Hex(1)HexNAc(1)NeuAc(2): Hex1HexNAc1NeuAc2.</summary>
    UNIMOD_Hex_1_HexNAc_1_NeuAc_2_ = 100000160,

    /// <summary>Hex(3)HexNAc(2)P(1): Hex3HexNAc2P1.</summary>
    UNIMOD_Hex_3_HexNAc_2_P_1_ = 100000161,

    /// <summary>Delta:S(-1)Se(1): Selenium replaces sulfur.</summary>
    UNIMOD_Delta_S__1_Se_1_ = 100000162,

    /// <summary>Delta:H(1)O(-1)18O(1): Glycosylated asparagine 18O labeling.</summary>
    UNIMOD_Delta_H_1_O__1_18O_1_ = 100000170,

    /// <summary>NBS:13C(6): Shimadzu NBS-13C.</summary>
    UNIMOD_NBS_13C_6_ = 100000171,

    /// <summary>NBS: Shimadzu NBS-12C.</summary>
    UNIMOD_NBS = 100000172,

    /// <summary>BHT: Michael addition of BHT quinone methide to Cysteine and Lysine.</summary>
    UNIMOD_BHT = 100000176,

    /// <summary>DAET: Phosphorylation to amine thiol.</summary>
    UNIMOD_DAET = 100000178,

    /// <summary>Label:13C(9): 13C(9) Silac label.</summary>
    UNIMOD_Label_13C_9_ = 100000184,

    /// <summary>Label:13C(9)+Phospho: C13 label (Phosphotyrosine).</summary>
    UNIMOD_Label_13C_9__Phospho = 100000185,

    /// <summary>HPG: Hydroxyphenylglyoxal arginine.</summary>
    UNIMOD_HPG = 100000186,

    /// <summary>2HPG: Bis(hydroxphenylglyoxal) arginine.</summary>
    UNIMOD_2HPG = 100000187,

    /// <summary>Label:13C(6): 13C(6) Silac label.</summary>
    UNIMOD_Label_13C_6_ = 100000188,

    /// <summary>Label:18O(2): O18 label at both C-terminal oxygens.</summary>
    UNIMOD_Label_18O_2_ = 100000193,

    /// <summary>AccQTag: 6-aminoquinolyl-N-hydroxysuccinimidyl carbamate.</summary>
    UNIMOD_AccQTag = 100000194,

    /// <summary>QAT: APTA-d0.</summary>
    UNIMOD_QAT = 100000195,

    /// <summary>QAT:2H(3): APTA d3.</summary>
    UNIMOD_QAT_2H_3_ = 100000196,

    /// <summary>EQAT: EAPTA d0.</summary>
    UNIMOD_EQAT = 100000197,

    /// <summary>EQAT:2H(5): EAPTA d5.</summary>
    UNIMOD_EQAT_2H_5_ = 100000198,

    /// <summary>Dimethyl:2H(4): DiMethyl-CHD2.</summary>
    UNIMOD_Dimethyl_2H_4_ = 100000199,

    /// <summary>Ethanedithiol: EDT.</summary>
    UNIMOD_Ethanedithiol = 100000200,

    /// <summary>Delta:H(6)C(6)O(1): Acrolein addition +94.</summary>
    UNIMOD_Delta_H_6_C_6_O_1_ = 100000205,

    /// <summary>Delta:H(4)C(3)O(1): Acrolein addition +56.</summary>
    UNIMOD_Delta_H_4_C_3_O_1_ = 100000206,

    /// <summary>Delta:H(2)C(3): Acrolein addition +38.</summary>
    UNIMOD_Delta_H_2_C_3_ = 100000207,

    /// <summary>Delta:H(4)C(6): Acrolein addition +76.</summary>
    UNIMOD_Delta_H_4_C_6_ = 100000208,

    /// <summary>Delta:H(8)C(6)O(2): Acrolein addition +112.</summary>
    UNIMOD_Delta_H_8_C_6_O_2_ = 100000209,

    /// <summary>NEIAA: N-ethyl iodoacetamide-d0.</summary>
    UNIMOD_NEIAA = 100000211,

    /// <summary>NEIAA:2H(5): N-ethyl iodoacetamide-d5.</summary>
    UNIMOD_NEIAA_2H_5_ = 100000212,

    /// <summary>ADP-Ribosyl: ADP Ribose addition.</summary>
    UNIMOD_ADP_Ribosyl = 100000213,

    /// <summary>iTRAQ4plex: Representative mass and accurate mass for 116 & 117.</summary>
    UNIMOD_iTRAQ4plex = 100000214,

    /// <summary>IGBP: Light IDBEST tag for quantitation.</summary>
    UNIMOD_IGBP = 100000243,

    /// <summary>Crotonaldehyde: Crotonaldehyde.</summary>
    UNIMOD_Crotonaldehyde = 100000253,

    /// <summary>Delta:H(2)C(2): Acetaldehyde +26.</summary>
    UNIMOD_Delta_H_2_C_2_ = 100000254,

    /// <summary>Delta:H(4)C(2): Acetaldehyde +28.</summary>
    UNIMOD_Delta_H_4_C_2_ = 100000255,

    /// <summary>Delta:H(4)C(3): Propionaldehyde +40.</summary>
    UNIMOD_Delta_H_4_C_3_ = 100000256,

    /// <summary>Label:18O(1): O18 Labeling.</summary>
    UNIMOD_Label_18O_1_ = 100000258,

    /// <summary>Label:13C(6)15N(2): 13C(6) 15N(2) Silac label.</summary>
    UNIMOD_Label_13C_6_15N_2_ = 100000259,

    /// <summary>Thiophospho: Thiophosphorylation.</summary>
    UNIMOD_Thiophospho = 100000260,

    /// <summary>SPITC: 4-sulfophenyl isothiocyanate.</summary>
    UNIMOD_SPITC = 100000261,

    /// <summary>Label:2H(3): Trideuteration.</summary>
    UNIMOD_Label_2H_3_ = 100000262,

    /// <summary>PET: Phosphorylation to pyridyl thiol.</summary>
    UNIMOD_PET = 100000264,

    /// <summary>Label:13C(6)15N(4): 13C(6) 15N(4) Silac label.</summary>
    UNIMOD_Label_13C_6_15N_4_ = 100000267,

    /// <summary>Label:13C(5)15N(1): 13C(5) 15N(1) Silac label.</summary>
    UNIMOD_Label_13C_5_15N_1_ = 100000268,

    /// <summary>Label:13C(9)15N(1): 13C(9) 15N(1) Silac label.</summary>
    UNIMOD_Label_13C_9_15N_1_ = 100000269,

    /// <summary>Cytopiloyne: Nucleophilic addtion to cytopiloyne.</summary>
    UNIMOD_Cytopiloyne = 100000270,

    /// <summary>Cytopiloyne+water: Nucleophilic addition to cytopiloyne+H2O.</summary>
    UNIMOD_Cytopiloyne_water = 100000271,

    /// <summary>CAF: Sulfonation of N-terminus.</summary>
    UNIMOD_CAF = 100000272,

    /// <summary>Xlink:SSD: Covalent modification of lysine by cross-linking reagent.</summary>
    UNIMOD_Xlink_SSD = 100000273,

    /// <summary>Nitrosyl: S-nitrosylation.</summary>
    UNIMOD_Nitrosyl = 100000275,

    /// <summary>AEBS: Aminoethylbenzenesulfonylation.</summary>
    UNIMOD_AEBS = 100000276,

    /// <summary>Ethanolyl: Ethanolation.</summary>
    UNIMOD_Ethanolyl = 100000278,

    /// <summary>Ethyl: Ethylation.</summary>
    UNIMOD_Ethyl = 100000280,

    /// <summary>CoenzymeA: Cysteine modified Coenzyme A.</summary>
    UNIMOD_CoenzymeA = 100000281,

    /// <summary>Methyl:2H(2): Deuterium Methylation of Lysine.</summary>
    UNIMOD_Methyl_2H_2_ = 100000284,

    /// <summary>SulfanilicAcid: Light Sulfanilic Acid (SA) C12.</summary>
    UNIMOD_SulfanilicAcid = 100000285,

    /// <summary>SulfanilicAcid:13C(6): Heavy Sulfanilic Acid (SA) C13.</summary>
    UNIMOD_SulfanilicAcid_13C_6_ = 100000286,

    /// <summary>Trp->Oxolactone: Tryptophan oxidation to oxolactone.</summary>
    UNIMOD_Trp__Oxolactone = 100000288,

    /// <summary>Biotin-PEO-Amine: Biotin polyethyleneoxide amine.</summary>
    UNIMOD_Biotin_PEO_Amine = 100000289,

    /// <summary>Biotin-HPDP: Pierce EZ-Link Biotin-HPDP.</summary>
    UNIMOD_Biotin_HPDP = 100000290,

    /// <summary>Delta:Hg(1): Mercury Mercaptan.</summary>
    UNIMOD_Delta_Hg_1_ = 100000291,

    /// <summary>IodoU-AMP: Cross-link of (Iodo)-uracil MP with W,F,Y.</summary>
    UNIMOD_IodoU_AMP = 100000292,

    /// <summary>CAMthiopropanoyl: 3-(carbamidomethylthio)propanoyl.</summary>
    UNIMOD_CAMthiopropanoyl = 100000293,

    /// <summary>IED-Biotin: Biotinoyl-iodoacetyl-ethylenediamine.</summary>
    UNIMOD_IED_Biotin = 100000294,

    /// <summary>dHex: Fucose.</summary>
    UNIMOD_dHex = 100000295,

    /// <summary>Methyl:2H(3): Deuterated methyl ester.</summary>
    UNIMOD_Methyl_2H_3_ = 100000298,

    /// <summary>Carboxy: Carboxylation.</summary>
    UNIMOD_Carboxy = 100000299,

    /// <summary>Bromobimane: Monobromobimane derivative.</summary>
    UNIMOD_Bromobimane = 100000301,

    /// <summary>Menadione: Menadione quinone derivative.</summary>
    UNIMOD_Menadione = 100000302,

    /// <summary>DeStreak: Cysteine mercaptoethanol.</summary>
    UNIMOD_DeStreak = 100000303,

    /// <summary>dHex(1)Hex(3)HexNAc(4): Fucosylated biantennary (-2 galactose).</summary>
    UNIMOD_dHex_1_Hex_3_HexNAc_4_ = 100000305,

    /// <summary>dHex(1)Hex(4)HexNAc(4): Fucosylated biantennary (-1 galactose).</summary>
    UNIMOD_dHex_1_Hex_4_HexNAc_4_ = 100000307,

    /// <summary>dHex(1)Hex(5)HexNAc(4): Fucosylated biantennary.</summary>
    UNIMOD_dHex_1_Hex_5_HexNAc_4_ = 100000308,

    /// <summary>Hex(3)HexNAc(4): Biantennary (-2 galactose).</summary>
    UNIMOD_Hex_3_HexNAc_4_ = 100000309,

    /// <summary>Hex(4)HexNAc(4): Biantennary (-1 galactose).</summary>
    UNIMOD_Hex_4_HexNAc_4_ = 100000310,

    /// <summary>Hex(5)HexNAc(4): Biantennary.</summary>
    UNIMOD_Hex_5_HexNAc_4_ = 100000311,

    /// <summary>Cysteinyl: Cysteinylation.</summary>
    UNIMOD_Cysteinyl = 100000312,

    /// <summary>Lys-loss: Loss of C-terminal K from Heavy Chain of MAb.</summary>
    UNIMOD_Lys_loss = 100000313,

    /// <summary>Nmethylmaleimide: Nmethylmaleimide.</summary>
    UNIMOD_Nmethylmaleimide = 100000314,

    /// <summary>DimethylpyrroleAdduct: 2,5-dimethypyrrole.</summary>
    UNIMOD_DimethylpyrroleAdduct = 100000316,

    /// <summary>Delta:H(2)C(5): MDA adduct +62.</summary>
    UNIMOD_Delta_H_2_C_5_ = 100000318,

    /// <summary>Delta:H(2)C(3)O(1): MDA adduct +54.</summary>
    UNIMOD_Delta_H_2_C_3_O_1_ = 100000319,

    /// <summary>Nethylmaleimide+water: Nethylmaleimidehydrolysis.</summary>
    UNIMOD_Nethylmaleimide_water = 100000320,

    /// <summary>Xlink:B10621: Bis-N-I-sulfonerahodamine.</summary>
    UNIMOD_Xlink_B10621 = 100000323,

    /// <summary>DTBP: Dimethyl 3,3\'-dithiobispropionimidate.</summary>
    UNIMOD_DTBP = 100000324,

    /// <summary>FP-Biotin: 10-ethoxyphosphinyl-N-(biotinamidopentyl)decanamide.</summary>
    UNIMOD_FP_Biotin = 100000325,

    /// <summary>Delta:H(4)C(2)O(-1)S(1): S-Ethylcystine from Serine.</summary>
    UNIMOD_Delta_H_4_C_2_O__1_S_1_ = 100000327,

    /// <summary>Methyl:2H(3)13C(1): Monomethylated arginine.</summary>
    UNIMOD_Methyl_2H_3_13C_1_ = 100000329,

    /// <summary>Dimethyl:2H(6)13C(2): Dimethylated arginine.</summary>
    UNIMOD_Dimethyl_2H_6_13C_2_ = 100000330,

    /// <summary>Thiophos-S-S-biotin: Thiophosphate labeled with biotin-HPDP.</summary>
    UNIMOD_Thiophos_S_S_biotin = 100000332,

    /// <summary>Can-FP-biotin: 6-N-biotinylaminohexyl isopropyl phosphate.</summary>
    UNIMOD_Can_FP_biotin = 100000333,

    /// <summary>HNE+Delta:H(2): Reduced 4-Hydroxynonenal.</summary>
    UNIMOD_HNE_Delta_H_2_ = 100000335,

    /// <summary>Methylamine: Michael addition with methylamine.</summary>
    UNIMOD_Methylamine = 100000337,

    /// <summary>Bromo: Bromination.</summary>
    UNIMOD_Bromo = 100000340,

    /// <summary>Amino: Tyrosine oxidation to 2-aminotyrosine.</summary>
    UNIMOD_Amino = 100000342,

    /// <summary>Argbiotinhydrazide: Oxidized Arginine biotinylated with biotin hydrazide.</summary>
    UNIMOD_Argbiotinhydrazide = 100000343,

    /// <summary>Arg->GluSA: Arginine oxidation to glutamic semialdehyde.</summary>
    UNIMOD_Arg__GluSA = 100000344,

    /// <summary>Trioxidation: Cysteine oxidation to cysteic acid.</summary>
    UNIMOD_Trioxidation = 100000345,

    /// <summary>His->Asn: His->Asn substitution.</summary>
    UNIMOD_His__Asn = 100000348,

    /// <summary>His->Asp: His->Asp substitution.</summary>
    UNIMOD_His__Asp = 100000349,

    /// <summary>Trp->Hydroxykynurenin: Tryptophan oxidation to hydroxykynurenin.</summary>
    UNIMOD_Trp__Hydroxykynurenin = 100000350,

    /// <summary>Trp->Kynurenin: Tryptophan oxidation to kynurenin.</summary>
    UNIMOD_Trp__Kynurenin = 100000351,

    /// <summary>Lys->Allysine: Lysine oxidation to aminoadipic semialdehyde.</summary>
    UNIMOD_Lys__Allysine = 100000352,

    /// <summary>Lysbiotinhydrazide: Oxidized Lysine biotinylated with biotin hydrazide.</summary>
    UNIMOD_Lysbiotinhydrazide = 100000353,

    /// <summary>Nitro: Oxidation to nitro.</summary>
    UNIMOD_Nitro = 100000354,

    /// <summary>probiotinhydrazide: Oxidized proline biotinylated with biotin hydrazide.</summary>
    UNIMOD_probiotinhydrazide = 100000357,

    /// <summary>Pro->pyro-Glu: Proline oxidation to pyroglutamic acid.</summary>
    UNIMOD_Pro__pyro_Glu = 100000359,

    /// <summary>Pro->Pyrrolidinone: Proline oxidation to pyrrolidinone.</summary>
    UNIMOD_Pro__Pyrrolidinone = 100000360,

    /// <summary>Thrbiotinhydrazide: Oxidized Threonine biotinylated with biotin hydrazide.</summary>
    UNIMOD_Thrbiotinhydrazide = 100000361,

    /// <summary>Diisopropylphosphate: O-Diisopropylphosphorylation.</summary>
    UNIMOD_Diisopropylphosphate = 100000362,

    /// <summary>Isopropylphospho: O-Isopropylphosphorylation.</summary>
    UNIMOD_Isopropylphospho = 100000363,

    /// <summary>ICPL:13C(6): Bruker Daltonics SERVA-ICPL(TM) quantification chemistry, heavy form.</summary>
    UNIMOD_ICPL_13C_6_ = 100000364,

    /// <summary>ICPL: Bruker Daltonics SERVA-ICPL(TM) quantification chemistry, light form.</summary>
    UNIMOD_ICPL = 100000365,

    /// <summary>Deamidated:18O(1): Deamidation in presence of O18.</summary>
    UNIMOD_Deamidated_18O_1_ = 100000366,

    /// <summary>Cys->Dha: Dehydroalanine (from Cysteine).</summary>
    UNIMOD_Cys__Dha = 100000368,

    /// <summary>Pro->Pyrrolidone: Pyrrolidone from Proline.</summary>
    UNIMOD_Pro__Pyrrolidone = 100000369,

    /// <summary>HMVK: Michael addition of hydroxymethylvinyl ketone to cysteine.</summary>
    UNIMOD_HMVK = 100000371,

    /// <summary>Arg->Orn: Ornithine from Arginine.</summary>
    UNIMOD_Arg__Orn = 100000372,

    /// <summary>Dehydro: Half of a disulfide bridge.</summary>
    UNIMOD_Dehydro = 100000374,

    /// <summary>Diphthamide: Diphthamide.</summary>
    UNIMOD_Diphthamide = 100000375,

    /// <summary>Hydroxyfarnesyl: Hydroxyfarnesyl.</summary>
    UNIMOD_Hydroxyfarnesyl = 100000376,

    /// <summary>Diacylglycerol: Diacylglycerol.</summary>
    UNIMOD_Diacylglycerol = 100000377,

    /// <summary>Carboxyethyl: Carboxyethyl.</summary>
    UNIMOD_Carboxyethyl = 100000378,

    /// <summary>Hypusine: Hypusine.</summary>
    UNIMOD_Hypusine = 100000379,

    /// <summary>Retinylidene: Retinal.</summary>
    UNIMOD_Retinylidene = 100000380,

    /// <summary>Lys->AminoadipicAcid: Alpha-amino adipic acid.</summary>
    UNIMOD_Lys__AminoadipicAcid = 100000381,

    /// <summary>Cys->PyruvicAcid: Pyruvic acid from N-term cys.</summary>
    UNIMOD_Cys__PyruvicAcid = 100000382,

    /// <summary>Ammonia-loss: Loss of ammonia.</summary>
    UNIMOD_Ammonia_loss = 100000385,

    /// <summary>Phycocyanobilin: Phycocyanobilin.</summary>
    UNIMOD_Phycocyanobilin = 100000387,

    /// <summary>Phycoerythrobilin: Phycoerythrobilin.</summary>
    UNIMOD_Phycoerythrobilin = 100000388,

    /// <summary>Phytochromobilin: Phytochromobilin.</summary>
    UNIMOD_Phytochromobilin = 100000389,

    /// <summary>Heme: Heme.</summary>
    UNIMOD_Heme = 100000390,

    /// <summary>Molybdopterin: Molybdopterin.</summary>
    UNIMOD_Molybdopterin = 100000391,

    /// <summary>Quinone: Quinone.</summary>
    UNIMOD_Quinone = 100000392,

    /// <summary>Glucosylgalactosyl: Glucosylgalactosyl hydroxylysine.</summary>
    UNIMOD_Glucosylgalactosyl = 100000393,

    /// <summary>GPIanchor: Glycosylphosphatidylinositol.</summary>
    UNIMOD_GPIanchor = 100000394,

    /// <summary>PhosphoribosyldephosphoCoA: Phosphoribosyl dephospho-coenzyme A.</summary>
    UNIMOD_PhosphoribosyldephosphoCoA = 100000395,

    /// <summary>GlycerylPE: Glycerylphosphorylethanolamine.</summary>
    UNIMOD_GlycerylPE = 100000396,

    /// <summary>Triiodothyronine: Triiodo.</summary>
    UNIMOD_Triiodothyronine = 100000397,

    /// <summary>Thyroxine: Tetraiodo.</summary>
    UNIMOD_Thyroxine = 100000398,

    /// <summary>Tyr->Dha: Dehydroalanine (from Tyrosine).</summary>
    UNIMOD_Tyr__Dha = 100000400,

    /// <summary>Didehydro: 2-amino-3-oxo-butanoic_acid.</summary>
    UNIMOD_Didehydro = 100000401,

    /// <summary>Cys->Oxoalanine: Oxoalanine.</summary>
    UNIMOD_Cys__Oxoalanine = 100000402,

    /// <summary>Ser->LacticAcid: Lactic acid from N-term Ser.</summary>
    UNIMOD_Ser__LacticAcid = 100000403,

    /// <summary>Phosphoadenosine: AMP binding site.</summary>
    UNIMOD_Phosphoadenosine = 100000405,

    /// <summary>Hydroxycinnamyl: Hydroxycinnamyl.</summary>
    UNIMOD_Hydroxycinnamyl = 100000407,

    /// <summary>Glycosyl: Glycosyl-L-hydroxyproline.</summary>
    UNIMOD_Glycosyl = 100000408,

    /// <summary>FMNH: Flavin mononucleotide.</summary>
    UNIMOD_FMNH = 100000409,

    /// <summary>Archaeol: S-diphytanylglycerol diether.</summary>
    UNIMOD_Archaeol = 100000410,

    /// <summary>Phenylisocyanate: Phenyl isocyanate.</summary>
    UNIMOD_Phenylisocyanate = 100000411,

    /// <summary>Phenylisocyanate:2H(5): D5-phenyl isocyanate.</summary>
    UNIMOD_Phenylisocyanate_2H_5_ = 100000412,

    /// <summary>Phosphoguanosine: Phospho-guanosine.</summary>
    UNIMOD_Phosphoguanosine = 100000413,

    /// <summary>Hydroxymethyl: Hydroxymethyl.</summary>
    UNIMOD_Hydroxymethyl = 100000414,

    /// <summary>MolybdopterinGD+Delta:S(-1)Se(1): L-selenocysteinyl molybdenum bis(molybdopterin guanine dinucleotide).</summary>
    UNIMOD_MolybdopterinGD_Delta_S__1_Se_1_ = 100000415,

    /// <summary>Dipyrrolylmethanemethyl: Dipyrrolylmethanemethyl.</summary>
    UNIMOD_Dipyrrolylmethanemethyl = 100000416,

    /// <summary>PhosphoUridine: Uridine phosphodiester.</summary>
    UNIMOD_PhosphoUridine = 100000417,

    /// <summary>Glycerophospho: Glycerophospho.</summary>
    UNIMOD_Glycerophospho = 100000419,

    /// <summary>Carboxy->Thiocarboxy: Thiocarboxylic acid.</summary>
    UNIMOD_Carboxy__Thiocarboxy = 100000420,

    /// <summary>Sulfide: Persulfide.</summary>
    UNIMOD_Sulfide = 100000421,

    /// <summary>PyruvicAcidIminyl: N-pyruvic acid 2-iminyl.</summary>
    UNIMOD_PyruvicAcidIminyl = 100000422,

    /// <summary>Delta:Se(1): Selenyl.</summary>
    UNIMOD_Delta_Se_1_ = 100000423,

    /// <summary>MolybdopterinGD: Molybdenum bis(molybdopterin guanine dinucleotide).</summary>
    UNIMOD_MolybdopterinGD = 100000424,

    /// <summary>Dioxidation: Dihydroxy.</summary>
    UNIMOD_Dioxidation = 100000425,

    /// <summary>Octanoyl: Octanoyl.</summary>
    UNIMOD_Octanoyl = 100000426,

    /// <summary>PhosphoHexNAc: N-acetylglucosamine-1-phosphoryl.</summary>
    UNIMOD_PhosphoHexNAc = 100000428,

    /// <summary>PhosphoHex: Phosphoglycosyl-D-mannose-1-phosphoryl.</summary>
    UNIMOD_PhosphoHex = 100000429,

    /// <summary>Palmitoleyl: Palmitoleyl.</summary>
    UNIMOD_Palmitoleyl = 100000431,

    /// <summary>Cholesterol: Cholesterol ester.</summary>
    UNIMOD_Cholesterol = 100000432,

    /// <summary>Didehydroretinylidene: 3,4-didehydroretinylidene.</summary>
    UNIMOD_Didehydroretinylidene = 100000433,

    /// <summary>CHDH: Cis-14-hydroxy-10,13-dioxo-7-heptadecenoic ester.</summary>
    UNIMOD_CHDH = 100000434,

    /// <summary>Methylpyrroline: 4-methyl-delta-1-pyrroline-5-carboxyl.</summary>
    UNIMOD_Methylpyrroline = 100000435,

    /// <summary>Hydroxyheme: Hydroxyheme.</summary>
    UNIMOD_Hydroxyheme = 100000436,

    /// <summary>MicrocinC7: (3-aminopropyl)(L-aspartyl-1-amino)phosphoryl-5-adenosine.</summary>
    UNIMOD_MicrocinC7 = 100000437,

    /// <summary>Cyano: Cyano.</summary>
    UNIMOD_Cyano = 100000438,

    /// <summary>Diironsubcluster: Hydrogenase diiron subcluster.</summary>
    UNIMOD_Diironsubcluster = 100000439,

    /// <summary>Amidino: Amidino.</summary>
    UNIMOD_Amidino = 100000440,

    /// <summary>FMN: O3-(riboflavin phosphoryl).</summary>
    UNIMOD_FMN = 100000442,

    /// <summary>FMNC: S-(4a-FMN).</summary>
    UNIMOD_FMNC = 100000443,

    /// <summary>CuSMo: Copper sulfido molybdopterin cytosine dinuncleotide.</summary>
    UNIMOD_CuSMo = 100000444,

    /// <summary>Hydroxytrimethyl: 5-hydroxy-N6,N6,N6-trimethyl.</summary>
    UNIMOD_Hydroxytrimethyl = 100000445,

    /// <summary>Deoxy: Reduction.</summary>
    UNIMOD_Deoxy = 100000447,

    /// <summary>Microcin: Microcin E492 siderophore ester from serine.</summary>
    UNIMOD_Microcin = 100000448,

    /// <summary>Decanoyl: Lipid.</summary>
    UNIMOD_Decanoyl = 100000449,

    /// <summary>Glu: Monoglutamyl.</summary>
    UNIMOD_Glu = 100000450,

    /// <summary>GluGlu: Diglutamyl.</summary>
    UNIMOD_GluGlu = 100000451,

    /// <summary>GluGluGlu: Triglutamyl.</summary>
    UNIMOD_GluGluGlu = 100000452,

    /// <summary>GluGluGluGlu: Tetraglutamyl.</summary>
    UNIMOD_GluGluGluGlu = 100000453,

    /// <summary>HexN: Hexosamine.</summary>
    UNIMOD_HexN = 100000454,

    /// <summary>Xlink:DMP-s: One end of crosslink attached, one end free.</summary>
    UNIMOD_Xlink_DMP_s = 100000455,

    /// <summary>Xlink:DMP: Both ends of crosslink attached to same peptide.</summary>
    UNIMOD_Xlink_DMP = 100000456,

    /// <summary>NDA: Naphthalene-2,3-dicarboxaldehyde.</summary>
    UNIMOD_NDA = 100000457,

    /// <summary>SPITC:13C(6): 4-sulfophenyl isothiocyanate (Heavy C13).</summary>
    UNIMOD_SPITC_13C_6_ = 100000464,

    /// <summary>AEC-MAEC: Aminoethylcysteine.</summary>
    UNIMOD_AEC_MAEC = 100000472,

    /// <summary>TMAB: 4-trimethyllammoniumbutyryl-.</summary>
    UNIMOD_TMAB = 100000476,

    /// <summary>TMAB:2H(9): D9-4-trimethyllammoniumbutyryl-.</summary>
    UNIMOD_TMAB_2H_9_ = 100000477,

    /// <summary>FTC: Fluorescein-5-thiosemicarbazide.</summary>
    UNIMOD_FTC = 100000478,

    /// <summary>Label:2H(4): 4,4,5,5-D4 Lysine.</summary>
    UNIMOD_Label_2H_4_ = 100000481,

    /// <summary>DHP: Dehydropyrrolizidine alkaloid (dehydroretronecine) on cysteines.</summary>
    UNIMOD_DHP = 100000488,

    /// <summary>Hep: Heptose.</summary>
    UNIMOD_Hep = 100000490,

    /// <summary>BADGE: Bisphenol A diglycidyl ether derivative.</summary>
    UNIMOD_BADGE = 100000493,

    /// <summary>CyDye-Cy3: Cy3 CyDye DIGE Fluor saturation dye.</summary>
    UNIMOD_CyDye_Cy3 = 100000494,

    /// <summary>CyDye-Cy5: Cy5 CyDye DIGE Fluor saturation dye.</summary>
    UNIMOD_CyDye_Cy5 = 100000495,

    /// <summary>BHTOH: Michael addition of t-butyl hydroxylated BHT (BHTOH) to C, H or K.</summary>
    UNIMOD_BHTOH = 100000498,

    /// <summary>IGBP:13C(2): Heavy IDBEST tag for quantitation.</summary>
    UNIMOD_IGBP_13C_2_ = 100000499,

    /// <summary>Nmethylmaleimide+water: Nmethylmaleimidehydrolysis.</summary>
    UNIMOD_Nmethylmaleimide_water = 100000500,

    /// <summary>PyMIC: 3-methyl-2-pyridyl isocyanate.</summary>
    UNIMOD_PyMIC = 100000501,

    /// <summary>LG-lactam-K: Levuglandinyl - lysine lactam adduct.</summary>
    UNIMOD_LG_lactam_K = 100000503,

    /// <summary>LG-Hlactam-K: Levuglandinyl - lysine hydroxylactam adduct.</summary>
    UNIMOD_LG_Hlactam_K = 100000504,

    /// <summary>LG-lactam-R: Levuglandinyl - arginine lactam adduct.</summary>
    UNIMOD_LG_lactam_R = 100000505,

    /// <summary>LG-Hlactam-R: Levuglandinyl - arginine hydroxylactam adduct.</summary>
    UNIMOD_LG_Hlactam_R = 100000506,

    /// <summary>Dimethyl:2H(4)13C(2): DiMethyl-C13HD2.</summary>
    UNIMOD_Dimethyl_2H_4_13C_2_ = 100000510,

    /// <summary>Hex(2): Lactosylation.</summary>
    UNIMOD_Hex_2_ = 100000512,

    /// <summary>C8-QAT: [3-(2,5)-Dioxopyrrolidin-1-yloxycarbonyl)-propyl]dimethyloctylammonium.</summary>
    UNIMOD_C8_QAT = 100000513,

    /// <summary>PropylNAGthiazoline: Propyl-1,2-dideoxy-2\'-methyl-alpha-D-glucopyranoso-[2,1-d]-Delta2\'-thiazoline.</summary>
    UNIMOD_PropylNAGthiazoline = 100000514,

    /// <summary>FNEM: Fluorescein-5-maleimide.</summary>
    UNIMOD_FNEM = 100000515,

    /// <summary>Diethyl: Diethylation, analogous to Dimethylation.</summary>
    UNIMOD_Diethyl = 100000518,

    /// <summary>BisANS: 4,4\'-dianilino-1,1\'-binaphthyl-5,5\'-disulfonic acid.</summary>
    UNIMOD_BisANS = 100000519,

    /// <summary>Piperidine: Piperidination.</summary>
    UNIMOD_Piperidine = 100000520,

    /// <summary>Maleimide-PEO2-Biotin: Maleimide-Biotin.</summary>
    UNIMOD_Maleimide_PEO2_Biotin = 100000522,

    /// <summary>Sulfo-NHS-LC-LC-Biotin: Biot_LC_LC.</summary>
    UNIMOD_Sulfo_NHS_LC_LC_Biotin = 100000523,

    /// <summary>CLIP_TRAQ_2: CLIP_TRAQ_2.</summary>
    UNIMOD_CLIP_TRAQ_2 = 100000525,

    /// <summary>Dethiomethyl: Prompt loss of side chain from oxidised Met.</summary>
    UNIMOD_Dethiomethyl = 100000526,

    /// <summary>Methyl+Deamidated: Deamidation followed by a methylation.</summary>
    UNIMOD_Methyl_Deamidated = 100000528,

    /// <summary>Delta:H(5)C(2): Dimethylation of proline residue.</summary>
    UNIMOD_Delta_H_5_C_2_ = 100000529,

    /// <summary>Cation:K: Replacement of proton by potassium.</summary>
    UNIMOD_Cation_K = 100000530,

    /// <summary>Cation:Cu[I]: Replacement of proton by copper.</summary>
    UNIMOD_Cation_Cu_I_ = 100000531,

    /// <summary>iTRAQ4plex114: Accurate mass for 114.</summary>
    UNIMOD_iTRAQ4plex114 = 100000532,

    /// <summary>iTRAQ4plex115: Accurate mass for 115.</summary>
    UNIMOD_iTRAQ4plex115 = 100000533,

    /// <summary>Dibromo: Dibromo.</summary>
    UNIMOD_Dibromo = 100000534,

    /// <summary>LeuArgGlyGly: Ubiquitination.</summary>
    UNIMOD_LeuArgGlyGly = 100000535,

    /// <summary>CLIP_TRAQ_3: CLIP_TRAQ_3.</summary>
    UNIMOD_CLIP_TRAQ_3 = 100000536,

    /// <summary>CLIP_TRAQ_4: CLIP_TRAQ_4.</summary>
    UNIMOD_CLIP_TRAQ_4 = 100000537,

    /// <summary>Biotin:Cayman-10141: Was 15dB-biotin.</summary>
    UNIMOD_Biotin_Cayman_10141 = 100000538,

    /// <summary>Biotin:Cayman-10013: Was PGA1-biotin.</summary>
    UNIMOD_Biotin_Cayman_10013 = 100000539,

    /// <summary>Ala->Ser: Ala->Ser substitution.</summary>
    UNIMOD_Ala__Ser = 100000540,

    /// <summary>Ala->Thr: Ala->Thr substitution.</summary>
    UNIMOD_Ala__Thr = 100000541,

    /// <summary>Ala->Asp: Ala->Asp substitution.</summary>
    UNIMOD_Ala__Asp = 100000542,

    /// <summary>Ala->Pro: Ala->Pro substitution.</summary>
    UNIMOD_Ala__Pro = 100000543,

    /// <summary>Ala->Gly: Ala->Gly substitution.</summary>
    UNIMOD_Ala__Gly = 100000544,

    /// <summary>Ala->Glu: Ala->Glu substitution.</summary>
    UNIMOD_Ala__Glu = 100000545,

    /// <summary>Ala->Val: Ala->Val substitution.</summary>
    UNIMOD_Ala__Val = 100000546,

    /// <summary>Cys->Phe: Cys->Phe substitution.</summary>
    UNIMOD_Cys__Phe = 100000547,

    /// <summary>Cys->Ser: Cys->Ser substitution.</summary>
    UNIMOD_Cys__Ser = 100000548,

    /// <summary>Cys->Trp: Cys->Trp substitution.</summary>
    UNIMOD_Cys__Trp = 100000549,

    /// <summary>Cys->Tyr: Cys->Tyr substitution.</summary>
    UNIMOD_Cys__Tyr = 100000550,

    /// <summary>Cys->Arg: Cys->Arg substitution.</summary>
    UNIMOD_Cys__Arg = 100000551,

    /// <summary>Cys->Gly: Cys->Gly substitution.</summary>
    UNIMOD_Cys__Gly = 100000552,

    /// <summary>Asp->Ala: Asp->Ala substitution.</summary>
    UNIMOD_Asp__Ala = 100000553,

    /// <summary>Asp->His: Asp->His substitution.</summary>
    UNIMOD_Asp__His = 100000554,

    /// <summary>Asp->Asn: Asp->Asn substitution.</summary>
    UNIMOD_Asp__Asn = 100000555,

    /// <summary>Asp->Gly: Asp->Gly substitution.</summary>
    UNIMOD_Asp__Gly = 100000556,

    /// <summary>Asp->Tyr: Asp->Tyr substitution.</summary>
    UNIMOD_Asp__Tyr = 100000557,

    /// <summary>Asp->Glu: Asp->Glu substitution.</summary>
    UNIMOD_Asp__Glu = 100000558,

    /// <summary>Asp->Val: Asp->Val substitution.</summary>
    UNIMOD_Asp__Val = 100000559,

    /// <summary>Glu->Ala: Glu->Ala substitution.</summary>
    UNIMOD_Glu__Ala = 100000560,

    /// <summary>Glu->Gln: Glu->Gln substitution.</summary>
    UNIMOD_Glu__Gln = 100000561,

    /// <summary>Glu->Asp: Glu->Asp substitution.</summary>
    UNIMOD_Glu__Asp = 100000562,

    /// <summary>Glu->Lys: Glu->Lys substitution.</summary>
    UNIMOD_Glu__Lys = 100000563,

    /// <summary>Glu->Gly: Glu->Gly substitution.</summary>
    UNIMOD_Glu__Gly = 100000564,

    /// <summary>Glu->Val: Glu->Val substitution.</summary>
    UNIMOD_Glu__Val = 100000565,

    /// <summary>Phe->Ser: Phe->Ser substitution.</summary>
    UNIMOD_Phe__Ser = 100000566,

    /// <summary>Phe->Cys: Phe->Cys substitution.</summary>
    UNIMOD_Phe__Cys = 100000567,

    /// <summary>Phe->Xle: Phe->Leu/Ile substitution.</summary>
    UNIMOD_Phe__Xle = 100000568,

    /// <summary>Phe->Tyr: Phe->Tyr substitution.</summary>
    UNIMOD_Phe__Tyr = 100000569,

    /// <summary>Phe->Val: Phe->Val substitution.</summary>
    UNIMOD_Phe__Val = 100000570,

    /// <summary>Gly->Ala: Gly->Ala substitution.</summary>
    UNIMOD_Gly__Ala = 100000571,

    /// <summary>Gly->Ser: Gly->Ser substitution.</summary>
    UNIMOD_Gly__Ser = 100000572,

    /// <summary>Gly->Trp: Gly->Trp substitution.</summary>
    UNIMOD_Gly__Trp = 100000573,

    /// <summary>Gly->Glu: Gly->Glu substitution.</summary>
    UNIMOD_Gly__Glu = 100000574,

    /// <summary>Gly->Val: Gly->Val substitution.</summary>
    UNIMOD_Gly__Val = 100000575,

    /// <summary>Gly->Asp: Gly->Asp substitution.</summary>
    UNIMOD_Gly__Asp = 100000576,

    /// <summary>Gly->Cys: Gly->Cys substitution.</summary>
    UNIMOD_Gly__Cys = 100000577,

    /// <summary>Gly->Arg: Gly->Arg substitution.</summary>
    UNIMOD_Gly__Arg = 100000578,

    /// <summary>His->Pro: His->Pro substitution.</summary>
    UNIMOD_His__Pro = 100000580,

    /// <summary>His->Tyr: His->Tyr substitution.</summary>
    UNIMOD_His__Tyr = 100000581,

    /// <summary>His->Gln: His->Gln substitution.</summary>
    UNIMOD_His__Gln = 100000582,

    /// <summary>His->Arg: His->Arg substitution.</summary>
    UNIMOD_His__Arg = 100000584,

    /// <summary>His->Xle: His->Leu/Ile substitution.</summary>
    UNIMOD_His__Xle = 100000585,

    /// <summary>Xle->Thr: Leu/Ile->Thr substitution.</summary>
    UNIMOD_Xle__Thr = 100000588,

    /// <summary>Xle->Asn: Leu/Ile->Asn substitution.</summary>
    UNIMOD_Xle__Asn = 100000589,

    /// <summary>Xle->Lys: Leu/Ile->Lys substitution.</summary>
    UNIMOD_Xle__Lys = 100000590,

    /// <summary>Lys->Thr: Lys->Thr substitution.</summary>
    UNIMOD_Lys__Thr = 100000594,

    /// <summary>Lys->Asn: Lys->Asn substitution.</summary>
    UNIMOD_Lys__Asn = 100000595,

    /// <summary>Lys->Glu: Lys->Glu substitution.</summary>
    UNIMOD_Lys__Glu = 100000596,

    /// <summary>Lys->Gln: Lys->Gln substitution.</summary>
    UNIMOD_Lys__Gln = 100000597,

    /// <summary>Lys->Met: Lys->Met substitution.</summary>
    UNIMOD_Lys__Met = 100000598,

    /// <summary>Lys->Arg: Lys->Arg substitution.</summary>
    UNIMOD_Lys__Arg = 100000599,

    /// <summary>Lys->Xle: Lys->Leu/Ile substitution.</summary>
    UNIMOD_Lys__Xle = 100000600,

    /// <summary>Xle->Ser: Leu/Ile->Ser substitution.</summary>
    UNIMOD_Xle__Ser = 100000601,

    /// <summary>Xle->Phe: Leu/Ile->Phe substitution.</summary>
    UNIMOD_Xle__Phe = 100000602,

    /// <summary>Xle->Trp: Leu/Ile->Trp substitution.</summary>
    UNIMOD_Xle__Trp = 100000603,

    /// <summary>Xle->Pro: Leu/Ile->Pro substitution.</summary>
    UNIMOD_Xle__Pro = 100000604,

    /// <summary>Xle->Val: Leu/Ile->Val substitution.</summary>
    UNIMOD_Xle__Val = 100000605,

    /// <summary>Xle->His: Leu/Ile->His substitution.</summary>
    UNIMOD_Xle__His = 100000606,

    /// <summary>Xle->Gln: Leu/Ile->Gln substitution.</summary>
    UNIMOD_Xle__Gln = 100000607,

    /// <summary>Xle->Met: Leu/Ile->Met substitution.</summary>
    UNIMOD_Xle__Met = 100000608,

    /// <summary>Xle->Arg: Leu/Ile->Arg substitution.</summary>
    UNIMOD_Xle__Arg = 100000609,

    /// <summary>Met->Thr: Met->Thr substitution.</summary>
    UNIMOD_Met__Thr = 100000610,

    /// <summary>Met->Arg: Met->Arg substitution.</summary>
    UNIMOD_Met__Arg = 100000611,

    /// <summary>Met->Lys: Met->Lys substitution.</summary>
    UNIMOD_Met__Lys = 100000613,

    /// <summary>Met->Xle: Met->Leu/Ile substitution.</summary>
    UNIMOD_Met__Xle = 100000614,

    /// <summary>Met->Val: Met->Val substitution.</summary>
    UNIMOD_Met__Val = 100000615,

    /// <summary>Asn->Ser: Asn->Ser substitution.</summary>
    UNIMOD_Asn__Ser = 100000616,

    /// <summary>Asn->Thr: Asn->Thr substitution.</summary>
    UNIMOD_Asn__Thr = 100000617,

    /// <summary>Asn->Lys: Asn->Lys substitution.</summary>
    UNIMOD_Asn__Lys = 100000618,

    /// <summary>Asn->Tyr: Asn->Tyr substitution.</summary>
    UNIMOD_Asn__Tyr = 100000619,

    /// <summary>Asn->His: Asn->His substitution.</summary>
    UNIMOD_Asn__His = 100000620,

    /// <summary>Asn->Asp: Asn->Asp substitution.</summary>
    UNIMOD_Asn__Asp = 100000621,

    /// <summary>Asn->Xle: Asn->Leu/Ile substitution.</summary>
    UNIMOD_Asn__Xle = 100000622,

    /// <summary>Pro->Ser: Pro->Ser substitution.</summary>
    UNIMOD_Pro__Ser = 100000623,

    /// <summary>Pro->Ala: Pro->Ala substitution.</summary>
    UNIMOD_Pro__Ala = 100000624,

    /// <summary>Pro->His: Pro->His substitution.</summary>
    UNIMOD_Pro__His = 100000625,

    /// <summary>Pro->Gln: Pro->Gln substitution.</summary>
    UNIMOD_Pro__Gln = 100000626,

    /// <summary>Pro->Thr: Pro->Thr substitution.</summary>
    UNIMOD_Pro__Thr = 100000627,

    /// <summary>Pro->Arg: Pro->Arg substitution.</summary>
    UNIMOD_Pro__Arg = 100000628,

    /// <summary>Pro->Xle: Pro->Leu/Ile substitution.</summary>
    UNIMOD_Pro__Xle = 100000629,

    /// <summary>Gln->Pro: Gln->Pro substitution.</summary>
    UNIMOD_Gln__Pro = 100000630,

    /// <summary>Gln->Lys: Gln->Lys substitution.</summary>
    UNIMOD_Gln__Lys = 100000631,

    /// <summary>Gln->Glu: Gln->Glu substitution.</summary>
    UNIMOD_Gln__Glu = 100000632,

    /// <summary>Gln->His: Gln->His substitution.</summary>
    UNIMOD_Gln__His = 100000633,

    /// <summary>Gln->Arg: Gln->Arg substitution.</summary>
    UNIMOD_Gln__Arg = 100000634,

    /// <summary>Gln->Xle: Gln->Leu/Ile substitution.</summary>
    UNIMOD_Gln__Xle = 100000635,

    /// <summary>Arg->Ser: Arg->Ser substitution.</summary>
    UNIMOD_Arg__Ser = 100000636,

    /// <summary>Arg->Trp: Arg->Trp substitution.</summary>
    UNIMOD_Arg__Trp = 100000637,

    /// <summary>Arg->Thr: Arg->Thr substitution.</summary>
    UNIMOD_Arg__Thr = 100000638,

    /// <summary>Arg->Pro: Arg->Pro substitution.</summary>
    UNIMOD_Arg__Pro = 100000639,

    /// <summary>Arg->Lys: Arg->Lys substitution.</summary>
    UNIMOD_Arg__Lys = 100000640,

    /// <summary>Arg->His: Arg->His substitution.</summary>
    UNIMOD_Arg__His = 100000641,

    /// <summary>Arg->Gln: Arg->Gln substitution.</summary>
    UNIMOD_Arg__Gln = 100000642,

    /// <summary>Arg->Met: Arg->Met substitution.</summary>
    UNIMOD_Arg__Met = 100000643,

    /// <summary>Arg->Cys: Arg->Cys substitution.</summary>
    UNIMOD_Arg__Cys = 100000644,

    /// <summary>Arg->Xle: Arg->Leu/Ile substitution.</summary>
    UNIMOD_Arg__Xle = 100000645,

    /// <summary>Arg->Gly: Arg->Gly substitution.</summary>
    UNIMOD_Arg__Gly = 100000646,

    /// <summary>Ser->Phe: Ser->Phe substitution.</summary>
    UNIMOD_Ser__Phe = 100000647,

    /// <summary>Ser->Ala: Ser->Ala substitution.</summary>
    UNIMOD_Ser__Ala = 100000648,

    /// <summary>Ser->Trp: Ser->Trp substitution.</summary>
    UNIMOD_Ser__Trp = 100000649,

    /// <summary>Ser->Thr: Ser->Thr substitution.</summary>
    UNIMOD_Ser__Thr = 100000650,

    /// <summary>Ser->Asn: Ser->Asn substitution.</summary>
    UNIMOD_Ser__Asn = 100000651,

    /// <summary>Ser->Pro: Ser->Pro substitution.</summary>
    UNIMOD_Ser__Pro = 100000652,

    /// <summary>Ser->Tyr: Ser->Tyr substitution.</summary>
    UNIMOD_Ser__Tyr = 100000653,

    /// <summary>Ser->Cys: Ser->Cys substitution.</summary>
    UNIMOD_Ser__Cys = 100000654,

    /// <summary>Ser->Arg: Ser->Arg substitution.</summary>
    UNIMOD_Ser__Arg = 100000655,

    /// <summary>Ser->Xle: Ser->Leu/Ile substitution.</summary>
    UNIMOD_Ser__Xle = 100000656,

    /// <summary>Ser->Gly: Ser->Gly substitution.</summary>
    UNIMOD_Ser__Gly = 100000657,

    /// <summary>Thr->Ser: Thr->Ser substitution.</summary>
    UNIMOD_Thr__Ser = 100000658,

    /// <summary>Thr->Ala: Thr->Ala substitution.</summary>
    UNIMOD_Thr__Ala = 100000659,

    /// <summary>Thr->Asn: Thr->Asn substitution.</summary>
    UNIMOD_Thr__Asn = 100000660,

    /// <summary>Thr->Lys: Thr->Lys substitution.</summary>
    UNIMOD_Thr__Lys = 100000661,

    /// <summary>Thr->Pro: Thr->Pro substitution.</summary>
    UNIMOD_Thr__Pro = 100000662,

    /// <summary>Thr->Met: Thr->Met substitution.</summary>
    UNIMOD_Thr__Met = 100000663,

    /// <summary>Thr->Xle: Thr->Leu/Ile substitution.</summary>
    UNIMOD_Thr__Xle = 100000664,

    /// <summary>Thr->Arg: Thr->Arg substitution.</summary>
    UNIMOD_Thr__Arg = 100000665,

    /// <summary>Val->Phe: Val->Phe substitution.</summary>
    UNIMOD_Val__Phe = 100000666,

    /// <summary>Val->Ala: Val->Ala substitution.</summary>
    UNIMOD_Val__Ala = 100000667,

    /// <summary>Val->Glu: Val->Glu substitution.</summary>
    UNIMOD_Val__Glu = 100000668,

    /// <summary>Val->Met: Val->Met substitution.</summary>
    UNIMOD_Val__Met = 100000669,

    /// <summary>Val->Asp: Val->Asp substitution.</summary>
    UNIMOD_Val__Asp = 100000670,

    /// <summary>Val->Xle: Val->Leu/Ile substitution.</summary>
    UNIMOD_Val__Xle = 100000671,

    /// <summary>Val->Gly: Val->Gly substitution.</summary>
    UNIMOD_Val__Gly = 100000672,

    /// <summary>Trp->Ser: Trp->Ser substitution.</summary>
    UNIMOD_Trp__Ser = 100000673,

    /// <summary>Trp->Cys: Trp->Cys substitution.</summary>
    UNIMOD_Trp__Cys = 100000674,

    /// <summary>Trp->Arg: Trp->Arg substitution.</summary>
    UNIMOD_Trp__Arg = 100000675,

    /// <summary>Trp->Gly: Trp->Gly substitution.</summary>
    UNIMOD_Trp__Gly = 100000676,

    /// <summary>Trp->Xle: Trp->Leu/Ile substitution.</summary>
    UNIMOD_Trp__Xle = 100000677,

    /// <summary>Tyr->Phe: Tyr->Phe substitution.</summary>
    UNIMOD_Tyr__Phe = 100000678,

    /// <summary>Tyr->Ser: Tyr->Ser substitution.</summary>
    UNIMOD_Tyr__Ser = 100000679,

    /// <summary>Tyr->Asn: Tyr->Asn substitution.</summary>
    UNIMOD_Tyr__Asn = 100000680,

    /// <summary>Tyr->His: Tyr->His substitution.</summary>
    UNIMOD_Tyr__His = 100000681,

    /// <summary>Tyr->Asp: Tyr->Asp substitution.</summary>
    UNIMOD_Tyr__Asp = 100000682,

    /// <summary>Tyr->Cys: Tyr->Cys substitution.</summary>
    UNIMOD_Tyr__Cys = 100000683,

    /// <summary>BDMAPP: Mass Defect Tag on lysine e-amino.</summary>
    UNIMOD_BDMAPP = 100000684,

    /// <summary>NA-LNO2: Nitroalkylation by Nitro Linoleic Acid.</summary>
    UNIMOD_NA_LNO2 = 100000685,

    /// <summary>NA-OA-NO2: Nitroalkylation by Nitro Oleic Acid.</summary>
    UNIMOD_NA_OA_NO2 = 100000686,

    /// <summary>ICPL:2H(4): Bruker Daltonics SERVA-ICPL(TM) quantification chemistry, medium form.</summary>
    UNIMOD_ICPL_2H_4_ = 100000687,

    /// <summary>Label:13C(6)15N(1): 13C(6) 15N(1) Silac label.</summary>
    UNIMOD_Label_13C_6_15N_1_ = 100000695,

    /// <summary>Label:2H(9)13C(6)15N(2): 13C(6) 15N(2) (D)9 SILAC label.</summary>
    UNIMOD_Label_2H_9_13C_6_15N_2_ = 100000696,

    /// <summary>NIC: Nicotinic Acid.</summary>
    UNIMOD_NIC = 100000697,

    /// <summary>dNIC: Deuterated Nicotinic Acid.</summary>
    UNIMOD_dNIC = 100000698,

    /// <summary>HNE-Delta:H(2)O: Dehydrated 4-hydroxynonenal.</summary>
    UNIMOD_HNE_Delta_H_2_O = 100000720,

    /// <summary>4-ONE: 4-Oxononenal (ONE).</summary>
    UNIMOD_4_ONE = 100000721,

    /// <summary>O-Dimethylphosphate: O-Dimethylphosphorylation.</summary>
    UNIMOD_O_Dimethylphosphate = 100000723,

    /// <summary>O-Methylphosphate: O-Methylphosphorylation.</summary>
    UNIMOD_O_Methylphosphate = 100000724,

    /// <summary>Diethylphosphate: O-Diethylphosphorylation.</summary>
    UNIMOD_Diethylphosphate = 100000725,

    /// <summary>Ethylphosphate: O-Ethylphosphorylation.</summary>
    UNIMOD_Ethylphosphate = 100000726,

    /// <summary>O-pinacolylmethylphosphonate: O-pinacolylmethylphosphonylation.</summary>
    UNIMOD_O_pinacolylmethylphosphonate = 100000727,

    /// <summary>Methylphosphonate: Methylphosphonylation.</summary>
    UNIMOD_Methylphosphonate = 100000728,

    /// <summary>O-Isopropylmethylphosphonate: O-Isopropylmethylphosphonylation.</summary>
    UNIMOD_O_Isopropylmethylphosphonate = 100000729,

    /// <summary>iTRAQ8plex: Representative mass and accurate mass for 113, 114, 116 & 117.</summary>
    UNIMOD_iTRAQ8plex = 100000730,

    /// <summary>iTRAQ8plex:13C(6)15N(2): Accurate mass for 115, 118, 119 & 121.</summary>
    UNIMOD_iTRAQ8plex_13C_6_15N_2_ = 100000731,

    /// <summary>Ethanolamine: Carboxyl modification with ethanolamine.</summary>
    UNIMOD_Ethanolamine = 100000734,

    /// <summary>DTT_ST: Dithiothreitol (DTT).</summary>
    UNIMOD_DTT_ST = 100000735,

    /// <summary>DTT_C: Dithiothreitol (DTT) on Cys.</summary>
    UNIMOD_DTT_C = 100000736,

    /// <summary>TMT6plex: Sixplex Tandem Mass Tag®.</summary>
    UNIMOD_TMT6plex = 100000737,

    /// <summary>TMT2plex: Duplex Tandem Mass Tag®.</summary>
    UNIMOD_TMT2plex = 100000738,

    /// <summary>TMT: Native Tandem Mass Tag®.</summary>
    UNIMOD_TMT = 100000739,

    /// <summary>ExacTagThiol: ExacTag Thiol label mass for 2-4-7-10 plex.</summary>
    UNIMOD_ExacTagThiol = 100000740,

    /// <summary>ExacTagAmine: ExacTag Amine label mass for 2-4-7-10 plex.</summary>
    UNIMOD_ExacTagAmine = 100000741,

    /// <summary>4-ONE+Delta:H(-2)O(-1): Dehydrated 4-Oxononenal Michael adduct.</summary>
    UNIMOD_4_ONE_Delta_H__2_O__1_ = 100000743,

    /// <summary>NO_SMX_SEMD: Nitroso Sulfamethoxazole Sulphenamide thiol adduct.</summary>
    UNIMOD_NO_SMX_SEMD = 100000744,

    /// <summary>NO_SMX_SMCT: Nitroso Sulfamethoxazole semimercaptal thiol adduct.</summary>
    UNIMOD_NO_SMX_SMCT = 100000745,

    /// <summary>NO_SMX_SIMD: Nitroso Sulfamethoxazole Sulfinamide thiol adduct.</summary>
    UNIMOD_NO_SMX_SIMD = 100000746,

    /// <summary>Malonyl: Malonylation of C and S residues.</summary>
    UNIMOD_Malonyl = 100000747,

    /// <summary>3sulfo: Derivatization by N-term modification using 3-Sulfobenzoic succinimidyl ester.</summary>
    UNIMOD_3sulfo = 100000748,

    /// <summary>trifluoro: Trifluoroleucine replacement of leucine.</summary>
    UNIMOD_trifluoro = 100000750,

    /// <summary>TNBS: Tri nitro benzene.</summary>
    UNIMOD_TNBS = 100000751,

    /// <summary>IDEnT: Isotope Distribution Encoded Tag.</summary>
    UNIMOD_IDEnT = 100000762,

    /// <summary>DTT_ST:2H(6): Isotopically labeled Dithiothreitol (DTT) modification of serines or threonines.</summary>
    UNIMOD_DTT_ST_2H_6_ = 100000763,

    /// <summary>DTT_C:2H(6): Isotopically labeled Dithiothreitol (DTT) modification of cysteines.</summary>
    UNIMOD_DTT_C_2H_6_ = 100000764,

    /// <summary>Met-loss: Removal of initiator methionine from protein N-terminus.</summary>
    UNIMOD_Met_loss = 100000765,

    /// <summary>Met-loss+Acetyl: Removal of initiator methionine from protein N-terminus, then acetylation of the new N-terminus.</summary>
    UNIMOD_Met_loss_Acetyl = 100000766,

    /// <summary>Menadione-HQ: Menadione hydroquinone derivative.</summary>
    UNIMOD_Menadione_HQ = 100000767,

    /// <summary>Methyl+Acetyl:2H(3): Mono-methylated lysine labelled with Acetyl_heavy.</summary>
    UNIMOD_Methyl_Acetyl_2H_3_ = 100000768,

    /// <summary>lapachenole: Lapachenole photochemically added to cysteine.</summary>
    UNIMOD_lapachenole = 100000771,

    /// <summary>Label:13C(5): 13C(5) Silac label.</summary>
    UNIMOD_Label_13C_5_ = 100000772,

    /// <summary>maleimide: Maleimide.</summary>
    UNIMOD_maleimide = 100000773,

    /// <summary>Biotin-phenacyl: Alkylation by biotinylated form of phenacyl bromide.</summary>
    UNIMOD_Biotin_phenacyl = 100000774,

    /// <summary>Carboxymethyl:13C(2): Iodoacetic acid derivative w/ 13C label.</summary>
    UNIMOD_Carboxymethyl_13C_2_ = 100000775,

    /// <summary>NEM:2H(5): D5 N-ethylmaleimide on cysteines.</summary>
    UNIMOD_NEM_2H_5_ = 100000776,

    /// <summary>AEC-MAEC:2H(4): Deuterium cysteamine modification to S or T.</summary>
    UNIMOD_AEC_MAEC_2H_4_ = 100000792,

    /// <summary>Hex1HexNAc1: Hex1HexNAc1.</summary>
    UNIMOD_Hex1HexNAc1 = 100000793,

    /// <summary>Label:13C(6)+GlyGly: 13C6 labeled ubiquitinylation residue.</summary>
    UNIMOD_Label_13C_6__GlyGly = 100000799,

    /// <summary>Biotin:Thermo-21345: Was PentylamineBiotin.</summary>
    UNIMOD_Biotin_Thermo_21345 = 100000800,

    /// <summary>Pentylamine: Labeling transglutaminase substrate on glutamine side chain.</summary>
    UNIMOD_Pentylamine = 100000801,

    /// <summary>Biotin:Thermo-21360: Was Biotin-PEO4-hydrazide.</summary>
    UNIMOD_Biotin_Thermo_21360 = 100000811,

    /// <summary>Cy3b-maleimide: Fluorescent dye that labels cysteines.</summary>
    UNIMOD_Cy3b_maleimide = 100000821,

    /// <summary>Gly-loss+Amide: Enzymatic glycine removal leaving an amidated C-terminus.</summary>
    UNIMOD_Gly_loss_Amide = 100000822,

    /// <summary>BMOE: Addition of BMOE crosslinker.</summary>
    UNIMOD_BMOE = 100000824,

    /// <summary>DFDNB: Addition of DFDNB crosslinker.</summary>
    UNIMOD_DFDNB = 100000825,

    /// <summary>TMPP-Ac: Tris(2,4,6-trimethoxyphenyl)phosphonium acetic acid N-hydroxysuccinimide ester derivative.</summary>
    UNIMOD_TMPP_Ac = 100000827,

    /// <summary>Dihydroxyimidazolidine: Dihydroxy methylglyoxal adduct.</summary>
    UNIMOD_Dihydroxyimidazolidine = 100000830,

    /// <summary>Label:2H(4)+Acetyl: Acetyl 4,4,5,5-D4 Lysine.</summary>
    UNIMOD_Label_2H_4__Acetyl = 100000834,

    /// <summary>Label:13C(6)+Acetyl: Acetyl 13C(6) Silac label.</summary>
    UNIMOD_Label_13C_6__Acetyl = 100000835,

    /// <summary>Label:13C(6)15N(2)+Acetyl: Acetyl_13C(6) 15N(2) Silac label.</summary>
    UNIMOD_Label_13C_6_15N_2__Acetyl = 100000836,

    /// <summary>Arg->Npo: Arginine replacement by Nitropyrimidyl ornithine.</summary>
    UNIMOD_Arg__Npo = 100000837,

    /// <summary>EQIGG: Sumo mutant Smt3-WT tail following trypsin digestion.</summary>
    UNIMOD_EQIGG = 100000846,

    /// <summary>Arg2PG: Adduct of phenylglyoxal with Arg.</summary>
    UNIMOD_Arg2PG = 100000848,

    /// <summary>cGMP: S-guanylation.</summary>
    UNIMOD_cGMP = 100000849,

    /// <summary>cGMP+RMP-loss: S-guanylation-2.</summary>
    UNIMOD_cGMP_RMP_loss = 100000851,

    /// <summary>Label:2H(4)+GlyGly: Ubiquitination 2H4 lysine.</summary>
    UNIMOD_Label_2H_4__GlyGly = 100000853,

    /// <summary>Label:13C(8)15N(2): 13C(8) 15N(2) Silac label.</summary>
    UNIMOD_Label_13C_8_15N_2_ = 100000854,

    /// <summary>MG-H1: Methylglyoxal-derived hydroimidazolone.</summary>
    UNIMOD_MG_H1 = 100000859,

    /// <summary>G-H1: Glyoxal-derived hydroimiadazolone.</summary>
    UNIMOD_G_H1 = 100000860,

    /// <summary>ZGB: NHS ester linked Green Fluorescent Bodipy Dye.</summary>
    UNIMOD_ZGB = 100000861,

    /// <summary>Label:13C(1)2H(3): SILAC.</summary>
    UNIMOD_Label_13C_1_2H_3_ = 100000862,

    /// <summary>Label:13C(6)15N(2)+GlyGly: 13C(6) 15N(2) Lysine glygly.</summary>
    UNIMOD_Label_13C_6_15N_2__GlyGly = 100000864,

    /// <summary>ICPL:13C(6)2H(4): Bruker Daltonics SERVA-ICPL(TM) quantification chemistry, +10 Da form.</summary>
    UNIMOD_ICPL_13C_6_2H_4_ = 100000866,

    /// <summary>QEQTGG: SUMOylation by SUMO-1.</summary>
    UNIMOD_QEQTGG = 100000876,

    /// <summary>QQQTGG: SUMOylation by SUMO-2/3.</summary>
    UNIMOD_QQQTGG = 100000877,

    /// <summary>Bodipy: Bodipy modifications onto cysteine.</summary>
    UNIMOD_Bodipy = 100000878,

    /// <summary>Biotin:Thermo-21325: Was ChromoBiotin.</summary>
    UNIMOD_Biotin_Thermo_21325 = 100000884,

    /// <summary>Label:13C(1)2H(3)+Oxidation: Oxidised methionine 13C(1)2H(3) SILAC label.</summary>
    UNIMOD_Label_13C_1_2H_3__Oxidation = 100000885,

    /// <summary>HydroxymethylOP: 2-ammonio-6-[4-(hydroxymethyl)-3-oxidopyridinium-1-yl]- hexanoate.</summary>
    UNIMOD_HydroxymethylOP = 100000886,

    /// <summary>MDCC: Covalent linkage of maleimidyl coumarin probe (Molecular Probes D-10253).</summary>
    UNIMOD_MDCC = 100000887,

    /// <summary>mTRAQ: MTRAQ light.</summary>
    UNIMOD_mTRAQ = 100000888,

    /// <summary>mTRAQ:13C(3)15N(1): MTRAQ medium.</summary>
    UNIMOD_mTRAQ_13C_3_15N_1_ = 100000889,

    /// <summary>DyLight-maleimide: Thiol-reactive dye for fluorescence labelling of proteins.</summary>
    UNIMOD_DyLight_maleimide = 100000890,

    /// <summary>Methyl-PEO12-Maleimide: Methyl-PEO12-Maleimide.</summary>
    UNIMOD_Methyl_PEO12_Maleimide = 100000891,

    /// <summary>CarbamidomethylDTT: Carbamidomethylated DTT modification of cysteine.</summary>
    UNIMOD_CarbamidomethylDTT = 100000893,

    /// <summary>CarboxymethylDTT: Carboxymethylated DTT modification of cysteine.</summary>
    UNIMOD_CarboxymethylDTT = 100000894,

    /// <summary>Biotin-PEG-PRA: Biotin polyethyleneoxide (n=3) alkyne.</summary>
    UNIMOD_Biotin_PEG_PRA = 100000895,

    /// <summary>Met->Aha: Methionine replacement by azido homoalanine.</summary>
    UNIMOD_Met__Aha = 100000896,

    /// <summary>Label:15N(4): SILAC 15N(4).</summary>
    UNIMOD_Label_15N_4_ = 100000897,

    /// <summary>pyrophospho: Pyrophosphorylation of Ser/Thr.</summary>
    UNIMOD_pyrophospho = 100000898,

    /// <summary>Met->Hpg: Methionine replacement by homopropargylglycine.</summary>
    UNIMOD_Met__Hpg = 100000899,

    /// <summary>4AcAllylGal: 2,3,4,6-tetra-O-Acetyl-1-allyl-alpha-D-galactopyranoside modification of cysteine.</summary>
    UNIMOD_4AcAllylGal = 100000901,

    /// <summary>DimethylArsino: Reaction with dimethylarsinous (AsIII) acid.</summary>
    UNIMOD_DimethylArsino = 100000902,

    /// <summary>Lys->CamCys: Lys->Cys substitution and carbamidomethylation.</summary>
    UNIMOD_Lys__CamCys = 100000903,

    /// <summary>Phe->CamCys: Phe->Cys substitution and carbamidomethylation.</summary>
    UNIMOD_Phe__CamCys = 100000904,

    /// <summary>Leu->MetOx: Leu->Met substitution and sulfoxidation.</summary>
    UNIMOD_Leu__MetOx = 100000905,

    /// <summary>Lys->MetOx: Lys->Met substitution and sulfoxidation.</summary>
    UNIMOD_Lys__MetOx = 100000906,

    /// <summary>Galactosyl: Galactosyl hydroxylysine.</summary>
    UNIMOD_Galactosyl = 100000907,

    /// <summary>SMCC-maleimide: Modified SMCC maleimide with 3-(dimethylamino)-1-propylamine.</summary>
    UNIMOD_SMCC_maleimide = 100000908,

    /// <summary>Bacillosamine: 2,4-diacetamido-2,4,6-trideoxyglucopyranose.</summary>
    UNIMOD_Bacillosamine = 100000910,

    /// <summary>MTSL: Cys modification by (1-oxyl-2,2,5,5-tetramethyl-3-pyrroline-3-methyl)methanesulfonate (MTSL).</summary>
    UNIMOD_MTSL = 100000911,

    /// <summary>HNE-BAHAH: 4-hydroxy-2-nonenal and biotinamidohexanoic acid hydrazide, reduced.</summary>
    UNIMOD_HNE_BAHAH = 100000912,

    /// <summary>Methylmalonylation: Methylmalonylation on Serine.</summary>
    UNIMOD_Methylmalonylation = 100000914,

    /// <summary>Ethoxyformyl: Ethoxyformylation.</summary>
    UNIMOD_Ethoxyformyl = 100000915,

    /// <summary>Label:13C(4)15N(2)+GlyGly: 13C(4) 15N(2) Lysine glygly.</summary>
    UNIMOD_Label_13C_4_15N_2__GlyGly = 100000923,

    /// <summary>ethylamino: Ethyl amino.</summary>
    UNIMOD_ethylamino = 100000926,

    /// <summary>MercaptoEthanol: 2-OH-ethyl thio-Ser.</summary>
    UNIMOD_MercaptoEthanol = 100000928,

    /// <summary>EthylAmide: Solvolysis of amide group on Asn or Gln by ethanol.</summary>
    UNIMOD_EthylAmide = 100000931,

    /// <summary>VFQQQTGG: SUMOylation by SUMO-2/3 (formic acid cleavage).</summary>
    UNIMOD_VFQQQTGG = 100000932,

    /// <summary>VIEVYQEQTGG: SUMOylation by SUMO-1 (formic acid cleavage).</summary>
    UNIMOD_VIEVYQEQTGG = 100000933,

    /// <summary>AMTzHexNAc2: Photocleavable Biotin + GalNAz on O-GlcNAc.</summary>
    UNIMOD_AMTzHexNAc2 = 100000934,

    /// <summary>Atto495Maleimide: High molecular absorption maleimide label for proteins.</summary>
    UNIMOD_Atto495Maleimide = 100000935,

    /// <summary>Chlorination: Chlorination of tyrosine residues.</summary>
    UNIMOD_Chlorination = 100000936,

    /// <summary>dichlorination: Dichlorination of tyrosine residues.</summary>
    UNIMOD_dichlorination = 100000937,

    /// <summary>AROD: Cysteine modifier.</summary>
    UNIMOD_AROD = 100000938,

    /// <summary>Cys->methylaminoAla: Carbamidomethylated Cys that undergoes beta-elimination and Michael addition of methylamine.</summary>
    UNIMOD_Cys__methylaminoAla = 100000939,

    /// <summary>Cys->ethylaminoAla: Carbamidomethylated Cys that undergoes beta-elimination and Michael addition of ethylamine.</summary>
    UNIMOD_Cys__ethylaminoAla = 100000940,

    /// <summary>DNPS: 2,4-Dinitrobenzenesulfenyl.</summary>
    UNIMOD_DNPS = 100000941,

    /// <summary>SulfoGMBS: High molecular absorption label for proteins.</summary>
    UNIMOD_SulfoGMBS = 100000942,

    /// <summary>DimethylamineGMBS: Modified GMBS X linker for proteins.</summary>
    UNIMOD_DimethylamineGMBS = 100000943,

    /// <summary>Label:15N(2)2H(9): SILAC label.</summary>
    UNIMOD_Label_15N_2_2H_9_ = 100000944,

    /// <summary>LG-anhydrolactam: Levuglandinyl-lysine anhydrolactam adduct.</summary>
    UNIMOD_LG_anhydrolactam = 100000946,

    /// <summary>LG-pyrrole: Levuglandinyl-lysine pyrrole adduct.</summary>
    UNIMOD_LG_pyrrole = 100000947,

    /// <summary>LG-anhyropyrrole: Levuglandinyl-lysine anhyropyrrole adduct.</summary>
    UNIMOD_LG_anhyropyrrole = 100000948,

    /// <summary>3-deoxyglucosone: Condensation product of 3-deoxyglucosone.</summary>
    UNIMOD_3_deoxyglucosone = 100000949,

    /// <summary>Cation:Li: Replacement of proton by lithium.</summary>
    UNIMOD_Cation_Li = 100000950,

    /// <summary>Cation:Ca[II]: Replacement of 2 protons by calcium.</summary>
    UNIMOD_Cation_Ca_II_ = 100000951,

    /// <summary>Cation:Fe[II]: Replacement of 2 protons by iron.</summary>
    UNIMOD_Cation_Fe_II_ = 100000952,

    /// <summary>Cation:Ni[II]: Replacement of 2 protons by nickel.</summary>
    UNIMOD_Cation_Ni_II_ = 100000953,

    /// <summary>Cation:Zn[II]: Replacement of 2 protons by zinc.</summary>
    UNIMOD_Cation_Zn_II_ = 100000954,

    /// <summary>Cation:Ag: Replacement of proton by silver.</summary>
    UNIMOD_Cation_Ag = 100000955,

    /// <summary>Cation:Mg[II]: Replacement of 2 protons by magnesium.</summary>
    UNIMOD_Cation_Mg_II_ = 100000956,

    /// <summary>2-succinyl: S-(2-succinyl) cysteine.</summary>
    UNIMOD_2_succinyl = 100000957,

    /// <summary>Propargylamine: Propargylamine.</summary>
    UNIMOD_Propargylamine = 100000958,

    /// <summary>Phosphopropargyl: Phospho-propargylamine.</summary>
    UNIMOD_Phosphopropargyl = 100000959,

    /// <summary>SUMO2135: SUMOylation by SUMO-1 after tryptic cleavage.</summary>
    UNIMOD_SUMO2135 = 100000960,

    /// <summary>SUMO3549: SUMOylation by SUMO-2/3 after tryptic cleavage.</summary>
    UNIMOD_SUMO3549 = 100000961,

    /// <summary>thioacylPA: Membrane protein extraction.</summary>
    UNIMOD_thioacylPA = 100000967,

    /// <summary>maleimide3: Maleimide-3-saccharide.</summary>
    UNIMOD_maleimide3 = 100000971,

    /// <summary>maleimide5: Maleimide-5-saccharide.</summary>
    UNIMOD_maleimide5 = 100000972,

    /// <summary>Puromycin: Puromycin.</summary>
    UNIMOD_Puromycin = 100000973,

    /// <summary>Chlorpyrifos: O,O-diethyl o-3,5,6-trichloro-2-pyridyl phosphorothioate.</summary>
    UNIMOD_Chlorpyrifos = 100000975,

    /// <summary>Carbofuran: 2,3-dihydro-2,2-dimethyl-7-benzofuranol N-methyl carbamate.</summary>
    UNIMOD_Carbofuran = 100000977,

    /// <summary>BITC: Benzyl isothiocyanate.</summary>
    UNIMOD_BITC = 100000978,

    /// <summary>PEITC: Phenethyl isothiocyanate.</summary>
    UNIMOD_PEITC = 100000979,

    /// <summary>glucosone: Condensation product of glucosone.</summary>
    UNIMOD_glucosone = 100000981,

    /// <summary>cysTMT: Native cysteine-reactive Tandem Mass Tag®.</summary>
    UNIMOD_cysTMT = 100000984,

    /// <summary>cysTMT6plex: Cysteine-reactive Sixplex Tandem Mass Tag®.</summary>
    UNIMOD_cysTMT6plex = 100000985,

    /// <summary>Label:13C(6)+Dimethyl: Dimethyl 13C(6) Silac label.</summary>
    UNIMOD_Label_13C_6__Dimethyl = 100000986,

    /// <summary>Label:13C(6)15N(2)+Dimethyl: Dimethyl 13C(6)15N(2) Silac label.</summary>
    UNIMOD_Label_13C_6_15N_2__Dimethyl = 100000987,

    /// <summary>Ammonium: Replacement of proton with ammonium ion.</summary>
    UNIMOD_Ammonium = 100000989,

    /// <summary>ISD_z+2_ion: ISD (z+2)-series.</summary>
    UNIMOD_ISD_z_2_ion = 100000991,

    /// <summary>Biotin:Sigma-B1267: Was Biotin-maleimide.</summary>
    UNIMOD_Biotin_Sigma_B1267 = 100000993,

    /// <summary>Label:15N(1): 15N(1).</summary>
    UNIMOD_Label_15N_1_ = 100000994,

    /// <summary>Label:15N(2): 15N(2).</summary>
    UNIMOD_Label_15N_2_ = 100000995,

    /// <summary>Label:15N(3): 15N(3).</summary>
    UNIMOD_Label_15N_3_ = 100000996,

    /// <summary>sulfo+amino: Aminotyrosine with sulfation.</summary>
    UNIMOD_sulfo_amino = 100000997,

    /// <summary>BHAc: N-biotinyl-6-aminohexanoyl.</summary>
    UNIMOD_BHAc = 100000998,

    /// <summary>AHA-Alkyne: Azidohomoalanine (AHA) bound to propargylglycine-NH2 (alkyne).</summary>
    UNIMOD_AHA_Alkyne = 100001000,

    /// <summary>AHA-Alkyne-KDDDD: Azidohomoalanine (AHA) bound to DDDDK-propargylglycine-NH2 (alkyne).</summary>
    UNIMOD_AHA_Alkyne_KDDDD = 100001001,

    /// <summary>EGCG1: (-)-epigallocatechin-3-gallate.</summary>
    UNIMOD_EGCG1 = 100001002,

    /// <summary>EGCG2: (-)-dehydroepigallocatechin.</summary>
    UNIMOD_EGCG2 = 100001003,

    /// <summary>Label:13C(6)15N(4)+Methyl: Monomethylated Arg13C(6) 15N(4).</summary>
    UNIMOD_Label_13C_6_15N_4__Methyl = 100001004,

    /// <summary>Label:13C(6)15N(4)+Dimethyl: Dimethylated Arg13C(6) 15N(4).</summary>
    UNIMOD_Label_13C_6_15N_4__Dimethyl = 100001005,

    /// <summary>Label:13C(6)15N(4)+Methyl:2H(3)13C(1): 2H(3) 13C(1) monomethylated Arg13C(6) 15N(4).</summary>
    UNIMOD_Label_13C_6_15N_4__Methyl_2H_3_13C_1_ = 100001006,

    /// <summary>Label:13C(6)15N(4)+Dimethyl:2H(6)13C(2): 2H(6) 13C(2) Dimethylated Arg13C(6) 15N(4).</summary>
    UNIMOD_Label_13C_6_15N_4__Dimethyl_2H_6_13C_2_ = 100001007,

    /// <summary>SecCarbamidomethyl: Sec Iodoacetamide derivative.</summary>
    UNIMOD_SecCarbamidomethyl = 100001008,

    /// <summary>Thiazolidine: Addition of Carbon to cysteine.</summary>
    UNIMOD_Thiazolidine = 100001009,

    /// <summary>DEDGFLYMVYASQETFG: Addition of DEDGFLYMVYASQETFG.</summary>
    UNIMOD_DEDGFLYMVYASQETFG = 100001010,

    /// <summary>Biotin:Invitrogen-M1602: Nalpha-(3-maleimidylpropionyl)biocytin.</summary>
    UNIMOD_Biotin_Invitrogen_M1602 = 100001012,

    /// <summary>glycidamide: Glycidamide adduct.</summary>
    UNIMOD_glycidamide = 100001014,

    /// <summary>Ahx2+Hsl: C-terminal homoserine lactone and two aminohexanoic acids.</summary>
    UNIMOD_Ahx2_Hsl = 100001015,

    /// <summary>DMPO: DMPO spin-trap nitrone adduct.</summary>
    UNIMOD_DMPO = 100001017,

    /// <summary>ICDID: Isotope-Coded Dimedone light form.</summary>
    UNIMOD_ICDID = 100001018,

    /// <summary>ICDID:2H(6): Isotope-Coded Dimedone heavy form.</summary>
    UNIMOD_ICDID_2H_6_ = 100001019,

    /// <summary>Xlink:DSS: Monolink of DSS/BS3 crosslinker to Lys or N-terminus.</summary>
    UNIMOD_Xlink_DSS = 100001020,

    /// <summary>Xlink:EGS: Monolink of EGS crosslinker to Lys or N-terminus.</summary>
    UNIMOD_Xlink_EGS = 100001021,

    /// <summary>Xlink:DST: Monolink of DST crosslinker to Lys or N-terminus.</summary>
    UNIMOD_Xlink_DST = 100001022,

    /// <summary>Xlink:DTSSP: Monolink of DSP/DTSSP crosslinker to Lys or N-terminus.</summary>
    UNIMOD_Xlink_DTSSP = 100001023,

    /// <summary>Xlink:SMCC: Monolink of sulfoSMCC/SMCC crosslinker to Cys.</summary>
    UNIMOD_Xlink_SMCC = 100001024,

    /// <summary>Xlink:DMP-de: Monolink of DMP crosslinker to Lys or N-terminus.</summary>
    UNIMOD_Xlink_DMP_de = 100001027,

    /// <summary>Xlink:EGScleaved: EGS crosslinker to Lys or N-terminus following hydroxylamine cleavage.</summary>
    UNIMOD_Xlink_EGScleaved = 100001028,

    /// <summary>Biotin:Thermo-88310: Desthiobiotin modification of lysine.</summary>
    UNIMOD_Biotin_Thermo_88310 = 100001031,

    /// <summary>2-nitrobenzyl: Tyrosine caged with 2-nitrobenzyl (ONB).</summary>
    UNIMOD_2_nitrobenzyl = 100001032,

    /// <summary>SecNEM: N-ethylmaleimide on selenocysteines.</summary>
    UNIMOD_SecNEM = 100001033,

    /// <summary>SecNEM:2H(5): D5 N-ethylmaleimide on selenocysteines.</summary>
    UNIMOD_SecNEM_2H_5_ = 100001034,

    /// <summary>Thiadiazole: Thiadiazolydation of Cys.</summary>
    UNIMOD_Thiadiazole = 100001035,

    /// <summary>Withaferin: Modification of cystein by withaferin.</summary>
    UNIMOD_Withaferin = 100001036,

    /// <summary>Biotin:Thermo-88317: Desthiobiotin fluorophosphonate.</summary>
    UNIMOD_Biotin_Thermo_88317 = 100001037,

    /// <summary>TAMRA-FP: TAMRA fluorophosphonate modification of serine.</summary>
    UNIMOD_TAMRA_FP = 100001038,

    /// <summary>Biotin:Thermo-21901+H2O: Maleimide-Biotin + Water.</summary>
    UNIMOD_Biotin_Thermo_21901_H2O = 100001039,

    /// <summary>Deoxyhypusine: Deoxyhypusine.</summary>
    UNIMOD_Deoxyhypusine = 100001041,

    /// <summary>Acetyldeoxyhypusine: Acetyldeoxyhypusine.</summary>
    UNIMOD_Acetyldeoxyhypusine = 100001042,

    /// <summary>Acetylhypusine: Acetylhypusine.</summary>
    UNIMOD_Acetylhypusine = 100001043,

    /// <summary>Ala->Cys: Ala->Cys substitution.</summary>
    UNIMOD_Ala__Cys = 100001044,

    /// <summary>Ala->Phe: Ala->Phe substitution.</summary>
    UNIMOD_Ala__Phe = 100001045,

    /// <summary>Ala->His: Ala->His substitution.</summary>
    UNIMOD_Ala__His = 100001046,

    /// <summary>Ala->Xle: Ala->Leu/Ile substitution.</summary>
    UNIMOD_Ala__Xle = 100001047,

    /// <summary>Ala->Lys: Ala->Lys substitution.</summary>
    UNIMOD_Ala__Lys = 100001048,

    /// <summary>Ala->Met: Ala->Met substitution.</summary>
    UNIMOD_Ala__Met = 100001049,

    /// <summary>Ala->Asn: Ala->Asn substitution.</summary>
    UNIMOD_Ala__Asn = 100001050,

    /// <summary>Ala->Gln: Ala->Gln substitution.</summary>
    UNIMOD_Ala__Gln = 100001051,

    /// <summary>Ala->Arg: Ala->Arg substitution.</summary>
    UNIMOD_Ala__Arg = 100001052,

    /// <summary>Ala->Trp: Ala->Trp substitution.</summary>
    UNIMOD_Ala__Trp = 100001053,

    /// <summary>Ala->Tyr: Ala->Tyr substitution.</summary>
    UNIMOD_Ala__Tyr = 100001054,

    /// <summary>Cys->Ala: Cys->Ala substitution.</summary>
    UNIMOD_Cys__Ala = 100001055,

    /// <summary>Cys->Asp: Cys->Asp substitution.</summary>
    UNIMOD_Cys__Asp = 100001056,

    /// <summary>Cys->Glu: Cys->Glu substitution.</summary>
    UNIMOD_Cys__Glu = 100001057,

    /// <summary>Cys->His: Cys->His substitution.</summary>
    UNIMOD_Cys__His = 100001058,

    /// <summary>Cys->Xle: Cys->Leu/Ile substitution.</summary>
    UNIMOD_Cys__Xle = 100001059,

    /// <summary>Cys->Lys: Cys->Lys substitution.</summary>
    UNIMOD_Cys__Lys = 100001060,

    /// <summary>Cys->Met: Cys->Met substitution.</summary>
    UNIMOD_Cys__Met = 100001061,

    /// <summary>Cys->Asn: Cys->Asn substitution.</summary>
    UNIMOD_Cys__Asn = 100001062,

    /// <summary>Cys->Pro: Cys->Pro substitution.</summary>
    UNIMOD_Cys__Pro = 100001063,

    /// <summary>Cys->Gln: Cys->Gln substitution.</summary>
    UNIMOD_Cys__Gln = 100001064,

    /// <summary>Cys->Thr: Cys->Thr substitution.</summary>
    UNIMOD_Cys__Thr = 100001065,

    /// <summary>Cys->Val: Cys->Val substitution.</summary>
    UNIMOD_Cys__Val = 100001066,

    /// <summary>Asp->Cys: Asp->Cys substitution.</summary>
    UNIMOD_Asp__Cys = 100001067,

    /// <summary>Asp->Phe: Asp->Phe substitution.</summary>
    UNIMOD_Asp__Phe = 100001068,

    /// <summary>Asp->Xle: Asp->Leu/Ile substitution.</summary>
    UNIMOD_Asp__Xle = 100001069,

    /// <summary>Asp->Lys: Asp->Lys substitution.</summary>
    UNIMOD_Asp__Lys = 100001070,

    /// <summary>Asp->Met: Asp->Met substitution.</summary>
    UNIMOD_Asp__Met = 100001071,

    /// <summary>Asp->Pro: Asp->Pro substitution.</summary>
    UNIMOD_Asp__Pro = 100001072,

    /// <summary>Asp->Gln: Asp->Gln substitution.</summary>
    UNIMOD_Asp__Gln = 100001073,

    /// <summary>Asp->Arg: Asp->Arg substitution.</summary>
    UNIMOD_Asp__Arg = 100001074,

    /// <summary>Asp->Ser: Asp->Ser substitution.</summary>
    UNIMOD_Asp__Ser = 100001075,

    /// <summary>Asp->Thr: Asp->Thr substitution.</summary>
    UNIMOD_Asp__Thr = 100001076,

    /// <summary>Asp->Trp: Asp->Trp substitution.</summary>
    UNIMOD_Asp__Trp = 100001077,

    /// <summary>Glu->Cys: Glu->Cys substitution.</summary>
    UNIMOD_Glu__Cys = 100001078,

    /// <summary>Glu->Phe: Glu->Phe substitution.</summary>
    UNIMOD_Glu__Phe = 100001079,

    /// <summary>Glu->His: Glu->His substitution.</summary>
    UNIMOD_Glu__His = 100001080,

    /// <summary>Glu->Xle: Glu->Leu/Ile substitution.</summary>
    UNIMOD_Glu__Xle = 100001081,

    /// <summary>Glu->Met: Glu->Met substitution.</summary>
    UNIMOD_Glu__Met = 100001082,

    /// <summary>Glu->Asn: Glu->Asn substitution.</summary>
    UNIMOD_Glu__Asn = 100001083,

    /// <summary>Glu->Pro: Glu->Pro substitution.</summary>
    UNIMOD_Glu__Pro = 100001084,

    /// <summary>Glu->Arg: Glu->Arg substitution.</summary>
    UNIMOD_Glu__Arg = 100001085,

    /// <summary>Glu->Ser: Glu->Ser substitution.</summary>
    UNIMOD_Glu__Ser = 100001086,

    /// <summary>Glu->Thr: Glu->Thr substitution.</summary>
    UNIMOD_Glu__Thr = 100001087,

    /// <summary>Glu->Trp: Glu->Trp substitution.</summary>
    UNIMOD_Glu__Trp = 100001088,

    /// <summary>Glu->Tyr: Glu->Tyr substitution.</summary>
    UNIMOD_Glu__Tyr = 100001089,

    /// <summary>Phe->Ala: Phe->Ala substitution.</summary>
    UNIMOD_Phe__Ala = 100001090,

    /// <summary>Phe->Asp: Phe->Asp substitution.</summary>
    UNIMOD_Phe__Asp = 100001091,

    /// <summary>Phe->Glu: Phe->Glu substitution.</summary>
    UNIMOD_Phe__Glu = 100001092,

    /// <summary>Phe->Gly: Phe->Gly substitution.</summary>
    UNIMOD_Phe__Gly = 100001093,

    /// <summary>Phe->His: Phe->His substitution.</summary>
    UNIMOD_Phe__His = 100001094,

    /// <summary>Phe->Lys: Phe->Lys substitution.</summary>
    UNIMOD_Phe__Lys = 100001095,

    /// <summary>Phe->Met: Phe->Met substitution.</summary>
    UNIMOD_Phe__Met = 100001096,

    /// <summary>Phe->Asn: Phe->Asn substitution.</summary>
    UNIMOD_Phe__Asn = 100001097,

    /// <summary>Phe->Pro: Phe->Pro substitution.</summary>
    UNIMOD_Phe__Pro = 100001098,

    /// <summary>Phe->Gln: Phe->Gln substitution.</summary>
    UNIMOD_Phe__Gln = 100001099,

    /// <summary>Phe->Arg: Phe->Arg substitution.</summary>
    UNIMOD_Phe__Arg = 100001100,

    /// <summary>Phe->Thr: Phe->Thr substitution.</summary>
    UNIMOD_Phe__Thr = 100001101,

    /// <summary>Phe->Trp: Phe->Trp substitution.</summary>
    UNIMOD_Phe__Trp = 100001102,

    /// <summary>Gly->Phe: Gly->Phe substitution.</summary>
    UNIMOD_Gly__Phe = 100001103,

    /// <summary>Gly->His: Gly->His substitution.</summary>
    UNIMOD_Gly__His = 100001104,

    /// <summary>Gly->Xle: Gly->Leu/Ile substitution.</summary>
    UNIMOD_Gly__Xle = 100001105,

    /// <summary>Gly->Lys: Gly->Lys substitution.</summary>
    UNIMOD_Gly__Lys = 100001106,

    /// <summary>Gly->Met: Gly->Met substitution.</summary>
    UNIMOD_Gly__Met = 100001107,

    /// <summary>Gly->Asn: Gly->Asn substitution.</summary>
    UNIMOD_Gly__Asn = 100001108,

    /// <summary>Gly->Pro: Gly->Pro substitution.</summary>
    UNIMOD_Gly__Pro = 100001109,

    /// <summary>Gly->Gln: Gly->Gln substitution.</summary>
    UNIMOD_Gly__Gln = 100001110,

    /// <summary>Gly->Thr: Gly->Thr substitution.</summary>
    UNIMOD_Gly__Thr = 100001111,

    /// <summary>Gly->Tyr: Gly->Tyr substitution.</summary>
    UNIMOD_Gly__Tyr = 100001112,

    /// <summary>His->Ala: His->Ala substitution.</summary>
    UNIMOD_His__Ala = 100001113,

    /// <summary>His->Cys: His->Cys substitution.</summary>
    UNIMOD_His__Cys = 100001114,

    /// <summary>His->Glu: His->Glu substitution.</summary>
    UNIMOD_His__Glu = 100001115,

    /// <summary>His->Phe: His->Phe substitution.</summary>
    UNIMOD_His__Phe = 100001116,

    /// <summary>His->Gly: His->Gly substitution.</summary>
    UNIMOD_His__Gly = 100001117,

    /// <summary>His->Lys: His->Lys substitution.</summary>
    UNIMOD_His__Lys = 100001119,

    /// <summary>His->Met: His->Met substitution.</summary>
    UNIMOD_His__Met = 100001120,

    /// <summary>His->Ser: His->Ser substitution.</summary>
    UNIMOD_His__Ser = 100001121,

    /// <summary>His->Thr: His->Thr substitution.</summary>
    UNIMOD_His__Thr = 100001122,

    /// <summary>His->Val: His->Val substitution.</summary>
    UNIMOD_His__Val = 100001123,

    /// <summary>His->Trp: His->Trp substitution.</summary>
    UNIMOD_His__Trp = 100001124,

    /// <summary>Xle->Ala: Leu/Ile->Ala substitution.</summary>
    UNIMOD_Xle__Ala = 100001125,

    /// <summary>Xle->Cys: Leu/Ile->Cys substitution.</summary>
    UNIMOD_Xle__Cys = 100001126,

    /// <summary>Xle->Asp: Leu/Ile->Asp substitution.</summary>
    UNIMOD_Xle__Asp = 100001127,

    /// <summary>Xle->Glu: Leu/Ile->Glu substitution.</summary>
    UNIMOD_Xle__Glu = 100001128,

    /// <summary>Xle->Gly: Leu/Ile->Gly substitution.</summary>
    UNIMOD_Xle__Gly = 100001129,

    /// <summary>Xle->Tyr: Leu/Ile->Tyr substitution.</summary>
    UNIMOD_Xle__Tyr = 100001130,

    /// <summary>Lys->Ala: Lys->Ala substitution.</summary>
    UNIMOD_Lys__Ala = 100001131,

    /// <summary>Lys->Cys: Lys->Cys substitution.</summary>
    UNIMOD_Lys__Cys = 100001132,

    /// <summary>Lys->Asp: Lys->Asp substitution.</summary>
    UNIMOD_Lys__Asp = 100001133,

    /// <summary>Lys->Phe: Lys->Phe substitution.</summary>
    UNIMOD_Lys__Phe = 100001134,

    /// <summary>Lys->Gly: Lys->Gly substitution.</summary>
    UNIMOD_Lys__Gly = 100001135,

    /// <summary>Lys->His: Lys->His substitution.</summary>
    UNIMOD_Lys__His = 100001136,

    /// <summary>Lys->Pro: Lys->Pro substitution.</summary>
    UNIMOD_Lys__Pro = 100001137,

    /// <summary>Lys->Ser: Lys->Ser substitution.</summary>
    UNIMOD_Lys__Ser = 100001138,

    /// <summary>Lys->Val: Lys->Val substitution.</summary>
    UNIMOD_Lys__Val = 100001139,

    /// <summary>Lys->Trp: Lys->Trp substitution.</summary>
    UNIMOD_Lys__Trp = 100001140,

    /// <summary>Lys->Tyr: Lys->Tyr substitution.</summary>
    UNIMOD_Lys__Tyr = 100001141,

    /// <summary>Met->Ala: Met->Ala substitution.</summary>
    UNIMOD_Met__Ala = 100001142,

    /// <summary>Met->Cys: Met->Cys substitution.</summary>
    UNIMOD_Met__Cys = 100001143,

    /// <summary>Met->Asp: Met->Asp substitution.</summary>
    UNIMOD_Met__Asp = 100001144,

    /// <summary>Met->Glu: Met->Glu substitution.</summary>
    UNIMOD_Met__Glu = 100001145,

    /// <summary>Met->Phe: Met->Phe substitution.</summary>
    UNIMOD_Met__Phe = 100001146,

    /// <summary>Met->Gly: Met->Gly substitution.</summary>
    UNIMOD_Met__Gly = 100001147,

    /// <summary>Met->His: Met->His substitution.</summary>
    UNIMOD_Met__His = 100001148,

    /// <summary>Met->Asn: Met->Asn substitution.</summary>
    UNIMOD_Met__Asn = 100001149,

    /// <summary>Met->Pro: Met->Pro substitution.</summary>
    UNIMOD_Met__Pro = 100001150,

    /// <summary>Met->Gln: Met->Gln substitution.</summary>
    UNIMOD_Met__Gln = 100001151,

    /// <summary>Met->Ser: Met->Ser substitution.</summary>
    UNIMOD_Met__Ser = 100001152,

    /// <summary>Met->Trp: Met->Trp substitution.</summary>
    UNIMOD_Met__Trp = 100001153,

    /// <summary>Met->Tyr: Met->Tyr substitution.</summary>
    UNIMOD_Met__Tyr = 100001154,

    /// <summary>Asn->Ala: Asn->Ala substitution.</summary>
    UNIMOD_Asn__Ala = 100001155,

    /// <summary>Asn->Cys: Asn->Cys substitution.</summary>
    UNIMOD_Asn__Cys = 100001156,

    /// <summary>Asn->Glu: Asn->Glu substitution.</summary>
    UNIMOD_Asn__Glu = 100001157,

    /// <summary>Asn->Phe: Asn->Phe substitution.</summary>
    UNIMOD_Asn__Phe = 100001158,

    /// <summary>Asn->Gly: Asn->Gly substitution.</summary>
    UNIMOD_Asn__Gly = 100001159,

    /// <summary>Asn->Met: Asn->Met substitution.</summary>
    UNIMOD_Asn__Met = 100001160,

    /// <summary>Asn->Pro: Asn->Pro substitution.</summary>
    UNIMOD_Asn__Pro = 100001161,

    /// <summary>Asn->Gln: Asn->Gln substitution.</summary>
    UNIMOD_Asn__Gln = 100001162,

    /// <summary>Asn->Arg: Asn->Arg substitution.</summary>
    UNIMOD_Asn__Arg = 100001163,

    /// <summary>Asn->Val: Asn->Val substitution.</summary>
    UNIMOD_Asn__Val = 100001164,

    /// <summary>Asn->Trp: Asn->Trp substitution.</summary>
    UNIMOD_Asn__Trp = 100001165,

    /// <summary>Pro->Cys: Pro->Cys substitution.</summary>
    UNIMOD_Pro__Cys = 100001166,

    /// <summary>Pro->Asp: Pro->Asp substitution.</summary>
    UNIMOD_Pro__Asp = 100001167,

    /// <summary>Pro->Glu: Pro->Glu substitution.</summary>
    UNIMOD_Pro__Glu = 100001168,

    /// <summary>Pro->Phe: Pro->Phe substitution.</summary>
    UNIMOD_Pro__Phe = 100001169,

    /// <summary>Pro->Gly: Pro->Gly substitution.</summary>
    UNIMOD_Pro__Gly = 100001170,

    /// <summary>Pro->Lys: Pro->Lys substitution.</summary>
    UNIMOD_Pro__Lys = 100001171,

    /// <summary>Pro->Met: Pro->Met substitution.</summary>
    UNIMOD_Pro__Met = 100001172,

    /// <summary>Pro->Asn: Pro->Asn substitution.</summary>
    UNIMOD_Pro__Asn = 100001173,

    /// <summary>Pro->Val: Pro->Val substitution.</summary>
    UNIMOD_Pro__Val = 100001174,

    /// <summary>Pro->Trp: Pro->Trp substitution.</summary>
    UNIMOD_Pro__Trp = 100001175,

    /// <summary>Pro->Tyr: Pro->Tyr substitution.</summary>
    UNIMOD_Pro__Tyr = 100001176,

    /// <summary>Gln->Ala: Gln->Ala substitution.</summary>
    UNIMOD_Gln__Ala = 100001177,

    /// <summary>Gln->Cys: Gln->Cys substitution.</summary>
    UNIMOD_Gln__Cys = 100001178,

    /// <summary>Gln->Asp: Gln->Asp substitution.</summary>
    UNIMOD_Gln__Asp = 100001179,

    /// <summary>Gln->Phe: Gln->Phe substitution.</summary>
    UNIMOD_Gln__Phe = 100001180,

    /// <summary>Gln->Gly: Gln->Gly substitution.</summary>
    UNIMOD_Gln__Gly = 100001181,

    /// <summary>Gln->Met: Gln->Met substitution.</summary>
    UNIMOD_Gln__Met = 100001182,

    /// <summary>Gln->Asn: Gln->Asn substitution.</summary>
    UNIMOD_Gln__Asn = 100001183,

    /// <summary>Gln->Ser: Gln->Ser substitution.</summary>
    UNIMOD_Gln__Ser = 100001184,

    /// <summary>Gln->Thr: Gln->Thr substitution.</summary>
    UNIMOD_Gln__Thr = 100001185,

    /// <summary>Gln->Val: Gln->Val substitution.</summary>
    UNIMOD_Gln__Val = 100001186,

    /// <summary>Gln->Trp: Gln->Trp substitution.</summary>
    UNIMOD_Gln__Trp = 100001187,

    /// <summary>Gln->Tyr: Gln->Tyr substitution.</summary>
    UNIMOD_Gln__Tyr = 100001188,

    /// <summary>Arg->Ala: Arg->Ala substitution.</summary>
    UNIMOD_Arg__Ala = 100001189,

    /// <summary>Arg->Asp: Arg->Asp substitution.</summary>
    UNIMOD_Arg__Asp = 100001190,

    /// <summary>Arg->Glu: Arg->Glu substitution.</summary>
    UNIMOD_Arg__Glu = 100001191,

    /// <summary>Arg->Asn: Arg->Asn substitution.</summary>
    UNIMOD_Arg__Asn = 100001192,

    /// <summary>Arg->Val: Arg->Val substitution.</summary>
    UNIMOD_Arg__Val = 100001193,

    /// <summary>Arg->Tyr: Arg->Tyr substitution.</summary>
    UNIMOD_Arg__Tyr = 100001194,

    /// <summary>Arg->Phe: Arg->Phe substitution.</summary>
    UNIMOD_Arg__Phe = 100001195,

    /// <summary>Ser->Asp: Ser->Asp substitution.</summary>
    UNIMOD_Ser__Asp = 100001196,

    /// <summary>Ser->Glu: Ser->Glu substitution.</summary>
    UNIMOD_Ser__Glu = 100001197,

    /// <summary>Ser->His: Ser->His substitution.</summary>
    UNIMOD_Ser__His = 100001198,

    /// <summary>Ser->Lys: Ser->Lys substitution.</summary>
    UNIMOD_Ser__Lys = 100001199,

    /// <summary>Ser->Met: Ser->Met substitution.</summary>
    UNIMOD_Ser__Met = 100001200,

    /// <summary>Ser->Gln: Ser->Gln substitution.</summary>
    UNIMOD_Ser__Gln = 100001201,

    /// <summary>Ser->Val: Ser->Val substitution.</summary>
    UNIMOD_Ser__Val = 100001202,

    /// <summary>Thr->Cys: Thr->Cys substitution.</summary>
    UNIMOD_Thr__Cys = 100001203,

    /// <summary>Thr->Asp: Thr->Asp substitution.</summary>
    UNIMOD_Thr__Asp = 100001204,

    /// <summary>Thr->Glu: Thr->Glu substitution.</summary>
    UNIMOD_Thr__Glu = 100001205,

    /// <summary>Thr->Phe: Thr->Phe substitution.</summary>
    UNIMOD_Thr__Phe = 100001206,

    /// <summary>Thr->Gly: Thr->Gly substitution.</summary>
    UNIMOD_Thr__Gly = 100001207,

    /// <summary>Thr->His: Thr->His substitution.</summary>
    UNIMOD_Thr__His = 100001208,

    /// <summary>Thr->Gln: Thr->Gln substitution.</summary>
    UNIMOD_Thr__Gln = 100001209,

    /// <summary>Thr->Val: Thr->Val substitution.</summary>
    UNIMOD_Thr__Val = 100001210,

    /// <summary>Thr->Trp: Thr->Trp substitution.</summary>
    UNIMOD_Thr__Trp = 100001211,

    /// <summary>Thr->Tyr: Thr->Tyr substitution.</summary>
    UNIMOD_Thr__Tyr = 100001212,

    /// <summary>Val->Cys: Val->Cys substitution.</summary>
    UNIMOD_Val__Cys = 100001213,

    /// <summary>Val->His: Val->His substitution.</summary>
    UNIMOD_Val__His = 100001214,

    /// <summary>Val->Lys: Val->Lys substitution.</summary>
    UNIMOD_Val__Lys = 100001215,

    /// <summary>Val->Asn: Val->Asn substitution.</summary>
    UNIMOD_Val__Asn = 100001216,

    /// <summary>Val->Pro: Val->Pro substitution.</summary>
    UNIMOD_Val__Pro = 100001217,

    /// <summary>Val->Gln: Val->Gln substitution.</summary>
    UNIMOD_Val__Gln = 100001218,

    /// <summary>Val->Arg: Val->Arg substitution.</summary>
    UNIMOD_Val__Arg = 100001219,

    /// <summary>Val->Ser: Val->Ser substitution.</summary>
    UNIMOD_Val__Ser = 100001220,

    /// <summary>Val->Thr: Val->Thr substitution.</summary>
    UNIMOD_Val__Thr = 100001221,

    /// <summary>Val->Trp: Val->Trp substitution.</summary>
    UNIMOD_Val__Trp = 100001222,

    /// <summary>Val->Tyr: Val->Tyr substitution.</summary>
    UNIMOD_Val__Tyr = 100001223,

    /// <summary>Trp->Ala: Trp->Ala substitution.</summary>
    UNIMOD_Trp__Ala = 100001224,

    /// <summary>Trp->Asp: Trp->Asp substitution.</summary>
    UNIMOD_Trp__Asp = 100001225,

    /// <summary>Trp->Glu: Trp->Glu substitution.</summary>
    UNIMOD_Trp__Glu = 100001226,

    /// <summary>Trp->Phe: Trp->Phe substitution.</summary>
    UNIMOD_Trp__Phe = 100001227,

    /// <summary>Trp->His: Trp->His substitution.</summary>
    UNIMOD_Trp__His = 100001228,

    /// <summary>Trp->Lys: Trp->Lys substitution.</summary>
    UNIMOD_Trp__Lys = 100001229,

    /// <summary>Trp->Met: Trp->Met substitution.</summary>
    UNIMOD_Trp__Met = 100001230,

    /// <summary>Trp->Asn: Trp->Asn substitution.</summary>
    UNIMOD_Trp__Asn = 100001231,

    /// <summary>Trp->Pro: Trp->Pro substitution.</summary>
    UNIMOD_Trp__Pro = 100001232,

    /// <summary>Trp->Gln: Trp->Gln substitution.</summary>
    UNIMOD_Trp__Gln = 100001233,

    /// <summary>Trp->Thr: Trp->Thr substitution.</summary>
    UNIMOD_Trp__Thr = 100001234,

    /// <summary>Trp->Val: Trp->Val substitution.</summary>
    UNIMOD_Trp__Val = 100001235,

    /// <summary>Trp->Tyr: Trp->Tyr substitution.</summary>
    UNIMOD_Trp__Tyr = 100001236,

    /// <summary>Tyr->Ala: Tyr->Ala substitution.</summary>
    UNIMOD_Tyr__Ala = 100001237,

    /// <summary>Tyr->Glu: Tyr->Glu substitution.</summary>
    UNIMOD_Tyr__Glu = 100001238,

    /// <summary>Tyr->Gly: Tyr->Gly substitution.</summary>
    UNIMOD_Tyr__Gly = 100001239,

    /// <summary>Tyr->Lys: Tyr->Lys substitution.</summary>
    UNIMOD_Tyr__Lys = 100001240,

    /// <summary>Tyr->Met: Tyr->Met substitution.</summary>
    UNIMOD_Tyr__Met = 100001241,

    /// <summary>Tyr->Pro: Tyr->Pro substitution.</summary>
    UNIMOD_Tyr__Pro = 100001242,

    /// <summary>Tyr->Gln: Tyr->Gln substitution.</summary>
    UNIMOD_Tyr__Gln = 100001243,

    /// <summary>Tyr->Arg: Tyr->Arg substitution.</summary>
    UNIMOD_Tyr__Arg = 100001244,

    /// <summary>Tyr->Thr: Tyr->Thr substitution.</summary>
    UNIMOD_Tyr__Thr = 100001245,

    /// <summary>Tyr->Val: Tyr->Val substitution.</summary>
    UNIMOD_Tyr__Val = 100001246,

    /// <summary>Tyr->Trp: Tyr->Trp substitution.</summary>
    UNIMOD_Tyr__Trp = 100001247,

    /// <summary>Tyr->Xle: Tyr->Leu/Ile substitution.</summary>
    UNIMOD_Tyr__Xle = 100001248,

    /// <summary>AHA-SS: Azidohomoalanine coupled to reductively cleaved tag.</summary>
    UNIMOD_AHA_SS = 100001249,

    /// <summary>AHA-SS_CAM: Carbamidomethylated form of reductively cleaved tag coupled to azidohomoalanine.</summary>
    UNIMOD_AHA_SS_CAM = 100001250,

    /// <summary>Biotin:Thermo-33033: Sulfo-SBED Label Photoreactive Biotin Crosslinker.</summary>
    UNIMOD_Biotin_Thermo_33033 = 100001251,

    /// <summary>Biotin:Thermo-33033-H: Sulfo-SBED Label Photoreactive Biotin Crosslinker minus Hydrogen.</summary>
    UNIMOD_Biotin_Thermo_33033_H = 100001252,

    /// <summary>2-monomethylsuccinyl: S-(2-monomethylsuccinyl) cysteine.</summary>
    UNIMOD_2_monomethylsuccinyl = 100001253,

    /// <summary>Saligenin: O-toluene.</summary>
    UNIMOD_Saligenin = 100001254,

    /// <summary>Cresylphosphate: O-toluyl-phosphorylation.</summary>
    UNIMOD_Cresylphosphate = 100001255,

    /// <summary>CresylSaligeninPhosphate: Cresyl-Saligenin-phosphorylation.</summary>
    UNIMOD_CresylSaligeninPhosphate = 100001256,

    /// <summary>Ub-Br2: Ub Bromide probe addition.</summary>
    UNIMOD_Ub_Br2 = 100001257,

    /// <summary>Ub-VME: Ubiquitin vinylmethylester.</summary>
    UNIMOD_Ub_VME = 100001258,

    /// <summary>Ub-amide: Ub amide probe addition.</summary>
    UNIMOD_Ub_amide = 100001260,

    /// <summary>Ub-fluorescein: Ub Fluorescein probe addition.</summary>
    UNIMOD_Ub_fluorescein = 100001261,

    /// <summary>2-dimethylsuccinyl: S-(2-dimethylsuccinyl) cysteine.</summary>
    UNIMOD_2_dimethylsuccinyl = 100001262,

    /// <summary>Gly: Addition of Glycine.</summary>
    UNIMOD_Gly = 100001263,

    /// <summary>pupylation: Addition of GGE.</summary>
    UNIMOD_pupylation = 100001264,

    /// <summary>Label:13C(4): 13C4 Methionine label.</summary>
    UNIMOD_Label_13C_4_ = 100001266,

    /// <summary>Label:13C(4)+Oxidation: Oxidised 13C4 labelled Methionine.</summary>
    UNIMOD_Label_13C_4__Oxidation = 100001267,

    /// <summary>HCysThiolactone: N-Homocysteine thiolactone.</summary>
    UNIMOD_HCysThiolactone = 100001270,

    /// <summary>HCysteinyl: S-homocysteinylation.</summary>
    UNIMOD_HCysteinyl = 100001271,

    /// <summary>UgiJoullie: Side reaction of HisTag.</summary>
    UNIMOD_UgiJoullie = 100001276,

    /// <summary>Dipyridyl: Cys modified with dipy ligand.</summary>
    UNIMOD_Dipyridyl = 100001277,

    /// <summary>Furan: Chemical modification of the iodinated sites of thyroglobulin by Suzuki reaction.</summary>
    UNIMOD_Furan = 100001278,

    /// <summary>Difuran: Chemical modification of the diiodinated sites of thyroglobulin by Suzuki reaction.</summary>
    UNIMOD_Difuran = 100001279,

    /// <summary>BMP-piperidinol: 1-methyl-3-benzoyl-4-hydroxy-4-phenylpiperidine.</summary>
    UNIMOD_BMP_piperidinol = 100001281,

    /// <summary>UgiJoullieProGly: Side reaction of PG with Side chain of aspartic or glutamic acid.</summary>
    UNIMOD_UgiJoullieProGly = 100001282,

    /// <summary>UgiJoullieProGlyProGly: Side reaction of PGPG with Side chain of aspartic or glutamic acid.</summary>
    UNIMOD_UgiJoullieProGlyProGly = 100001283,

    /// <summary>IMEHex(2)NeuAc: Glycosylation with IME linked Hex(2) NeuAc.</summary>
    UNIMOD_IMEHex_2_NeuAc = 100001286,

    /// <summary>Arg-loss: Loss of arginine due to transpeptidation.</summary>
    UNIMOD_Arg_loss = 100001287,

    /// <summary>Arg: Addition of arginine due to transpeptidation.</summary>
    UNIMOD_Arg = 100001288,

    /// <summary>Butyryl: Butyryl.</summary>
    UNIMOD_Butyryl = 100001289,

    /// <summary>Dicarbamidomethyl: Double Carbamidomethylation.</summary>
    UNIMOD_Dicarbamidomethyl = 100001290,

    /// <summary>Dimethyl:2H(6): Dimethyl-Medium.</summary>
    UNIMOD_Dimethyl_2H_6_ = 100001291,

    /// <summary>GGQ: SUMOylation leaving GlyGlyGln.</summary>
    UNIMOD_GGQ = 100001292,

    /// <summary>QTGG: SUMOylation leaving GlnThrGlyGly.</summary>
    UNIMOD_QTGG = 100001293,

    /// <summary>Label:13C(3): 13C3 label for SILAC.</summary>
    UNIMOD_Label_13C_3_ = 100001296,

    /// <summary>Label:13C(3)15N(1): 13C3 15N1 label for SILAC.</summary>
    UNIMOD_Label_13C_3_15N_1_ = 100001297,

    /// <summary>Label:13C(4)15N(1): 13C4 15N1 label for SILAC.</summary>
    UNIMOD_Label_13C_4_15N_1_ = 100001298,

    /// <summary>Label:2H(10): 2H(10) label.</summary>
    UNIMOD_Label_2H_10_ = 100001299,

    /// <summary>Label:2H(4)13C(1): Label:2H(4)13C(1).</summary>
    UNIMOD_Label_2H_4_13C_1_ = 100001300,

    /// <summary>Lys: Addition of lysine due to transpeptidation.</summary>
    UNIMOD_Lys = 100001301,

    /// <summary>mTRAQ:13C(6)15N(2): MTRAQ heavy.</summary>
    UNIMOD_mTRAQ_13C_6_15N_2_ = 100001302,

    /// <summary>NeuAc: N-acetyl neuraminic acid.</summary>
    UNIMOD_NeuAc = 100001303,

    /// <summary>NeuGc: N-glycoyl neuraminic acid.</summary>
    UNIMOD_NeuGc = 100001304,

    /// <summary>Propyl: Propyl.</summary>
    UNIMOD_Propyl = 100001305,

    /// <summary>Propyl:2H(6): Propyl:2H(6).</summary>
    UNIMOD_Propyl_2H_6_ = 100001306,

    /// <summary>Propiophenone: Propiophenone.</summary>
    UNIMOD_Propiophenone = 100001310,

    /// <summary>AhaEthynylUridine: 1,3 dipolar cycloaddition of ethynyluridine to azidohomoalanine.</summary>
    UNIMOD_AhaEthynylUridine = 100001311,

    /// <summary>Delta:H(6)C(3)O(1): Reduced acrolein addition +58.</summary>
    UNIMOD_Delta_H_6_C_3_O_1_ = 100001312,

    /// <summary>Delta:H(8)C(6)O(1): Reduced acrolein addition +96.</summary>
    UNIMOD_Delta_H_8_C_6_O_1_ = 100001313,

    /// <summary>biotinAcrolein298: Biotin hydrazide labeled acrolein addition +298.</summary>
    UNIMOD_biotinAcrolein298 = 100001314,

    /// <summary>MM-diphenylpentanone: 3-methyl-5-(methylamino)-1,3-diphenylpentan-1-one.</summary>
    UNIMOD_MM_diphenylpentanone = 100001315,

    /// <summary>EHD-diphenylpentanone: 2-ethyl-3-hydroxy-1,3-diphenylpentan-1-one.</summary>
    UNIMOD_EHD_diphenylpentanone = 100001317,

    /// <summary>Bipyridine: Cys modified with bipy ligand.</summary>
    UNIMOD_Bipyridine = 100001318,

    /// <summary>VinylPMS: Serine modified with p-vinylphenylmethylsulfonyl group.</summary>
    UNIMOD_VinylPMS = 100001319,

    /// <summary>Biotin:Thermo-21901+2H2O: Maleimide-Biotin + 2Water.</summary>
    UNIMOD_Biotin_Thermo_21901_2H2O = 100001320,

    /// <summary>DiLeu4plex115: Accurate mass for DiLeu 115 isobaric tag.</summary>
    UNIMOD_DiLeu4plex115 = 100001321,

    /// <summary>DiLeu4plex: Accurate mass for DiLeu 116 isobaric tag.</summary>
    UNIMOD_DiLeu4plex = 100001322,

    /// <summary>DiLeu4plex117: Accurate mass for DiLeu 117 isobaric tag.</summary>
    UNIMOD_DiLeu4plex117 = 100001323,

    /// <summary>DiLeu4plex118: Accurate mass for DiLeu 118 isobaric tag.</summary>
    UNIMOD_DiLeu4plex118 = 100001324,

    /// <summary>NEMsulfur: N-ethylmaleimideSulfur.</summary>
    UNIMOD_NEMsulfur = 100001326,

    /// <summary>SulfurDioxide: SulfurDioxide.</summary>
    UNIMOD_SulfurDioxide = 100001327,

    /// <summary>NEMsulfurWater: N-ethylmaleimideSulfurWater.</summary>
    UNIMOD_NEMsulfurWater = 100001328,

    /// <summary>unit: A unit of measurement is a standardized quantity of a physical quality.</summary>
    UO_unit = 200000000,

    /// <summary>length unit: A unit which is a standard measure of the distance between two points.</summary>
    UO_length_unit = 200000001,

    /// <summary>mass unit: A unit which is a standard measure of the amount of matter/energy of a physical object.</summary>
    UO_mass_unit = 200000002,

    /// <summary>time unit: A unit which is a standard measure of the dimension in which events occur in sequence.</summary>
    UO_time_unit = 200000003,

    /// <summary>electric current unit: A unit which is a standard measure of the flow of electric charge.</summary>
    UO_electric_current_unit = 200000004,

    /// <summary>temperature unit: A unit which is a standard measure of the average kinetic energy of the particles in a sample of matter.</summary>
    UO_temperature_unit = 200000005,

    /// <summary>substance unit: A unit which is a standardised quantity of an element or compound with uniform composition.</summary>
    UO_substance_unit = 200000006,

    /// <summary>luminous intensity unit: A unit which is a standard measure of the wavelength-weighted power emitted by a light source in a particular direction.</summary>
    UO_luminous_intensity_unit = 200000007,

    /// <summary>meter: A length unit which is equal to the length of the path traveled by light in vacuum during a time interval of 1/299 792 458 of a second.</summary>
    UO_meter = 200000008,

    /// <summary>kilogram: A mass unit which is equal to the mass of the International Prototype Kilogram kept by the BIPM at Svres, France.</summary>
    UO_kilogram = 200000009,

    /// <summary>second: A time unit which is equal to the duration of 9 192 631 770 periods of the radiation corresponding to the transition between the two hyperfine levels of the ground state of the caesium 133 atom.</summary>
    UO_second = 200000010,

    /// <summary>ampere: An electric current unit which is equal to the constant current which, if maintained in two straight parallel conductors of infinite length, of negligible circular cross-section, and placed 1 m apart in vacuum, would produce between these conductors a force equal to 2 x 10^[-7] newton per meter of length.</summary>
    UO_ampere = 200000011,

    /// <summary>kelvin: A thermodynamic temperature unit which is equal to the fraction 1/273.16 of the thermodynamic temperature of the triple point of water.</summary>
    UO_kelvin = 200000012,

    /// <summary>mole: A substance unit which is equal to the amount of substance of a molecular system which contains as many elementary entities as there are atoms in 0.012 kilogram of carbon 12.</summary>
    UO_mole = 200000013,

    /// <summary>candela: A luminous intensity unit which equal to the luminous intensity, in a given direction, of a source that emits monochromatic radiation of frequency 540 x 1012 hertz and that has a radiant intensity in that direction of 1/683 watt per steradian.</summary>
    UO_candela = 200000014,

    /// <summary>centimeter: A length unit which is equal to one hundredth of a meter or 10^[-2] m.</summary>
    UO_centimeter = 200000015,

    /// <summary>millimeter: A length unit which is equal to one thousandth of a meter or 10^[-3] m.</summary>
    UO_millimeter = 200000016,

    /// <summary>micrometer: A length unit which is equal to one millionth of a meter or 10^[-6] m.</summary>
    UO_micrometer = 200000017,

    /// <summary>nanometer: A length unit which is equal to one thousandth of one millionth of a meter or 10^[-9] m.</summary>
    UO_nanometer = 200000018,

    /// <summary>angstrom: A length unit which is equal to 10 [-10] m.</summary>
    UO_angstrom = 200000019,

    /// <summary>picometer: A length unit which is equal to 10^[-12] m.</summary>
    UO_picometer = 200000020,

    /// <summary>gram: A mass unit which is equal to one thousandth of a kilogram or 10^[-3] kg.</summary>
    UO_gram = 200000021,

    /// <summary>milligram: A mass unit which is equal to one thousandth of a gram or 10^[-3] g.</summary>
    UO_milligram = 200000022,

    /// <summary>microgram: A mass unit which is equal to one millionth of a gram or 10^[-6] g.</summary>
    UO_microgram = 200000023,

    /// <summary>nanogram: A mass unit which is equal to one thousandth of one millionth of a gram or 10^[-9] g.</summary>
    UO_nanogram = 200000024,

    /// <summary>picogram: A mass unit which is equal to 10^[-12] g.</summary>
    UO_picogram = 200000025,

    /// <summary>femtogram: A mass unit which is equal to 10^[-15] g.</summary>
    UO_femtogram = 200000026,

    /// <summary>degree celsius: A temperature unit which is equal to one Kelvin degree. However, they have their zeros at different points. The Centigrade scale has its zero at 273.15 K.</summary>
    UO_degree_celsius = 200000027,

    /// <summary>millisecond: A time unit which is equal to one thousandth of a second or 10^[-3] s.</summary>
    UO_millisecond = 200000028,

    /// <summary>microsecond: A time unit which is equal to one millionth of a second or 10^[-6] s.</summary>
    UO_microsecond = 200000029,

    /// <summary>picosecond: A time unit which is equal to 10^[-12] s.</summary>
    UO_picosecond = 200000030,

    /// <summary>minute: A time unit which is equal to 60 seconds.</summary>
    UO_minute = 200000031,

    /// <summary>hour: A time unit which is equal to 3600 seconds or 60 minutes.</summary>
    UO_hour = 200000032,

    /// <summary>day: A time unit which is equal to 24 hours.</summary>
    UO_day = 200000033,

    /// <summary>week: A time unit which is equal to 7 days.</summary>
    UO_week = 200000034,

    /// <summary>month: A time unit which is approximately equal to the length of time of one of cycle of the moon's phases which in science is taken to be equal to 30 days.</summary>
    UO_month = 200000035,

    /// <summary>year: A time unit which is equal to 12 months which is science is taken to be equal to 365.25 days.</summary>
    UO_year = 200000036,

    /// <summary>milliampere: An electric current unit current which is equal to one thousandth of an ampere or 10^[-3] A.</summary>
    UO_milliampere = 200000037,

    /// <summary>microampere: An electric current unit current which is equal to one millionth of an ampere or 10^[-6] A.</summary>
    UO_microampere = 200000038,

    /// <summary>micromole: A substance unit equal to a millionth of a mol or 10^[-6] mol.</summary>
    UO_micromole = 200000039,

    /// <summary>millimole: A substance unit equal to a thousandth of a mol or 10^[-3] mol.</summary>
    UO_millimole = 200000040,

    /// <summary>nanomole: A substance unit equal to one thousandth of one millionth of a mole or 10^[-9] mol.</summary>
    UO_nanomole = 200000041,

    /// <summary>picomole: A substance unit equal to 10^[-12] mol.</summary>
    UO_picomole = 200000042,

    /// <summary>femtomole: A substance unit equal to 10^[-15] mol.</summary>
    UO_femtomole = 200000043,

    /// <summary>attomole: A substance unit equal to 10^[-18] mol.</summary>
    UO_attomole = 200000044,

    /// <summary>base unit: A unit which is one of a particular measure to which all measures of that type can be related.</summary>
    UO_base_unit = 200000045,

    /// <summary>derived unit: A unit which is derived from base units.</summary>
    UO_derived_unit_OBSOLETE = 200000046,

    /// <summary>area unit: A unit which is a standard measure of the amount of a 2-dimensional flat surface.</summary>
    UO_area_unit = 200000047,

    /// <summary>acceleration unit: A unit which is a standard measure of the rate of change of velocity in either speed or direction.</summary>
    UO_acceleration_unit = 200000048,

    /// <summary>angular velocity unit: A unit which is a standard measure of the rate of angular movement about an axis; the angle rotated in a given time.</summary>
    UO_angular_velocity_unit = 200000049,

    /// <summary>angular acceleration unit: A unit which is a standard measure of the rate of change of angular velocity.</summary>
    UO_angular_acceleration_unit = 200000050,

    /// <summary>concentration unit: A unit which represents a standard measurement of how much of a given substance there is mixed with another substance.</summary>
    UO_concentration_unit = 200000051,

    /// <summary>mass density unit: A density unit which is a standard measure of the mass of a substance in a given volume.</summary>
    UO_mass_density_unit = 200000052,

    /// <summary>luminance unit: A unit which is a standard measure of the luminous intensity impinging on a given area.</summary>
    UO_luminance_unit = 200000053,

    /// <summary>area density unit: A density unit which is a standard measure of the mass exerting an influence on a given area.</summary>
    UO_area_density_unit = 200000054,

    /// <summary>molar mass unit: A unit which is a standard measure of the mass of a homogeneous substance containing 6.02 x 1023 atoms or molecules.</summary>
    UO_molar_mass_unit = 200000055,

    /// <summary>molar volume unit: A unit which is a standard measure of the volume of a homogeneous substance containing 6.02 x 1023 atoms or molecules.</summary>
    UO_molar_volume_unit = 200000056,

    /// <summary>momentum unit: A unit which is a standard measure of the quantity of motion measured by the product of mass and velocity.</summary>
    UO_momentum_unit = 200000057,

    /// <summary>rotational frequency unit: A unit which is a standard measure of the number of rotations in a given time.</summary>
    UO_rotational_frequency_unit = 200000058,

    /// <summary>specific volume unit: A unit which is a standard measure of the volume of a given mass of substance (the reciprocal of density).</summary>
    UO_specific_volume_unit = 200000059,

    /// <summary>speed/velocity unit: A unit which is a standard measure of the rate of movement. Speed is measured in the same physical units of measurement as velocity, but does not contain the element of direction that velocity has. Speed is thus the magnitude component of velocity.</summary>
    UO_speed_velocity_unit = 200000060,

    /// <summary>unit of molarity: A concentration unit which is a standard measure of the number of moles of a given substance per liter of solution.</summary>
    UO_unit_of_molarity = 200000061,

    /// <summary>molar: A unit of concentration which expresses a concentration of 1 mole of solute per liter of solution (mol/L).</summary>
    UO_molar = 200000062,

    /// <summary>millimolar: A unit of molarity which is equal to one thousandth of a molar or 10^[-3] M.</summary>
    UO_millimolar = 200000063,

    /// <summary>micromolar: A unit of molarity which is equal to one millionth of a molar or 10^[-6] M.</summary>
    UO_micromolar = 200000064,

    /// <summary>nanomolar: A unit of molarity which is equal to one thousandth of one millionth of a molar or 10^[-9] M.</summary>
    UO_nanomolar = 200000065,

    /// <summary>picomolar: A unit of molarity which is equal to 10^[-12] M.</summary>
    UO_picomolar = 200000066,

    /// <summary>unit of molality: A concentration unit which is a standard measure of the number of moles of a given substance per kilogram of solvent.</summary>
    UO_unit_of_molality = 200000067,

    /// <summary>molal: A unit of concentration which expresses a concentration of a solution of 1 mole per kilogram of solvent (mol/kg).</summary>
    UO_molal = 200000068,

    /// <summary>millimolal: A molality unit which is equal to one thousandth of a molal or 10^[-3] m.</summary>
    UO_millimolal = 200000069,

    /// <summary>micromolal: A molality unit which is equal to one millionth of a molal or 10^[-6] m.</summary>
    UO_micromolal = 200000070,

    /// <summary>nanomolal: A molality unit which is equal to one thousandth of one millionth of a molal or 10^[-9] m.</summary>
    UO_nanomolal = 200000071,

    /// <summary>picomolal: A molality unit which is equal to 10^[-12] m.</summary>
    UO_picomolal = 200000072,

    /// <summary>femtomolar: A unit of molarity which is equal to 10^[-15] M.</summary>
    UO_femtomolar = 200000073,

    /// <summary>unit of normality: A unit of concentration which highlights the chemical nature of salts.</summary>
    UO_unit_of_normality = 200000074,

    /// <summary>normal: A unit of concentration which is one gram equivalent of a solute per liter of solution. A gram equivalent weight or equivalent is a measure of the reactive capacity of a given molecule.</summary>
    UO_normal = 200000075,

    /// <summary>mole fraction: A concentration unit which denotes the number of moles of solute as a proportion of the total number of moles in a solution.</summary>
    UO_mole_fraction = 200000076,

    /// <summary>meter per second per second: An acceleration unit which is equal to the acceleration an object changing its velocity by 1meter/s over a time period that equals one second.</summary>
    UO_meter_per_second_per_second = 200000077,

    /// <summary>radian per second per second: An angular unit acceleration which is equal to the angular acceleration of an object changing its angular velocity by 1rad/s over a time period that equals one second.</summary>
    UO_radian_per_second_per_second = 200000078,

    /// <summary>radian per second: An angular unit velocity which is equal to about 9.54930 rpm (revolutions per minute).</summary>
    UO_radian_per_second = 200000079,

    /// <summary>square meter: An area unit which is equal to an area enclosed by a square with sides each 1 meter long.</summary>
    UO_square_meter = 200000080,

    /// <summary>square centimeter: An area unit which is equal to one thousand of square meter or 10^[-3] m^[2].</summary>
    UO_square_centimeter = 200000081,

    /// <summary>square millimeter: An area unit which is equal to one millionth of a square meter or 10^[-6] m^[2].</summary>
    UO_square_millimeter = 200000082,

    /// <summary>kilogram per cubic meter: A mass unit density which is equal to mass of an object in kilograms divided by the volume in cubic meters.</summary>
    UO_kilogram_per_cubic_meter = 200000083,

    /// <summary>gram per cubic centimeter: A mass unit density which is equal to mass of an object in grams divided by the volume in cubic centimeters.</summary>
    UO_gram_per_cubic_centimeter = 200000084,

    /// <summary>candela per square meter: A luminance unit which is equal to a luminous intensity of one candela radiating from a surface whose area is one square meter.</summary>
    UO_candela_per_square_meter = 200000085,

    /// <summary>kilogram per square meter: An area density unit which is equal to the mass of an object in kilograms divided by the surface area in meters squared.</summary>
    UO_kilogram_per_square_meter = 200000086,

    /// <summary>kilogram per mole: A molar mass unit which is equal to one kilogram of mass of one mole of chemical element or chemical compound.</summary>
    UO_kilogram_per_mole = 200000087,

    /// <summary>gram per mole: A molar mass unit which is equal to one gram of mass of one mole of chemical element or chemical compound.</summary>
    UO_gram_per_mole = 200000088,

    /// <summary>cubic meter per mole: A molar volume unit which is equal to 1 cubic meter occupied by one mole of a substance in the form of a solid, liquid, or gas.</summary>
    UO_cubic_meter_per_mole = 200000089,

    /// <summary>cubic centimeter per mole: A molar volume unit which is equal to 1 cubic centimeter occupied by one mole of a substance in the form of a solid, liquid, or gas.</summary>
    UO_cubic_centimeter_per_mole = 200000090,

    /// <summary>kilogram meter per second: A momentum unit which is equal to the momentum of a one kilogram mass object with a speed of one meter per second.</summary>
    UO_kilogram_meter_per_second = 200000091,

    /// <summary>turns per second: A rotational frequency unit which is equal to the number complete turn in a period of time that equals to 1 second.</summary>
    UO_turns_per_second = 200000092,

    /// <summary>cubic meter per kilogram: A specific volume unit which is equal to one cubic meter volume occupied by one kilogram of a particular substance.</summary>
    UO_cubic_meter_per_kilogram = 200000093,

    /// <summary>meter per second: A speed/velocity unit which is equal to the speed of an object traveling 1 meter distance in one second.</summary>
    UO_meter_per_second = 200000094,

    /// <summary>volume unit: A unit which is a standard measure of the amount of space occupied by any substance, whether solid, liquid, or gas.</summary>
    UO_volume_unit = 200000095,

    /// <summary>cubic meter: A volume unit which is equal to the volume of a cube with edges one meter in length. One cubic meter equals to 1000 liters.</summary>
    UO_cubic_meter = 200000096,

    /// <summary>cubic centimeter: A volume unit which is equal to one millionth of a cubic meter or 10^[-9] m^[3], or to 1 ml.</summary>
    UO_cubic_centimeter = 200000097,

    /// <summary>milliliter: A volume unit which is equal to one thousandth of a liter or 10^[-3] L, or to 1 cubic centimeter.</summary>
    UO_milliliter = 200000098,

    /// <summary>liter: A volume unit which is equal to one thousandth of a cubic meter or 10^[-3] m^[3], or to 1 decimeter.</summary>
    UO_liter = 200000099,

    /// <summary>cubic decimeter: A volume unit which is equal to one thousand of a cubic meter or 10^[-3] m^[3], or to 1 L.</summary>
    UO_cubic_decimeter = 200000100,

    /// <summary>microliter: A volume unit which is equal to one millionth of a liter or 10^[-6] L.</summary>
    UO_microliter = 200000101,

    /// <summary>nanoliter: A volume unit which is equal to one thousandth of one millionth of a liter or 10^[-9] L.</summary>
    UO_nanoliter = 200000102,

    /// <summary>picoliter: A volume unit which is equal to 10^[-12] L.</summary>
    UO_picoliter = 200000103,

    /// <summary>femtoliter: A volume unit which is equal to 10^[-15] L.</summary>
    UO_femtoliter = 200000104,

    /// <summary>frequency unit: A unit which is a standard measure of the number of repetitive actions in a particular time.</summary>
    UO_frequency_unit = 200000105,

    /// <summary>hertz: A frequency unit which is equal to 1 complete cycle of a recurring phenomenon in 1 second.</summary>
    UO_hertz = 200000106,

    /// <summary>force unit: A unit which is a standard measure of the force is applied when a mass is accelerated.</summary>
    UO_force_unit = 200000107,

    /// <summary>newton: A force unit which is equal to the force required to cause an acceleration of 1m/s2 of a mass of 1 Kg in the direction of the force.</summary>
    UO_newton = 200000108,

    /// <summary>pressure unit: A unit which is a standard measure of the force applied to a given area.</summary>
    UO_pressure_unit = 200000109,

    /// <summary>pascal: A pressure unit which is equal to the pressure or stress on a surface caused by a force of 1 newton spread over a surface of 1 m^[2].</summary>
    UO_pascal = 200000110,

    /// <summary>energy unit: A unit which is a standard measure of the work done by a certain force (gravitational, electric, magnetic, force of inertia, etc).</summary>
    UO_energy_unit = 200000111,

    /// <summary>joule: An energy unit which is equal to the energy required when a force of 1 newton moves an object 1 meter in the direction of the force.</summary>
    UO_joule = 200000112,

    /// <summary>power unit: A unit which is a standard measure power or the rate of doing work.</summary>
    UO_power_unit = 200000113,

    /// <summary>watt: A power unit which is equal to the power used when work is done at the rate of 1 joule per second.</summary>
    UO_watt = 200000114,

    /// <summary>illuminance unit: A unit which is a standard measure of the luminous flux incident on a unit area.</summary>
    UO_illuminance_unit = 200000115,

    /// <summary>lux: An illuminance unit which is equal to the illuminance produced by 1 lumen evenly spread over an area 1 m^[2].</summary>
    UO_lux = 200000116,

    /// <summary>luminous flux unit: A unit which is a standard measure of the flow of radiant energy.</summary>
    UO_luminous_flux_unit = 200000117,

    /// <summary>lumen: A luminous flux unit which is equal to the luminous flux emitted into 1 steradian by a point source of 1 candela.</summary>
    UO_lumen = 200000118,

    /// <summary>catalytic activity unit: A unit which is a standard measure of the amount of the action of a catalyst.</summary>
    UO_catalytic_activity_unit = 200000119,

    /// <summary>katal: A catalytic unit activity which is equal to the activity of a catalyst in moles per second, such as the amount of an enzyme needed to transform one mole of substrate per second.</summary>
    UO_katal = 200000120,

    /// <summary>angle unit: A unit which is a standard measure of the figure or space formed by the junction of two lines or planes.</summary>
    UO_angle_unit = 200000121,

    /// <summary>plane angle unit: A unit which is a standard measure of the angle formed by two straight lines in the same plane.</summary>
    UO_plane_angle_unit = 200000122,

    /// <summary>radian: A plane angle unit which is equal to the angle subtended at the center of a circle by an arc equal in length to the radius of the circle, approximately 57 degrees 17 minutes and 44.6 seconds.</summary>
    UO_radian = 200000123,

    /// <summary>solid angle unit: A unit which is a standard measure of the angle formed by three or more planes intersecting at a common point.</summary>
    UO_solid_angle_unit = 200000124,

    /// <summary>steradian: A solid angle unit which is equal to the solid angle subtended at the center of a sphere by an area on the surface of the sphere that is equal to the radius squared.</summary>
    UO_steradian = 200000125,

    /// <summary>radiation unit: A unit which is a standard measure of the amount of radiation emitted by a given radiation source as well as the amount of radiation absorbed or deposited in a specific material by a radiation source.</summary>
    UO_radiation_unit = 200000127,

    /// <summary>activity (of a radionuclide) unit: A unit which is a standard measure of the transformation (disintegration) rate of a radioactive substance.</summary>
    UO_activity__of_a_radionuclide__unit = 200000128,

    /// <summary>absorbed dose unit: A unit which is a standard measure of the energy imparted by ionizing radiation to unit mass of matter such as tissue.</summary>
    UO_absorbed_dose_unit = 200000129,

    /// <summary>dose equivalent unit: A unit which is a standard measure of the expression of dose in terms of its biological effect.</summary>
    UO_dose_equivalent_unit = 200000130,

    /// <summary>exposure unit: A unit which is a standard measure of the quantity that expresses the ability of radiation to ionize air and thereby create electric charges which can be collected and measured.</summary>
    UO_exposure_unit = 200000131,

    /// <summary>becquerel: An activity (of a radionuclide) unit which is equal to the activity of a quantity of radioactive material in which one nucleus decays per second or there is one atom disintegration per second (dps).</summary>
    UO_becquerel = 200000132,

    /// <summary>curie: An activity (of a radionuclide) unit which is equal to the activity of a quantity of radioactive material in which there are 3.7 x 10^[10] atom disintegration per second (dps).</summary>
    UO_curie = 200000133,

    /// <summary>gray: An absorbed dose unit which is equal to the absorption of one joule of radiation energy by one kilogram of matter.</summary>
    UO_gray = 200000134,

    /// <summary>rad: An absorbed dose unit which is equal to 0.01 gray (Gy).</summary>
    UO_rad = 200000135,

    /// <summary>roentgen: An exposure unit which is equal to the amount of radiation required to liberate positive and negative charges of one electrostatic unit of charge in 1 cm^[3] of air at standard temperature and pressure (STP). This corresponds to the generation of approximately 2.0810^[9] ion pairs.</summary>
    UO_roentgen = 200000136,

    /// <summary>sievert: A dose equivalent unit which is equal to the absorption of one joule of radiation energy by one kilogram of matter.</summary>
    UO_sievert = 200000137,

    /// <summary>millisievert: A dose equivalent unit which is equal to one thousandth of a sievert or 10^[-3] Sv.</summary>
    UO_millisievert = 200000138,

    /// <summary>microsievert: A dose equivalent unit which is equal to one millionth of a sievert or 10^[-6] Sv.</summary>
    UO_microsievert = 200000139,

    /// <summary>Roentgen equivalent man: A dose equivalent unit which when multiplied by hundred is equal to one sievert or 1 Sv. 1 Sv is equal to 100 rem.</summary>
    UO_Roentgen_equivalent_man = 200000140,

    /// <summary>microgray: An absorbed dose unit which is equal to one millionth of a gray or 10^[-6] Gy.</summary>
    UO_microgray = 200000141,

    /// <summary>milligray: An absorbed dose unit which is equal to one thousandth of a gray or 10^[-3] Gy.</summary>
    UO_milligray = 200000142,

    /// <summary>nanogray: An absorbed dose unit which is equal to one thousandth of a millionth of a gray or 10^[-9] Gy.</summary>
    UO_nanogray = 200000143,

    /// <summary>nanosievert: A dose equivalent unit which is equal to one thousandth of a millionth of a sievert or 10^[-9] Sv.</summary>
    UO_nanosievert = 200000144,

    /// <summary>millicurie: An activity (of a radionuclide) unit which is equal to one thousandth of a curie or 10^[-3] Ci.</summary>
    UO_millicurie = 200000145,

    /// <summary>microcurie: An activity (of a radionuclide) unit which is equal to one millionth of a curie or 10^[-6] Ci.</summary>
    UO_microcurie = 200000146,

    /// <summary>disintegrations per minute: An activity (of a radionuclide) unit which is equal to the activity of a quantity of radioactive material in which one nucleus decays per minute or there is one atom disintegration per minute.</summary>
    UO_disintegrations_per_minute = 200000147,

    /// <summary>counts per minute: An activity (of a radionuclide) unit which is equal to the number of light emissions produced by ionizing radiation in one minute.</summary>
    UO_counts_per_minute = 200000148,

    /// <summary>nanosecond: A time unit which is equal to one thousandth of one millionth of a second or 10^[-9] s.</summary>
    UO_nanosecond = 200000150,

    /// <summary>century: A time unit which is equal to 100 years.</summary>
    UO_century = 200000151,

    /// <summary>half life: A time unit which represents the period over which the activity or concentration of a specified chemical or element falls to half its original activity or concentration.</summary>
    UO_half_life = 200000152,

    /// <summary>foot candle: An illuminance unit which is equal to the illuminance produced by 1 lumen evenly spread over an area 1 foot^[2]. One footcandle is equal to 10.76 lux.</summary>
    UO_foot_candle = 200000153,

    /// <summary>irradiance unit: A unit which is a standard measure of the power of electromagnetic radiation at a surface, per unit area.</summary>
    UO_irradiance_unit = 200000154,

    /// <summary>watt per square meter: An irradiance unit which is equal to 1 watt of radiant power incident per one square meter surface area.</summary>
    UO_watt_per_square_meter = 200000155,

    /// <summary>einstein per square meter per second: An irradiance unit which is equal to one einstein per square meter per second. One einstein is one mole of photons, regardless of their frequency. Therefore, the number of photons in an einstein is Avogadro's number.</summary>
    UO_einstein_per_square_meter_per_second = 200000156,

    /// <summary>light unit: A unit which is a standard measure of the intensity of light.</summary>
    UO_light_unit = 200000157,

    /// <summary>watt per steradian per square meter: A radiance unit which is equal to one watt of radiant power incident per steradian solid angle per one square meter projected area of the source, as viewed from the given direction.</summary>
    UO_watt_per_steradian_per_square_meter = 200000158,

    /// <summary>radiant intensity unit: A unit which is a standard measure of the intensity of electromagnetic radiation.</summary>
    UO_radiant_intensity_unit = 200000159,

    /// <summary>microeinstein per square meter per second: An irradiance unit which is equal to one microeinstein per square meter per second or 10^[-6] microeinstein/sm^[2].</summary>
    UO_microeinstein_per_square_meter_per_second = 200000160,

    /// <summary>radiance unit: A unit which is a standard measure of the power of electromagnetic radiation through space or through a material medium in the form of electromagnetic waves.</summary>
    UO_radiance_unit = 200000161,

    /// <summary>watt per steradian: A radiant intensity unit which is equal to one kilogram meter squared per second cubed per steradian.</summary>
    UO_watt_per_steradian = 200000162,

    /// <summary>mass percentage: A dimensionless concentration unit which denotes the mass of a substance in a mixture as a percentage of the mass of the entire mixture.</summary>
    UO_mass_percentage = 200000163,

    /// <summary>mass volume percentage: A dimensionless concentration unit which denotes the mass of the substance in a mixture as a percentage of the volume of the entire mixture.</summary>
    UO_mass_volume_percentage = 200000164,

    /// <summary>volume percentage: A dimensionless concentration unit which denotes the volume of the solute in mL per 100 mL of the resulting solution.</summary>
    UO_volume_percentage = 200000165,

    /// <summary>parts per notation unit: A dimensionless concentration notation which describes the amount of one substance in another. It is the ratio of the amount of the substance of interest to the amount of that substance plus the amount of the substance.</summary>
    UO_parts_per_notation_unit = 200000166,

    /// <summary>parts per hundred: A dimensionless concentration notation which denotes the amount of a given substance in a total amount of 100 regardless of the units of measure as long as they are the same.</summary>
    UO_parts_per_hundred = 200000167,

    /// <summary>parts per thousand: A dimensionless concentration notation which denotes the amount of a given substance in a total amount of 1000 regardless of the units of measure as long as they are the same.</summary>
    UO_parts_per_thousand = 200000168,

    /// <summary>parts per million: A dimensionless concentration notation which denotes the amount of a given substance in a total amount of 1,000,000 regardless of the units of measure used as long as they are the same or 1 part in 10^[6].</summary>
    UO_parts_per_million = 200000169,

    /// <summary>parts per billion: A dimensionless concentration notation which denotes the amount of a given substance in a total amount of 1,000,000,000 regardless of the units of measure as long as they are the same or 1 part in 10^[9].</summary>
    UO_parts_per_billion = 200000170,

    /// <summary>parts per trillion: A dimensionless concentration notation which denotes the amount of a given substance in a total amount of 1,000,000,000 regardless of the units of measure used as long as they are the same or 1 part in 10^[12].</summary>
    UO_parts_per_trillion = 200000171,

    /// <summary>parts per quadrillion: A dimensionless concentration notation which denotes the amount of a given substance in a total amount of 1,000,000,000,000 regardless of the units of measure used as long as they are the same or 1 part in 10^[15].</summary>
    UO_parts_per_quadrillion = 200000172,

    /// <summary>gram per milliliter: A mass unit density which is equal to mass of an object in grams divided by the volume in milliliter.</summary>
    UO_gram_per_milliliter = 200000173,

    /// <summary>kilogram per liter: A mass unit density which is equal to mass of an object in kilograms divided by the volume in liters.</summary>
    UO_kilogram_per_liter = 200000174,

    /// <summary>gram per liter: A mass unit density which is equal to mass of an object in grams divided by the volume in liters.</summary>
    UO_gram_per_liter = 200000175,

    /// <summary>milligram per milliliter: A mass unit density which is equal to mass of an object in milligrams divided by the volume in milliliters.</summary>
    UO_milligram_per_milliliter = 200000176,

    /// <summary>unit per volume unit: A concentration unit which is a standard measure of the number of units, as an agreed arbitrary amount, of a given substance per a specific volume of solution.</summary>
    UO_unit_per_volume_unit = 200000177,

    /// <summary>unit per milliliter: A unit per milliliter unit which is equal to one unit of an agreed arbitrary amount per one milliliter.</summary>
    UO_unit_per_milliliter = 200000178,

    /// <summary>unit per liter: A unit per milliliter unit which is equal to one unit of an agreed arbitrary amount per one liter.</summary>
    UO_unit_per_liter = 200000179,

    /// <summary>mass per unit volume: A concentration unit which is a standard measure of the mass of a substance in a given volume (density).</summary>
    UO_mass_per_unit_volume = 200000180,

    /// <summary>enzyme unit: A catalytic unit activity which is equal to the amount of the enzyme that catalyzes the conversion of 1 micro mole of substrate per minute.</summary>
    UO_enzyme_unit = 200000181,

    /// <summary>density unit: A unit which is a standard measure of the influence exerted by some mass.</summary>
    UO_density_unit = 200000182,

    /// <summary>linear density unit: A density unit which is a standard measure of the mass exerting an influence on a one-dimensional object.</summary>
    UO_linear_density_unit = 200000183,

    /// <summary>kilogram per meter: An area density unit which is equal to the mass of an object in kilograms divided by one meter.</summary>
    UO_kilogram_per_meter = 200000184,

    /// <summary>degree: A plane angle unit which is equal to 1/360 of a full rotation or 1.7453310^[-2] rad.</summary>
    UO_degree = 200000185,

    /// <summary>dimensionless unit: A unit which is a standard measure of physical quantity consisting of only a numerical number without any units.</summary>
    UO_dimensionless_unit = 200000186,

    /// <summary>percent: A dimensionless ratio unit which denotes numbers as fractions of 100.</summary>
    UO_percent = 200000187,

    /// <summary>pi: A dimensionless unit which denoted an irrational real number, approximately equal to 3.14159 which is the ratio of a circle's circumference to its diameter in Euclidean geometry.</summary>
    UO_pi = 200000188,

    /// <summary>count unit: A dimensionless unit which denotes a simple count of things.</summary>
    UO_count_unit = 200000189,

    /// <summary>ratio: A dimensionless unit which denotes an amount or magnitude of one quantity relative to another.</summary>
    UO_ratio = 200000190,

    /// <summary>fraction: A dimensionless ratio unit which relates the part (the numerator) to the whole (the denominator).</summary>
    UO_fraction = 200000191,

    /// <summary>molecule count: A dimensionless count unit which denotes the number of molecules.</summary>
    UO_molecule_count = 200000192,

    /// <summary>purity percentage: A dimensionless percent unit which denotes the homogeneity of a biomaterial.</summary>
    UO_purity_percentage = 200000193,

    /// <summary>confluence percentage: A dimensionless percent unit which denotes the density of an attached or monolayer culture (e.g., cell culture).</summary>
    UO_confluence_percentage = 200000194,

    /// <summary>degree fahrenheit: A temperature unit which is equal to 5/9ths of a kelvin. Negative 40 degrees Fahrenheit is equal to negative 40 degrees Celsius.</summary>
    UO_degree_fahrenheit = 200000195,

    /// <summary>pH: A dimensionless concentration notation which denotes the acidity of a solution in terms of activity of hydrogen ions (H+).</summary>
    UO_pH = 200000196,

    /// <summary>liter per kilogram: A specific volume unit which is equal to one liter volume occupied by one kilogram of a particular substance.</summary>
    UO_liter_per_kilogram = 200000197,

    /// <summary>milliliter per kilogram: A specific volume unit which is equal to a thousandth of a liter per kilogram or 10^[-3] l/kg.</summary>
    UO_milliliter_per_kilogram = 200000198,

    /// <summary>microliter per kilogram: A specific volume unit which is equal to one millionth of a liter per kilogram or 10^[-6] l/kg.</summary>
    UO_microliter_per_kilogram = 200000199,

    /// <summary>cell concentration unit: A concentration unit which denotes the average cell number in a given volume.</summary>
    UO_cell_concentration_unit = 200000200,

    /// <summary>cells per milliliter: A unit of cell concentration which is equal to one cell in a volume of 1 milliliter.</summary>
    UO_cells_per_milliliter = 200000201,

    /// <summary>catalytic (activity) concentration unit: A concentration unit which is a standard measure of the amount of the action of a catalyst in a given volume.</summary>
    UO_catalytic__activity__concentration_unit = 200000202,

    /// <summary>katal per cubic meter: A catalytic (activity) concentration unit which is equal to 1 katal activity of a catalyst in a given volume of one cubic meter.</summary>
    UO_katal_per_cubic_meter = 200000203,

    /// <summary>katal per liter: A catalytic (activity) concentration unit which is equal to 1 katal activity of a catalyst in a given volume of one thousandth of a cubic meter.</summary>
    UO_katal_per_liter = 200000204,

    /// <summary>volume per unit volume: A dimensionless concentration unit which denotes the given volume of the solute in the total volume of the resulting solution.</summary>
    UO_volume_per_unit_volume = 200000205,

    /// <summary>milliliter per cubic meter: A volume per unit volume unit which is equal to one millionth of a liter of solute in one cubic meter of solution.</summary>
    UO_milliliter_per_cubic_meter = 200000206,

    /// <summary>milliliter per liter: A volume per unit volume unit which is equal to one millionth of a liter of solute in one liter of solution.</summary>
    UO_milliliter_per_liter = 200000207,

    /// <summary>gram per deciliter: A mass density unit which is equal to mass of an object in grams divided by the volume in deciliters.</summary>
    UO_gram_per_deciliter = 200000208,

    /// <summary>deciliter: A volume unit which is equal to one tenth of a liter or 10^[-1] L.</summary>
    UO_deciliter = 200000209,

    /// <summary>colony forming unit: A dimensionless count unit which a measure of viable bacterial numbers.</summary>
    UO_colony_forming_unit = 200000210,

    /// <summary>plaque forming unit: A dimensionless count unit which a measure of plague forming units in a given volume.</summary>
    UO_plaque_forming_unit = 200000211,

    /// <summary>colony forming unit per volume: A concentration unit which a measure of viable bacterial numbers in a given volume.</summary>
    UO_colony_forming_unit_per_volume = 200000212,

    /// <summary>colony forming unit per milliliter: A colony forming unit which a measure of viable bacterial numbers in one milliliter.</summary>
    UO_colony_forming_unit_per_milliliter = 200000213,

    /// <summary>plaque forming unit per volume: A concentration unit which a measure of plague forming units in a given volume.</summary>
    UO_plaque_forming_unit_per_volume = 200000214,

    /// <summary>plaque forming unit per milliliter: A concentration unit which a measure of plague forming units in one milliliter.</summary>
    UO_plaque_forming_unit_per_milliliter = 200000215,

    /// <summary>disintegrations per second: An activity (of a radionuclide) unit which is equal to the activity of a quantity of radioactive material in which one nucleus decays per second or there is one atom disintegration per second.</summary>
    UO_disintegrations_per_second = 200000216,

    /// <summary>electric potential difference unit: A unit which is a standard measure of the work done per unit charge as a charge is moved between two points in an electric field.</summary>
    UO_electric_potential_difference_unit = 200000217,

    /// <summary>volt: An electric potential difference unit which is equal to the work per unit charge. One volt is the potential difference required to move one coulomb of charge between two points in a circuit while using one joule of energy.</summary>
    UO_volt = 200000218,

    /// <summary>electric charge: A unit which is a standard measure of the quantity of unbalanced electricity in a body (either positive or negative) and construed as an excess or deficiency of electrons.</summary>
    UO_electric_charge = 200000219,

    /// <summary>coulomb: An electrical charge unit which is equal to the amount of charge transferred by a current of 1 ampere in 1 second.</summary>
    UO_coulomb = 200000220,

    /// <summary>dalton: An independently to the base SI units defined mass unit which is equal to one twelfth of the mass of an unbound atom of the carbon-12 nuclide, at rest and in its ground state.</summary>
    UO_dalton = 200000221,

    /// <summary>kilodalton: A mass unit which is equal to one thousand daltons.</summary>
    UO_kilodalton = 200000222,

    /// <summary>watt-hour: An energy unit which is equal to the amount of electrical energy equivalent to a one-watt load drawing power for one hour.</summary>
    UO_watt_hour = 200000223,

    /// <summary>kilowatt-hour: An energy unit which is equal to 1,000 watt-hours.</summary>
    UO_kilowatt_hour = 200000224,

    /// <summary>magnetic flux unit: A unit which is a standard measure of quantity of magnetism, taking account of the strength and the extent of a magnetic field.</summary>
    UO_magnetic_flux_unit = 200000225,

    /// <summary>weber: A magnetic flux unit which is equal to the amount of flux that when linked with a single turn of wire for an interval of one second will induce an electromotive force of one volt.</summary>
    UO_weber = 200000226,

    /// <summary>magnetic flux density unit: A unit which is a standard measure of the strength of a magnetic field.</summary>
    UO_magnetic_flux_density_unit = 200000227,

    /// <summary>tesla: A magnetic flux density unit which is equal to one weber per square meter.</summary>
    UO_tesla = 200000228,

    /// <summary>volt-hour: A magnetic flux unit which is equal to 3600 Wb.</summary>
    UO_volt_hour = 200000229,

    /// <summary>kilovolt-hour: A magnetic flux unit which is equal to one thousand volt-hours.</summary>
    UO_kilovolt_hour = 200000230,

    /// <summary>information unit: A unit which is a standard measure of the amount of information.</summary>
    UO_information_unit = 200000231,

    /// <summary>bit: An information unit which refers to a digit in the binary numeral system, which consists of base 2 digits (ie there are only 2 possible values: 0 or 1).</summary>
    UO_bit = 200000232,

    /// <summary>byte: An information unit which is equal to 8 bits.</summary>
    UO_byte = 200000233,

    /// <summary>kilobyte: An information unit which is equal to 1000 bytes.</summary>
    UO_kilobyte = 200000234,

    /// <summary>megabyte: An information unit which is equal to 1000 kB.</summary>
    UO_megabyte = 200000235,

    /// <summary>image resolution unit: An information unit which is a standard measure of the detail an image holds.</summary>
    UO_image_resolution_unit = 200000236,

    /// <summary>chroma sampling unit: An image resolution unit which is a standard measure of the amount of spatial detail in an image.</summary>
    UO_chroma_sampling_unit = 200000237,

    /// <summary>dynamic range unit: An image resolution unit which is a standard measure of the amount of contrast available in a pixel.</summary>
    UO_dynamic_range_unit = 200000238,

    /// <summary>spatial resolution unit: An image resolution unit which is a standard measure of the way luminance and chrominance may be sampled at different levels.</summary>
    UO_spatial_resolution_unit = 200000239,

    /// <summary>dots per inch: A spatial resolution unit which is a standard measure of the printing resolution, in particular the number of individual dots of ink a printer or toner can produce within a linear one-inch space.</summary>
    UO_dots_per_inch = 200000240,

    /// <summary>micron pixel: A spatial resolution unit which is equal to a pixel size of one micrometer.</summary>
    UO_micron_pixel = 200000241,

    /// <summary>pixels per inch: A spatial resolution unit which is a standard measure of the resolution of a computer display, related to the size of the display in inches and the total number of pixels in the horizontal and vertical directions.</summary>
    UO_pixels_per_inch = 200000242,

    /// <summary>pixels per millimeter: A spatial resolution unit which is a standard measure of the number of pixels in one millimeter length or width of a digital image divided by the physical length or width of a printed image.</summary>
    UO_pixels_per_millimeter = 200000243,

    /// <summary>base pair: A count unit which contains one nucleotide.</summary>
    UO_base_pair = 200000244,

    /// <summary>kibibyte: An information unit which is equal to 1024 B.</summary>
    UO_kibibyte = 200000245,

    /// <summary>mebibyte: An information unit which is equal to 1024 KiB.</summary>
    UO_mebibyte = 200000246,

    /// <summary>millivolt: An electric potential difference unit which is equal to one thousandth of a volt or 10^[-3] V.</summary>
    UO_millivolt = 200000247,

    /// <summary>kilovolt: An electric potential difference unit which is equal to one thousand volts or 10^[3] V.</summary>
    UO_kilovolt = 200000248,

    /// <summary>microvolt: An electric potential difference unit which is equal to one millionth of a volt or 10^[-6] V.</summary>
    UO_microvolt = 200000249,

    /// <summary>nanovolt: An electric potential difference unit which is equal to one billionth of a volt or 10^[-12] V.</summary>
    UO_nanovolt = 200000250,

    /// <summary>picovolt: An electric potential difference unit which is equal to one trillionth of a volt or 10^[-12] V.</summary>
    UO_picovolt = 200000251,

    /// <summary>megavolt: An electric potential difference unit which is equal to one million volts or 10^[6] V.</summary>
    UO_megavolt = 200000252,

    /// <summary>surface tension unit: A unit which is a standard measure of the ability of a liguid to attraction of molecules at its surface as a result of unbalanced molecular cohesive forces.</summary>
    UO_surface_tension_unit = 200000253,

    /// <summary>newton per meter: A surface tension unit which is equal to one newton per meter.</summary>
    UO_newton_per_meter = 200000254,

    /// <summary>dyne per cm: A surface tension unit which is equal to one dyne per centimeter.</summary>
    UO_dyne_per_cm = 200000255,

    /// <summary>viscosity unit: A unit which is a standard measure of the internal resistance of fluids to flow.</summary>
    UO_viscosity_unit = 200000256,

    /// <summary>pascal second: A viscosity unit which is equal to one pascale per second.</summary>
    UO_pascal_second = 200000257,

    /// <summary>poise: A viscosity unit which is equal to one dyne second per square centimeter.</summary>
    UO_poise = 200000258,

    /// <summary>decibel: An ratio unit which is an indicator of sound power per unit area.</summary>
    UO_decibel = 200000259,

    /// <summary>effective dose unit: A unit which is a standard measure of the estimate of the stochastic effect that a non-uniform radiation dose has on a human.</summary>
    UO_effective_dose_unit = 200000260,

    /// <summary>conduction unit: A unit which represents a standard measurement of the transmission of an entity through a medium.</summary>
    UO_conduction_unit = 200000261,

    /// <summary>electrical conduction unit: A unit which represents a standard measurement of the movement of electrically charged particles through a transmission medium (electrical conductor).</summary>
    UO_electrical_conduction_unit = 200000262,

    /// <summary>heat conduction unit: A unit which represents a standard measurement of the spontaneous transfer of thermal energy through matter, from a region of higher temperature to a region of lower temperature.</summary>
    UO_heat_conduction_unit = 200000263,

    /// <summary>siemens: An electrical conduction unit which is equal to A/V.</summary>
    UO_siemens = 200000264,

    /// <summary>watt per meter kelvin: An heat conduction unit which is equal to one watt divided by meter kelvin.</summary>
    UO_watt_per_meter_kelvin = 200000265,

    /// <summary>electronvolt: A non-SI unit of energy (eV) defined as the energy acquired by a single unbound electron when it passes through an electrostatic potential difference of one volt. An electronvolt is equal to 1.602 176 53(14) x 10^-19 J.</summary>
    UO_electronvolt = 200000266,

    /// <summary>electric field strength unit: The electric field strength is a unit which is a measure of the potential difference between two points some distance apart.</summary>
    UO_electric_field_strength_unit = 200000267,

    /// <summary>volt per meter: The volt per meter is a unit of electric field strength equal to the a potential difference of 1 volt existing between two points that are 1 meter apart.</summary>
    UO_volt_per_meter = 200000268,

    /// <summary>absorbance unit: A dimensionless logarithmic unit assigned to a measure of absorbance of light through a partially absorbing substance, defined as -log10(I/I_0) where I = transmitted light and I_0 = incident light.</summary>
    UO_absorbance_unit = 200000269,

    /// <summary>volumetric flow rate unit: A unit which is a standard measure of the volume of fluid which passes through a given surface per unit time .</summary>
    UO_volumetric_flow_rate_unit = 200000270,

    /// <summary>microliters per minute: A volumetric flow rate unit which is equal to one microliter volume through a given surface in one minute.</summary>
    UO_microliters_per_minute = 200000271,

    /// <summary>millimetres of mercury: A unit of pressure equal to the amount of fluid pressure one millimeter deep in mercury at zero degrees centigrade on Earth.</summary>
    UO_millimetres_of_mercury = 200000272,

    /// <summary>milligram per liter: A mass unit density which is equal to mass of an object in milligrams divided by the volume in liters.</summary>
    UO_milligram_per_liter = 200000273,

    /// <summary>microgram per milliliter: A mass unit density which is equal to mass of an object in micrograms divided by the volume in millliters.</summary>
    UO_microgram_per_milliliter = 200000274,

    /// <summary>nanogram per milliliter: A mass unit density which is equal to mass of an object in nanograms divided by the volume in milliliters.</summary>
    UO_nanogram_per_milliliter = 200000275,

    /// <summary>amount per container: A concentration unit which is a standard measure of the amount of a substance in a given container.</summary>
    UO_amount_per_container = 200000276,

    /// <summary>ug/disk: A unit which is equal to one microgram per disk, where a disk is some physical surface/container upon which the substance is deposited.</summary>
    UO_ug_disk = 200000277,

    /// <summary>nmole/disk: A unit which is equal to one nanomole per disk, where a disk is some physical surface/container upon which the substance is deposited.</summary>
    UO_nmole_disk = 200000278,

    /// <summary>milliunits per milliliter: A unit per milliliter unit which is equal to one thousandth of a unit of an agreed arbitrary amount per one milliliter.</summary>
    UO_milliunits_per_milliliter = 200000279,

    /// <summary>rate unit: A unit which represents a standard measurement occurrence of a process per unit time.</summary>
    UO_rate_unit = 200000280,

    /// <summary>count per nanomolar second: A rate unit which is equal to one over one nanomolar second.</summary>
    UO_count_per_nanomolar_second = 200000281,

    /// <summary>count per molar second: A rate unit which is equal to one over one molar second.</summary>
    UO_count_per_molar_second = 200000282,

    /// <summary>kilogram per hectare: An area density unit which is equal to the mass of an object in kilograms divided by the surface area in hectares..</summary>
    UO_kilogram_per_hectare = 200000283,

    /// <summary>count per nanomolar: A rate unit which is equal to one over one nanomolar.</summary>
    UO_count_per_nanomolar = 200000284,

    /// <summary>count per molar: A rate unit which is equal to one over one molar.</summary>
    UO_count_per_molar = 200000285,

    /// <summary>microgram per liter: A mass unit density which is equal to mass of an object in micrograms divided by the volume in liters.</summary>
    UO_microgram_per_liter = 200000301
}; // enum CVID


/// <summary>
/// A list of enumerated CVIDs (CV terms); implements IList&lt;CVID&gt;
/// </summary>
public DEFINE_STD_VECTOR_WRAPPER_FOR_VALUE_TYPE(CVIDList, pwiz::cv::CVID, CVID, NATIVE_VALUE_TO_CLI, CLI_VALUE_TO_NATIVE_VALUE);

/// <summary>
/// A list of enumerated System.Strings; implements IList&lt;System.String&gt;
/// </summary>
public DEFINE_STD_VECTOR_WRAPPER_FOR_REFERENCE_TYPE(StringList, std::string, System::String, STD_STRING_TO_CLI_STRING, CLI_STRING_TO_STD_STRING);

/// <summary>
/// A utility class for getting detailed information about a particular CVID (CV term)
/// </summary>
public ref class CVTermInfo
{
    DEFINE_INTERNAL_BASE_CODE(CVTermInfo, pwiz::cv::CVTermInfo);
    public:

    /// <summary>
    /// returns the CVID corresponding this CVTermInfo instance describes
    /// <para>- note: for PSI-MS terms, the CVID always corresponds with the accession number
    /// </summary>
    property CVID cvid { CVID get() {return (CVID) base_->cvid;} }

    /// <summary>
    /// returns the accession id of the term in the form: "prefix:number", e.g. "MS:1000001"
    /// </summary>
    property System::String^ id { System::String^ get() {return gcnew System::String(base_->id.c_str());} }

    /// <summary>
    /// the full name of the term, e.g. "sample number"
    /// </summary>
    property System::String^ name { System::String^ get() {return gcnew System::String(base_->name.c_str());} }

    /// <summary>
    /// returns the full text definition of the term
    /// </summary>
    property System::String^ def { System::String^ get() {return gcnew System::String(base_->def.c_str());} }

    /// <summary>
    /// returns true if the term is obsolete
    /// </summary>
    property bool isObsolete { bool get() {return base_->isObsolete;} }

    /// <summary>
    /// returns a list of terms which this term has an IS_A relationship with
    /// </summary>
    property CVIDList^ parentsIsA { CVIDList^ get() {return gcnew CVIDList(&base_->parentsIsA);} }

    /// <summary>
    /// returns a list of terms which this term has a PART_OF relationship with
    /// </summary>
    property CVIDList^ parentsPartOf { CVIDList^ get() {return gcnew CVIDList(&base_->parentsPartOf);} }

    /// <summary>
    /// returns a list of term names synonymous with this term
    /// </summary>
    property StringList^ exactSynonyms { StringList^ get() {return gcnew StringList(&base_->exactSynonyms);} }

    CVTermInfo() : base_(new pwiz::cv::CVTermInfo()) {}

    /// <summary>
    /// returns CV term info for the specified CVID
    /// </summary>
    CVTermInfo(CVID cvid) : base_(new pwiz::cv::CVTermInfo(pwiz::cv::cvTermInfo((pwiz::cv::CVID) cvid))) {}

    /// <summary>
    /// returns CV term info for the specified id in the form: "prefix:number"
    /// </summary>
    CVTermInfo(System::String^ id) : base_(new pwiz::cv::CVTermInfo(pwiz::cv::cvTermInfo(ToStdString(id)))) {}

    /// <summary>
    /// returns the shortest synonym from exactSynonyms()
    /// </summary>
    System::String^ shortName() {return gcnew System::String(base_->shortName().c_str());}

    /// <summary>
    /// returns the prefix of the id, e.g. "MS" for PSI-MS terms
    /// </summary>
    System::String^ prefix() {return gcnew System::String(base_->prefix().c_str());}
};


} // namespace cv
} // namespace CLI
} // namespace pwiz


#endif // _CV_HPP_CLI_


